---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome  - edgeR

### Libraries
```{r, warning=FALSE, message=FALSE}
library(dplyr)
library(edgeR)
library(ggfortify)
library(gplots)
```

### Data
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) 

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1)
```


### Differential Gene Expression Analysis

#### DGElist
```{r}
DGE <- DGEList(counts=Raw, group=SampleInfo$sex)
DGE$samples
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```

### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")
```

#### Cluster dendrogram
Males = 1274, 1407, 1402, 270
Unknown = 1265
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```


### Test for DEGs
```{r}
et <- exactTest(DGE, pair=c("male", "female"))
```

#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results <- topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05)
nrow(results)

write.csv(results, "Output Files/gnome_sex_edgeR_degs.csv")
```

#### DEG Heatmap
```{r}
jpeg("Figures/genome_sex_edgeR_heatmap.jpg", width=9, height=9, units='in', res=900)
heatmap.2(as.matrix(CPM[rownames(CPM) %in% rownames(results), ]),
          scale = "row", trace = "none", margins=c(5,8), 
          cexRow = 0.5, labRow=FALSE)
dev.off()
```
