--- 
title: "insert title"
date: 'Last Updated: `r Sys.Date()`'
description: McCosker Dissertation Chapter 1 Statistical Analyses
site: bookdown::bookdown_site
documentclass: book
suppresss-bibliography: true
---

# Publication Information

## Authors 
Christina McCosker^1^, Ebru Unal, Wendy Puryear^3^, Jon Runstadler^3^, Kimberly Murray^4^, Kristina M. Cammen^1^  
*^1^School of Marine Sciences, University of Maine, Orono, ME, USA*  
*^3^ Cummings School of Veterinary Medicine, Tufts University, North Grafton, MA, USA*  
*^4^ NOAA Northeast Fisheries Science Center, Woods Hole, MA, USA*  

## Abstract
xxx


<!--chapter:end:index.Rmd-->

---
title: ''
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# LOC Gene Identification

## Load required libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(Biostrings)
library(BSgenome)
```

## Load data
```{r}
txome_genes <- 
  read.table("Input Files/txome_genecounts.txt") %>% 
  rownames_to_column(var="gene")
txome_map <- read.table("Input Files/txome_transcript-gene_map.txt", 
                        header=1)
txome_seqs <- readDNAStringSet("Input Files/txome_denovo_GG.okay_trim.txt", 
                   format = "fasta", seek.first.rec = TRUE, use.names = TRUE)

gnome_genes <- 
  read.table("Input Files/gnome_genecounts.txt") %>% 
  rownames_to_column(var="gene")
gnome_map <- read.table("Input Files/gnome_transcript-gene_map.txt", 
                        header=1)
gnome_seqs <- readDNAStringSet("Input Files/gnome_Transcript_Seqs_exons_trim.txt", 
                               format = "fasta", seek.first.rec = TRUE, use.names = TRUE)
```


## Extract gene sequences

### Extract LOC genes from gene count matrices
```{r}
txome_loc <- 
  txome_genes %>% 
  select(gene) %>% 
  filter(str_detect(gene, "^LOC"))

gnome_loc <-
  gnome_genes %>% 
  select(gene) %>% 
  filter(str_detect(gene, "^LOC"))
```

### Extract transcript IDs for all sequence IDs
```{r}
txome_loc_txpts <- 
  merge(txome_loc, txome_map, by.x="gene", by.y="gene_id", all.x=TRUE)
  
gnome_loc_txpts <- 
  merge(gnome_loc, gnome_map, by.x="gene", by.y="gene_id", all.x=TRUE)
```

### Extract transcript sequences for blast
```{r}
txome_loc_seq <- getSeq(x = txome_seqs, txome_loc_txpts$transcript_id)

gnome_loc_seq <- getSeq(x = gnome_seqs, gnome_loc_txpts$transcript_id)
```

### Export sequences
```{r}
writeXStringSet(txome_loc_seq, "Output Files/txome_loc_seqs.fasta")
writeXStringSet(gnome_loc_seq, "Output Files/gnome_loc_seqs.fasta")
```


## Blast sequences
blastn against nt database with code:  

- blastn -db nt -query input.fasta -max_target_seqs 2 -max_hsps 1 -evalue 0.01 -perc_identity 80 -outfmt "10 std qcovs qcovhsp stitle" -out output.csv -remote


## Process BLASTn Results

### Transcriptome

#### Read in & tidy results
```{r, message=FALSE}
txome_blast <- 
  read.table("Input Files/txome_loc_blastresults.csv",
             header=FALSE, sep=",", col.names=paste0("V", seq_len(24)), 
             fill=TRUE, quote="") %>%
  mutate_all(as.character) %>% 
  mutate_if(is.character, ~replace_na(.,"")) %>% 
  mutate(V15=paste0(V15, V16, V17, V18, V19, V20, V21, V22, V23, V24)) %>% 
  select(!c(16:24)) %>% 
  setNames(c("transcript_id", "sseqid", "pident", "length", "mismatch", "gapopen", "qstart", 
           "qend", "sstart", "send", "evalue", "bitscore", "qcovs", "qcovhsp", "stitle")) %>% 
  select(transcript_id, pident, qcovs, evalue, stitle) %>% 
  left_join(., txome_map)
```

#### Prepare output dataframes
```{r}
txome_geneassignments <- 
  data.frame(matrix(ncol = 6, nrow = 0)) %>% 
  setNames(c("transcript_id", "pident", "qcovs", "evalue", "stitle", "gene_id"))

txome_needcheck <- 
  data.frame(matrix(ncol = 6, nrow = 0)) %>% 
  setNames(c("transcript_id", "pident", "qcovs", "evalue", "stitle", "gene_id"))
```

#### Loop
```{r}
for (i in 1:nrow(txome_loc)) {
  data <- txome_blast[txome_blast$gene_id==txome_loc[i,1],]
  
# Remove lines with vague terms in stitle title
data <- data[grep("uncharacterized", data$stitle, invert = TRUE),]
data <- data[grep("genome assembly", data$stitle, invert = TRUE),]
data <- data[grep(" chromosome", data$stitle, invert=TRUE),]

if (nrow(data)==0) {
  next
}

# Extract gene names in ()
for (i in 1:nrow(data)) {
  stitle <- data[i,]$stitle
  {if(grepl('\\(', stitle)) {
    newstitle <- str_extract(stitle, "(?<=\\()([^()]*?)(?=\\)[^()]*$)")
    }
    else{newstitle <- NA}
  }
  data[i,]$stitle <- newstitle
}

# Select first stitle for each ID in transcript
data <- data[ !duplicated(data$transcript_id), ]

# Select only unique stitles for each gene (across transcripts)
data <- data %>% distinct(stitle, .keep_all=TRUE)

# Exclude rows with irrelevant gene names (NA, LOCs, long names)
data <- 
  data %>% 
  filter(!stitle=="NA" &
         !stitle=="" & 
         !grepl("^LOC", stitle) &
         !nchar(stitle) > 20)

# If there is one row left of the dataframe, save to a list
{if((nrow(data))==1) {
  txome_geneassignments <- rbind(data, txome_geneassignments)
}
  else{txome_needcheck <- rbind(data, txome_needcheck)}
}

# End Loop

}
```

### Genome

#### Read in & configure gnome blastn files
```{r, message=FALSE}
gnome_blast <- 
  read.csv("Input Files/gnome_loc_blastresults.csv", 
           header=FALSE, sep=",", col.names=paste0("V", seq_len(21)), 
           fill=TRUE, quote="") %>%
  mutate_all(as.character) %>% 
  mutate_if(is.character, ~replace_na(.,"")) %>% 
  mutate(V15=paste0(V15, V16, V17, V18, V19, V20, V21)) %>% 
  select(!c(16:21)) %>% 
  setNames(c("transcript_id", "sseqid", "pident", "length", "mismatch", "gapopen", "qstart", 
           "qend", "sstart", "send", "evalue", "bitscore", "qcovs", "qcovhsp", "stitle")) %>% 
  select(transcript_id, pident, qcovs, evalue, stitle) %>% 
  left_join(., gnome_map)
```

#### Prepare output dataframes
```{r}
gnome_geneassignments <- 
  data.frame(matrix(ncol = 6, nrow = 0)) %>% 
  setNames(c("transcript_id", "pident", "qcovs", "evalue", "stitle", "gene_id"))

gnome_needcheck <- 
  data.frame(matrix(ncol = 6, nrow = 0)) %>% 
  setNames(c("transcript_id", "pident", "qcovs", "evalue", "stitle", "gene_id"))
```

#### Loop
```{r}
for (i in 1:nrow(gnome_loc)) {
  data <- gnome_blast[gnome_blast$gene_id==gnome_loc[i,1],]
  
# Remove lines with vague terms in stitle title
data <- data[grep("uncharacterized", data$stitle, invert = TRUE),]
data <- data[grep("genome assembly", data$stitle, invert = TRUE),]
data <- data[grep(" chromosome", data$stitle, invert=TRUE),]

if (nrow(data)==0) {
  next
}

# Extract gene names in ()
for (i in 1:nrow(data)) {
  stitle <- data[i,]$stitle
  {if(grepl('\\(', stitle)) {
    newstitle <- str_extract(stitle, "(?<=\\()([^()]*?)(?=\\)[^()]*$)")
    }
    else{newstitle <- NA}
  }
  data[i,]$stitle <- newstitle
}

# Select first stitle for each ID in transcript
data <- data[ !duplicated(data$transcript_id), ]

# Select only unique stitles for each gene (across transcripts)
data <- data %>% distinct(stitle, .keep_all=TRUE)

# Exclude rows with irrelevant gene names (NA, LOCs, long names)
data <- 
  data %>% 
  filter(!stitle=="NA" &
         !stitle=="" & 
         !grepl("^LOC", stitle) &
         !nchar(stitle) > 20 &
         pident > 89)

# If there is one row left of the dataframe, save to a list
{if((nrow(data))==1) {
  gnome_geneassignments <- rbind(data, gnome_geneassignments)
}
  else{gnome_needcheck <- rbind(data, gnome_needcheck)}
}

# End Loop

}
```

### Combine Transcriptome & Genome Blast Results
Note: For duplicates, only using genes for which both T & G blasts yielded the same result (distinct())
```{r}
blast_all <-
  bind_rows(txome_geneassignments, gnome_geneassignments) %>% 
  select(stitle, gene_id) %>%
  distinct() %>%
  distinct(gene_id, .keep_all=TRUE) %>% 
  mutate(stitle=toupper(stitle))
```


## Combine with prior LOC ID master list
LOCs identified through blastn & manually through extracting NCBI's description
```{r}
null_loc <-
  read.csv("Input Files/TG_nullset_masterLOC.csv") %>% 
  mutate(symbol=ifelse(gene=="LOC118528561", "ANKRD13A", symbol)) %>% 
  filter(!grepl('LOC', symbol))

loc_update <- 
  merge(blast_all, null_loc, by.x="gene_id", by.y="gene", all=TRUE) %>% 
  mutate_all(~replace(., is.na(.), 0))

for (i in 1:nrow(loc_update)) {
  if(loc_update$stitle[i]=="0" && !loc_update$symbol[i]=="0") {
    loc_update$stitle[i] <- loc_update$symbol[i]
  }
  else{loc_update$stitle[i] <- loc_update$stitle[i]}
}

loc_update <- 
  loc_update %>% 
  select(!symbol)
```


## Update & tidy gene count matrices
Replace LOCs with new IDs & aggregate gene counts
```{r}
txome_genes_update <- 
  txome_genes %>% 
  mutate(gene=ifelse(gene %in% loc_update$gene_id, 
                     loc_update[match(gene, loc_update$gene_id),2],
                     gene)) %>% 
  aggregate(. ~ gene, sum)

gnome_genes_update <- 
  gnome_genes %>% 
  mutate(gene=ifelse(gene %in% loc_update$gene_id, 
                     loc_update[match(gene, loc_update$gene_id),2],
                     gene)) %>% 
  aggregate(. ~ gene, sum)
```


## Remove hemoglobin genes
Hemoglobin genes identified from Harrington et al. 2020 (suppl table 1) and from NCBI gray seal gene search for "hemoglobin"
```{r}
hemoglobin_genes <-
  c("HBA1", "HBA2", "HBB", "HBBP1", "HBD", "HBE1", "HBG1", "HBG2", "HBM", "HBQ1",
    "HBZ", "HBZP1","AHSP", "RHEX", "LOC118528782", "LOC118528783", "LOC118528784", 
    "LOC118528786", "LOC118528797", "LOC118530184", "LOC118530187")

txome_hemo <- 
  txome_genes_update %>% 
  filter(!gene %in% hemoglobin_genes)

gnome_hemo <- 
  gnome_genes_update %>% 
  filter(!gene %in% hemoglobin_genes)
```


## Export new gene count matrices 
```{r}
write.csv(txome_hemo, "Output Files/txome_genecounts_locid_nohemo.csv", row.names=FALSE)
write.csv(gnome_hemo, "Output Files/gnome_genecounts_locid_nohemo.csv", row.names=FALSE)
```


<!--chapter:end:01-LOC_Gene_Identification.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Male v Female DGE

## Transcriptome - edgeR

### Libraries
```{r, warning=FALSE, message=FALSE}
library(dplyr)
library(edgeR)
library(ggfortify)
library(gplots)
```

### Data
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) 

Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1)
```


### Differential Gene Expression Analysis

#### DGElist
```{r}
DGE <- DGEList(counts=Raw, group=SampleInfo$sex)
DGE$samples
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```

### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")
```

#### Cluster dendrogram
Males = 1274, 1407, 1402, 270
Unknown = 1265
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```

### Test for DEGs
```{r}
et <- exactTest(DGE, pair=c("male", "female"))
```

#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results <- topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05)
nrow(results)

write.csv(results, "Output Files/txome_sex_edgeR_degs.csv")
```

#### DEG Heatmap
```{r}
jpeg("Figures/transcriptome_sex_edgeR_heatmap.jpg", width=9, height=9, units='in', res=900)
heatmap.2(as.matrix(CPM[rownames(CPM) %in% rownames(results), ]),
          scale = "row", trace = "none", margins=c(5,8), 
          cexRow = 0.5, labRow=FALSE)
dev.off()
```

<!--chapter:end:02-T_Sex_edgeR.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome  - edgeR

### Libraries
```{r, warning=FALSE, message=FALSE}
library(dplyr)
library(edgeR)
library(ggfortify)
library(gplots)
```

### Data
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) 

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1)
```


### Differential Gene Expression Analysis

#### DGElist
```{r}
DGE <- DGEList(counts=Raw, group=SampleInfo$sex)
DGE$samples
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```

### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")
```

#### Cluster dendrogram
Males = 1274, 1407, 1402, 270
Unknown = 1265
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```


### Test for DEGs
```{r}
et <- exactTest(DGE, pair=c("male", "female"))
```

#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results <- topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05)
nrow(results)

write.csv(results, "Output Files/gnome_sex_edgeR_degs.csv")
```

#### DEG Heatmap
```{r}
jpeg("Figures/genome_sex_edgeR_heatmap.jpg", width=9, height=9, units='in', res=900)
heatmap.2(as.matrix(CPM[rownames(CPM) %in% rownames(results), ]),
          scale = "row", trace = "none", margins=c(5,8), 
          cexRow = 0.5, labRow=FALSE)
dev.off()
```

<!--chapter:end:03-G_Sex_edgeR.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Virus+ vs Virus- DGE

## Transcriptome - edgeR - Females

### Libraries
```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(edgeR)
library(ggfortify)
library(kableExtra)
library(knitr)
library(WebGestaltR)
```

### Data
```{r}
SampleInfo <- read.csv("Input Files/metadata.csv", 
                       stringsAsFactors = FALSE) %>% 
  mutate(RNA=ifelse(disease_stage=="acute" | disease_stage=="peak", "virus+", "virus-")) %>% 
  filter(!sex=="male")
  
#Raw RSEM Gene Counts
Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Differential Gene Expression Analysis


#### DGElist
```{r}
#Create DGE Object with counts & sample groups
DGE <- DGEList(counts = Raw, group = SampleInfo$RNA)

#Check DGE Object Samples, Groups, Lib Size, and Norm Factor
DGE$samples 
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))

postfilter <- rownames(DGE$counts)
write.table(postfilter, "Output Files/txome_female_edgeR_viralRNA_postfilter.txt",
            row.name=FALSE, col.name=FALSE, quote=FALSE)
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```

### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")

write.csv(CPM, "Output Files/txome_female_edgeR_viralRNA_normcounts.csv")
```

#### Cluster dendrogram
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="RNA") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="RNA") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```

### Test for DEGs
```{r}
et <- exactTest(DGE, pair=c("virus-", "virus+"))

write.csv(et, "Output Files/txome_female_edgeR_viralRNA_DGE_results.csv")
```

#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results <- 
  topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results)

write.csv(results, "Output Files/txome_female_edgeR_viralRNA_DEG.csv")
```

#### Make sure genes are up/down regulated in Virus+ vs Virus-
Visually compare gene expression across viral RNA status to ensure directional change is correct.
```{r}
CPM %>% 
  rownames_to_column(var="gene") %>% 
  filter(gene %in% results$gene) %>% 
  pivot_longer(cols=!gene, names_to="sample", values_to="expression") %>% 
  mutate(RNA=ifelse(sample %in% SampleInfo$sample, 
                              SampleInfo[match(sample, SampleInfo$sample), 8],
                              NA)) %>% 
  filter(gene %in% results$gene[1:10]) %>% 
  mutate(RNA=factor(RNA, levels=c("virus-", "virus+"))) %>% 
  ggplot(., aes(x=RNA, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene)
```

### GO Enrichment Analysis

#### Configure Gene Lists
```{r}
background <- rownames(CPM)

up <- 
  results %>% 
  filter(logFC>=1)

down <-
  results %>% 
  filter(logFC < 1)
```


#### Up-regulated genes

##### Biological Process
```{r}
go_bp_up <-
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         go="bp")
```

##### Molecular Function
```{r}
go_mf_up <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
```

##### Cellular Component
```{r}
go_cc_up <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
```

#### Down-regulated Genes

##### Biological Process
```{r}
go_bp_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         go="mf")
```

##### Cellular Component
```{r}
go_cc_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         go="cc")
```


#### Save Results
```{r}
go <-
  bind_rows(go_bp_down, go_bp_up,
            go_cc_down, 
            go_mf_down)

go <- 
  go %>% 
  select(!link & !overlapId) 

go %>% 
  mutate(across(where(is.numeric), \(x) round(x, digits=5))) %>% 
  kable() %>% 
  kable_styling("basic", font_size=9)

write.csv(go, "Output Files/txome_female_edgeR_viralRNA_go.csv",
          row.names=FALSE)
```

### KEGG Analysis

```{r}
kegg <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE) 

write.csv(kegg, "Output Files/txome_female_edgeR_viralRNA_kegg.csv")
```

<!--chapter:end:04-T_F_edgeR_viralRNA.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome - edgeR - Females 

### Libraries
```{r, message=FALSE, warning=FALSE}
library(tidyverse)
library(edgeR)
library(ggfortify)
library(kableExtra)
library(knitr)
library(WebGestaltR)
```

### Data
```{r}
SampleInfo <- read.csv("Input Files/metadata.csv", 
                       stringsAsFactors = FALSE) %>% 
  mutate(RNA=ifelse(disease_stage=="acute" | disease_stage=="peak", "virus+", "virus-")) %>% 
  filter(!sex=="male")
  
#Raw RSEM Gene Counts
Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Differential Gene Expression Analysis

#### DGElist
```{r}
#Create DGE Object with counts & sample groups
DGE <- DGEList(counts = Raw, group = SampleInfo$RNA)

#Check DGE Object Samples, Groups, Lib Size, and Norm Factor
DGE$samples 
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))

postfilter <- rownames(DGE$counts)
write.table(postfilter, "Output Files/gnome_female_edgeR_viralRNA_postfilter.txt",
            row.name=FALSE, col.name=FALSE, quote=FALSE)
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```

### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")

write.csv(CPM, "Output Files/gnome_female_edgeR_viralRNA_normcounts.csv")
```

#### Cluster dendrogram
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="RNA") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="RNA") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```

### Test for DEGs
```{r}
et <- exactTest(DGE, pair=c("virus-", "virus+"))

write.csv(et, "Output Files/gnome_female_edgeR_viralRNA_DGE_results.csv")
```

#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results <- 
  topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results)

write.csv(results, "Output Files/gnome_female_edgeR_viralRNA_DEG.csv")
```

#### Make sure genes are up/down regulated in Virus+ vs Virus-
Visually compare gene expression across viral RNA status to ensure directional change is correct.
```{r}
CPM %>% 
  rownames_to_column(var="gene") %>% 
  filter(gene %in% results$gene) %>% 
  pivot_longer(cols=!gene, names_to="sample", values_to="expression") %>% 
  mutate(RNA=ifelse(sample %in% SampleInfo$sample, 
                              SampleInfo[match(sample, SampleInfo$sample), 8],
                              NA)) %>% 
  filter(gene %in% results$gene[1:10]) %>% 
  mutate(RNA=factor(RNA, levels=c("virus-", "virus+"))) %>% 
  ggplot(., aes(x=RNA, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene)
```

### GO Enrichment Analysis

#### Configure Gene Lists
```{r}
background <- rownames(CPM)

up <- 
  results %>% 
  filter(logFC>=1)
down <-
  results %>% 
  filter(logFC < 1)
```

#### Up-regulated genes

##### Biological Process
```{r}
go_bp_up <-
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         go="bp")
```

##### Molecular Function
```{r}
go_mf_up <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
```

##### Cellular Component
```{r}
go_cc_up <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
```

#### Down-regulated Genes

##### Biological Process
```{r}
go_bp_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         go="mf")
```

##### Cellular Component
```{r}
go_cc_down <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         go="cc")
```


#### Save Results
```{r}
go <-
  bind_rows(go_bp_down, go_bp_up,
            go_cc_down,
            go_mf_down)

go <- 
  go %>% 
  select(!link & !overlapId) 

go %>% 
  mutate(across(where(is.numeric), \(x) round(x, digits=5))) %>% 
  kable() %>% 
  kable_styling("basic", font_size=9)


write.csv(go, "Output Files/gnome_female_edgeR_viralRNA_go.csv")
```

### KEGG Analysis

```{r}
kegg <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE) 

write.csv(kegg, "Output Files/gnome_female_edgeR_viralRNA_kegg.csv")
```

<!--chapter:end:05-G_F_edgeR_viralRNA.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Transcriptome Null Set - edgeR - Females

```{r include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- read.csv("Input Files/metadata.csv", 
                       stringsAsFactors = FALSE) %>% 
  mutate(RNA=ifelse(disease_stage=="acute" | disease_stage=="peak", "virus+", "virus-")) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Loop Differential Gene Expression Analysis

#### Configure inputs & set up results lists
```{r}
rna <- SampleInfo$RNA

degenes <- setNames(vector('list', 1000), 1:1000)

postfilter <- setNames(vector('list', 1000), 1:1000)
```

#### Loop
Random number generator used to generate 9 for set.seed() function.
```{r}
set.seed(9)

for (i in 1:1000) {

#Shuffle group assignments without replacement
groups <- sample(rna, replace = FALSE, prob = NULL)

# Create DGE Object
DGE <- DGEList(counts=Raw, group=groups)

# Filter out lowly expressed genes
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]

postfilter[[i]] <- 
  DGE %>% 
  '[['(1) %>% 
  rownames()

# Normalize based on library sizes & calculate dispersion
DGE <- calcNormFactors(DGE)
DGE <- estimateCommonDisp(DGE)
DGE <- estimateTrendedDisp(DGE)
DGE <- estimateTagwiseDisp(DGE)

# Test for DEGs
et <- exactTest(DGE, pair=c("virus-", "virus+"))

# Select DEGs by fold change & pvalue
results <- 
  topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

# Save output to list
degenes[[i]] <- results$gene

# End loop
}
```


#### Save lists of DEGs
```{r}
# Postfilter Lists
max_length <- max(unlist(lapply(postfilter, length)))
postfilter_filled <-
  lapply(postfilter, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
postfilter_df <- do.call(cbind, postfilter_filled)
write.csv(postfilter_df, file = "Output Files/txome_female_edgeR_viralRNA_null_DEG_postfilter.csv", row.names = FALSE)

# DEGs
max_length <- max(unlist(lapply(degenes, length)))
filled <-
  lapply(degenes, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
df <- do.call(cbind, filled)
write.csv(df, file = "Output Files/txome_female_edgeR_viralRNA_null_DEG.csv", row.names = FALSE)
```

### GO Enrichment Analysis

#### Import data 
```{r}
degenes <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_DEG.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)])

postfilter <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_DEG_postfilter.csv")
```

#### Set up results lists
```{r}
go_bp <- setNames(vector('list', 1000), 1:1000)
go_mf <- setNames(vector('list', 1000), 1:1000)
go_cc <- setNames(vector('list', 1000), 1:1000)
```

#### Loop
```{r}
for (i in 1:1000) {
  genes <- degenes[[i]]
  background <- postfilter[[i]]
  bp_test <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
go_bp[[i]] <- bp_test$description
  
mf_test <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
go_mf[[i]] <- mf_test$description

cc_test <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)    
go_cc[[i]] <- cc_test$description
}
```

#### Save files
```{r}
max_length <- max(unlist(lapply(go_bp, length)))
go_bp_filled <-
  lapply(go_bp, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_final <- do.call(cbind, go_bp_filled)
write.csv(go_bp_final, file = "Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf, length)))
go_mf_filled <-
  lapply(go_mf, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_final <- do.call(cbind, go_mf_filled)
write.csv(go_mf_final, file = "Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc, length)))
go_cc_filled <-
  lapply(go_cc, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_final <- do.call(cbind, go_cc_filled)
write.csv(go_cc_final, file = "Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv", row.names = FALSE)
```

### KEGG Analysis

#### Set up results lists
```{r}
kegg <- setNames(vector('list', 1000), 1:1000)
```


#### Loop
```{r}
for (i in 164:1000) {
genes <- degenes[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg[[i]] <- NA, 
       kegg[[i]] <- test$description)
}
```

#### Save results
```{r}
max_length <- max(unlist(lapply(kegg, length)))
kegg <-
  lapply(kegg, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg <- do.call(cbind, kegg)
write.csv(kegg, file = "Output Files/txome_female_edgeR_viralRNA_null_kegg.csv", row.names = FALSE)
```



<!--chapter:end:06-T_F_edgeR_viralRNA_Null.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome Null Set - edgeR -  Females
```{r include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- read.csv("Input Files/metadata.csv", 
                       stringsAsFactors = FALSE) %>% 
  mutate(RNA=ifelse(disease_stage=="acute" | disease_stage=="peak", "virus+", "virus-")) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Loop Differential Gene Expression Analysis

#### Configure inputs & set up results lists
```{r}
rna <- SampleInfo$RNA

degenes <- setNames(vector('list', 1000), 1:1000)

postfilter <- setNames(vector('list', 1000), 1:1000)
```


#### Loop
Random number generator used to generate 9 for set.seed() function.

```{r}
set.seed(9)

for (i in 1:1000) {

#Shuffle group assignments without replacement
groups <- sample(rna, replace = FALSE, prob = NULL)

# Create DGE Object
DGE <- DGEList(counts=Raw, group=groups)

# Filter out lowly expressed genes
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]

postfilter[[i]] <- 
  DGE %>% 
  '[['(1) %>% 
  rownames()

# Normalize based on library sizes & calculate dispersion
DGE <- calcNormFactors(DGE)
DGE <- estimateCommonDisp(DGE)
DGE <- estimateTrendedDisp(DGE)
DGE <- estimateTagwiseDisp(DGE)

# Test for DEGs
et <- exactTest(DGE, pair=c("virus-", "virus+"))

# Select DEGs by fold change & pvalue
results <- 
  topTags(et, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

# Save output to list
degenes[[i]] <- results$gene

# End loop
}
```


#### Save lists of DEGs
```{r}
# Postfilter Lists
max_length <- max(unlist(lapply(postfilter, length)))
postfilter_filled <-
  lapply(postfilter, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
postfilter_df <- do.call(cbind, postfilter_filled)
write.csv(postfilter_df, file = "Output Files/gnome_female_edgeR_viralRNA_null_DEG_postfilter.csv", row.names = FALSE)

# DEGs
max_length <- max(unlist(lapply(degenes, length)))
filled <-
  lapply(degenes, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
df <- do.call(cbind, filled)
write.csv(df, file = "Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv", row.names = FALSE)
```


### GO Enrichment Analysis

#### Import data 
```{r}
degenes <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv")

postfilter <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_DEG_postfilter.csv")
```

#### Set up results lists
```{r}
go_bp <- setNames(vector('list', 1000), 1:1000)
go_mf <- setNames(vector('list', 1000), 1:1000)
go_cc <- setNames(vector('list', 1000), 1:1000)
```


#### Loop
```{r}
for (i in 1:1000) {
  genes <- degenes[[i]]
  background <- postfilter[[i]]
  bp_test <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
go_bp[[i]] <- bp_test$description

mf_test <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)
go_mf[[i]] <- mf_test$description

cc_test <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE)    
go_cc[[i]] <- cc_test$description
}
```

#### Save files
```{r}
max_length <- max(unlist(lapply(go_bp, length)))
go_bp_filled <-
  lapply(go_bp, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_final <- do.call(cbind, go_bp_filled)
write.csv(go_bp_final, file = "Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf, length)))
go_mf_filled <-
  lapply(go_mf, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_final <- do.call(cbind, go_mf_filled)
write.csv(go_mf_final, file = "Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc, length)))
go_cc_filled <-
  lapply(go_cc, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_final <- do.call(cbind, go_cc_filled)
write.csv(go_cc_final, file = "Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv", row.names = FALSE)
```


### KEGG Analysis

#### Set up results lists
```{r}
kegg <- setNames(vector('list', 1000), 1:1000)
```

#### Loop
```{r}
for (i in 1:1000) {
genes <- degenes[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg[[i]] <- NA, 
       kegg[[i]] <- test$description)
}
```

#### Save results
```{r}
max_length <- max(unlist(lapply(kegg, length)))
kegg <-
  lapply(kegg, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg <- do.call(cbind, kegg)
write.csv(kegg, file = "Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv", row.names = FALSE)
```

<!--chapter:end:07-G_F_edgeR_viralRNA_Null.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Refine GO/KEGG
*Based on Null Set results

### Libraries
```{r}
library(tidyverse)
library(kableExtra)
library(knitr)
library(ggh4x)
library(gridExtra)
```

### Data
#### GO/KEGG Data
```{r}
txome_go <- read.csv("Output Files/txome_female_edgeR_viralRNA_go.csv")
t_null_bp <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv")
t_null_mf <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv")
t_null_cc <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv")


txome_kegg <- read.csv("Output Files/txome_female_edgeR_viralRNA_kegg.csv")
t_kegg <- read.csv("Output Files/txome_female_edgeR_viralRNA_null_kegg.csv")

gnome_go <- read.csv("Output Files/gnome_female_edgeR_viralRNA_go.csv", row.names=1)
g_null_bp <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv")
g_null_mf <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv")
g_null_cc <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv")

gnome_kegg <- read.csv("Output Files/gnome_female_edgeR_viralRNA_kegg.csv")
g_kegg <- read.csv("Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv")
```

#### Raw Gene Expression
```{r}
t_raw <- read.csv("Output Files/txome_female_edgeR_viralRNA_normcounts.csv") %>% 
  dplyr::rename("gene" = "X")

g_raw <- read.csv("Output Files/gnome_female_edgeR_viralRNA_normcounts.csv") %>% 
  dplyr::rename("gene" = "X")

meta <- read.csv("Input Files/metadata.csv")
```

### Transcriptome

#### GO Term

##### What terms are overrepresented in >10% of nullsets?
```{r}
# BP
t_null_bp_10 <-
  t_null_bp %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
t_null_mf_10 <-
  t_null_mf %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
t_null_cc_10 <-
  t_null_cc %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter results
```{r}
final_t_bp <- 
  txome_go %>% 
  filter(go=="bp" &
         !description %in% t_null_bp_10$value)
         
final_t_mf <-
  txome_go %>% 
  filter(go=="mf" &
         !description %in% t_null_mf_10$value)

final_t_cc <-
  txome_go %>% 
  filter(go=="cc" &
         !description %in% t_null_cc_10$value)
```

#### KEGG Pathway
##### What pathways are overrepreesented  in >10% of nullsets?
```{r}
t_kegg_10 <-
  t_kegg %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter Results
```{r}
txome_kegg_final <- 
  txome_kegg %>% 
  filter(!description %in% t_kegg_10$value)
txome_kegg %>% 
  kable()
```


### Genome

#### GO Term

##### What terms are overrepresented in >10% of nullsets?
```{r}
# BP
g_null_bp_10 <-
  g_null_bp %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
g_null_mf_10 <-
  g_null_mf %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
g_null_cc_10 <-
  g_null_cc %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter results
```{r}
final_g_bp <- 
  gnome_go %>% 
  filter(go=="bp" &
         !description %in% g_null_bp_10$value)
         
final_g_mf <-
  gnome_go %>% 
  filter(go=="mf" &
         !description %in% g_null_mf_10$value)

final_g_cc <-
  gnome_go %>% 
  filter(go=="cc" &
         !description %in% g_null_cc_10$value)
```

#### KEGG Pathway
##### What pathways are overrepreesented  in >10% of nullsets?
```{r}
g_kegg_10 <-
  g_kegg %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter Results
```{r}
gnome_kegg_final <- 
  gnome_kegg %>% 
  filter(!description %in% g_kegg_10$value) 
gnome_kegg %>% 
  kable()
```

### GO Term Figure
```{r}
t_final <- 
  rbind(final_t_bp, final_t_mf, final_t_cc) %>% 
  mutate(approach="Transcriptome")
write.csv(t_final, "Output Files/txome_female_edgeR_viralRNA_go_refined.csv", row.names=FALSE)

g_final <-
  rbind(final_g_bp, final_g_mf, final_g_cc) %>% 
  mutate(approach= "Genome")
write.csv(g_final, "Output Files/gnome_female_edgeR_viralRNA_go_refined.csv", row.names=FALSE)

go_all <-
  rbind(t_final, g_final)

go_fig <-
  go_all %>% 
  mutate(go=factor(
           ifelse(go=="bp", "BP", 
                  ifelse(go=="cc", "CC", "MF")),
           levels=c("BP", "CC", "MF")), 
         enrichmentRatio=ifelse(direction=="down", enrichmentRatio*-1, enrichmentRatio)) %>% 
  
  ggplot(data=., aes(x=fct_reorder(description, enrichmentRatio, .desc=TRUE), 
                     y=enrichmentRatio, fill=factor(approach))) +
  geom_bar(stat="identity", position="dodge") +
  scale_fill_manual(values=c("slategray3", "slategrey"), name=NULL) +
  labs(x="GO Term", y="Enrichment Ratio") +
  scale_x_discrete(labels = function(description) str_wrap(description, width = 65)) +
  scale_y_continuous(n.breaks=10) +
  facet_grid(vars(go), scales="free_y", space="free_y") +
  theme_bw() +
  theme(panel.grid=element_blank(), text=element_text(size=11), 
        legend.position="bottom") +
  coord_flip() 
  
ggsave(filename="Figures/viralRNA_goterm.jpg", go_fig, width=9, height=8, units="in")
```


### Immunne GO term Genes 

#### Extracting Immune GO Genes
Manually look up function of each gene & categorize. Select subset of 16 genes for presentation in manuscript. 
```{r}
immune_go_genes <- 
  go_all %>% 
  filter(grepl("immune|viral|inflammatory|interferon|leukocyte", description)) %>% 
  select(userId) %>% 
  mutate(genes = str_split_fixed(userId, ";", n=18)) %>% 
  select(!userId) %>% 
  unlist() %>% 
  data.frame() %>% 
  setNames("genes") %>% 
  filter(!duplicated(genes)) %>% 
  filter(!genes=="") %>% 
  arrange(genes) %>% 
  mutate(description = c(
    "Innate Immunity", 
    "extra", 
    "Antiviral", 
    "Innate Immunity & adaptive", 
    "extra", 
    "extra", 
    "Complement", 
    "Complement", 
    "Complement", 
    "Complement", 
    "adaptive", 
    "Complement", 
    "Complement",
    "Immunoregulation",
    "Innate Immunity", 
    "inflammation", 
    "mhc", 
    "Innate Immunity",
    "Immunoregulation",
    "Immunoregulation", 
    "Innate Immunity", 
    "Immunoregulation", 
    "Innate Immunity", 
    "extra",
    "extra", 
    "Antiviral", 
    "Antiviral", 
    "Innate Immunity", 
    "extra", 
    "extra", 
    "extra", 
    "Antiviral", 
    "Immunoregulation", 
    "inflammation", 
    "extra", 
    "Innate Immunity", 
    "Immunoregulation"
  ))

immune_go_genes_subset <- 
  immune_go_genes %>% 
  filter(description == "Innate Immunity" |
         description == "Complement" |
         description == "Immunoregulation" |
         description == "Antiviral") %>% 
  filter(!genes == "C2" & 
         !genes == "CFD",
         !genes == "HSPA8",
         !genes == "ACTB", 
         !genes == "DHX58") %>% 
  mutate(description = factor(description, levels=c("Innate Immunity", "Immunoregulation", "Antiviral", "Complement")))
```

#### Extract Expression Data
```{r}
t_immunegenes_subset <- 
  t_raw %>% 
  filter(gene %in% immune_go_genes_subset$genes) %>% 
  pivot_longer(!gene, names_to="sample", values_to="expression") %>% 
  mutate(disease=ifelse(sample %in% meta$sample,
                     meta[match(sample, meta$sample),4],
                     sample),
         disease=factor(ifelse(disease=="acute", "A", 
                               ifelse(disease=="peak", "P", 
                                      ifelse(disease=="late", "L", "C"))),
                        levels=c("C", "A", "P", "L")),
         description = ifelse(gene %in% immune_go_genes$genes, 
                              immune_go_genes[match(gene, immune_go_genes$genes),2],
                              gene)) 

t_immune_genes_all <-
  t_raw %>% 
  filter(gene %in% immune_go_genes$genes) %>% 
  pivot_longer(!gene, names_to="sample", values_to="expression") %>% 
  mutate(disease=ifelse(sample %in% meta$sample,
                     meta[match(sample, meta$sample),4],
                     sample),
         disease=factor(ifelse(disease=="acute", "A", 
                               ifelse(disease=="peak", "P", 
                                      ifelse(disease=="late", "L", "C"))),
                        levels=c("C", "A", "P", "L")))

g_immune_genes_all <- 
  g_raw %>% 
  filter(gene %in% immune_go_genes$genes) %>% 
  pivot_longer(!gene, names_to="sample", values_to="expression") %>% 
  mutate(disease=ifelse(sample %in% meta$sample,
                     meta[match(sample, meta$sample),4],
                     sample),
         disease=factor(ifelse(disease=="acute", "A", 
                               ifelse(disease=="peak", "P", 
                                      ifelse(disease=="late", "L", "C"))),
                        levels=c("C", "A", "P", "L"))) 
```

#### Plot for manuscript
```{r}
t_plot_manuscript <-
  t_immunegenes_subset %>% 
  ggplot(., aes(x=disease, y=expression)) +
  geom_boxplot() +
  facet_nested_wrap(vars(description, gene),
                    nest_line=element_line(colour="black")) +
  theme(ggh4x.facet.nestline = element_line(linetype = 1)) +
  labs(x = "Influenza Stage", y = "log2(CPM)") +
  theme_bw() +
  theme(panel.grid=element_blank())
t_plot_manuscript

ggsave("Figures/T_female_edgeR_immunegenes.jpg", t_plot_manuscript, 
       width=8, height=8, units="in")
```

#### Plot for supplemental
```{r}
t_plot_supplemental <-
  t_immune_genes_all %>% 
  ggplot(aes(x=disease, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene, scales="free", ncol=5) +
  labs(title=NULL,
       x="Influenza Stage", y="log2(CPM)") +
  theme_bw() +
  theme(panel.grid=element_blank())

ggsave("Figures/supplemental_T_female_edgeR_immunegenes.jpg", t_plot_supplemental,
          width=8, height=10, units="in")

g_plot_supplemental <-
  g_immune_genes_all %>% 
  ggplot(aes(x=disease, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene, scales="free", ncol=5) +
  labs(title=NULL,
       x="Influenza Stage", y="log2(CPM)") +
  theme_bw() +
  theme(panel.grid=element_blank())

ggsave("Figures/supplemental_G_female_edgeR_immunegenes.jpg", g_plot_supplemental,
          width=8, height=10, units="in")
```


### KEGG Table - Manuscript
```{r}
t_kegg_table <- 
  txome_kegg_final %>% 
  mutate(`Pairwise Comparison (Approach)` = "Virus+ vs Virus- (Transcriptome)",
         `KEGG Pathway` = description,
         `Enrichment Ratio (FDR)` = paste0(round(enrichmentRatio, digits=3), " (", round(FDR, digits=4), ")"),
         Genes=userId) %>% 
  select(`Pairwise Comparison (Approach)`, `KEGG Pathway`, `Enrichment Ratio (FDR)`, Genes)

g_kegg_table <-
  gnome_kegg_final %>% 
  mutate(`Pairwise Comparison (Approach)` = "Virus+ vs Virus- (Genome)",
         `KEGG Pathway` = description,
         `Enrichment Ratio (FDR)` = paste0(round(enrichmentRatio, digits=2), " (", round(FDR, digits=4), ")"),
         Genes=userId) %>% 
  select(`Pairwise Comparison (Approach)`, `KEGG Pathway`, `Enrichment Ratio (FDR)`, Genes)


kegg_table <- 
  rbind(t_kegg_table, g_kegg_table)

write.csv(kegg_table, "Output Files/manuscript_keggtable.csv", row.names=FALSE)
```


### Supplemental Table

#### Biological Process
```{r}
t_sup_bp <- 
  txome_go %>% 
  filter(go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_bp <- 
  gnome_go %>% 
  filter(go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_bp <- 
  merge(t_sup_bp, g_sup_bp, all=TRUE) %>% 
  mutate(go="BP")
```


#### Cellular Component
```{r}
t_sup_cc <- 
  txome_go %>% 
  filter(go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_cc <- 
  gnome_go %>% 
  filter(go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_cc <- 
  merge(t_sup_cc, g_sup_cc, all=TRUE) %>% 
  mutate(go="CC")
```


#### Molecular Function
```{r}
t_sup_mf <- 
  txome_go %>% 
  filter(go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_mf <- 
  gnome_go %>% 
  filter(go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_mf <- 
  merge(t_sup_mf, g_sup_mf, all=TRUE) %>% 
  mutate(go="MF")
```

#### Save GO Tables
```{r}
sup <- 
  rbind(sup_bp, sup_cc, sup_mf)

sup %>% 
  kable()

write.csv(sup, "Output Files/supplemental_female_edgeR_viralRNA_go.csv", row.names=FALSE)
```

#### KEGG
```{r}
t_sup_kegg <- 
  txome_kegg %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Transcriptome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_kegg$value, "x", ""),
         Transcriptome_Genes = userId) %>% 
  select(KEGG, Transcriptome, Transcriptome_null, Transcriptome_Genes)

g_sup_kegg <- 
  gnome_kegg %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Genome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_kegg$value, "x", ""),
         Genome_Genes = userId) %>% 
  select(KEGG, Genome, Genome_null, Genome_Genes)


sup_kegg <- 
  merge(t_sup_kegg, g_sup_kegg, all=TRUE)

sup_kegg %>% 
  kable()

write.csv(sup_kegg, "Output Files/supplemental_female_edgeR_viralRNA_kegg.csv", row.names=FALSE)
```

<!--chapter:end:08-TG_F_edgeR_viralRNA_nullcomp.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Influenza Stage DGE 

## Transcriptome - edgeR - Females

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(ggfortify)
library(kableExtra)
library(knitr)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Differential Gene Expression Analysis

#### DGElist
```{r}
DGE <- DGEList(counts=Raw, group=SampleInfo$disease_stage)
DGE$samples
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))

postfilter <- rownames(DGE$counts)
write.table(postfilter, "Output Files/txome_female_edgeR_postfilter.txt",
            row.name=FALSE, col.name=FALSE, quote=FALSE)
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```


### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")

write.csv(CPM, "Output Files/txome_female_edgeR_normcounts.csv")
```

#### Cluster dendrogram
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```


### Test for DEGs
```{r}
et_acute <- exactTest(DGE, pair=c("control", "acute"))
et_peak <- exactTest(DGE, pair=c("control", "peak"))
et_late <- exactTest(DGE, pair=c("control", "late"))

# Save for DESeq2 comparisons
write.csv(et_acute, "Output Files/txome_female_edgeR_DGE_results_acute.csv")
write.csv(et_peak, "Output Files/txome_female_edgeR_DGE_results_peak.csv")
write.csv(et_late, "Output Files/txome_female_edgeR_DGE_results_late.csv")
```


#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results_acute <- 
  topTags(et_acute, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_acute)

results_peak <- 
  topTags(et_peak, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_peak)

results_late <- 
  topTags(et_late, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_late)

write.csv(results_acute, "Output Files/txome_female_edgeR_DEG_acute.csv")
write.csv(results_peak, "Output Files/txome_female_edgeR_DEG_peak.csv")
write.csv(results_late, "Output Files/txome_female_edgeR_DEG_late.csv")
```

#### Make sure genes are up/down regulated in APL vs Control
Visually compare gene expression across disease stages to DEGs in acute, peak, late to ensure directional change is correct.
```{r}
CPM %>% 
  rownames_to_column(var="gene") %>% 
  filter(gene %in% results_acute$gene) %>% 
  pivot_longer(cols=!gene, names_to="sample", values_to="expression") %>% 
  mutate(disease_stage=ifelse(sample %in% SampleInfo$sample, 
                              SampleInfo[match(sample, SampleInfo$sample), 4],
                              NA)) %>% 
  filter(gene %in% results_acute$gene[1:10]) %>% 
  mutate(disease_stage=factor(disease_stage, levels=c("control", "acute", "peak", "late"))) %>% 
  ggplot(., aes(x=disease_stage, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene)
```


### GO Enrichment Analysis

#### Configure Gene Lists
```{r}
background <- rownames(CPM)

acute_up <- 
  results_acute %>% 
  filter(logFC>=1)
acute_down <-
  results_acute %>% 
  filter(logFC < 1)

peak_up <- 
  results_peak %>% 
  filter(logFC>=1)
peak_down <-
  results_peak %>% 
  filter(logFC < 1)

late_up <- 
  results_late %>% 
  filter(logFC>=1)
late_down <-
  results_late %>% 
  filter(logFC < 1)
```


#### Up-regulated genes

##### Biological Process
```{r}
go_bp_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="bp")


go_bp_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         disease_stage="peak",
         go="bp")

go_bp_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         disease_stage="late",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="mf")

go_mf_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="peak",
         go="mf")

go_mf_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="late",
         go="mf")
```

##### Cellular Component
```{r}
go_cc_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="cc")

go_cc_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="peak",
         go="cc")

go_cc_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="late",
         go="cc")
```


#### Down-regulated genes

##### Biological Process
```{r}
go_bp_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="bp")

go_bp_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="bp")

go_bp_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="mf")

go_mf_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="mf")

go_mf_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="mf")
```

##### Cellular Component
```{r}
go_cc_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="cc")

go_cc_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="cc")

go_cc_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="cc")
```

#### Save results
```{r}
go <-
  bind_rows(go_bp_down_acute,
            go_bp_up_peak, go_bp_down_peak,
            go_bp_up_late,
            go_mf_down_acute,
            go_mf_up_peak, go_mf_down_peak,
            go_mf_up_late,
            go_cc_down_acute,
            go_cc_up_peak, go_cc_down_peak,
            go_cc_up_late)
go <- 
  go %>% 
  select(!link & !overlapId) 

go %>% 
  mutate(across(where(is.numeric), \(x) round(x, digits=5))) %>% 
  kable() %>% 
  kable_styling("basic", font_size=9)

write.csv(go, "Output Files/txome_female_edgeR_go.csv")
```

### KEGG Analysis

#### Acute
```{r}
kegg_acute <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_acute$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE) %>% 
  mutate(disease_stage="acute")
```

#### Peak
```{r}
kegg_peak <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_peak$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)  %>% 
  mutate(disease_stage="peak")
```

#### Late
```{r}
kegg_late <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_late$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)  #%>% 
  #mutate(disease_stage="late")
```

#### Save results
```{r}
kegg <- 
  bind_rows(kegg_acute, kegg_peak) 

write.csv(kegg, "Output Files/txome_female_edgeR_kegg.csv")
```


<!--chapter:end:09-T_F_edgeR.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome - edgeR - Females


### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(ggfortify)
library(kableExtra)
library(knitr)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Differential Gene Expression Analysis

#### DGElist
```{r}
DGE <- DGEList(counts=Raw, group=SampleInfo$disease_stage)
DGE$samples
```

#### Filter out lowly expressed genes
```{r}
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]
length(rownames(DGE$counts))

postfilter <- rownames(DGE$counts)
write.table(postfilter, "Output Files/gnome_female_edgeR_postfilter.txt",
            row.name=FALSE, col.name=FALSE, quote=FALSE)
```

#### Normalize based on library sizes
```{r}
DGE <- calcNormFactors(DGE)
```

#### Estimate Dispersion
On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%
```{r}
# Overall common dispersion
DGE <- estimateCommonDisp(DGE, verbose = TRUE)

# Dispersion trend based on gene abundance
DGE <- estimateTrendedDisp(DGE)

# Tagwise dispersion - allows different dispersion for each gene
DGE <- estimateTagwiseDisp(DGE, verbose = TRUE)

plotBCV(DGE)
```


### Exploratory Plots

#### Normalized counts
```{r}
CPM <- 
  cpm(DGE, normalized.lib.sizes = TRUE, log = TRUE) %>% 
  data.frame()
boxplot(CPM, las = 2, ylab = "log2 CPM", main = "Normalized Data")

write.csv(CPM, "Output Files/gnome_female_edgeR_normcounts.csv")
```

#### Cluster dendrogram
Males = 1274, 1407, 1402, 270
Unknown = 1265
```{r}
RawDist <- dist(t(CPM), method = "euclidean")
plot(hclust(RawDist, method = "average"), xlab="Average Euclidean Distance")
```

#### PCA Plots
```{r}
PCA <- prcomp(t(RawDist))

# PCs 1-2
autoplot(PCA, data=SampleInfo, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, colour="molt") +
  theme(legend.position="bottom")

# PCs 3-4
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="disease_stage") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="sex") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="year") +
  theme(legend.position="bottom")
autoplot(PCA, data=SampleInfo, x=3, y=4, colour="molt") +
  theme(legend.position="bottom")
```


### Test for DEGs
```{r}
et_acute <- exactTest(DGE, pair=c("control", "acute")) 
et_peak <- exactTest(DGE, pair=c("control", "peak"))
et_late <- exactTest(DGE, pair=c("control", "late"))

# Save for DESeq2 comparisons
write.csv(et_acute, "Output Files/gnome_female_edgeR_DGE_results_acute.csv")
write.csv(et_peak, "Output Files/gnome_female_edgeR_DGE_results_peak.csv")
write.csv(et_late, "Output Files/gnome_female_edgeR_DGE_results_late.csv")
```


#### Select DEGs by fold change & pvalue
Uncorrected p < 0.05
log2FC > 1 (2-fold change)
```{r}
results_acute <- 
  topTags(et_acute, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_acute)

results_peak <- 
  topTags(et_peak, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_peak)

results_late <- 
  topTags(et_late, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")
nrow(results_late)

write.csv(results_acute, "Output Files/gnome_female_edgeR_DEG_acute.csv")
write.csv(results_peak, "Output Files/gnome_female_edgeR_DEG_peak.csv")
write.csv(results_late, "Output Files/gnome_female_edgeR_DEG_late.csv")
```


#### Make sure genes are up/down regulated in APL vs Control
Visually compare gene expression across disease stages to DEGs in acute, peak, late to ensure directional change is correct.
```{r}
CPM %>% 
  rownames_to_column(var="gene") %>% 
  filter(gene %in% results_acute$gene) %>% 
  pivot_longer(cols=!gene, names_to="sample", values_to="expression") %>% 
  mutate(disease_stage=ifelse(sample %in% SampleInfo$sample, 
                              SampleInfo[match(sample, SampleInfo$sample), 4],
                              NA)) %>% 
  filter(gene %in% results_acute$gene[1:10]) %>% 
  mutate(disease_stage=factor(disease_stage, levels=c("control", "acute", "peak", "late"))) %>% 
  ggplot(., aes(x=disease_stage, y=expression)) +
  geom_boxplot() +
  facet_wrap(~gene)
```


### GO Enrichment Analysis

#### Configure Gene Lists
```{r}
background <- rownames(CPM)

acute_up <- 
  results_acute %>% 
  filter(logFC>=1)
acute_down <-
  results_acute %>% 
  filter(logFC < 1)

peak_up <- 
  results_peak %>% 
  filter(logFC>=1)
peak_down <-
  results_peak %>% 
  filter(logFC < 1)

late_up <- 
  results_late %>% 
  filter(logFC>=1)
late_down <-
  results_late %>% 
  filter(logFC < 1)
```


#### Up-regulated genes

##### Biological Process
```{r}
go_bp_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="bp")


go_bp_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         disease_stage="peak",
         go="bp")

go_bp_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up",
         disease_stage="late",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="mf")

go_mf_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="peak",
         go="mf")

go_mf_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="late",
         go="mf")
```

##### Cellular Component
```{r}
go_cc_up_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = acute_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="acute",
         go="cc")

go_cc_up_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = peak_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="peak",
         go="cc")

go_cc_up_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = late_up$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="up", 
         disease_stage="late",
         go="cc")
```


#### Down-regulated genes

##### Biological Process
```{r}
go_bp_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="bp")

go_bp_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="bp")

go_bp_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="bp")
```


##### Molecular Function
```{r}
go_mf_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="mf")

go_mf_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="mf")

go_mf_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="mf")
```

##### Cellular Component
```{r}
go_cc_down_acute <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = acute_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="acute",
         go="cc")

go_cc_down_peak <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = peak_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="peak",
         go="cc")

go_cc_down_late <- 
    WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = late_down$gene, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) %>% 
  mutate(direction="down", 
         disease_stage="late",
         go="cc")
```

#### Save results
```{r}
go <-
  bind_rows(
            go_bp_up_peak,
            go_bp_up_late,
            
            go_mf_up_peak,
            go_mf_up_late,
            go_cc_down_acute,
            go_cc_up_peak
            )
go <- 
  go %>% 
  select(!link & !overlapId) 

go %>% 
  mutate(across(where(is.numeric), \(x) round(x, digits=5))) %>% 
  kable() %>% 
  kable_styling("basic", font_size=9)

write.csv(go, "Output Files/gnome_female_edgeR_go.csv")
```

### KEGG Analysis

#### Acute
```{r}
kegg_acute <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_acute$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE) 
```

#### Peak
```{r}
kegg_peak <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_peak$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)  %>% 
  mutate(disease_stage="peak")
```

#### Late
```{r}
kegg_late <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = results_late$gene, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)
```

#### Save results
```{r}
kegg <- 
  bind_rows(kegg_peak) 

write.csv(kegg, "Output Files/gnome_female_edgeR_kegg.csv")
```

<!--chapter:end:10-G_F_edgeR.Rmd-->

---
title: ''
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Comapre T & G Approach - edgeR - Females

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(VennDiagram)
library(gridExtra)
```

### Data

#### Transcriptome results
```{r}
t_virus <- read.csv("Output Files/txome_female_edgeR_viralRNA_DGE_results.csv") %>% 
  dplyr::rename("gene" = "X")

t_acute <- 
  read.csv("Output Files/txome_female_edgeR_DGE_results_acute.csv") %>% 
  dplyr::rename("gene" = "X")

t_peak <-
  read.csv("Output Files/txome_female_edgeR_DGE_results_peak.csv") %>% 
  dplyr::rename("gene" = "X")

t_late <- 
  read.csv("Output Files/txome_female_edgeR_DGE_results_late.csv") %>% 
  dplyr::rename("gene" = "X")
```

#### Transcriptome DEGs
```{r}
tvirus_up <- 
  read.csv("Output Files/txome_female_edgeR_viralRNA_DEG.csv", row.name=1) %>% 
  filter(logFC>0)
tvirus_down <- 
  read.csv("Output Files/txome_female_edgeR_viralRNA_DEG.csv", row.name=1) %>% 
  filter(logFC<0)

t_acute_up <- 
  read.csv("Output Files/txome_female_edgeR_DEG_acute.csv", row.name=1) %>% 
  filter(logFC>0)
t_acute_down <- 
  read.csv("Output Files/txome_female_edgeR_DEG_acute.csv", row.name=1) %>% 
  filter(logFC<0)
t_peak_up <- 
  read.csv("Output Files/txome_female_edgeR_DEG_peak.csv", row.name=1) %>% 
  filter(logFC>0)
t_peak_down <- 
  read.csv("Output Files/txome_female_edgeR_DEG_peak.csv", row.name=1) %>% 
  filter(logFC<0)
t_late_up <- 
  read.csv("Output Files/txome_female_edgeR_DEG_late.csv", row.name=1) %>% 
  filter(logFC>0)
t_late_down <- 
  read.csv("Output Files/txome_female_edgeR_DEG_late.csv", row.name=1) %>% 
  filter(logFC<0)
```

#### Genome results
```{r}
g_virus <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_DGE_results.csv") %>% 
  dplyr::rename("gene"= "X")

g_acute <- 
  read.csv("Output Files/gnome_female_edgeR_DGE_results_acute.csv") %>% 
  dplyr::rename("gene" = "X")

g_peak <-
  read.csv("Output Files/gnome_female_edgeR_DGE_results_peak.csv") %>% 
  dplyr::rename("gene" = "X")

g_late <- 
  read.csv("Output Files/gnome_female_edgeR_DGE_results_late.csv") %>% 
  dplyr::rename("gene" = "X")
```

#### Genome DEGs
```{r}
gvirus_up <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_DEG.csv", row.name=1) %>% 
  filter(logFC>0)
gvirus_down <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_DEG.csv", row.name=1) %>% 
  filter(logFC<0)

g_acute_up <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_acute.csv", row.name=1) %>% 
  filter(logFC>0)
g_acute_down <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_acute.csv", row.name=1) %>% 
  filter(logFC<0)
g_peak_up <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_peak.csv", row.name=1) %>% 
  filter(logFC>0)
g_peak_down <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_peak.csv", row.name=1) %>% 
  filter(logFC<0)
g_late_up <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_late.csv", row.name=1) %>% 
  filter(logFC>0)
g_late_down <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_late.csv", row.name=1) %>% 
  filter(logFC<0)
```


### Number of up & down reg genes

#### Set up data frame
```{r}
gene_num <- 
  data.frame(
    approach = c("T", "T", "T", "T",
                 "T", "T", "T", "T",
                 "G", "G", "G", "G", 
                 "G", "G", "G", "G"),
    stage = c("V+ v V-", "V+ v V-", 
              "Acute", "Acute", "Peak", "Peak", "Late", "Late",
              "V+ v V-", "V+ v V-", 
              "Acute", "Acute", "Peak", "Peak", "Late", "Late"),
    direction = c("Up", "Down", "Up", "Down",
                  "Up", "Down", "Up", "Down",
                  "Up", "Down", "Up", "Down",
                   "Up", "Down", "Up", "Down"),
    genes = c(nrow(tvirus_up), nrow(tvirus_down),
              nrow(t_acute_up), nrow(t_acute_down), 
              nrow(t_peak_up), nrow(t_peak_down), 
              nrow(t_late_up), nrow(t_late_down),
              nrow(gvirus_up), nrow(gvirus_down),
              nrow(g_acute_up), nrow(g_acute_down), 
              nrow(g_peak_up), nrow(g_peak_down), 
              nrow(g_late_up), nrow(g_late_down))) %>% 
  mutate(stage=factor(stage, levels=c("V+ v V-", "Acute", "Peak", "Late")),
         direction=factor(direction, levels=c("Up", "Down")),
         approach=factor(approach, levels=c("T", "G")))
```

#### Plot
```{r}
gene_plot <- 
  ggplot(gene_num, aes(x=approach, y=genes, fill=direction)) +
  geom_col() +
  facet_grid(~stage) +
  labs(x="Approach", y = "Number of Genes") +
  scale_fill_manual(values=c("gray75", "gray42"), name=NULL) +
  theme_bw() +
  theme(panel.grid=element_blank())

ggsave("Figures/TG_F_edgeR_approachcomp.jpeg", gene_plot, 
       width=4, height=3, units="in")
```


### Gene identities
#### Set up lists
```{r}
gene_list <- 
  list(
    t_acute_up = t_acute_up$gene,
    t_acute_down = t_acute_down$gene,
    t_peak_up = t_peak_up$gene,
    t_peak_down = t_peak_down$gene,
    t_late_up = t_late_up$gene,
    t_late_down = t_late_down$gene,
    t_virus_up = tvirus_up$gene,
    t_virus_down = tvirus_down$gene,
    g_acute_up = g_acute_up$gene,
    g_acute_down = g_acute_down$gene,
    g_peak_up = g_peak_up$gene,
    g_peak_down = g_peak_down$gene,
    g_late_up = g_late_up$gene,
    g_late_down = g_late_down$gene,
    g_virus_up = gvirus_up$gene,
    g_virus_down = gvirus_down$gene
  )
```

#### Venn diagram
```{r, message=FALSE, warnings=FALSE}
virus_up <- 
   venn.diagram(gene_list[c(7,15)], 
               filename=NULL,
               disable.logging=TRUE,
               main="V+ v V- Up", main.pos=c(0.5,0.85),
               category.names = c("T", "G"), 
               cat.pos=c(0,0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               invert=TRUE,
               margin=0.3,
               height=3, 
               width=3,
               units="in")

virus_down <- 
   venn.diagram(gene_list[c(8,16)], 
               filename=NULL,
               disable.logging=TRUE,
               main="V+ v V- Down", main.pos=c(0.5,0.78),
               category.names = c("T", "G"), 
               cat.pos=c(0, 0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               margin=0.3,
               height=3, 
               width=3,
               units="in")

acute_up <- 
   venn.diagram(gene_list[c(1,9)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Acute Up", main.pos=c(0.5,0.85),
               category.names = c("T", "G"), 
               cat.pos=c(0, 0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               invert=TRUE,
               margin=0.3,
               height=3, 
               width=3,
               units="in")
 
acute_down<- 
    venn.diagram(gene_list[c(2,10)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Acute Down", main.pos=c(0.5,0.8),
               category.names = c("T", "G"), 
               cat.pos=c(0, 0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               margin=0.3,
               height=3, 
               width=3,
               units="in")
peak_up <- 
   venn.diagram(gene_list[c(3,11)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Peak Up", main.pos=c(0.5,0.85),
               category.names = c("T", "G"), 
               cat.pos=c(0,0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               invert=TRUE,
               margin=0.3,
               height=3, 
               width=3,
               units="in")
peak_down <- 
   venn.diagram(gene_list[c(4,12)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Peak Down", main.pos=c(0.5,0.8),
               category.names = c("T", "G"), 
               cat.pos=c(0,0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               margin=0.3,
               height=3, 
               width=3,
               units="in")

late_up <- 
   venn.diagram(gene_list[c(5,13)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Late Up", main.pos=c(0.5,0.85),
               category.names = c("T", "G"), 
               cat.pos=c(0,0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               invert=TRUE,
               margin=0.3,
               height=3, 
               width=3,
               units="in")

late_down <- 
   venn.diagram(gene_list[c(6,14)], 
               filename=NULL,
               disable.logging=TRUE,
               main="Late Down", main.pos=c(0.5,0.8),
               category.names = c("T", "G"), 
               cat.pos=c(0,0),
               cat.just=list(c(0.5,0) , c(0.5,0)),
               invert=TRUE,
               margin=0.3,
               height=3, 
               width=3,
               units="in")

venn <- 
  grid.arrange(gTree(children=virus_up),
               gTree(children=acute_up),
               gTree(children=peak_up),
               gTree(children=late_up),
               gTree(children=virus_down),
               gTree(children=acute_down),
               gTree(children=peak_down),
               gTree(children=late_down),
               ncol=4,
               rectGrob(x=c(0.5, 1.5, 2.5, 3.5,
                            0.5, 1.5, 2.5, 3.5), 
                        y=c(2.5, 2.5, 2.5, 2.5,
                            1.5, 1.5, 1.5, 1.5), 
                        gp=gpar(lwd=1, fill=NA))) 

ggsave("Figures/TG_F_edgeR_approachcomp_venn.jpeg", 
       venn, width=8, height=5, units="in")
```

<!--chapter:end:11-TG_F_edgeR_approachcomp.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Transcriptome Null Set - edgeR - Females

```{r include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Loop Differential Gene Expression Analysis

#### Configure inputs & set up results lists
```{r}
diseasestage <- SampleInfo$disease_stage

acute <- setNames(vector('list', 1000), 1:1000)
peak <- setNames(vector('list', 1000), 1:1000)
late <- setNames(vector('list', 1000), 1:1000)

postfilter <- setNames(vector('list', 1000), 1:1000)
```

#### Loop
Random number generator used to generate 9 for set.seed() function.
```{r}
set.seed(9)

for (i in 1:1000) {

#Shuffle group assignments without replacement
groups <- sample(diseasestage, replace = FALSE, prob = NULL)

# Create DGE Object
DGE <- DGEList(counts=Raw, group=groups)

# Filter out lowly expressed genes
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]

postfilter[[i]] <- 
  DGE %>% 
  '[['(1) %>% 
  rownames()

# Normalize based on library sizes & calculate dispersion
DGE <- calcNormFactors(DGE)
DGE <- estimateCommonDisp(DGE)
DGE <- estimateTrendedDisp(DGE)
DGE <- estimateTagwiseDisp(DGE)

# Test for DEGs
et_acute <- exactTest(DGE, pair=c("control", "acute"))
et_peak <- exactTest(DGE, pair=c("control", "peak"))
et_late <- exactTest(DGE, pair=c("control", "late"))

# Select DEGs by fold change & pvalue
results_acute <- 
  topTags(et_acute, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

results_peak <- 
  topTags(et_peak, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

results_late <- 
  topTags(et_late, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

# Save output to list
acute[[i]] <- results_acute$gene
peak[[i]] <- results_peak$gene
late[[i]] <- results_late$gene

# End loop
}
```

#### Save lists of DEGs
```{r}
# Postfilter Lists
max_length <- max(unlist(lapply(postfilter, length)))
postfilter_filled <-
  lapply(postfilter, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
postfilter_df <- do.call(cbind, postfilter_filled)
write.csv(postfilter_df, file = "Output Files/txome_female_edgeR_null_DEG_postfilter.csv", row.names = FALSE)

# Acute
max_length <- max(unlist(lapply(acute, length)))
acute_filled <-
  lapply(acute, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
acute_df <- do.call(cbind, acute_filled)
write.csv(acute_df, file = "Output Files/txome_female_edgeR_null_DEG_acute.csv", row.names = FALSE)

# Peak
max_length <- max(unlist(lapply(peak, length)))
peak_filled <-
  lapply(peak, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
peak_df <- do.call(cbind, peak_filled)
write.csv(peak_df, file = "Output Files/txome_female_edgeR_null_DEG_peak.csv", row.names = FALSE)

# Late
max_length <- max(unlist(lapply(late, length)))
late_filled <-
  lapply(late, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
late_df <- do.call(cbind, late_filled)
write.csv(late_df, file = "Output Files/txome_female_edgeR_null_DEG_late.csv", row.names = FALSE)
```

### GO Enrichment Analysis

#### Import data 
```{r}
acute <- read.csv("Output Files/txome_female_edgeR_null_DEG_acute.csv")
peak <- read.csv("Output Files/txome_female_edgeR_null_DEG_peak.csv")
late <- read.csv("Output Files/txome_female_edgeR_null_DEG_late.csv")

postfilter <- read.csv("Output Files/txome_female_edgeR_null_DEG_postfilter.csv")
```

#### Set up results lists
```{r}
go_bp_acute <- setNames(vector('list', 1000), 1:1000)
go_bp_peak <- setNames(vector('list', 1000), 1:1000)
go_bp_late <- setNames(vector('list', 1000), 1:1000)

go_mf_acute <- setNames(vector('list', 1000), 1:1000)
go_mf_peak <- setNames(vector('list', 1000), 1:1000)
go_mf_late <- setNames(vector('list', 1000), 1:1000)

go_cc_acute <- setNames(vector('list', 1000), 1:1000)
go_cc_peak <- setNames(vector('list', 1000), 1:1000)
go_cc_late <- setNames(vector('list', 1000), 1:1000)
```

#### Loops!
```{r}
# Acute
for (i in 1:1000) {
  genes <- acute[[i]]
  background <- postfilter[[i]]
  bp_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_acute[[i]] <- bp_test_acute$description
  
  mf_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_acute[[i]] <- mf_test_acute$description

  cc_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_acute[[i]] <- cc_test_acute$description
}


# Peak
for (i in 1:1000) {
  genes <- peak[[i]]
  background <- postfilter[[i]]
  bp_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_peak[[i]] <- bp_test_peak$description
  
  mf_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_peak[[i]] <- mf_test_peak$description

  cc_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_peak[[i]] <- cc_test_peak$description
}

## Late
for (i in 1:1000) {
  genes <- late[[i]]
  background <- postfilter[[i]]
  bp_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_late[[i]] <- bp_test_late$description
  
  mf_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_late[[i]] <- mf_test_late$description

  cc_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_late[[i]] <- cc_test_late$description
}
```

#### Save acute files
```{r}
max_length <- max(unlist(lapply(go_bp_acute, length)))

go_bp_acute_filled <-
  lapply(go_bp_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_acute_final <- do.call(cbind, go_bp_acute_filled)
write.csv(go_bp_acute_final, file = "Output Files/txome_female_edgeR_null_GO_bp_acute.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_acute, length)))
go_mf_acute_filled <-
  lapply(go_mf_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_acute_final <- do.call(cbind, go_mf_acute_filled)
write.csv(go_mf_acute_final, file = "Output Files/txome_female_edgeR_null_GO_mf_acute.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_acute, length)))
go_cc_acute_filled <-
  lapply(go_cc_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_acute_final <- do.call(cbind, go_cc_acute_filled)
write.csv(go_cc_acute_final, file = "Output Files/txome_female_edgeR_null_GO_cc_acute.csv", row.names = FALSE)
```

#### Save peak files
```{r}
max_length <- max(unlist(lapply(go_bp_peak, length)))
go_bp_peak_filled <-
  lapply(go_bp_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_peak_final <- do.call(cbind, go_bp_peak_filled)
write.csv(go_bp_peak_final, file = "Output Files/txome_female_edgeR_null_GO_bp_peak.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_peak, length)))
go_mf_peak_filled <-
  lapply(go_mf_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_peak_final <- do.call(cbind, go_mf_peak_filled)
write.csv(go_mf_peak_final, file = "Output Files/txome_female_edgeR_null_GO_mf_peak.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_peak, length)))
go_cc_peak_filled <-
  lapply(go_cc_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_peak_final <- do.call(cbind, go_cc_peak_filled)
write.csv(go_cc_peak_final, file = "Output Files/txome_female_edgeR_null_GO_cc_peak.csv", row.names = FALSE)
```

#### Save late files
```{r}
max_length <- max(unlist(lapply(go_bp_late, length)))
go_bp_late_filled <-
  lapply(go_bp_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_late_final <- do.call(cbind, go_bp_late_filled)
write.csv(go_bp_late_final, file = "Output Files/txome_female_edgeR_null_GO_bp_late.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_late, length)))
go_mf_late_filled <-
  lapply(go_mf_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_late_final <- do.call(cbind, go_mf_late_filled)
write.csv(go_mf_late_final, file = "Output Files/txome_female_edgeR_null_GO_mf_late.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_late, length)))
go_cc_late_filled <-
  lapply(go_cc_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_late_final <- do.call(cbind, go_cc_late_filled)
write.csv(go_cc_late_final, file = "Output Files/txome_female_edgeR_null_GO_cc_late.csv", row.names = FALSE)
```

### KEGG Analysis

#### Set up results lists
```{r}
kegg_acute <- setNames(vector('list', 1000), 1:1000)
kegg_peak <- setNames(vector('list', 1000), 1:1000)
kegg_late <-setNames(vector('list', 1000), 1:1000)
```


#### Acute
```{r}
for (i in 1:1000) {
genes <- acute[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg_acute[[i]] <- NA, 
       kegg_acute[[i]] <- test$description)
}

# Save results
max_length <- max(unlist(lapply(kegg_acute, length)))
kegg_acute <-
  lapply(kegg_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_acute <- do.call(cbind, kegg_acute)
write.csv(kegg_acute, file = "Output Files/txome_female_edgeR_null_kegg_acute.csv", row.names = FALSE)
```

#### Peak
```{r}
for (i in 1:1000) {
genes <- peak[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg_peak[[i]] <- NA, 
       kegg_peak[[i]] <- test$description)
}

# Save results
max_length <- max(unlist(lapply(kegg_peak, length)))
kegg_peak <-
  lapply(kegg_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_peak <- do.call(cbind, kegg_peak)
write.csv(kegg_peak, file = "Output Files/txome_female_edgeR_null_kegg_peak.csv", row.names = FALSE)
```

#### Late
```{r}
for (i in 1:1000) {
genes <- late[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg_late[[i]] <- NA, 
       kegg_late[[i]] <- test$description)
}

# Save results 
max_length <- max(unlist(lapply(kegg_late, length)))
kegg_late <-
  lapply(kegg_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_late <- do.call(cbind, kegg_late)
write.csv(kegg_late, file = "Output Files/txome_female_edgeR_null_kegg_late.csv", row.names = FALSE)
```


<!--chapter:end:12-T_F_edgeR_Null.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome Null Set - edgeR - Females
```{r include=FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(edgeR)
library(WebGestaltR)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male")

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)
```

### Loop Differential Gene Expression Analysis

#### Configure inputs & set up results lists
```{r}
diseasestage <- SampleInfo$disease_stage

acute <- setNames(vector('list', 1000), 1:1000)
peak <- setNames(vector('list', 1000), 1:1000)
late <- setNames(vector('list', 1000), 1:1000)

postfilter <- setNames(vector('list', 1000), 1:1000)
```

#### Loop
Random number generator used to generate 9 for set.seed() function.
```{r}
set.seed(9)

for (i in 1:1000) {

#Shuffle group assignments without replacement
groups <- sample(diseasestage, replace = FALSE, prob = NULL)

# Create DGE Object
DGE <- DGEList(counts=Raw, group=groups)

# Filter out lowly expressed genes
keep <- filterByExpr(DGE)
DGE <- DGE[keep, , keep.lib.sizes=FALSE]

postfilter[[i]] <- 
  DGE %>% 
  '[['(1) %>% 
  rownames()

# Normalize based on library sizes & calculate dispersion
DGE <- calcNormFactors(DGE)
DGE <- estimateCommonDisp(DGE)
DGE <- estimateTrendedDisp(DGE)
DGE <- estimateTagwiseDisp(DGE)

# Test for DEGs
et_acute <- exactTest(DGE, pair=c("control", "acute"))
et_peak <- exactTest(DGE, pair=c("control", "peak"))
et_late <- exactTest(DGE, pair=c("control", "late"))

# Select DEGs by fold change & pvalue
results_acute <- 
  topTags(et_acute, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

results_peak <- 
  topTags(et_peak, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

results_late <- 
  topTags(et_late, n = dim(DGE)[1]) %>% 
  data.frame() %>% 
  filter(abs(logFC) > 1 & PValue < 0.05) %>% 
  rownames_to_column(var="gene")

# Save output to list
acute[[i]] <- results_acute$gene
peak[[i]] <- results_peak$gene
late[[i]] <- results_late$gene

# End loop
}
```

#### Save lists of DEGs
```{r}
# Postfilter Lists
max_length <- max(unlist(lapply(postfilter, length)))
postfilter_filled <-
  lapply(postfilter, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
postfilter_df <- do.call(cbind, postfilter_filled)
write.csv(postfilter_df, file = "Output Files/gnome_female_edgeR_DEG_postfilter.csv", row.names = FALSE)

# Acute
max_length <- max(unlist(lapply(acute, length)))
acute_filled <-
  lapply(acute, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
acute_df <- do.call(cbind, acute_filled)
write.csv(acute_df, file = "Output Files/gnome_female_edgeR_null_DEG_acute.csv", row.names = FALSE)

# Peak
max_length <- max(unlist(lapply(peak, length)))
peak_filled <-
  lapply(peak, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
peak_df <- do.call(cbind, peak_filled)
write.csv(peak_df, file = "Output Files/gnome_female_edgeR_null_DEG_peak.csv", row.names = FALSE)

# Late
max_length <- max(unlist(lapply(late, length)))
late_filled <-
  lapply(late, function(x) {ans <- rep(NA,length=max_length);
    ans[0:length(x)]<- x;
    return(ans)})
late_df <- do.call(cbind, late_filled)
write.csv(late_df, file = "Output Files/gnome_female_edgeR_null_DEG_late.csv", row.names = FALSE)
```

### GO Enrichment Analysis

#### Import data 
```{r}
acute <- read.csv("Output Files/gnome_female_edgeR_null_DEG_acute.csv")
peak <- read.csv("Output Files/gnome_female_edgeR_null_DEG_peak.csv")
late <- read.csv("Output Files/gnome_female_edgeR_null_DEG_late.csv")

postfilter <- read.csv("Output Files/gnome_female_edgeR_null_DEG_postfilter.csv")
```

#### Set up results lists
```{r}
go_bp_acute <- setNames(vector('list', 1000), 1:1000)
go_bp_peak <- setNames(vector('list', 1000), 1:1000)
go_bp_late <- setNames(vector('list', 1000), 1:1000)

go_mf_acute <- setNames(vector('list', 1000), 1:1000)
go_mf_peak <- setNames(vector('list', 1000), 1:1000)
go_mf_late <- setNames(vector('list', 1000), 1:1000)

go_cc_acute <- setNames(vector('list', 1000), 1:1000)
go_cc_peak <- setNames(vector('list', 1000), 1:1000)
go_cc_late <- setNames(vector('list', 1000), 1:1000)
```

#### Acute Loop
```{r}
for (i in 1:1000) {
  genes <- acute[[i]]
  background <- postfilter[[i]]
  bp_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_acute[[i]] <- bp_test_acute$description
  
  mf_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_acute[[i]] <- mf_test_acute$description

  cc_test_acute <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_acute[[i]] <- cc_test_acute$description
}

for (i in 1:1000) {
  genes <- peak[[i]]
  background <- postfilter[[i]]
  bp_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_peak[[i]] <- bp_test_peak$description
  
  mf_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_peak[[i]] <- mf_test_peak$description

  cc_test_peak <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_peak[[i]] <- cc_test_peak$description
}

for (i in 1:1000) {
  genes <- late[[i]]
  background <- postfilter[[i]]
  bp_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Biological_Process",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_bp_late[[i]] <- bp_test_late$description
  
  mf_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Molecular_Function",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_mf_late[[i]] <- mf_test_late$description

  cc_test_late <- 
        WebGestaltR(enrichMethod="ORA", organism="hsapiens", 
                enrichDatabase="geneontology_Cellular_Component",
                interestGene = genes, interestGeneType = "genesymbol", 
                referenceGene = background, referenceGeneType = "genesymbol", 
                sigMethod = "fdr", fdrThr = 0.05, 
                isOutput = FALSE) 
  go_cc_late[[i]] <- cc_test_late$description
}
```

#### Save acute files
```{r}
max_length <- max(unlist(lapply(go_bp_acute, length)))
go_bp_acute_filled <-
  lapply(go_bp_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_acute_final <- do.call(cbind, go_bp_acute_filled)
write.csv(go_bp_acute_final, file = "Output Files/gnome_female_edgeR_null_GO_bp_acute.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_acute, length)))
go_mf_acute_filled <-
  lapply(go_mf_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_acute_final <- do.call(cbind, go_mf_acute_filled)
write.csv(go_mf_acute_final, file = "Output Files/gnome_female_edgeR_null_GO_mf_acute.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_acute, length)))
go_cc_acute_filled <-
  lapply(go_cc_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_acute_final <- do.call(cbind, go_cc_acute_filled)
write.csv(go_cc_acute_final, file = "Output Files/gnome_female_edgeR_null_GO_cc_acute.csv", row.names = FALSE)
```

#### Save peak files
```{r}
max_length <- max(unlist(lapply(go_bp_peak, length)))
go_bp_peak_filled <-
  lapply(go_bp_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_peak_final <- do.call(cbind, go_bp_peak_filled)
write.csv(go_bp_peak_final, file = "Output Files/gnome_female_edgeR_null_GO_bp_peak.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_peak, length)))
go_mf_peak_filled <-
  lapply(go_mf_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_peak_final <- do.call(cbind, go_mf_peak_filled)
write.csv(go_mf_peak_final, file = "Output Files/gnome_female_edgeR_null_GO_mf_peak.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_peak, length)))
go_cc_peak_filled <-
  lapply(go_cc_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_peak_final <- do.call(cbind, go_cc_peak_filled)
write.csv(go_cc_peak_final, file = "Output Files/gnome_female_edgeR_null_GO_cc_peak.csv", row.names = FALSE)
```


#### Save late files
```{r}
max_length <- max(unlist(lapply(go_bp_late, length)))
go_bp_late_filled <-
  lapply(go_bp_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_bp_late_final <- do.call(cbind, go_bp_late_filled)
write.csv(go_bp_late_final, file = "Output Files/gnome_female_edgeR_null_GO_bp_late.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_mf_late, length)))
go_mf_late_filled <-
  lapply(go_mf_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_mf_late_final <- do.call(cbind, go_mf_late_filled)
write.csv(go_mf_late_final, file = "Output Files/gnome_female_edgeR_null_GO_mf_late.csv", row.names = FALSE)

max_length <- max(unlist(lapply(go_cc_late, length)))
go_cc_late_filled <-
  lapply(go_cc_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
go_cc_late_final <- do.call(cbind, go_cc_late_filled)
write.csv(go_cc_late_final, file = "Output Files/gnome_female_edgeR_null_GO_cc_late.csv", row.names = FALSE)
```


### KEGG Analysis

#### Set up results lists
```{r}
kegg_acute <- setNames(vector('list', 1000), 1:1000)
kegg_peak <- setNames(vector('list', 1000), 1:1000)
kegg_late <-setNames(vector('list', 1000), 1:1000)
```


#### Acute
```{r}
for (i in 1:1000) {
genes <- acute[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg_acute[[i]] <- NA, 
       kegg_acute[[i]] <- test$description)
}

# Save results
max_length <- max(unlist(lapply(kegg_acute, length)))
kegg_acute <-
  lapply(kegg_acute, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_acute <- do.call(cbind, kegg_acute)
write.csv(kegg_acute, file = "Output Files/gnome_female_edgeR_null_kegg_acute.csv", row.names = FALSE)
```

#### Peak
```{r}
for (i in 1:1000) {
genes <- peak[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)

ifelse(is.data.frame(test)=="FALSE", 
       kegg_peak[[i]] <- NA, 
       kegg_peak[[i]] <- test$description)
}

# Save results
max_length <- max(unlist(lapply(kegg_peak, length)))
kegg_peak <-
  lapply(kegg_peak, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_peak <- do.call(cbind, kegg_peak)
write.csv(kegg_peak, file = "Output Files/gnome_female_edgeR_null_kegg_peak.csv", row.names = FALSE)
```

#### Late
```{r}
for (i in 1:1000) {
genes <- late[[i]]
background <- postfilter[[i]]
test <-
  WebGestaltR(enrichMethod = "ORA", organism = "hsapiens", enrichDatabase = "pathway_KEGG",
              interestGene = genes, interestGeneType = "genesymbol", 
              referenceGene = background, referenceGeneType = "genesymbol", 
              minNum = 5, sigMethod = "fdr", isOutput = FALSE)
ifelse(is.data.frame(test)=="FALSE", 
       kegg_late[[i]] <- NA, 
       kegg_late[[i]] <- test$description)
}

# Save results
max_length <- max(unlist(lapply(kegg_late, length)))
kegg_late <-
  lapply(kegg_late, function(x) {ans <- rep(NA,length=max_length);
  ans[0:length(x)]<- x;
  return(ans)})
kegg_late <- do.call(cbind, kegg_late)
write.csv(kegg_late, file = "Output Files/gnome_female_edgeR_null_kegg_late.csv", row.names = FALSE)
```


<!--chapter:end:13-G_F_edgeR_Null.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Refine GO/KEGG
*Based on Null Set results


### Libraries
```{r}
library(tidyverse)
library(kableExtra)
library(knitr)
library(ggh4x)
library(gridExtra)
```

### Data

#### GO and KEGG
```{r}
txome_go <- read.csv("Output Files/txome_female_edgeR_go.csv")

t_null_bp_acute <- read.csv("Output Files/txome_female_edgeR_null_GO_bp_acute.csv")
t_null_mf_acute <- read.csv("Output Files/txome_female_edgeR_null_GO_mf_acute.csv")
t_null_cc_acute <- read.csv("Output Files/txome_female_edgeR_null_GO_cc_acute.csv")

t_null_bp_peak <- read.csv("Output Files/txome_female_edgeR_null_GO_bp_peak.csv")
t_null_mf_peak <- read.csv("Output Files/txome_female_edgeR_null_GO_mf_peak.csv")
t_null_cc_peak <- read.csv("Output Files/txome_female_edgeR_null_GO_cc_peak.csv")

t_null_bp_late <- read.csv("Output Files/txome_female_edgeR_null_GO_bp_late.csv")
t_null_mf_late <- read.csv("Output Files/txome_female_edgeR_null_GO_mf_late.csv")
t_null_cc_late <- read.csv("Output Files/txome_female_edgeR_null_GO_cc_late.csv")

txome_kegg <- read.csv("Output Files/txome_female_edgeR_kegg.csv")
t_kegg_acute <- read.csv("Output Files/txome_female_edgeR_null_kegg_acute.csv")
t_kegg_peak <- read.csv("Output Files/txome_female_edgeR_null_kegg_peak.csv")
t_kegg_late <- read.csv("Output Files/txome_female_edgeR_null_kegg_late.csv")

gnome_go <- read.csv("Output Files/gnome_female_edgeR_go.csv")

g_null_bp_acute <- read.csv("Output Files/gnome_female_edgeR_null_GO_bp_acute.csv")
g_null_mf_acute <- read.csv("Output Files/gnome_female_edgeR_null_GO_mf_acute.csv")
g_null_cc_acute <- read.csv("Output Files/gnome_female_edgeR_null_GO_cc_acute.csv")

g_null_bp_peak <- read.csv("Output Files/gnome_female_edgeR_null_GO_bp_peak.csv")
g_null_mf_peak <- read.csv("Output Files/gnome_female_edgeR_null_GO_mf_peak.csv")
g_null_cc_peak <- read.csv("Output Files/gnome_female_edgeR_null_GO_cc_peak.csv")

g_null_bp_late <- read.csv("Output Files/gnome_female_edgeR_null_GO_bp_late.csv")
g_null_mf_late <- read.csv("Output Files/gnome_female_edgeR_null_GO_mf_late.csv")
g_null_cc_late <- read.csv("Output Files/gnome_female_edgeR_null_GO_cc_late.csv")

gnome_kegg <- read.csv("Output Files/gnome_female_edgeR_kegg.csv")
g_kegg_acute <- read.csv("Output Files/gnome_female_edgeR_null_kegg_acute.csv")
g_kegg_peak <- read.csv("Output Files/gnome_female_edgeR_null_kegg_peak.csv")
g_kegg_late <- read.csv("Output Files/gnome_female_edgeR_null_kegg_late.csv")
```

#### Raw Gene Expression
```{r}
t_raw <- read.csv("Output Files/txome_female_edgeR_normcounts.csv") %>% 
  dplyr::rename("gene" = "X")

meta <- read.csv("Input Files/metadata.csv")
```


### Transcriptome

#### GO Term

##### What terms are overrepresented in >10% of nullsets?

###### Acute
```{r}
# BP
t_null_bp_acute_10 <-
  t_null_bp_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
t_null_mf_acute_10 <-
  t_null_mf_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
t_null_cc_acute_10 <-
  t_null_cc_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

###### Peak
```{r}
# BP
t_null_bp_peak_10 <-
  t_null_bp_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
t_null_mf_peak_10 <-
  t_null_mf_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
t_null_cc_peak_10 <-
  t_null_cc_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

###### Late
```{r}
# BP
t_null_bp_late_10 <-
  t_null_bp_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
t_null_mf_late_10 <-
  t_null_mf_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
t_null_cc_late_10 <-
  t_null_cc_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter results

###### Acute
```{r}
final_t_bp_acute <- 
  txome_go %>% 
  filter(disease_stage=="acute" & 
         go=="bp" &
         !description %in% t_null_bp_acute_10$value)
         
final_t_mf_acute <-
  txome_go %>% 
  filter(disease_stage=="acute" &
         go=="mf" &
         !description %in% t_null_mf_acute_10$value)

final_t_cc_acute <-
  txome_go %>% 
  filter(disease_stage=="acute" &
         go=="cc" &
         !description %in% t_null_cc_acute_10$value)
```

###### Peak
```{r}
final_t_bp_peak <- 
  txome_go %>% 
  filter(disease_stage=="peak" & 
         go=="bp" &
         !description %in% t_null_bp_peak_10$value)
         
final_t_mf_peak <-
  txome_go %>% 
  filter(disease_stage=="peak" &
         go=="mf" &
         !description %in% t_null_mf_peak_10$value)

final_t_cc_peak <-
  txome_go %>% 
  filter(disease_stage=="peak" &
         go=="cc" &
         !description %in% t_null_cc_peak_10$value)
```

###### Late
```{r}
final_t_bp_late <- 
  txome_go %>% 
  filter(disease_stage=="late" & 
         go=="bp" &
         !description %in% t_null_bp_late_10$value)
         
final_t_mf_late <-
  txome_go %>% 
  filter(disease_stage=="late" &
         go=="mf" &
         !description %in% t_null_mf_late_10$value)

final_t_cc_late <-
  txome_go %>% 
  filter(disease_stage=="late" &
         go=="cc" &
         !description %in% t_null_cc_late_10$value)
```


#### KEGG Pathway
##### What pathways are overrepreesented  in >10% of nullsets?
```{r}
t_kegg_acute <-
  t_kegg_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

t_kegg_peak <-
  t_kegg_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100)

t_kegg_late <-
  t_kegg_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100)
```

##### Filter Results
```{r}
txome_kegg_acute <- 
  txome_kegg %>% 
  filter(disease_stage=="acute" &
         !description %in% t_kegg_acute$value)
txome_kegg_acute %>% 
  kable()

txome_kegg_peak <- 
  txome_kegg %>% 
  filter(disease_stage=="peak" &
         !description %in% t_kegg_peak$value)
txome_kegg_peak %>% 
  kable()
```


### Genome 

#### GO Term

##### What terms are overrepresented in >10% of nullsets?

###### Acute
```{r}
# BP
g_null_bp_acute_10 <-
  g_null_bp_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
g_null_mf_acute_10 <-
  g_null_mf_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
g_null_cc_acute_10 <-
  g_null_cc_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

###### Peak
```{r}
# BP
g_null_bp_peak_10 <-
  g_null_bp_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
g_null_mf_peak_10 <-
  g_null_mf_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
g_null_cc_peak_10 <-
  g_null_cc_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

###### Late
```{r}
# BP
g_null_bp_late_10 <-
  g_null_bp_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# MF
g_null_mf_late_10 <-
  g_null_mf_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

# CC
g_null_cc_late_10 <-
  g_null_cc_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 
```

##### Filter results

###### Acute
```{r}
final_g_bp_acute <- 
  gnome_go %>% 
  filter(disease_stage=="acute" & 
         go=="bp" &
         !description %in% g_null_bp_acute_10$value)
         
final_g_mf_acute <-
  gnome_go %>% 
  filter(disease_stage=="acute" &
         go=="mf" &
         !description %in% g_null_mf_acute_10$value)

final_g_cc_acute <-
  gnome_go %>% 
  filter(disease_stage=="acute" &
         go=="cc" &
         !description %in% g_null_cc_acute_10$value)
```

###### Peak
```{r}
final_g_bp_peak <- 
  gnome_go %>% 
  filter(disease_stage=="peak" & 
         go=="bp" &
         !description %in% g_null_bp_peak_10$value)
         
final_g_mf_peak <-
  gnome_go %>% 
  filter(disease_stage=="peak" &
         go=="mf" &
         !description %in% g_null_mf_peak_10$value)

final_g_cc_peak <-
  gnome_go %>% 
  filter(disease_stage=="peak" &
         go=="cc" &
         !description %in% g_null_cc_peak_10$value)
```

###### Late
```{r}
final_g_bp_late <- 
  gnome_go %>% 
  filter(disease_stage=="late" & 
         go=="bp" &
         !description %in% g_null_bp_late_10$value)
         
final_g_mf_late <-
  gnome_go %>% 
  filter(disease_stage=="late" &
         go=="mf" &
         !description %in% g_null_mf_late_10$value)

final_g_cc_late <-
  gnome_go %>% 
  filter(disease_stage=="late" &
         go=="cc" &
         !description %in% g_null_cc_late_10$value)
```


#### KEGG Pathway
##### What pathways are overrepreesented  in >10% of nullsets?
```{r}
g_kegg_acute <-
  g_kegg_acute %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100) 

g_kegg_peak <-
  g_kegg_peak %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100)

g_kegg_late <-
  g_kegg_late %>% 
  pivot_longer(cols=1:1000) %>% 
  na.omit(value) %>% 
  group_by(value) %>% 
  summarise(count=n()) %>% 
  filter(count>100)
```

##### Filter Results
```{r}
gnome_kegg_peak <- 
  gnome_kegg %>% 
  filter(disease_stage=="peak" &
         !description %in% g_kegg_peak$value)
gnome_kegg_peak %>% 
  kable()
```


### GO Term Figure
```{r}
t_final <- 
  rbind(final_t_bp_acute, final_t_bp_peak, final_t_bp_late,
        final_t_mf_acute, final_t_mf_peak, final_t_mf_late,
        final_t_cc_acute, final_t_cc_peak, final_t_cc_late) %>% 
  mutate(approach="Transcriptome")
write.csv(t_final, "Output Files/txome_female_edgeR_go_refined.csv", row.names=FALSE)

g_final <-
  rbind(final_g_bp_acute, final_g_bp_peak, final_g_bp_late,
        final_g_mf_acute, final_g_mf_peak, final_g_mf_late,
        final_g_cc_acute, final_g_cc_peak, final_g_cc_late) %>% 
  mutate(approach= "Genome")
write.csv(g_final, "Output Files/gnome_female_edgeR_go_refined.csv", row.names=FALSE)


go_fig <-
  rbind(t_final, g_final) %>% 
  mutate(disease_stage=factor(
          ifelse(disease_stage=="acute", "A", 
             ifelse(disease_stage=="peak", "P", "L")),
          levels=c("A", "P", "L")),
         go=factor(
           ifelse(go=="bp", "BP", 
                  ifelse(go=="cc", "CC", "MF")),
           levels=c("BP", "CC", "MF")), 
         enrichmentRatio=ifelse(direction=="down", enrichmentRatio*-1, enrichmentRatio)) %>%
  
  ggplot(data=., aes(x=fct_reorder(description, enrichmentRatio, .desc=TRUE), 
                     y=enrichmentRatio, fill=factor(approach))) +
  geom_bar(stat="identity", position="dodge") +
  scale_fill_manual(values=c("slategray3", "slategrey"), name=NULL) +

  labs(x="GO Term", y="Enrichment Ratio") +
  
  scale_x_discrete(labels = function(description) str_wrap(description, width = 65)) +
  #scale_y_continuous(limits=c(-5,37), expand=c(0,0), n.breaks=8) +
  
  facet_nested_wrap(vars(disease_stage, go), dir="v", scales="free_y",
                    strip.position="right", ncol=1, drop=TRUE) +
  
  force_panelsizes(rows = c(3, 2.5, 2, 4.8, 0.8)) +
  
  theme_bw() +
  theme(panel.grid=element_blank(), text=element_text(size=11), 
        legend.position="bottom", legend.justification="left", 
        axis.title.x=element_text(vjust=-3), axis.title.y=element_text(vjust=2)) +
  
  geom_hline(yintercept=0, colour="gray40") +
  geom_vline(xintercept=c(1.5:20.5), colour="gray88") +
  coord_flip() 

go_fig
ggsave(filename="Figures/goterm.jpg", width=10, height=9, units="in")
```


### Supplemental Table

#### Acute GO

##### Biological Process
```{r}
t_sup_a_bp <- 
  txome_go %>% 
  filter(disease_stage=="acute" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_acute_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_a_bp <- 
  gnome_go %>% 
  filter(disease_stage=="acute" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_acute_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_a_bp <- 
  merge(t_sup_a_bp, g_sup_a_bp, all=TRUE) %>% 
  mutate(go="BP")
```


##### Cellular Component
```{r}
t_sup_a_cc <- 
  txome_go %>% 
  filter(disease_stage=="acute" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_cc_acute_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_a_cc <- 
  gnome_go %>% 
  filter(disease_stage=="acute" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_cc_acute_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_a_cc <- 
  merge(t_sup_a_cc, g_sup_a_cc, all=TRUE) %>% 
  mutate(go="CC")
```


##### Molecular Function
```{r}
t_sup_a_mf <- 
  txome_go %>% 
  filter(disease_stage=="acute" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_mf_acute_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_a_mf <- 
  gnome_go %>% 
  filter(disease_stage=="acute" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_mf_acute_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_a_mf <- 
  merge(t_sup_a_mf, g_sup_a_mf, all=TRUE) %>% 
  mutate(go="MF")
```

##### Save Tables
```{r}
sup_a <- 
  rbind(sup_a_bp, sup_a_cc, sup_a_mf)

sup_a %>% 
  kable()

write.csv(sup_a, "Output Files/supplemental_female_edgeR_go_acute.csv", row.names=FALSE)
```



#### Peak GO

##### Biological Process
```{r}
t_sup_p_bp <- 
  txome_go %>% 
  filter(disease_stage=="peak" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_peak_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_p_bp <- 
  gnome_go %>% 
  filter(disease_stage=="peak" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_peak_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_p_bp <- 
  merge(t_sup_p_bp, g_sup_p_bp, all=TRUE) %>% 
  mutate(go="BP")
```


##### Cellular Component
```{r}
t_sup_p_cc <- 
  txome_go %>% 
  filter(disease_stage=="peak" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_cc_peak_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_p_cc <- 
  gnome_go %>% 
  filter(disease_stage=="peak" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_cc_peak_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_p_cc <- 
  merge(t_sup_p_cc, g_sup_p_cc, all=TRUE) %>% 
  mutate(go="CC")
```


##### Molecular Function
```{r}
t_sup_p_mf <- 
  txome_go %>% 
  filter(disease_stage=="peak" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_mf_peak_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_p_mf <- 
  gnome_go %>% 
  filter(disease_stage=="peak" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_mf_peak_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_p_mf <- 
  merge(t_sup_p_mf, g_sup_p_mf, all=TRUE) %>% 
  mutate(go="MF")
```

##### Save Tables
```{r}
sup_p <- 
  rbind(sup_p_bp, sup_p_cc, sup_p_mf)

sup_p %>% 
  kable()

write.csv(sup_p, "Output Files/supplemental_female_edgeR_go_peak.csv", row.names=FALSE)
```


#### Late GO

##### Biological Process
```{r}
t_sup_l_bp <- 
  txome_go %>% 
  filter(disease_stage=="late" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_bp_late_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_l_bp <- 
  gnome_go %>% 
  filter(disease_stage=="late" &
         go=="bp") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_bp_late_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_l_bp <- 
  merge(t_sup_l_bp, g_sup_l_bp, all=TRUE) %>% 
  mutate(go="BP")
```


##### Cellular Component
```{r}
t_sup_l_cc <- 
  txome_go %>% 
  filter(disease_stage=="late" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_cc_late_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_l_cc <- 
  gnome_go %>% 
  filter(disease_stage=="late" &
         go=="cc") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_cc_late_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_l_cc <- 
  merge(t_sup_l_cc, g_sup_l_cc, all=TRUE) %>% 
  mutate(go="CC")
```


##### Molecular Function
```{r}
t_sup_l_mf <- 
  txome_go %>% 
  filter(disease_stage=="late" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Transcriptome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_null_mf_late_10$value, "x", "")) %>% 
  select(GOTerm, Transcriptome, Transcriptome_null)


g_sup_l_mf <- 
  gnome_go %>% 
  filter(disease_stage=="late" &
         go=="mf") %>% 
  mutate(GOTerm = paste(description, " ", "(", geneSet, ")", sep=""), 
         enrichmentRatio = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         enrichmentRatio = ifelse(direction=="down", paste("-", enrichmentRatio, sep=""),
                                  paste("+", enrichmentRatio, sep="")),
         Genome = paste(enrichmentRatio, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_null_mf_late_10$value, "x", "")) %>% 
  select(GOTerm, Genome, Genome_null)

sup_l_mf <- 
  merge(t_sup_l_mf, g_sup_l_mf, all=TRUE) %>% 
  mutate(go="MF")
```

##### Save Tables
```{r}
sup_l <- 
  rbind(sup_l_bp, sup_l_cc, sup_l_mf)

sup_l %>% 
  kable()

write.csv(sup_l, "Output Files/supplemental_female_edgeR_go_late.csv", row.names=FALSE)
```


#### KEGG

##### Acute
```{r}
t_sup_kegg_a <- 
  txome_kegg %>% 
  filter(disease_stage=="acute") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Transcriptome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_kegg_acute$value, "x", ""),
         Transcriptome_Genes = userId) %>% 
  select(KEGG, Transcriptome, Transcriptome_null, Transcriptome_Genes)

g_sup_kegg_a <- 
  gnome_kegg %>% 
  filter(disease_stage=="acute") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Genome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_kegg_acute$value, "x", ""),
         Genome_Genes = userId) %>% 
  select(KEGG, Genome, Genome_null, Genome_Genes)

sup_kegg_a <- 
  merge(t_sup_kegg_a, g_sup_kegg_a, all=TRUE)

sup_kegg_a %>% 
  kable()

write.csv(sup_kegg_a, "Output Files/supplemental_female_edgeR_kegg_acute.csv", row.names=FALSE)
```

##### Peak
```{r}
t_sup_kegg_p <- 
  txome_kegg %>% 
  filter(disease_stage=="peak") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Transcriptome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_kegg_peak$value, "x", ""),
         Transcriptome_Genes = userId) %>% 
  select(KEGG, Transcriptome, Transcriptome_null, Transcriptome_Genes)

g_sup_kegg_p <- 
  gnome_kegg %>% 
  filter(disease_stage=="peak") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Genome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_kegg_peak$value, "x", ""),
         Genome_Genes = userId) %>% 
  select(KEGG, Genome, Genome_null, Genome_Genes)

sup_kegg_p <- 
  merge(t_sup_kegg_p, g_sup_kegg_p, all=TRUE)

sup_kegg_p %>% 
  kable()

write.csv(sup_kegg_p, "Output Files/supplemental_female_edgeR_kegg_peak.csv", row.names=FALSE)
```

##### Late
```{r}
t_sup_kegg_l <- 
  txome_kegg %>% 
  filter(disease_stage=="late") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Transcriptome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Transcriptome_null = ifelse(description %in% t_kegg_late$value, "x", ""),
         Transcriptome_Genes = userId) %>% 
  select(KEGG, Transcriptome, Transcriptome_null, Transcriptome_Genes)

g_sup_kegg_l <- 
  gnome_kegg %>% 
  filter(disease_stage=="late") %>% 
  mutate(KEGG = paste(description, " ", "(", geneSet, ")", sep=""), 
         Enrichment = round(enrichmentRatio, digits=1),
         FDR = round(FDR, digits=3),
         Genome = paste(Enrichment, " ", "(", FDR, ")", sep=""), 
         Genome_null = ifelse(description %in% g_kegg_late$value, "x", ""),
         Genome_Genes = userId) %>% 
  select(KEGG, Genome, Genome_null, Genome_Genes)

sup_kegg_l <- 
  merge(t_sup_kegg_l, g_sup_kegg_l, all=TRUE)

sup_kegg_l %>% 
  kable()

write.csv(sup_kegg_l, "Output Files/supplemental_female_edgeR_kegg_late.csv", row.names=FALSE)
```

<!--chapter:end:14-TG_F_edgeR_nullcomp.Rmd-->

---
title: "Untitled"
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Supplemental Null Set Exploratory Statistics

## Libaries
```{r}
library(tidyverse)
library(ggh4x)
```

## Number of DEGs
### Data
```{r}
t_virus_genes <- 
  read.csv("Output Files/txome_female_edgeR_viralRNA_null_DEG.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("genes" = ".")

t_acute_genes <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute") %>% 
  dplyr::rename("genes" = ".")

t_peak_genes <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak") %>% 
  dplyr::rename("genes" = ".")

t_late_genes <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Late") %>% 
  dplyr::rename("genes" = ".")

g_virus_genes <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("genes" = ".")

g_acute_genes <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute") %>% 
  dplyr::rename("genes" = ".")

g_peak_genes <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak") %>% 
  dplyr::rename("genes" = ".")

g_late_genes <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late") %>% 
  dplyr::rename("genes" = ".")


genes <- 
  rbind(t_virus_genes, g_virus_genes,
        t_acute_genes, t_peak_genes, t_late_genes,
        g_acute_genes, g_peak_genes, g_late_genes) %>% 
  mutate(approach=factor(approach, levels=c("Transcriptome", "Genome")),
         stage=factor(stage, levels=c("V+ vs V-", "Acute", "Peak", "Late")))
```

### Quick Stats
```{r}
range(t_virus_genes$genes)
range(g_virus_genes$genes)

range(t_acute_genes$genes)
range(t_peak_genes$genes)
range(t_late_genes$genes)

range(g_acute_genes$genes)
range(g_peak_genes$genes)
range(g_late_genes$genes)

range(t_acute_genefreq$Freq)
range(t_peak_genefreq$Freq)
range(t_late_genefreq$Freq)

range(g_acute_genefreq$Freq)
range(g_peak_genefreq$Freq)
range(g_late_genefreq$Freq)
```


### Plot
```{r}
null_genes_plot <-
  ggplot(genes, aes(x=stage, y=log(genes))) +
  geom_boxplot() +
  facet_grid(~approach) +
  labs(x= NULL, y="log(# DEGs) per Trial") +
  theme_bw() +
  theme(panel.grid=element_blank())
```

## DEG Frequency
### Data - Import & Configure
```{r}
t_virus_genefreq <- 
  read.csv("Output Files/txome_female_edgeR_viralRNA_null_DEG.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("genes" = ".")

t_acute_genefreq <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute") %>% 
  dplyr::rename("genes" = ".")

t_peak_genefreq <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak") %>% 
  dplyr::rename("genes" = ".")

t_late_genefreq <- 
  read.csv("Output Files/txome_female_edgeR_null_DEG_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>%  
  mutate(approach="Transcriptome", 
         stage="Late") %>% 
  dplyr::rename("genes" = ".")


g_virus_genefreq <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("genes" = ".")

g_acute_genefreq <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute") %>% 
  dplyr::rename("genes" = ".")

g_peak_genefreq <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak") %>% 
  dplyr::rename("genes" = ".")

g_late_genefreq <- 
  read.csv("Output Files/gnome_female_edgeR_null_DEG_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  unlist() %>% 
  table() %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late") %>% 
  dplyr::rename("genes" = ".")

genefreq <- 
  rbind(t_virus_genefreq, g_virus_genefreq, 
        t_acute_genefreq, t_peak_genefreq, t_late_genefreq,
        g_acute_genefreq, g_peak_genefreq, g_late_genefreq) %>% 
  mutate(approach=factor(approach, levels=c("Transcriptome", "Genome")),
         stage=factor(stage, levels=c("V+ vs V-", "Acute", "Peak", "Late")))
```

### Plot
```{r}
null_genefreq_plot <-
  ggplot(genefreq, aes(x=stage, y=log(Freq))) +
  geom_boxplot() +
  facet_grid(~approach) +
  labs(x = NULL, y="log(Gene Frequency)") +
  theme_bw() +
  theme(panel.grid=element_blank())
```

### Combine gene number & frequency plots
```{r}
gene_numfreq <-
  grid.arrange(null_genes_plot, null_genefreq_plot,
               ncol=1, bottom="Influenza Stage")

ggsave("Figures/supplemental_TG_F_null_genenum-freq.jpg", gene_numfreq, 
       width=6, height=6, units="in")
```


## Number of GO terms
### Data - Import & Configure
```{r}
t_virus_bp <-
   read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

t_virus_cc <-
   read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

t_virus_mf <-
   read.csv("Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

t_acute_bp <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_bp_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

t_acute_cc <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_cc_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

t_acute_mf <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_mf_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

t_peak_bp <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_bp_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

t_peak_cc <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_cc_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

t_peak_mf <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_mf_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

t_late_bp <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_bp_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Late",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

t_late_cc <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_cc_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Late",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

t_late_mf <- 
  read.csv("Output Files/txome_female_edgeR_null_GO_mf_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Late",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")


g_virus_bp <-
   read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

g_virus_cc <-
   read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

g_virus_mf <-
   read.csv("Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

g_acute_bp <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_bp_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

g_acute_cc <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_cc_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

g_acute_mf <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_mf_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

g_peak_bp <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_bp_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

g_peak_cc <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_cc_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

g_peak_mf <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_mf_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")

g_late_bp <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_bp_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late",
         go="Biological Process") %>% 
  dplyr::rename("terms" = ".")

g_late_cc <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_cc_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late",
         go="Cellular Component") %>% 
  dplyr::rename("terms" = ".")

g_late_mf <- 
  read.csv("Output Files/gnome_female_edgeR_null_GO_mf_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late",
         go="Molecular Function") %>% 
  dplyr::rename("terms" = ".")


goterms <- rbind(t_virus_bp, t_virus_cc, t_virus_mf,
                 t_acute_bp, t_acute_cc, t_acute_mf,
                 t_peak_bp, t_peak_cc, t_peak_mf, 
                 t_late_bp, t_late_cc, t_late_mf,
                 g_virus_bp, g_virus_cc, g_virus_mf,
                 g_acute_bp, g_acute_cc, g_acute_mf,
                 g_peak_bp, g_peak_cc, g_peak_mf, 
                 g_late_bp, g_late_cc, g_late_mf) %>% 
  mutate(approach=factor(approach, levels=c("Transcriptome", "Genome")),
         stage=factor(stage, levels=c("V+ vs V-", "Acute", "Peak", "Late")))
```

### Plot
```{r}
null_go_plot <-
  ggplot(goterms, aes(x=stage, y=log(terms))) +
  geom_boxplot() +
  facet_nested_wrap(vars(approach, go), dir="h", 
                    strip.position="top", ncol=3, drop=TRUE) +
  labs(x="Influenza Stage", y="log(# GO Terms) per Trial") +
  theme_bw() +
  theme(panel.grid=element_blank())

ggsave("Figures/supplemental_TG_F_null_#gotermsplot.jpg", null_go_plot, 
       width=5, height=7, units="in")
```


## Number of KEGG Pathways
### Data
```{r}
t_virus_kegg <-
   read.csv("Output Files/txome_female_edgeR_viralRNA_null_kegg.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("terms" = ".")

t_acute_kegg <- 
  read.csv("Output Files/txome_female_edgeR_null_kegg_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Acute") %>% 
  dplyr::rename("terms" = ".")

t_peak_kegg <- 
  read.csv("Output Files/txome_female_edgeR_null_kegg_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Peak") %>% 
  dplyr::rename("terms" = ".")

t_late_kegg <- 
  read.csv("Output Files/txome_female_edgeR_null_kegg_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Transcriptome", 
         stage="Late") %>% 
  dplyr::rename("terms" = ".")


g_virus_kegg <-
   read.csv("Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="V+ vs V-") %>% 
  dplyr::rename("terms" = ".")

g_acute_kegg <- 
  read.csv("Output Files/gnome_female_edgeR_null_kegg_acute.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Acute") %>% 
  dplyr::rename("terms" = ".")

g_peak_kegg <- 
  read.csv("Output Files/gnome_female_edgeR_null_kegg_peak.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Peak") %>% 
  dplyr::rename("terms" = ".")

g_late_kegg <- 
  read.csv("Output Files/gnome_female_edgeR_null_kegg_late.csv") %>% 
  as.list() %>% 
  lapply(., function(x) x[!is.na(x)]) %>% 
  lengths(., use.names=TRUE) %>% 
  data.frame() %>% 
  mutate(approach="Genome", 
         stage="Late") %>% 
  dplyr::rename("terms" = ".")


kegg <- rbind(t_virus_kegg,
              t_acute_kegg, 
              t_peak_kegg, 
              t_late_kegg,
              g_virus_kegg,
              g_acute_kegg, 
              g_peak_kegg,
              g_late_kegg) %>% 
  mutate(approach=factor(approach, levels=c("Transcriptome", "Genome")),
         stage=factor(stage, levels=c( "V+ vs V-", "Acute", "Peak", "Late")))
```

### Plot
```{r}
null_kegg_plot <-
  ggplot(kegg, aes(x=stage, y=log(terms))) +
  geom_boxplot() +
  facet_grid(~approach) +
  labs(x="Influenza Stage", y="log(# KEGG Pathways) per Trial") +
  theme_bw() +
  theme(panel.grid=element_blank())

ggsave("Figures/supplemental_TG_F_null_#keggplot.jpg", null_kegg_plot, 
       width=6, height=3, units="in")
```

<!--chapter:end:15-TG_F_edgeR_APL_viralRNA_nullstats.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Influenza DGE DESeq2 

## Transcriptome - DESeq2 - Females 

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(DESeq2)
library(ggpubr)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male") %>% 
  mutate(RNA = ifelse(disease_stage == "acute", "viruspos", 
                        ifelse(disease_stage == "peak", "viruspos", "virusneg")))

Raw <- read.csv(file = "Output Files/txome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)

SampleInfo <- 
  SampleInfo %>% 
  column_to_rownames(var="sample")
```

### Viral RNA

#### Create DESeq Dataset (dds)
```{r}
count_matrix <- as.matrix(Raw)

dds_virus <- DESeqDataSetFromMatrix(countData = round(count_matrix), 
                              colData = SampleInfo, 
                              design = ~ RNA)
dds_virus
```

#### Filter to remove lowly expressed genes (<10 counts)
```{r}
keep <- rowSums(counts(dds_virus)) >= 10
dds_virus <- dds_virus[keep,]
```

#### Set reference level to virus neg group - which group to compare against
```{r}
dds_virus$RNA <- factor(dds_virus$RNA, levels = c("virusneg","viruspos"))
dds_virus$RNA <- relevel(dds_virus$RNA, ref = "virusneg")
```

#### Test for differentially expressed genes
cooksCutoff = results function automatically flags genes that contain a Cook's distance above a cutoff for samples which have 3 or more replicates
```{r}
dds_virus <- DESeq(dds_virus)

results_virus <- 
  results(dds_virus, contrast=c("RNA", "viruspos", "virusneg"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")
```

#### Save results
```{r}
write.csv(results_virus, "Output Files/txome_female_deseq_viralRNA_DGE_results.csv", row.names=FALSE)
```

#### Select DE genes with absolute logFC>1 and p-value<0.05
```{r}
de_virus <- 
  results_virus %>% 
  filter(pvalue<0.05 & abs(log2FoldChange)>1) 
```

#### Save results
```{r}
write.csv(de_virus, "Output Files/txome_female_deseq_viralRNA_DEG.csv", row.names=FALSE)
```



### Influenza Stage

#### Create DESeq Dataset (dds)
```{r}
count_matrix <- as.matrix(Raw)

dds <- DESeqDataSetFromMatrix(countData = round(count_matrix), 
                              colData = SampleInfo, 
                              design = ~ disease_stage)
dds
```

#### Filter to remove lowly expressed genes (<10 counts)
```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

#### Set reference level to control group - which group to compare against
```{r}
dds$disease_stage <- factor(dds$disease_stage, levels = c("control","acute","peak","late"))
dds$disease_stage <- relevel(dds$disease_stage, ref = "control")
```

#### Test for differentially expressed genes
cooksCutoff = results function automatically flags genes that contain a Cook's distance above a cutoff for samples which have 3 or more replicates
```{r}
dds <- DESeq(dds)

results_acute <- 
  results(dds, contrast=c("disease_stage", "acute", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")

results_peak <- 
  results(dds, contrast=c("disease_stage", "peak", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")

results_late <- 
  results(dds, contrast=c("disease_stage", "late", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")
```

#### Save results
```{r}
write.csv(results_acute, "Output Files/txome_female_deseq_DGE_results_acute.csv", row.names=FALSE)
write.csv(results_peak, "Output Files/txome_female_deseq_DGE_results_peak.csv", row.names=FALSE)
write.csv(results_late, "Output Files/txome_female_deseq_DGE_results_late.csv", row.names=FALSE)
```

#### Select DE genes with absolute logFC>1 and p-value<0.05
```{r}
de_acute <- 
  results_acute %>% 
  filter(pvalue<0.05 & abs(log2FoldChange)>1) 

de_peak <-
  results_peak %>% 
  filter(pvalue<0.05, abs(log2FoldChange)>1)

de_late <-
  results_late %>% 
  filter(pvalue<0.05, abs(log2FoldChange)>1) 
```

#### Save results
```{r}
write.csv(de_acute, "Output Files/txome_female_deseq_DEG_acute.csv", row.names=FALSE)
write.csv(de_peak, "Output Files/txome_female_deseq_DEG_peak.csv", row.names=FALSE)
write.csv(de_late, "Output Files/txome_female_deseq_DEG_late.csv", row.names=FALSE)
```

<!--chapter:end:16-T_F_DESeq2.Rmd-->

---
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

## Genome - DESeq2 - Females

### Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(DESeq2)
library(ggpubr)
```

### Data
Unknown = female
```{r}
SampleInfo <- 
  read.csv("Input Files/metadata.csv", 
           stringsAsFactors = FALSE) %>% 
  mutate(year=as.factor(year)) %>% 
  filter(!sex=="male") %>% 
  mutate(RNA = ifelse(disease_stage == "acute", "viruspos", 
                        ifelse(disease_stage == "peak", "viruspos", "virusneg")))

Raw <- read.csv(file = "Output Files/gnome_genecounts_locid_nohemo.csv", 
                stringsAsFactors = FALSE,
                header = TRUE, 
                row.names = 1) %>% 
  select(SampleInfo$sample)

SampleInfo <- 
  SampleInfo %>% 
  column_to_rownames(var="sample")
```

### Viral RNA

#### Create DESeq Dataset (dds)
```{r}
count_matrix <- as.matrix(Raw)

dds_virus <- DESeqDataSetFromMatrix(countData = round(count_matrix), 
                              colData = SampleInfo, 
                              design = ~ RNA)
dds_virus
```

#### Filter to remove lowly expressed genes (<10 counts)
```{r}
keep <- rowSums(counts(dds_virus)) >= 10
dds_virus <- dds_virus[keep,]
```

#### Set reference level to virus neg group - which group to compare against
```{r}
dds_virus$RNA <- factor(dds_virus$RNA, levels = c("virusneg","viruspos"))
dds_virus$RNA <- relevel(dds_virus$RNA, ref = "virusneg")
```

#### Test for differentially expressed genes
cooksCutoff = results function automatically flags genes that contain a Cook's distance above a cutoff for samples which have 3 or more replicates
```{r}
dds_virus <- DESeq(dds_virus)

results_virus <- 
  results(dds_virus, contrast=c("RNA", "viruspos", "virusneg"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")
```

#### Save results
```{r}
write.csv(results_virus, "Output Files/gnome_female_deseq_viralRNA_DGE_results.csv", row.names=FALSE)
```

#### Select DE genes with absolute logFC>1 and p-value<0.05
```{r}
de_virus <- 
  results_virus %>% 
  filter(pvalue<0.05 & abs(log2FoldChange)>1) 
```

#### Save results
```{r}
write.csv(de_virus, "Output Files/gnome_female_deseq_viralRNA_DEG.csv", row.names=FALSE)
```



### Influenza Stage

#### Create DESeq Dataset (dds)
```{r}
count_matrix <- as.matrix(Raw)

dds <- DESeqDataSetFromMatrix(countData = round(count_matrix), 
                              colData = SampleInfo, 
                              design = ~ disease_stage)
dds
```

#### Filter to remove lowly expressed genes (<10 counts)
```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

#### Set reference level to control group - which group to compare against
```{r}
dds$disease_stage <- factor(dds$disease_stage, levels = c("control","acute","peak","late"))
dds$disease_stage <- relevel(dds$disease_stage, ref = "control")
```

#### Test for differentially expressed genes
cooksCutoff = results function automatically flags genes that contain a Cook's distance above a cutoff for samples which have 3 or more replicates
```{r}
dds <- DESeq(dds)

results_acute <- 
  results(dds, contrast=c("disease_stage", "acute", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")

results_peak <- 
  results(dds, contrast=c("disease_stage", "peak", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")

results_late <- 
  results(dds, contrast=c("disease_stage", "late", "control"), 
          cooksCutoff=FALSE) %>% 
  as.data.frame() %>% 
  rownames_to_column(var="gene")
```

#### Save results
```{r}
write.csv(results_acute, "Output Files/gnome_female_deseq_DGE_results_acute.csv", row.names=FALSE)
write.csv(results_peak, "Output Files/gnome_female_deseq_DGE_results_peak.csv", row.names=FALSE)
write.csv(results_late, "Output Files/gnome_female_deseq_DGE_results_late.csv", row.names=FALSE)
```

#### Select DE genes with absolute logFC>1 and p-value<0.05
```{r}
de_acute <- 
  results_acute %>% 
  filter(pvalue<0.05 & abs(log2FoldChange)>1) 

de_peak <-
  results_peak %>% 
  filter(pvalue<0.05, abs(log2FoldChange)>1)

de_late <-
  results_late %>% 
  filter(pvalue<0.05, abs(log2FoldChange)>1) 
```

#### Save results
```{r}
write.csv(de_acute, "Output Files/gnome_female_deseq_DEG_acute.csv", row.names=FALSE)
write.csv(de_peak, "Output Files/gnome_female_deseq_DEG_peak.csv", row.names=FALSE)
write.csv(de_late, "Output Files/gnome_female_deseq_DEG_late.csv", row.names=FALSE)
```

<!--chapter:end:17-G_F_DESeq2.Rmd-->

---
title: ''
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# R Package Comparison - edgeR vs DESeq2

## Libraries
```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(ggpubr)
library(VennDiagram)
library(ggfortify)
library(gridExtra)
```

## Data
### Metadata
```{r}
metadata <- read.csv("Input Files/metadata.csv") %>% 
  filter(!sex=="male") %>% 
  mutate(disease_stage=factor(disease_stage,
                              levels=c("control","acute","peak","late")),
         year=factor(year),
         RNA = ifelse(disease_stage == "acute", "virus+", 
                        ifelse(disease_stage == "peak", "virus+", "virus-"))) 
```

### Results from DGE testing
```{r}
txome_edge_virus <- 
  read.csv("Output Files/txome_female_edgeR_viralRNA_DGE_results.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .))

txome_edge_acute <- 
  read.csv("Output Files/txome_female_edgeR_DGE_results_acute.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .))
txome_edge_peak <-
  read.csv("Output Files/txome_female_edgeR_DGE_results_peak.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .))
txome_edge_late <- 
  read.csv("Output Files/txome_female_edgeR_DGE_results_late.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .)) 

txome_deseq_virus <- 
  read.csv("Output Files/txome_female_deseq_viralRNA_DGE_results.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
txome_deseq_acute <- 
  read.csv("Output Files/txome_female_deseq_DGE_results_acute.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
txome_deseq_peak <-
  read.csv("Output Files/txome_female_deseq_DGE_results_peak.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>%  
  rename_with(~paste0("deseq_", .))
txome_deseq_late <- 
  read.csv("Output Files/txome_female_deseq_DGE_results_late.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))

gnome_edge_virus <- 
  read.csv("Output Files/gnome_female_edgeR_viralRNA_DGE_results.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .))
gnome_edge_acute <- 
  read.csv("Output Files/gnome_female_edgeR_DGE_results_acute.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .)) 
gnome_edge_peak <-
  read.csv("Output Files/gnome_female_edgeR_DGE_results_peak.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .)) 
gnome_edge_late <- 
  read.csv("Output Files/gnome_female_edgeR_DGE_results_late.csv") %>% 
  dplyr::rename("gene" = "X") %>% 
  rename_with(~paste0("edge_", .)) 

gnome_deseq_virus <- 
  read.csv("Output Files/gnome_female_deseq_viralRNA_DGE_results.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_acute <- 
  read.csv("Output Files/gnome_female_deseq_DGE_results_acute.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_peak <-
  read.csv("Output Files/gnome_female_deseq_DGE_results_peak.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_late <- 
  read.csv("Output Files/gnome_female_deseq_DGE_results_late.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>%  
  rename_with(~paste0("deseq_", .))
```

### Differentially expressed genes only
```{r}
txome_edge_DEG_virus <-
  read.csv("Output Files/txome_female_edgeR_viralRNA_DEG.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .))
txome_edge_DEG_acute <- 
  read.csv("Output Files/txome_female_edgeR_DEG_acute.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .))
txome_edge_DEG_peak <-
  read.csv("Output Files/txome_female_edgeR_DEG_peak.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .))
txome_edge_DEG_late <- 
  read.csv("Output Files/txome_female_edgeR_DEG_late.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .)) 

txome_deseq_DEG_virus <-
  read.csv("Output Files/txome_female_deseq_viralRNA_DEG.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
txome_deseq_DEG_acute <- 
  read.csv("Output Files/txome_female_deseq_DEG_acute.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
txome_deseq_DEG_peak <-
  read.csv("Output Files/txome_female_deseq_DEG_peak.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>%  
  rename_with(~paste0("deseq_", .))
txome_deseq_DEG_late <- 
  read.csv("Output Files/txome_female_deseq_DEG_late.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))

gnome_edge_DEG_virus <-
  read.csv("Output Files/gnome_female_edgeR_viralRNA_DEG.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .))
gnome_edge_DEG_acute <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_acute.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .)) 
gnome_edge_DEG_peak <-
  read.csv("Output Files/gnome_female_edgeR_DEG_peak.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .)) 
gnome_edge_DEG_late <- 
  read.csv("Output Files/gnome_female_edgeR_DEG_late.csv", row.names=1) %>% 
  rename_with(~paste0("edge_", .)) 

gnome_deseq_DEG_virus <-
  read.csv("Output Files/gnome_female_deseq_viralRNA_DEG.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_DEG_acute <- 
  read.csv("Output Files/gnome_female_deseq_DEG_acute.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_DEG_peak <-
  read.csv("Output Files/gnome_female_deseq_DEG_peak.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>% 
  rename_with(~paste0("deseq_", .))
gnome_deseq_DEG_late <- 
  read.csv("Output Files/gnome_female_deseq_DEG_late.csv") %>% 
  mutate(logbaseMean=log(baseMean)) %>%  
  rename_with(~paste0("deseq_", .))
```

### Pre-filter lists
```{r}
txome_raw <- read.table("Input Files/txome_genecounts.txt")
gnome_raw <- read.table("Input Files/gnome_genecounts.txt")
```

### Normalized gene counts
```{r}
txome_norm <- 
  read.csv("Output Files/txome_female_edgeR_normcounts.csv", row.names=1)
txome_norm_virus <-
  read.csv("Output Files/txome_female_edgeR_viralRNA_normcounts.csv", row.names=1)

gnome_norm <- 
  read.csv("Output Files/gnome_female_edgeR_normcounts.csv", row.names=1)
gnome_norm_virus <-
  read.csv("Output Files/gnome_female_edgeR_viralRNA_normcounts.csv", row.names=1)
```


## edgeR vs DESeq2 

### Number of Pre vs Post Filter Genes
```{r}
nrow(txome_raw)

nrow(txome_edge_virus)
nrow(txome_raw)-nrow(txome_edge_virus)
(nrow(txome_raw)-nrow(txome_edge_virus))/nrow(txome_raw)

nrow(txome_edge_acute)
nrow(txome_raw)-nrow(txome_edge_acute)
(nrow(txome_raw)-nrow(txome_edge_acute))/nrow(txome_raw)

nrow(txome_deseq_virus)
nrow(txome_raw)-nrow(txome_deseq_virus)
(nrow(txome_raw)-nrow(txome_deseq_virus))/nrow(txome_raw)

nrow(txome_deseq_acute)
nrow(txome_raw)-nrow(txome_deseq_acute)
(nrow(txome_raw)-nrow(txome_deseq_acute))/nrow(txome_raw)

nrow(gnome_raw)

nrow(gnome_edge_virus)
nrow(gnome_raw)-nrow(gnome_edge_virus)
(nrow(gnome_raw)-nrow(gnome_edge_virus))/nrow(gnome_raw)

nrow(gnome_edge_acute)
nrow(gnome_raw)-nrow(gnome_edge_acute)
(nrow(gnome_raw)-nrow(gnome_edge_acute))/nrow(gnome_raw)

nrow(gnome_deseq_virus)
nrow(gnome_raw)-nrow(gnome_deseq_virus)
(nrow(gnome_raw)-nrow(gnome_deseq_virus))/nrow(gnome_raw)

nrow(gnome_deseq_acute)
nrow(gnome_raw)-nrow(gnome_deseq_acute)
(nrow(gnome_raw)-nrow(gnome_deseq_acute))/nrow(gnome_raw)
```


### Merge results
```{r}
txome_virus <-
  merge(txome_edge_virus, txome_deseq_virus,
        all.x=FALSE, by.x="edge_gene", by.y="deseq_gene")
txome_acute <- 
  merge(txome_edge_acute, txome_deseq_acute, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")
txome_peak <- 
  merge(txome_edge_peak, txome_deseq_peak, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")
txome_late <- 
  merge(txome_edge_late, txome_deseq_late, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")

gnome_virus <-
  merge(gnome_edge_virus, gnome_deseq_virus,
        all.x=FALSE, by.x="edge_gene", by.y="deseq_gene")
gnome_acute <- 
  merge(gnome_edge_acute, gnome_deseq_acute, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")
gnome_peak <- 
  merge(gnome_edge_peak, gnome_deseq_peak, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")
gnome_late <- 
  merge(gnome_edge_late, gnome_deseq_late, 
        all.x=FALSE, by.x ="edge_gene", by.y="deseq_gene")
```

### p-value
#### Transcriptome
```{r}
txome_pvalue_virus <-
  ggscatter(txome_virus, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Transcriptome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

txome_pvalue_acute <-
  ggscatter(txome_acute, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Transcriptome Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

txome_pvalue_peak <-
  ggscatter(txome_peak, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Transcriptome Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

txome_pvalue_late <-
  ggscatter(txome_late, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Transcriptome Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

txome_pvalue <- 
  ggarrange(txome_pvalue_virus, 
            txome_pvalue_acute, txome_pvalue_peak, txome_pvalue_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Genome
```{r}
gnome_pvalue_virus <-
  ggscatter(gnome_virus, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Genome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

gnome_pvalue_acute <-
  ggscatter(gnome_acute, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Genome Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

gnome_pvalue_peak <-
  ggscatter(gnome_peak, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Genome Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

gnome_pvalue_late <-
  ggscatter(gnome_late, x = "edge_PValue", y = "deseq_pvalue", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR p-value", ylab = "DESeq2 p-value", 
          main = "Genome Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

gnome_pvalue <- 
  ggarrange(gnome_pvalue_virus,
            gnome_pvalue_acute, gnome_pvalue_peak, gnome_pvalue_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r}
method_pvalue <-
  ggarrange(txome_pvalue, gnome_pvalue, ncol=1) 
ggsave("Figures/supplemental_TG_methodcomp_pvalue.jpeg", method_pvalue, 
       width=18, height=9, units="in")
```


### log(FC)
#### Transcriptome
```{r}
txome_fc_virus <-
  ggscatter(txome_virus, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Transcriptome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

txome_fc_acute <-
  ggscatter(txome_acute, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Transcriptome Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

txome_fc_peak <-
  ggscatter(txome_peak, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Transcriptome Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

txome_fc_late <-
  ggscatter(txome_late, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Transcriptome Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

txome_fc <- 
  ggarrange(txome_fc_virus,
            txome_fc_acute, txome_fc_peak, txome_fc_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Genome
```{r}
gnome_fc_virus <-
  ggscatter(gnome_virus, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Genome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

gnome_fc_acute <-
  ggscatter(gnome_acute, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Genome Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

gnome_fc_peak <-
  ggscatter(gnome_peak, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Genome Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

gnome_fc_late <-
  ggscatter(gnome_late, x = "edge_logFC", y = "deseq_log2FoldChange", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "edgeR log2(Fold Change)", ylab = "DESeq2 log2(Fold Change)", 
          main = "Genome Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

gnome_fc <- 
  ggarrange(gnome_fc_virus,
            gnome_fc_acute, gnome_fc_peak, gnome_fc_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r}
method_fc <-
  ggarrange(txome_fc, gnome_fc, ncol=1) 
ggsave("Figures/supplemental_TG_methodcomp_foldchange.jpeg", 
       method_fc, width=18, height=9, units="in")
```

### log mean expression
#### Transcriptome
```{r}
txome_virus_exp <-
  ggscatter(txome_virus, x = "edge_logCPM", y = "deseq_logbaseMean", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "edgeR log2(CPM)", ylab = "DESeq2 log2(baseMean)", 
          main = "Transcriptome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

txome_exp <-
    ggscatter(txome_acute, x = "edge_logCPM", y = "deseq_logbaseMean", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "edgeR log2(CPM)", ylab = "DESeq2 log2(baseMean)", 
          main = "Transcriptome Control vs Acute, Peak, Late") + 
  theme(plot.background = element_rect(color = "black"))
```

#### Genome
```{r}
gnome_virus_exp <-
    ggscatter(gnome_virus, x = "edge_logCPM", y = "deseq_logbaseMean", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "edgeR log2(CPM)", ylab = "DESeq2 log2(baseMean)", 
          main = "Genome Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

gnome_exp <-
    ggscatter(gnome_acute, x = "edge_logCPM", y = "deseq_logbaseMean", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "edgeR log2(CPM)", ylab = "DESeq2 log2(baseMean)", 
          main = "Genome Control vs Acute, Peak, Late") + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r}
method_exp <-
  ggarrange(txome_virus_exp, txome_exp, 
            gnome_virus_exp, gnome_exp, 
            ncol=2, nrow=2) 
ggsave("Figures/supplemental_TG_methodcomp_expression.jpeg",
       method_exp, width=10, height=10, units="in")
```


### DGE explortory stats
#### Pre vs Post Filter
```{r}
filter_stats <- 
  data.frame(
    approach=c("Transcriptome", 
               "Transcriptome", "Transcriptome", 
               "Transcriptome", "Transcriptome",
               "Genome", 
               "Genome", "Genome", 
               "Genome", "Genome"),
    method=c("Raw",
             "edgeR_Virus", "DESeq2_Virus", 
             "edgeR_CAPL", "DESeq2_CAPL", 
             "Raw", 
             "edgeR_Virus", "DESeq2_Virus", 
             "edgeR_CAPL", "DESeq2_CAPL"),
    genes=c(nrow(txome_raw), 
            nrow(txome_edge_virus), nrow(txome_deseq_virus), 
            nrow(txome_edge_acute), nrow(txome_deseq_acute),
            nrow(gnome_raw), 
            nrow(gnome_edge_virus), nrow(gnome_deseq_virus),
            nrow(gnome_edge_acute), nrow(gnome_deseq_acute))) %>% 
  mutate(method=factor(method, levels=c("Raw", 
                                        "edgeR_Virus", "DESeq2_Virus", 
                                        "edgeR_CAPL", "DESeq2_CAPL")),
         approach=factor(approach, levels=c("Transcriptome", "Genome")))

filter_stats_plot <-
  ggplot(filter_stats, aes(x=method, y=genes)) +
  geom_col() +
  facet_grid(~approach) +
  labs(x=NULL, y = "Number of Genes") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        axis.text.x = element_text(angle=45, hjust=0.9))

ggsave("Figures/TG_method_prepostfilter.jpg", plot=filter_stats_plot, 
       width=5, height=4, units="in")
```

#### Method comparison
```{r}
method_stats <-
  data.frame(
    approach=c("Transcriptome", "Transcriptome", "Transcriptome", "Transcriptome", 
               "Transcriptome", "Transcriptome", "Transcriptome", "Transcriptome", 
               "Genome", "Genome", "Genome", "Genome", 
               "Genome", "Genome", "Genome", "Genome"),
    method=c("edgeR", "edgeR", "edgeR", "edgeR",
             "DESeq2", "DESeq2", "DESeq2", "DESeq2",
             "edgeR", "edgeR", "edgeR", "edgeR", 
             "DESeq2", "DESeq2", "DESeq2", "DESeq2"),
    disease=c("V+ vs V-", "Acute", "Peak", "Late", 
              "V+ vs V-", "Acute", "Peak", "Late", 
              "V+ vs V-", "Acute", "Peak", "Late", 
              "V+ vs V-", "Acute", "Peak", "Late"),
    genes=c(nrow(txome_edge_DEG_virus),
            nrow(txome_edge_DEG_acute), nrow(txome_edge_DEG_peak), nrow(txome_edge_DEG_late),
            nrow(txome_deseq_DEG_virus),
            nrow(txome_deseq_DEG_acute), nrow(txome_deseq_DEG_peak), nrow(txome_deseq_DEG_late), 
            nrow(gnome_edge_DEG_virus),
            nrow(gnome_edge_DEG_acute), nrow(gnome_edge_DEG_peak), nrow(gnome_edge_DEG_late),
            nrow(gnome_deseq_DEG_virus),
            nrow(gnome_deseq_DEG_acute), nrow(gnome_deseq_DEG_peak), nrow(gnome_deseq_DEG_late))) %>%
  mutate(approach=factor(approach, levels=c("Transcriptome", "Genome")),
         method=factor(method, levels=c("edgeR", "DESeq2")), 
         disease=factor(disease, levels=c("V+ vs V-", "Acute", "Peak", "Late")))

method_stats %>% 
  pivot_wider(names_from=approach, values_from=genes)

method_stats_plot <-
  ggplot(method_stats, aes(x=method, y=genes, fill=disease)) +
  geom_col(position="dodge") +
  scale_fill_manual(values=c("darkslategray2", "darkslategray3", "darkslategray4", "darkslategray"),
                    name="Disease Stage") +
  facet_grid(~approach) +
  labs(x="R Package", y="Number of DEGs") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom")

ggsave("Figures/TG_method_degcomp.jpg", plot=method_stats_plot,
       width=5, height=4, units="in")
```

### Method Gene IDs
#### Virus
```{r, message=FALSE}
virus <-
  list(`T-edgeR` = txome_edge_DEG_virus$edge_gene,       
       `T-DESeq2` = txome_deseq_DEG_virus$deseq_gene,
       `G-edgeR` = gnome_edge_DEG_virus$edge_gene, 
       `G-DESeq2` = gnome_deseq_DEG_virus$deseq_gene)

venn.diagram(virus, 
             filename="Figures/TG_method_degID_virus.jpg",
             disable.logging=TRUE,
             main="Virus+ vs Virus-", main.pos=c(0.5,0.77),
             fill=c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3"),
             margin=0.3,
             height=6, 
             width=8,
             units="in")
```

#### Acute
```{r}
acute <- 
  list(`T-edgeR` = txome_edge_DEG_acute$edge_gene, 
       `T-DESeq2` = txome_deseq_DEG_acute$deseq_gene,
       `G-edgeR` = gnome_edge_DEG_acute$edge_gene, 
       `G-DESeq2` = gnome_deseq_DEG_acute$deseq_gene)

venn.diagram(acute, 
             filename="Figures/TG_method_degID_acute.jpg",
             disable.logging=TRUE,
             main="Control vs Acute", main.pos=c(0.5,0.77),
             fill=c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3"),
             margin=0.3,
             height=6, 
             width=8,
             units="in")
```

#### Peak
```{r}
peak <- 
  list(`T-edgeR` = txome_edge_DEG_peak$edge_gene, 
       `T-DESeq2` = txome_deseq_DEG_peak$deseq_gene,
       `G-edgeR` = gnome_edge_DEG_peak$edge_gene, 
       `G-DESeq2` = gnome_deseq_DEG_peak$deseq_gene)

venn.diagram(peak, 
             filename="Figures/TG_method_degID_peak.jpg",
             disable.logging=TRUE,
             main="Control vs Peak", main.pos=c(0.5,0.77),
             fill=c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3"),
             margin=0.3,
             height=6, 
             width=8,
             units="in")
```

#### Late
```{r}
late <- 
  list(`T-edgeR` = txome_edge_DEG_late$edge_gene, 
       `T-DESeq2` = txome_deseq_DEG_late$deseq_gene,
       `G-edgeR` = gnome_edge_DEG_late$edge_gene, 
       `G-DESeq2` = gnome_deseq_DEG_late$deseq_gene)

venn.diagram(late, 
             filename="Figures/TG_method_degID_late.jpg",
             disable.logging=TRUE,
             main="Control vs Late", main.pos=c(0.5,0.77),
             fill=c("#66c2a5", "#fc8d62", "#8da0cb", "#e78ac3"),
             margin=0.3,
             height=6, 
             width=8,
             units="in")
```


## Transcriptome vs Genome
### Merge results
```{r}
edge_virus <-
  merge(txome_edge_virus, gnome_edge_virus, all.x=FALSE, by="edge_gene")
edge_acute <-
  merge(txome_edge_acute, gnome_edge_acute, all.x=FALSE, by="edge_gene")
edge_peak <-
  merge(txome_edge_peak, gnome_edge_peak, all.x=FALSE, by="edge_gene")
edge_late <-
  merge(txome_edge_late, gnome_edge_late, all.x=FALSE, by="edge_gene")

deseq_virus <-
  merge(txome_deseq_virus, gnome_deseq_virus, all.x=FALSE, by="deseq_gene")
deseq_acute <- 
  merge(txome_deseq_acute, gnome_deseq_acute, all.x=FALSE, by="deseq_gene")
deseq_peak <- 
  merge(txome_deseq_peak, gnome_deseq_peak, all.x=FALSE, by="deseq_gene")
deseq_late <- 
  merge(txome_deseq_late, gnome_deseq_late, all.x=FALSE, by="deseq_gene")
```

### p-value
#### edgeR
```{r}
edge_pvalue_virus <-
  ggscatter(edge_virus, x = "edge_PValue.x", y = "edge_PValue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "edgeR Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

edge_pvalue_acute <-
  ggscatter(edge_acute, x = "edge_PValue.x", y = "edge_PValue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "edgeR Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

edge_pvalue_peak <-
  ggscatter(edge_peak, x = "edge_PValue.x", y = "edge_PValue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "edgeR Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

edge_pvalue_late <-
  ggscatter(edge_late, x = "edge_PValue.x", y = "edge_PValue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "edgeR Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

edge_pvalue <- 
  ggarrange(edge_pvalue_virus, 
            edge_pvalue_acute, edge_pvalue_peak, edge_pvalue_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### DESeq2
```{r}
deseq_pvalue_virus <-
  ggscatter(deseq_virus, x = "deseq_pvalue.x", y = "deseq_pvalue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "DESeq2 Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

deseq_pvalue_acute <-
  ggscatter(deseq_acute, x = "deseq_pvalue.x", y = "deseq_pvalue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "DESeq2 Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

deseq_pvalue_peak <-
  ggscatter(deseq_peak, x = "deseq_pvalue.x", y = "deseq_pvalue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "DESeq2 Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

deseq_pvalue_late <-
  ggscatter(deseq_late, x = "deseq_pvalue.x", y = "deseq_pvalue.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), cor.coef.coord = c(0, 1.05),
          xlab = "Transcriptome p-value", ylab = "Genome p-value", 
          main = "DESeq2 Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

deseq_pvalue <- 
  ggarrange(deseq_pvalue_virus,
            deseq_pvalue_acute, deseq_pvalue_peak, deseq_pvalue_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r}
approach_pvalue <-
  ggarrange(edge_pvalue, deseq_pvalue, ncol=1) 

ggsave("Figures/supplemental_TG_approachcomp_pvalue.jpeg", 
       approach_pvalue, width=18, height=9, units="in")
```


### log(FC)
#### edgeR
```{r}
edge_fc_virus <-
  ggscatter(edge_virus, x = "edge_logFC.x", y = "edge_logFC.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "edgeR Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

edge_fc_acute <-
  ggscatter(edge_acute, x = "edge_logFC.x", y = "edge_logFC.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "edgeR Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

edge_fc_peak <-
  ggscatter(edge_peak, x = "edge_logFC.x", y = "edge_logFC.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "edgeR Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

edge_fc_late <-
  ggscatter(edge_late, x = "edge_logFC.x", y = "edge_logFC.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), 
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "edgeR Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

edge_fc <- 
  ggarrange(edge_fc_virus,
            edge_fc_acute, edge_fc_peak, edge_fc_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### DESeq2
```{r}
deseq_fc_virus <-
  ggscatter(deseq_virus, x = "deseq_log2FoldChange.x", y = "deseq_log2FoldChange.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "DESeq2 Virus+ vs Virus-") + 
  theme(plot.background = element_rect(color = "black"))

deseq_fc_acute <-
  ggscatter(deseq_acute, x = "deseq_log2FoldChange.x", y = "deseq_log2FoldChange.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "DESeq2 Control vs Acute") + 
  theme(plot.background = element_rect(color = "black"))

deseq_fc_peak <-
  ggscatter(deseq_peak, x = "deseq_log2FoldChange.x", y = "deseq_log2FoldChange.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"), 
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "DESeq2 Control vs Peak") + 
  theme(plot.background = element_rect(color = "black"))

deseq_fc_late <-
  ggscatter(deseq_late, x = "deseq_log2FoldChange.x", y = "deseq_log2FoldChange.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome log2(Fold Change)", ylab = "Genome log2(Fold Change)", 
          main = "DESeq2 Control vs Late") + 
  theme(plot.background = element_rect(color = "black"))

deseq_fc <- 
  ggarrange(deseq_fc_virus,
            deseq_fc_acute, deseq_fc_peak, deseq_fc_late, 
            ncol=4) + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r}
approach_fc <-
  ggarrange(edge_fc, deseq_fc, ncol=1) 

ggsave("Figures/supplemental_TG_approachcomp_foldchange.jpeg", 
       approach_fc, width=18, height=9, units="in")
```

### log mean expression
#### edgeR
```{r}
edge_virus_exp <-
    ggscatter(edge_virus, x = "edge_logCPM.x", y = "edge_logCPM.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome", 
          ylab = "Genome", 
          main = " Virus+ vs Virus- edgeR log2(CPM)") + 
  theme(plot.background = element_rect(color = "black"))

edge_exp <-
    ggscatter(edge_acute, x = "edge_logCPM.x", y = "edge_logCPM.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome", 
          ylab = "Genome", 
          main = "C vs A,P,L edgeR log2(CPM)") + 
  theme(plot.background = element_rect(color = "black"))
```

#### DESeq2
```{r}
deseq_virus_exp <- 
      ggscatter(deseq_virus, x = "deseq_logbaseMean.x", y = "deseq_logbaseMean.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome", 
          ylab = "Genome", 
          main = " Virus+ vs Virus- DESeq2 log2(baseMean)") + 
  theme(plot.background = element_rect(color = "black"))

deseq_exp <- 
      ggscatter(deseq_acute, x = "deseq_logbaseMean.x", y = "deseq_logbaseMean.y", 
          add = "reg.line", add.params = list(color = "darkslategray4"), conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          cor.coeff.args = list(color="black"),
          xlab = "Transcriptome", 
          ylab = "Genome", 
          main = " C vs A,P,L DESeq2 log2(baseMean)") + 
  theme(plot.background = element_rect(color = "black"))
```

#### Combine
```{r, warning=FALSE}
approach_exp <-
  ggarrange(edge_virus_exp, edge_exp, 
            deseq_virus_exp, deseq_exp, 
            ncol=2, nrow=2) + 
  theme(plot.background = element_rect(color = "black"))
ggsave("Figures/supplemental_TG_approachcomp_expression.jpeg", 
       approach_exp, width=10, height=10, units="in")
```


### PCA plots
#### Viral RNA
##### PCA
```{r}
tPCA_virus <- prcomp(t(txome_norm_virus))
gPCA_virus <- prcomp(t(gnome_norm_virus))
```

##### Manuscript
```{r}
t_pca_disease_virus <- 
  autoplot(tPCA_virus, data=metadata, shape="RNA") +
  scale_shape_manual(values=c(1,16), 
                     labels=c("Virus-", "Virus+"),
                     name="IAV RNA") +
  labs(title="A. Transcriptome") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

rna_legend <- 
  get_legend(t_pca_disease_virus)

t_pca_disease_virus <-
  t_pca_disease_virus +
  theme(legend.position="none")

g_pca_disease_virus <- 
  autoplot(gPCA_virus, data=metadata, shape="RNA") +
  scale_shape_manual(values=c(1,16), 
                     labels=c("Virus-", "Virus+"),
                     name="IAV RNA") +
  labs(title="B. Genome") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

pca_disease_virus <- 
  grid.arrange(arrangeGrob(t_pca_disease_virus, g_pca_disease_virus, ncol=2),
             arrangeGrob(rna_legend, ncol=1),
             heights=c(5, 1))

ggsave("Figures/TG_pca_disease_virus.jpg", pca_disease_virus, width=8, height=4, units="in")
```

##### Supplemental
###### Transcriptome - Year
```{r}
a_virus <- 
  autoplot(tPCA_virus, data=metadata, shape="year") +
  scale_shape_manual(values=c(0, 8, 6, 16, 3), 
                     name=NULL) +
  labs(title="A. Year") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

year_legend <- 
  get_legend(a_virus)

a_virus <-
  a_virus +
  theme(legend.position="none")
```

###### Transcriptome - Molt Stage
```{r}
b_virus <- 
  autoplot(tPCA_virus, data=metadata, shape="molt") +
  scale_shape_manual(values=c(0, 16, 8), 
                     name=NULL) +
  labs(title="B. Molt Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

molt_legend <- 
  get_legend(b_virus)

b_virus <-
  b_virus +
  theme(legend.position="none")
```

###### Transcriptome - Location
```{r}
c_virus <- 
  autoplot(tPCA_virus, data=metadata, shape="location") +
  scale_shape_manual(values=c(16, 8), 
                     name=NULL,
                     labels=c("Monomoy", "Muskeget")) +
  labs(title="C. Location") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

location_legend <- 
  get_legend(c_virus)

c_virus <-
  c_virus +
  theme(legend.position="none")
```

###### Genome - Year
```{r}
d_virus <- 
  autoplot(gPCA_virus, data=metadata, shape="year") +
  scale_shape_manual(values=c(0, 8, 6, 16, 3), 
                     name="Year") +
  labs(title="D. Year") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

###### Genome - Molt Stage
```{r}
e_virus <- 
  autoplot(gPCA_virus, data=metadata, shape="molt") +
  scale_shape_manual(values=c(0, 16, 8), 
                     name="Molt Stage") +
  labs(title="E. Molt Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

###### Genome - Location
```{r}
f_virus <- 
  autoplot(gPCA_virus, data=metadata, shape="location") +
  scale_shape_manual(values=c(16, 8), 
                     name="Location") +
  labs(title="F. Location") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

###### Combine
```{r}
pca_grid <-
  grid.arrange(arrangeGrob(a_virus, b_virus, c_virus, ncol=3), 
               arrangeGrob(year_legend, molt_legend, location_legend, ncol=3), 
               arrangeGrob(d_virus, e_virus, f_virus, ncol=3), 
               nrow=3, heights=c(8,2,8))
ggsave("Figures/supplemental_TG_pca_grid_virus.jpg", 
       pca_grid, width=14, height=7, units="in")
```



#### Influenza Stages
##### PCA
```{r}
tPCA <- prcomp(t(txome_norm))
gPCA <- prcomp(t(gnome_norm))
```

##### Transcriptome - Influenza Stages
```{r}
a <- 
  autoplot(tPCA, data=metadata, shape="disease_stage") +
  scale_shape_manual(values=c(2,17,16,1), 
                     labels=c("C", "A", "P", "L"),
                     name=NULL) +
  labs(title="A. Influenza Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

disease_legend <- 
  get_legend(a)

a <-
  a +
  theme(legend.position="none")
```

##### Transcriptome - Year
```{r}
b <- 
  autoplot(tPCA, data=metadata, shape="year") +
  scale_shape_manual(values=c(0, 8, 6, 16, 3), 
                     name=NULL) +
  labs(title="B. Year") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

year_legend <- 
  get_legend(b)

b <-
  b +
  theme(legend.position="none")
```

##### Transcriptome - Molt Stage
```{r}
c <- 
  autoplot(tPCA, data=metadata, shape="molt") +
  scale_shape_manual(values=c(0, 16, 8), 
                     name=NULL) +
  labs(title="C. Molt Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

molt_legend <- 
  get_legend(c)

c <-
  c +
  theme(legend.position="none")
```

##### Transcriptome - Location
```{r}
d <- 
  autoplot(tPCA, data=metadata, shape="location") +
  scale_shape_manual(values=c(16, 8), 
                     name=NULL,
                     labels=c("Monomoy", "Muskeget")) +
  labs(title="D. Location") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(linewidth=0.5, linetype="solid", colour ="black"))

location_legend <- 
  get_legend(d)

d <-
  d +
  theme(legend.position="none")
```

##### Genome - Influenza Stages
```{r}
e <- 
  autoplot(gPCA, data=metadata, shape="disease_stage") +
  scale_shape_manual(values=c(2,17,16,1), 
                     labels=c("Control", "Acute", "Peak", "Late"),
                     name="Influenza Stage") +
  labs(title="E. Influenza Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

##### Genome - Year
```{r}
f <- 
  autoplot(gPCA, data=metadata, shape="year") +
  scale_shape_manual(values=c(0, 8, 6, 16, 3), 
                     name="Year") +
  labs(title="F. Year") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

##### Transcriptome - Molt Stage
```{r}
g <- 
  autoplot(gPCA, data=metadata, shape="molt") +
  scale_shape_manual(values=c(0, 16, 8), 
                     name="Molt Stage") +
  labs(title="G. Molt Stage") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

##### Transcriptome - Location
```{r}
h <- 
  autoplot(gPCA, data=metadata, shape="location") +
  scale_shape_manual(values=c(16, 8), 
                     name="Location") +
  labs(title="H. Location") +
  theme_bw() +
  theme(panel.grid=element_blank(),
        legend.position="none")
```

##### Combine
```{r}
pca_grid <-
  grid.arrange(arrangeGrob(a, b, c, d, ncol=4), 
               arrangeGrob(disease_legend, year_legend, molt_legend, location_legend, ncol=4), 
               arrangeGrob(e, f, g, h, ncol=4), 
               nrow=3, heights=c(8,2,8))
ggsave("Figures/supplemental_TG_pca_grid_influenzastage.jpg", 
       pca_grid, width=14, height=7, units="in")
```

<!--chapter:end:18-TG_F_methodcomp.Rmd-->

---
title: "Untitled"
author: "Christina McCosker"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# IAV Kegg Pathway Candidate Gene Analysis

## Libraries
```{r}
library(limma)
library(org.Hs.eg.db)
library(AnnotationDbi) 
library(tidyverse)
library(ggkegg)
library(tidygraph)
```

## Data
```{r}
t_postfilter <- read.table("Output Files/txome_female_edgeR_postfilter.txt")
  
t_acute <- read.csv("Output Files/txome_female_edgeR_DEG_acute.csv", row.names=1)
t_peak <- read.csv("Output Files/txome_female_edgeR_DEG_peak.csv", row.names=1)
t_late <- read.csv("Output Files/txome_female_edgeR_DEG_late.csv", row.names=1)

g_postfilter <- read.table("Output Files/gnome_female_edgeR_postfilter.txt")
g_acute <- read.csv("Output Files/gnome_female_edgeR_DEG_acute.csv", row.names=1)
g_peak <- read.csv("Output Files/gnome_female_edgeR_DEG_peak.csv", row.names=1)
g_late <- read.csv("Output Files/gnome_female_edgeR_DEG_late.csv", row.names=1)

t_virus_postfilter <- read.table("Output Files/txome_female_edgeR_viralRNA_postfilter.txt")
t_virus_degs <- read.csv("Output Files/txome_female_edgeR_viralRNA_DEG.csv", row.names=1)

g_virus_postfilter <- read.table("Output Files/gnome_female_edgeR_viralRNA_postfilter.txt")
g_virus_degs <- read.csv("Output Files/gnome_female_edgeR_viralRNA_DEG.csv", row.names=1)
```

## Extract genes from KEGG Pathway
```{r}
iav <- 
  getGeneKEGGLinks(species="hsa") %>% 
  filter(PathwayID == "hsa05164")

iav$Symbol <- mapIds(org.Hs.eg.db, iav$GeneID,
                     column="SYMBOL", keytype="ENTREZID")
```

## Overlap with Gene Lists
### Postfilter
```{r}
postfilter <- 
  rbind(t_postfilter, g_postfilter, t_virus_postfilter) %>% 
  filter(!duplicated(V1))

iav_postfilter <-
  intersect(iav$Symbol, postfilter$V1)
```

### Differentially Expressed Genes
```{r}
acute <- 
  data.frame(
    genes=c(intersect(iav$Symbol, t_acute$gene), 
            intersect(iav$Symbol, g_acute$gene)), 
    category="acute")

peak <- 
  data.frame(
    genes=c(intersect(iav$Symbol, t_peak$gene), 
            intersect(iav$Symbol, g_peak$gene)), 
    category="peak")

late <- 
  data.frame(
    genes=c(intersect(iav$Symbol, t_late$gene),
            intersect(iav$Symbol, g_late$gene)),
    category="late") %>% 
  filter(!duplicated(genes))

virus <- 
  data.frame(
    genes=c(intersect(iav$Symbol, t_virus_degs$gene),
            intersect(iav$Symbol, g_virus_degs$gene)),
    category="virus") %>% 
  filter(!duplicated(genes))
```

### Compile gene IDs
```{r}
deg <-
  rbind(acute, peak, late, virus) %>% 
  mutate(id = paste0("hsa:", ifelse(genes %in% iav$Symbol, 
                     iav[match(genes, iav$Symbol),1],
                     genes)))

expressed <-
  data.frame(genes=iav_postfilter) %>%  
  filter(!genes %in% deg$genes) %>% 
  mutate(id = paste0("hsa:", ifelse(genes %in% iav$Symbol, 
                     iav[match(genes, iav$Symbol),1],
                     genes)))
```

## Plot pathway map
```{r}
graph_info <-
  pathway("hsa05164") %>% 
  activate(nodes) %>% 
  mutate(convert_hsa=convert_id("hsa"), 
         convert_map=convert_id("pathway"),
         bgcolor=ifelse(convert_hsa %in% expressed$genes, "#ffffe0", 
                        ifelse(convert_hsa %in% deg$genes, "#fbc69d", "lightgray")))

graph <-
  ggraph(graph_info, x=x, y=y) +
   geom_node_rect(aes(filter=type=="gene", 
                      fill=I(bgcolor)),
                 color="black") + 
  overlay_raw_map() +
  theme_void() 
graph 


ggkeggsave(filename="Figures/iavkegg.jpg", graph, dpi=300)
```

### Create Legend
```{r}
png(filename="Figures/iavkegglegend.png", width = 10, height=5, units="in", res=300)
  plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', , xlim=c(0,100), ylim=c(0,10))
  legend("topleft", legend =c('Ex', 'DE', 'NE'), 
         pch=15, pt.cex=5, cex=2, bty='o',
       col = c("#ffffe0", "#fbc69d", "lightgray"), horiz=T)
dev.off()
```


<!--chapter:end:19-IAV-KEGG_CandidateGenes.Rmd-->

