<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 IAV Kegg Pathway Candidate Gene Analysis | Molecular mechanisms underlying response to influenza in grey seals (Halichoerus grypus), a potential wild reservoir</title>
  <meta name="description" content="9 IAV Kegg Pathway Candidate Gene Analysis | Molecular mechanisms underlying response to influenza in grey seals (Halichoerus grypus), a potential wild reservoir" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="9 IAV Kegg Pathway Candidate Gene Analysis | Molecular mechanisms underlying response to influenza in grey seals (Halichoerus grypus), a potential wild reservoir" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 IAV Kegg Pathway Candidate Gene Analysis | Molecular mechanisms underlying response to influenza in grey seals (Halichoerus grypus), a potential wild reservoir" />
  
  
  

<meta name="author" content="Christina McCosker" />


<meta name="date" content="2025-07-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-package-comparison---edger-vs-deseq2.html"/>
<link rel="next" href="outlier-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hg-RNA Stats</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Publication Information</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors"><i class="fa fa-check"></i><b>1.1</b> Authors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#abstract"><i class="fa fa-check"></i><b>1.2</b> Abstract</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html"><i class="fa fa-check"></i><b>2</b> LOC Gene Identification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#libraries"><i class="fa fa-check"></i><b>2.1</b> Libraries</a></li>
<li class="chapter" data-level="2.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#load-data"><i class="fa fa-check"></i><b>2.2</b> Load data</a></li>
<li class="chapter" data-level="2.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-gene-sequences"><i class="fa fa-check"></i><b>2.3</b> Extract gene sequences</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-loc-genes-from-gene-count-matrices"><i class="fa fa-check"></i><b>2.3.1</b> Extract LOC genes from gene count matrices</a></li>
<li class="chapter" data-level="2.3.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-ids-for-all-sequence-ids"><i class="fa fa-check"></i><b>2.3.2</b> Extract transcript IDs for all sequence IDs</a></li>
<li class="chapter" data-level="2.3.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-sequences-for-blast"><i class="fa fa-check"></i><b>2.3.3</b> Extract transcript sequences for blast</a></li>
<li class="chapter" data-level="2.3.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-sequences"><i class="fa fa-check"></i><b>2.3.4</b> Export sequences</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#blast-sequences"><i class="fa fa-check"></i><b>2.4</b> Blast sequences</a></li>
<li class="chapter" data-level="2.5" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#process-blastn-results"><i class="fa fa-check"></i><b>2.5</b> Process BLASTn Results</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#transcriptome"><i class="fa fa-check"></i><b>2.5.1</b> Transcriptome</a></li>
<li class="chapter" data-level="2.5.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#genome"><i class="fa fa-check"></i><b>2.5.2</b> Genome</a></li>
<li class="chapter" data-level="2.5.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-transcriptome-genome-blast-results"><i class="fa fa-check"></i><b>2.5.3</b> Combine Transcriptome &amp; Genome Blast Results</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-with-prior-loc-id-master-list"><i class="fa fa-check"></i><b>2.6</b> Combine with prior LOC ID master list</a></li>
<li class="chapter" data-level="2.7" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#update-tidy-gene-count-matrices"><i class="fa fa-check"></i><b>2.7</b> Update &amp; tidy gene count matrices</a></li>
<li class="chapter" data-level="2.8" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#hemoglobin"><i class="fa fa-check"></i><b>2.8</b> Hemoglobin</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.1</b> Extract hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#prop-of-total-gene-counts"><i class="fa fa-check"></i><b>2.8.2</b> Prop of total gene counts</a></li>
<li class="chapter" data-level="2.8.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#remove-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.3</b> Remove hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-new-gene-count-matrices"><i class="fa fa-check"></i><b>2.8.4</b> Export new gene count matrices</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html"><i class="fa fa-check"></i><b>3</b> Male v Female DGE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#transcriptome---edger"><i class="fa fa-check"></i><b>3.1</b> Transcriptome - edgeR</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-1"><i class="fa fa-check"></i><b>3.1.1</b> Libraries</a></li>
<li class="chapter" data-level="3.1.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data"><i class="fa fa-check"></i><b>3.1.2</b> Data</a></li>
<li class="chapter" data-level="3.1.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.1.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots"><i class="fa fa-check"></i><b>3.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.1.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs"><i class="fa fa-check"></i><b>3.1.5</b> Test for DEGs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#genome---edger"><i class="fa fa-check"></i><b>3.2</b> Genome - edgeR</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-2"><i class="fa fa-check"></i><b>3.2.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data-1"><i class="fa fa-check"></i><b>3.2.2</b> Data</a></li>
<li class="chapter" data-level="3.2.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>3.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.2.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots-1"><i class="fa fa-check"></i><b>3.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.2.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs-1"><i class="fa fa-check"></i><b>3.2.5</b> Test for DEGs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html"><i class="fa fa-check"></i><b>4</b> Virus+ vs Virus- DGE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome---edger---females"><i class="fa fa-check"></i><b>4.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-3"><i class="fa fa-check"></i><b>4.1.1</b> Libraries</a></li>
<li class="chapter" data-level="4.1.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-2"><i class="fa fa-check"></i><b>4.1.2</b> Data</a></li>
<li class="chapter" data-level="4.1.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>4.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.1.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-2"><i class="fa fa-check"></i><b>4.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.1.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-2"><i class="fa fa-check"></i><b>4.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.1.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis"><i class="fa fa-check"></i><b>4.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.1.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis"><i class="fa fa-check"></i><b>4.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome---edger---females"><i class="fa fa-check"></i><b>4.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-4"><i class="fa fa-check"></i><b>4.2.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-3"><i class="fa fa-check"></i><b>4.2.2</b> Data</a></li>
<li class="chapter" data-level="4.2.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>4.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.2.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-3"><i class="fa fa-check"></i><b>4.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.2.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-3"><i class="fa fa-check"></i><b>4.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.2.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-1"><i class="fa fa-check"></i><b>4.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.2.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-1"><i class="fa fa-check"></i><b>4.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-null-set---edger---females"><i class="fa fa-check"></i><b>4.3</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-5"><i class="fa fa-check"></i><b>4.3.1</b> Libraries</a></li>
<li class="chapter" data-level="4.3.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-4"><i class="fa fa-check"></i><b>4.3.2</b> Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-2"><i class="fa fa-check"></i><b>4.3.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.3.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-2"><i class="fa fa-check"></i><b>4.3.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-null-set---edger---females"><i class="fa fa-check"></i><b>4.4</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-6"><i class="fa fa-check"></i><b>4.4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-5"><i class="fa fa-check"></i><b>4.4.2</b> Data</a></li>
<li class="chapter" data-level="4.4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>4.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-3"><i class="fa fa-check"></i><b>4.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-3"><i class="fa fa-check"></i><b>4.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#refine-gokegg"><i class="fa fa-check"></i><b>4.5</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-7"><i class="fa fa-check"></i><b>4.5.1</b> Libraries</a></li>
<li class="chapter" data-level="4.5.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-6"><i class="fa fa-check"></i><b>4.5.2</b> Data</a></li>
<li class="chapter" data-level="4.5.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-1"><i class="fa fa-check"></i><b>4.5.3</b> Transcriptome</a></li>
<li class="chapter" data-level="4.5.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-1"><i class="fa fa-check"></i><b>4.5.4</b> Genome</a></li>
<li class="chapter" data-level="4.5.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-term-figure"><i class="fa fa-check"></i><b>4.5.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="4.5.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#immunne-go-term-genes"><i class="fa fa-check"></i><b>4.5.6</b> Immunne GO term Genes</a></li>
<li class="chapter" data-level="4.5.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-table---manuscript"><i class="fa fa-check"></i><b>4.5.7</b> KEGG Table - Manuscript</a></li>
<li class="chapter" data-level="4.5.8" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#supplemental-table"><i class="fa fa-check"></i><b>4.5.8</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html"><i class="fa fa-check"></i><b>5</b> Influenza Stage DGE</a>
<ul>
<li class="chapter" data-level="5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome---edger---females-1"><i class="fa fa-check"></i><b>5.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-8"><i class="fa fa-check"></i><b>5.1.1</b> Libraries</a></li>
<li class="chapter" data-level="5.1.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-7"><i class="fa fa-check"></i><b>5.1.2</b> Data</a></li>
<li class="chapter" data-level="5.1.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-4"><i class="fa fa-check"></i><b>5.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-4"><i class="fa fa-check"></i><b>5.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.1.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-4"><i class="fa fa-check"></i><b>5.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.1.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-4"><i class="fa fa-check"></i><b>5.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.1.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-4"><i class="fa fa-check"></i><b>5.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome---edger---females-1"><i class="fa fa-check"></i><b>5.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-9"><i class="fa fa-check"></i><b>5.2.1</b> Libraries</a></li>
<li class="chapter" data-level="5.2.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-8"><i class="fa fa-check"></i><b>5.2.2</b> Data</a></li>
<li class="chapter" data-level="5.2.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-5"><i class="fa fa-check"></i><b>5.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.2.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-5"><i class="fa fa-check"></i><b>5.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.2.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-5"><i class="fa fa-check"></i><b>5.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.2.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-5"><i class="fa fa-check"></i><b>5.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.2.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-5"><i class="fa fa-check"></i><b>5.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#comapre-t-g-approach---edger---females"><i class="fa fa-check"></i><b>5.3</b> Comapre T &amp; G Approach - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-10"><i class="fa fa-check"></i><b>5.3.1</b> Libraries</a></li>
<li class="chapter" data-level="5.3.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-9"><i class="fa fa-check"></i><b>5.3.2</b> Data</a></li>
<li class="chapter" data-level="5.3.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#number-of-up-down-reg-genes"><i class="fa fa-check"></i><b>5.3.3</b> Number of up &amp; down reg genes</a></li>
<li class="chapter" data-level="5.3.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#gene-identities"><i class="fa fa-check"></i><b>5.3.4</b> Gene identities</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.4</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-11"><i class="fa fa-check"></i><b>5.4.1</b> Libraries</a></li>
<li class="chapter" data-level="5.4.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-10"><i class="fa fa-check"></i><b>5.4.2</b> Data</a></li>
<li class="chapter" data-level="5.4.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-6"><i class="fa fa-check"></i><b>5.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-6"><i class="fa fa-check"></i><b>5.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.5</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-12"><i class="fa fa-check"></i><b>5.5.1</b> Libraries</a></li>
<li class="chapter" data-level="5.5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-11"><i class="fa fa-check"></i><b>5.5.2</b> Data</a></li>
<li class="chapter" data-level="5.5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>5.5.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-7"><i class="fa fa-check"></i><b>5.5.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-7"><i class="fa fa-check"></i><b>5.5.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#refine-gokegg-1"><i class="fa fa-check"></i><b>5.6</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-13"><i class="fa fa-check"></i><b>5.6.1</b> Libraries</a></li>
<li class="chapter" data-level="5.6.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-12"><i class="fa fa-check"></i><b>5.6.2</b> Data</a></li>
<li class="chapter" data-level="5.6.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-2"><i class="fa fa-check"></i><b>5.6.3</b> Transcriptome</a></li>
<li class="chapter" data-level="5.6.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-2"><i class="fa fa-check"></i><b>5.6.4</b> Genome</a></li>
<li class="chapter" data-level="5.6.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-term-figure-1"><i class="fa fa-check"></i><b>5.6.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="5.6.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#supplemental-table-1"><i class="fa fa-check"></i><b>5.6.6</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html"><i class="fa fa-check"></i><b>6</b> Supplemental Null Set Exploratory Statistics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#libaries"><i class="fa fa-check"></i><b>6.1</b> Libaries</a></li>
<li class="chapter" data-level="6.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-degs"><i class="fa fa-check"></i><b>6.2</b> Number of DEGs</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-13"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats"><i class="fa fa-check"></i><b>6.2.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.2.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-1"><i class="fa fa-check"></i><b>6.2.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#deg-frequency"><i class="fa fa-check"></i><b>6.3</b> DEG Frequency</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-14"><i class="fa fa-check"></i><b>6.3.1</b> Data</a></li>
<li class="chapter" data-level="6.3.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats-1"><i class="fa fa-check"></i><b>6.3.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.3.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-2"><i class="fa fa-check"></i><b>6.3.3</b> Plot</a></li>
<li class="chapter" data-level="6.3.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#combine-gene-number-frequency-plots"><i class="fa fa-check"></i><b>6.3.4</b> Combine gene number &amp; frequency plots</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-go-terms"><i class="fa fa-check"></i><b>6.4</b> Number of GO terms</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data---import-configure"><i class="fa fa-check"></i><b>6.4.1</b> Data - Import &amp; Configure</a></li>
<li class="chapter" data-level="6.4.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-3"><i class="fa fa-check"></i><b>6.4.2</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-kegg-pathways"><i class="fa fa-check"></i><b>6.5</b> Number of KEGG Pathways</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-15"><i class="fa fa-check"></i><b>6.5.1</b> Data</a></li>
<li class="chapter" data-level="6.5.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-4"><i class="fa fa-check"></i><b>6.5.2</b> Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html"><i class="fa fa-check"></i><b>7</b> Influenza DGE DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#transcriptome---deseq2---females"><i class="fa fa-check"></i><b>7.1</b> Transcriptome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-14"><i class="fa fa-check"></i><b>7.1.1</b> Libraries</a></li>
<li class="chapter" data-level="7.1.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-16"><i class="fa fa-check"></i><b>7.1.2</b> Data</a></li>
<li class="chapter" data-level="7.1.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna"><i class="fa fa-check"></i><b>7.1.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.1.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage"><i class="fa fa-check"></i><b>7.1.4</b> Influenza Stage</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#genome---deseq2---females"><i class="fa fa-check"></i><b>7.2</b> Genome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-15"><i class="fa fa-check"></i><b>7.2.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-17"><i class="fa fa-check"></i><b>7.2.2</b> Data</a></li>
<li class="chapter" data-level="7.2.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna-1"><i class="fa fa-check"></i><b>7.2.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.2.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage-1"><i class="fa fa-check"></i><b>7.2.4</b> Influenza Stage</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html"><i class="fa fa-check"></i><b>8</b> R Package Comparison - edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#libraries-16"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#data-18"><i class="fa fa-check"></i><b>8.2</b> Data</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#metadata"><i class="fa fa-check"></i><b>8.2.1</b> Metadata</a></li>
<li class="chapter" data-level="8.2.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#results-from-dge-testing"><i class="fa fa-check"></i><b>8.2.2</b> Results from DGE testing</a></li>
<li class="chapter" data-level="8.2.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#differentially-expressed-genes-only"><i class="fa fa-check"></i><b>8.2.3</b> Differentially expressed genes only</a></li>
<li class="chapter" data-level="8.2.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pre-filter-lists"><i class="fa fa-check"></i><b>8.2.4</b> Pre-filter lists</a></li>
<li class="chapter" data-level="8.2.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#normalized-gene-counts"><i class="fa fa-check"></i><b>8.2.5</b> Normalized gene counts</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#edger-vs-deseq2"><i class="fa fa-check"></i><b>8.3</b> edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#number-of-pre-vs-post-filter-genes"><i class="fa fa-check"></i><b>8.3.1</b> Number of Pre vs Post Filter Genes</a></li>
<li class="chapter" data-level="8.3.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results"><i class="fa fa-check"></i><b>8.3.2</b> Merge results</a></li>
<li class="chapter" data-level="8.3.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value"><i class="fa fa-check"></i><b>8.3.3</b> p-value</a></li>
<li class="chapter" data-level="8.3.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc"><i class="fa fa-check"></i><b>8.3.4</b> log(FC)</a></li>
<li class="chapter" data-level="8.3.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression"><i class="fa fa-check"></i><b>8.3.5</b> log mean expression</a></li>
<li class="chapter" data-level="8.3.6" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#dge-explortory-stats"><i class="fa fa-check"></i><b>8.3.6</b> DGE explortory stats</a></li>
<li class="chapter" data-level="8.3.7" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#method-gene-ids"><i class="fa fa-check"></i><b>8.3.7</b> Method Gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#transcriptome-vs-genome"><i class="fa fa-check"></i><b>8.4</b> Transcriptome vs Genome</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results-1"><i class="fa fa-check"></i><b>8.4.1</b> Merge results</a></li>
<li class="chapter" data-level="8.4.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value-1"><i class="fa fa-check"></i><b>8.4.2</b> p-value</a></li>
<li class="chapter" data-level="8.4.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc-1"><i class="fa fa-check"></i><b>8.4.3</b> log(FC)</a></li>
<li class="chapter" data-level="8.4.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression-1"><i class="fa fa-check"></i><b>8.4.4</b> log mean expression</a></li>
<li class="chapter" data-level="8.4.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pca-plots-6"><i class="fa fa-check"></i><b>8.4.5</b> PCA plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html"><i class="fa fa-check"></i><b>9</b> IAV Kegg Pathway Candidate Gene Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#libraries-17"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#data-19"><i class="fa fa-check"></i><b>9.2</b> Data</a></li>
<li class="chapter" data-level="9.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#extract-genes-from-kegg-pathway"><i class="fa fa-check"></i><b>9.3</b> Extract genes from KEGG Pathway</a></li>
<li class="chapter" data-level="9.4" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#overlap-with-gene-lists"><i class="fa fa-check"></i><b>9.4</b> Overlap with Gene Lists</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#postfilter"><i class="fa fa-check"></i><b>9.4.1</b> Postfilter</a></li>
<li class="chapter" data-level="9.4.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#differentially-expressed-genes"><i class="fa fa-check"></i><b>9.4.2</b> Differentially Expressed Genes</a></li>
<li class="chapter" data-level="9.4.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#compile-gene-ids"><i class="fa fa-check"></i><b>9.4.3</b> Compile gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#plot-pathway-map"><i class="fa fa-check"></i><b>9.5</b> Plot pathway map</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#create-legend"><i class="fa fa-check"></i><b>9.5.1</b> Create Legend</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="outlier-analysis.html"><a href="outlier-analysis.html"><i class="fa fa-check"></i><b>10</b> Outlier Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#libraries-18"><i class="fa fa-check"></i><b>10.1</b> Libraries</a></li>
<li class="chapter" data-level="10.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#data-20"><i class="fa fa-check"></i><b>10.2</b> Data</a></li>
<li class="chapter" data-level="10.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#pca-2"><i class="fa fa-check"></i><b>10.3</b> PCA</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#transcriptome-6"><i class="fa fa-check"></i><b>10.3.1</b> Transcriptome</a></li>
<li class="chapter" data-level="10.3.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#genome-6"><i class="fa fa-check"></i><b>10.3.2</b> Genome</a></li>
<li class="chapter" data-level="10.3.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#number-of-reads"><i class="fa fa-check"></i><b>10.3.3</b> Number of Reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html"><i class="fa fa-check"></i><b>11</b> Downsample Late Group to test # DEGs</a>
<ul>
<li class="chapter" data-level="11.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#libraries-19"><i class="fa fa-check"></i><b>11.1</b> Libraries</a></li>
<li class="chapter" data-level="11.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#data-21"><i class="fa fa-check"></i><b>11.2</b> Data</a></li>
<li class="chapter" data-level="11.3" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#transcriptome-7"><i class="fa fa-check"></i><b>11.3</b> Transcriptome</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop"><i class="fa fa-check"></i><b>11.3.1</b> Downsample &amp; Loop</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#genome-7"><i class="fa fa-check"></i><b>11.4</b> Genome</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop-1"><i class="fa fa-check"></i><b>11.4.1</b> Downsample &amp; Loop</a></li>
<li class="chapter" data-level="11.4.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#summarize-degs"><i class="fa fa-check"></i><b>11.4.2</b> Summarize # DEGs</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular mechanisms underlying response to influenza in grey seals (<em>Halichoerus grypus</em>), a potential wild reservoir</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="iav-kegg-pathway-candidate-gene-analysis" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> IAV Kegg Pathway Candidate Gene Analysis<a href="iav-kegg-pathway-candidate-gene-analysis.html#iav-kegg-pathway-candidate-gene-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="libraries-17" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Libraries<a href="iav-kegg-pathway-candidate-gene-analysis.html#libraries-17" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-1" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb445-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-2" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb445-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-3" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi) </span>
<span id="cb445-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb445-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-5" tabindex="-1"></a><span class="fu">library</span>(ggkegg)</span>
<span id="cb445-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb445-6" tabindex="-1"></a><span class="fu">library</span>(tidygraph)</span></code></pre></div>
</div>
<div id="data-19" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Data<a href="iav-kegg-pathway-candidate-gene-analysis.html#data-19" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-1" tabindex="-1"></a>t_postfilter <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;Output Files/txome_female_edgeR_postfilter.txt&quot;</span>)</span>
<span id="cb446-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-2" tabindex="-1"></a>  </span>
<span id="cb446-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-3" tabindex="-1"></a>t_acute <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_DEG_acute.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-4" tabindex="-1"></a>t_peak <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_DEG_peak.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-5" tabindex="-1"></a>t_late <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_DEG_late.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-6" tabindex="-1"></a></span>
<span id="cb446-7"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-7" tabindex="-1"></a>g_postfilter <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_postfilter.txt&quot;</span>)</span>
<span id="cb446-8"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-8" tabindex="-1"></a>g_acute <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_DEG_acute.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-9"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-9" tabindex="-1"></a>g_peak <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_DEG_peak.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-10"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-10" tabindex="-1"></a>g_late <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_DEG_late.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-11"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-11" tabindex="-1"></a></span>
<span id="cb446-12"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-12" tabindex="-1"></a>t_virus_postfilter <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_postfilter.txt&quot;</span>)</span>
<span id="cb446-13"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-13" tabindex="-1"></a>t_virus_degs <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_DEG.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb446-14"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-14" tabindex="-1"></a></span>
<span id="cb446-15"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-15" tabindex="-1"></a>g_virus_postfilter <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_postfilter.txt&quot;</span>)</span>
<span id="cb446-16"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb446-16" tabindex="-1"></a>g_virus_degs <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_DEG.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="extract-genes-from-kegg-pathway" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Extract genes from KEGG Pathway<a href="iav-kegg-pathway-candidate-gene-analysis.html#extract-genes-from-kegg-pathway" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-1" tabindex="-1"></a>iav <span class="ot">&lt;-</span> </span>
<span id="cb447-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-2" tabindex="-1"></a>  <span class="fu">getGeneKEGGLinks</span>(<span class="at">species=</span><span class="st">&quot;hsa&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb447-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-3" tabindex="-1"></a>  <span class="fu">filter</span>(PathwayID <span class="sc">==</span> <span class="st">&quot;hsa05164&quot;</span>)</span>
<span id="cb447-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-4" tabindex="-1"></a></span>
<span id="cb447-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-5" tabindex="-1"></a>iav<span class="sc">$</span>Symbol <span class="ot">&lt;-</span> <span class="fu">mapIds</span>(org.Hs.eg.db, iav<span class="sc">$</span>GeneID,</span>
<span id="cb447-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb447-6" tabindex="-1"></a>                     <span class="at">column=</span><span class="st">&quot;SYMBOL&quot;</span>, <span class="at">keytype=</span><span class="st">&quot;ENTREZID&quot;</span>)</span></code></pre></div>
</div>
<div id="overlap-with-gene-lists" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Overlap with Gene Lists<a href="iav-kegg-pathway-candidate-gene-analysis.html#overlap-with-gene-lists" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="postfilter" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Postfilter<a href="iav-kegg-pathway-candidate-gene-analysis.html#postfilter" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-1" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> </span>
<span id="cb448-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-2" tabindex="-1"></a>  <span class="fu">rbind</span>(t_postfilter, g_postfilter, t_virus_postfilter) <span class="sc">%&gt;%</span> </span>
<span id="cb448-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(V1))</span>
<span id="cb448-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-4" tabindex="-1"></a></span>
<span id="cb448-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-5" tabindex="-1"></a>iav_postfilter <span class="ot">&lt;-</span></span>
<span id="cb448-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb448-6" tabindex="-1"></a>  <span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, postfilter<span class="sc">$</span>V1)</span></code></pre></div>
</div>
<div id="differentially-expressed-genes" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> Differentially Expressed Genes<a href="iav-kegg-pathway-candidate-gene-analysis.html#differentially-expressed-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb449-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-1" tabindex="-1"></a>acute <span class="ot">&lt;-</span> </span>
<span id="cb449-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb449-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-3" tabindex="-1"></a>    <span class="at">genes=</span><span class="fu">c</span>(<span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, t_acute<span class="sc">$</span>gene), </span>
<span id="cb449-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-4" tabindex="-1"></a>            <span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, g_acute<span class="sc">$</span>gene)), </span>
<span id="cb449-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-5" tabindex="-1"></a>    <span class="at">category=</span><span class="st">&quot;acute&quot;</span>)</span>
<span id="cb449-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-6" tabindex="-1"></a></span>
<span id="cb449-7"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-7" tabindex="-1"></a>peak <span class="ot">&lt;-</span> </span>
<span id="cb449-8"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-8" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb449-9"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-9" tabindex="-1"></a>    <span class="at">genes=</span><span class="fu">c</span>(<span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, t_peak<span class="sc">$</span>gene), </span>
<span id="cb449-10"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-10" tabindex="-1"></a>            <span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, g_peak<span class="sc">$</span>gene)), </span>
<span id="cb449-11"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-11" tabindex="-1"></a>    <span class="at">category=</span><span class="st">&quot;peak&quot;</span>)</span>
<span id="cb449-12"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-12" tabindex="-1"></a></span>
<span id="cb449-13"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-13" tabindex="-1"></a>late <span class="ot">&lt;-</span> </span>
<span id="cb449-14"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-14" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb449-15"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-15" tabindex="-1"></a>    <span class="at">genes=</span><span class="fu">c</span>(<span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, t_late<span class="sc">$</span>gene),</span>
<span id="cb449-16"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-16" tabindex="-1"></a>            <span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, g_late<span class="sc">$</span>gene)),</span>
<span id="cb449-17"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-17" tabindex="-1"></a>    <span class="at">category=</span><span class="st">&quot;late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb449-18"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-18" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(genes))</span>
<span id="cb449-19"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-19" tabindex="-1"></a></span>
<span id="cb449-20"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-20" tabindex="-1"></a>virus <span class="ot">&lt;-</span> </span>
<span id="cb449-21"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-21" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb449-22"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-22" tabindex="-1"></a>    <span class="at">genes=</span><span class="fu">c</span>(<span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, t_virus_degs<span class="sc">$</span>gene),</span>
<span id="cb449-23"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-23" tabindex="-1"></a>            <span class="fu">intersect</span>(iav<span class="sc">$</span>Symbol, g_virus_degs<span class="sc">$</span>gene)),</span>
<span id="cb449-24"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-24" tabindex="-1"></a>    <span class="at">category=</span><span class="st">&quot;virus&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb449-25"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb449-25" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(genes))</span></code></pre></div>
</div>
<div id="compile-gene-ids" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> Compile gene IDs<a href="iav-kegg-pathway-candidate-gene-analysis.html#compile-gene-ids" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-1" tabindex="-1"></a>deg <span class="ot">&lt;-</span></span>
<span id="cb450-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-2" tabindex="-1"></a>  <span class="fu">rbind</span>(acute, peak, late, virus) <span class="sc">%&gt;%</span> </span>
<span id="cb450-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">paste0</span>(<span class="st">&quot;hsa:&quot;</span>, <span class="fu">ifelse</span>(genes <span class="sc">%in%</span> iav<span class="sc">$</span>Symbol, </span>
<span id="cb450-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-4" tabindex="-1"></a>                     iav[<span class="fu">match</span>(genes, iav<span class="sc">$</span>Symbol),<span class="dv">1</span>],</span>
<span id="cb450-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-5" tabindex="-1"></a>                     genes)))</span>
<span id="cb450-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-6" tabindex="-1"></a></span>
<span id="cb450-7"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-7" tabindex="-1"></a>expressed <span class="ot">&lt;-</span></span>
<span id="cb450-8"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-8" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">genes=</span>iav_postfilter) <span class="sc">%&gt;%</span>  </span>
<span id="cb450-9"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-9" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>genes <span class="sc">%in%</span> deg<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb450-10"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">paste0</span>(<span class="st">&quot;hsa:&quot;</span>, <span class="fu">ifelse</span>(genes <span class="sc">%in%</span> iav<span class="sc">$</span>Symbol, </span>
<span id="cb450-11"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-11" tabindex="-1"></a>                     iav[<span class="fu">match</span>(genes, iav<span class="sc">$</span>Symbol),<span class="dv">1</span>],</span>
<span id="cb450-12"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-12" tabindex="-1"></a>                     genes)))</span>
<span id="cb450-13"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-13" tabindex="-1"></a></span>
<span id="cb450-14"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-14" tabindex="-1"></a>notexpressed <span class="ot">&lt;-</span> </span>
<span id="cb450-15"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-15" tabindex="-1"></a>  <span class="fu">data.frame</span>(iav) <span class="sc">%&gt;%</span> </span>
<span id="cb450-16"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-16" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>Symbol <span class="sc">%in%</span> expressed<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb450-17"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-17" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>Symbol <span class="sc">%in%</span> deg<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb450-18"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">paste0</span>(<span class="st">&quot;hsa:&quot;</span>, <span class="fu">ifelse</span>(Symbol <span class="sc">%in%</span> iav<span class="sc">$</span>Symbol, </span>
<span id="cb450-19"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-19" tabindex="-1"></a>                     iav[<span class="fu">match</span>(Symbol, iav<span class="sc">$</span>Symbol),<span class="dv">1</span>],</span>
<span id="cb450-20"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb450-20" tabindex="-1"></a>                     Symbol)))</span></code></pre></div>
</div>
</div>
<div id="plot-pathway-map" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Plot pathway map<a href="iav-kegg-pathway-candidate-gene-analysis.html#plot-pathway-map" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>** no longer used - cannot figure out coloring for some genes. Created map online.</p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-1" tabindex="-1"></a>graph_info <span class="ot">&lt;-</span></span>
<span id="cb451-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-2" tabindex="-1"></a>  <span class="fu">pathway</span>(<span class="st">&quot;hsa05164&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb451-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-3" tabindex="-1"></a>  <span class="fu">activate</span>(nodes) <span class="sc">%&gt;%</span> </span>
<span id="cb451-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">convert_hsa=</span><span class="fu">convert_id</span>(<span class="st">&quot;hsa&quot;</span>), </span>
<span id="cb451-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-5" tabindex="-1"></a>         <span class="at">convert_map=</span><span class="fu">convert_id</span>(<span class="st">&quot;pathway&quot;</span>))</span>
<span id="cb451-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-6" tabindex="-1"></a>         bgcolor<span class="ot">=</span><span class="fu">ifelse</span>(convert_hsa <span class="sc">%in%</span> expressed<span class="sc">$</span>id, <span class="st">&quot;#FFFFE0&quot;</span>, bgcolor)<span class="er">)</span></span>
<span id="cb451-7"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-7" tabindex="-1"></a></span>
<span id="cb451-8"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-8" tabindex="-1"></a>graph <span class="ot">&lt;-</span></span>
<span id="cb451-9"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-9" tabindex="-1"></a>  <span class="fu">ggraph</span>(graph_info, <span class="at">x=</span>x, <span class="at">y=</span>y) <span class="sc">+</span></span>
<span id="cb451-10"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-10" tabindex="-1"></a>   <span class="fu">geom_node_rect</span>(<span class="fu">aes</span>(<span class="at">filter=</span>type<span class="sc">==</span><span class="st">&quot;gene&quot;</span>, </span>
<span id="cb451-11"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-11" tabindex="-1"></a>                      <span class="at">color=</span>bgcolor)) <span class="sc">+</span> </span>
<span id="cb451-12"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-12" tabindex="-1"></a>  <span class="fu">overlay_raw_map</span>() <span class="sc">+</span></span>
<span id="cb451-13"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-13" tabindex="-1"></a>  <span class="fu">theme_void</span>() </span>
<span id="cb451-14"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-14" tabindex="-1"></a>graph </span>
<span id="cb451-15"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-15" tabindex="-1"></a></span>
<span id="cb451-16"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb451-16" tabindex="-1"></a><span class="co">#ggkeggsave(filename=&quot;Figures/iavkegg.jpg&quot;, graph, dpi=300)</span></span></code></pre></div>
<div id="create-legend" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Create Legend<a href="iav-kegg-pathway-candidate-gene-analysis.html#create-legend" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-1" tabindex="-1"></a><span class="fu">png</span>(<span class="at">filename=</span><span class="st">&quot;Figures/iavkeggleg.png&quot;</span>, <span class="at">width =</span> <span class="dv">6</span>, <span class="at">height=</span><span class="dv">7</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>, <span class="at">res=</span><span class="dv">300</span>)</span>
<span id="cb452-2"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-2" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="cn">NULL</span> ,<span class="at">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="at">yaxt=</span><span class="st">&#39;n&#39;</span>,<span class="at">bty=</span><span class="st">&#39;n&#39;</span>,<span class="at">ylab=</span><span class="st">&#39;&#39;</span>,<span class="at">xlab=</span><span class="st">&#39;&#39;</span>, , <span class="at">xlim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>), <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">10</span>))</span>
<span id="cb452-3"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-3" tabindex="-1"></a>  <span class="fu">legend</span>(<span class="st">&quot;topleft&quot;</span>, </span>
<span id="cb452-4"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-4" tabindex="-1"></a>         <span class="at">legend =</span><span class="fu">c</span>(<span class="st">&quot;Acute&quot;</span>, <span class="st">&quot;Peak&quot;</span>, <span class="st">&quot;Late&quot;</span>, <span class="st">&quot;Virus&quot;</span>, <span class="st">&quot;Expressed&quot;</span>, <span class="st">&quot;Not Expressed&quot;</span>), </span>
<span id="cb452-5"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-5" tabindex="-1"></a>         <span class="at">pch=</span><span class="dv">15</span>, <span class="at">pt.cex=</span><span class="dv">7</span>, <span class="at">cex=</span><span class="dv">3</span>, <span class="at">bty=</span><span class="st">&#39;o&#39;</span>,</span>
<span id="cb452-6"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-6" tabindex="-1"></a>         <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;darkslategray1&quot;</span>, <span class="st">&quot;darkslategray3&quot;</span>, <span class="st">&quot;darkslategray4&quot;</span>, <span class="st">&quot;gray80&quot;</span>, <span class="st">&quot;coral&quot;</span>, <span class="st">&quot;wheat2&quot;</span>), </span>
<span id="cb452-7"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-7" tabindex="-1"></a>         <span class="at">horiz=</span>F)</span>
<span id="cb452-8"><a href="iav-kegg-pathway-candidate-gene-analysis.html#cb452-8" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="r-package-comparison---edger-vs-deseq2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="outlier-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section",
    "scroll_highlight": true
  },
  "info": false
});
});
</script>

</body>

</html>
