<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Supplemental Null Set Exploratory Statistics | Untitled</title>
  <meta name="description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Supplemental Null Set Exploratory Statistics | Untitled" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Supplemental Null Set Exploratory Statistics | Untitled" />
  
  <meta name="twitter:description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  

<meta name="author" content="Christina McCosker" />


<meta name="date" content="2025-01-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="influenza-stage-dge.html"/>
<link rel="next" href="influenza-dge-deseq2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hg-RNA Stats</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Publication Information</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors"><i class="fa fa-check"></i><b>1.1</b> Authors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#abstract"><i class="fa fa-check"></i><b>1.2</b> Abstract</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html"><i class="fa fa-check"></i><b>2</b> LOC Gene Identification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#libraries"><i class="fa fa-check"></i><b>2.1</b> Libraries</a></li>
<li class="chapter" data-level="2.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#load-data"><i class="fa fa-check"></i><b>2.2</b> Load data</a></li>
<li class="chapter" data-level="2.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-gene-sequences"><i class="fa fa-check"></i><b>2.3</b> Extract gene sequences</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-loc-genes-from-gene-count-matrices"><i class="fa fa-check"></i><b>2.3.1</b> Extract LOC genes from gene count matrices</a></li>
<li class="chapter" data-level="2.3.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-ids-for-all-sequence-ids"><i class="fa fa-check"></i><b>2.3.2</b> Extract transcript IDs for all sequence IDs</a></li>
<li class="chapter" data-level="2.3.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-sequences-for-blast"><i class="fa fa-check"></i><b>2.3.3</b> Extract transcript sequences for blast</a></li>
<li class="chapter" data-level="2.3.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-sequences"><i class="fa fa-check"></i><b>2.3.4</b> Export sequences</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#blast-sequences"><i class="fa fa-check"></i><b>2.4</b> Blast sequences</a></li>
<li class="chapter" data-level="2.5" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#process-blastn-results"><i class="fa fa-check"></i><b>2.5</b> Process BLASTn Results</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#transcriptome"><i class="fa fa-check"></i><b>2.5.1</b> Transcriptome</a></li>
<li class="chapter" data-level="2.5.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#genome"><i class="fa fa-check"></i><b>2.5.2</b> Genome</a></li>
<li class="chapter" data-level="2.5.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-transcriptome-genome-blast-results"><i class="fa fa-check"></i><b>2.5.3</b> Combine Transcriptome &amp; Genome Blast Results</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-with-prior-loc-id-master-list"><i class="fa fa-check"></i><b>2.6</b> Combine with prior LOC ID master list</a></li>
<li class="chapter" data-level="2.7" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#update-tidy-gene-count-matrices"><i class="fa fa-check"></i><b>2.7</b> Update &amp; tidy gene count matrices</a></li>
<li class="chapter" data-level="2.8" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#hemoglobin"><i class="fa fa-check"></i><b>2.8</b> Hemoglobin</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.1</b> Extract hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#prop-of-total-gene-counts"><i class="fa fa-check"></i><b>2.8.2</b> Prop of total gene counts</a></li>
<li class="chapter" data-level="2.8.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#remove-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.3</b> Remove hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-new-gene-count-matrices"><i class="fa fa-check"></i><b>2.8.4</b> Export new gene count matrices</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html"><i class="fa fa-check"></i><b>3</b> Male v Female DGE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#transcriptome---edger"><i class="fa fa-check"></i><b>3.1</b> Transcriptome - edgeR</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-1"><i class="fa fa-check"></i><b>3.1.1</b> Libraries</a></li>
<li class="chapter" data-level="3.1.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data"><i class="fa fa-check"></i><b>3.1.2</b> Data</a></li>
<li class="chapter" data-level="3.1.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.1.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots"><i class="fa fa-check"></i><b>3.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.1.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs"><i class="fa fa-check"></i><b>3.1.5</b> Test for DEGs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#genome---edger"><i class="fa fa-check"></i><b>3.2</b> Genome - edgeR</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-2"><i class="fa fa-check"></i><b>3.2.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data-1"><i class="fa fa-check"></i><b>3.2.2</b> Data</a></li>
<li class="chapter" data-level="3.2.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>3.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.2.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots-1"><i class="fa fa-check"></i><b>3.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.2.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs-1"><i class="fa fa-check"></i><b>3.2.5</b> Test for DEGs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html"><i class="fa fa-check"></i><b>4</b> Virus+ vs Virus- DGE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome---edger---females"><i class="fa fa-check"></i><b>4.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-3"><i class="fa fa-check"></i><b>4.1.1</b> Libraries</a></li>
<li class="chapter" data-level="4.1.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-2"><i class="fa fa-check"></i><b>4.1.2</b> Data</a></li>
<li class="chapter" data-level="4.1.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>4.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.1.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-2"><i class="fa fa-check"></i><b>4.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.1.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-2"><i class="fa fa-check"></i><b>4.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.1.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis"><i class="fa fa-check"></i><b>4.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.1.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis"><i class="fa fa-check"></i><b>4.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome---edger---females"><i class="fa fa-check"></i><b>4.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-4"><i class="fa fa-check"></i><b>4.2.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-3"><i class="fa fa-check"></i><b>4.2.2</b> Data</a></li>
<li class="chapter" data-level="4.2.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>4.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.2.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-3"><i class="fa fa-check"></i><b>4.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.2.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-3"><i class="fa fa-check"></i><b>4.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.2.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-1"><i class="fa fa-check"></i><b>4.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.2.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-1"><i class="fa fa-check"></i><b>4.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-null-set---edger---females"><i class="fa fa-check"></i><b>4.3</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-5"><i class="fa fa-check"></i><b>4.3.1</b> Libraries</a></li>
<li class="chapter" data-level="4.3.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-4"><i class="fa fa-check"></i><b>4.3.2</b> Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-2"><i class="fa fa-check"></i><b>4.3.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.3.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-2"><i class="fa fa-check"></i><b>4.3.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-null-set---edger---females"><i class="fa fa-check"></i><b>4.4</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-6"><i class="fa fa-check"></i><b>4.4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-5"><i class="fa fa-check"></i><b>4.4.2</b> Data</a></li>
<li class="chapter" data-level="4.4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>4.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-3"><i class="fa fa-check"></i><b>4.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-3"><i class="fa fa-check"></i><b>4.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#refine-gokegg"><i class="fa fa-check"></i><b>4.5</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-7"><i class="fa fa-check"></i><b>4.5.1</b> Libraries</a></li>
<li class="chapter" data-level="4.5.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-6"><i class="fa fa-check"></i><b>4.5.2</b> Data</a></li>
<li class="chapter" data-level="4.5.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-1"><i class="fa fa-check"></i><b>4.5.3</b> Transcriptome</a></li>
<li class="chapter" data-level="4.5.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-1"><i class="fa fa-check"></i><b>4.5.4</b> Genome</a></li>
<li class="chapter" data-level="4.5.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-term-figure"><i class="fa fa-check"></i><b>4.5.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="4.5.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#immunne-go-term-genes"><i class="fa fa-check"></i><b>4.5.6</b> Immunne GO term Genes</a></li>
<li class="chapter" data-level="4.5.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-table---manuscript"><i class="fa fa-check"></i><b>4.5.7</b> KEGG Table - Manuscript</a></li>
<li class="chapter" data-level="4.5.8" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#supplemental-table"><i class="fa fa-check"></i><b>4.5.8</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html"><i class="fa fa-check"></i><b>5</b> Influenza Stage DGE</a>
<ul>
<li class="chapter" data-level="5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome---edger---females-1"><i class="fa fa-check"></i><b>5.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-8"><i class="fa fa-check"></i><b>5.1.1</b> Libraries</a></li>
<li class="chapter" data-level="5.1.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-7"><i class="fa fa-check"></i><b>5.1.2</b> Data</a></li>
<li class="chapter" data-level="5.1.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-4"><i class="fa fa-check"></i><b>5.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-4"><i class="fa fa-check"></i><b>5.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.1.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-4"><i class="fa fa-check"></i><b>5.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.1.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-4"><i class="fa fa-check"></i><b>5.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.1.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-4"><i class="fa fa-check"></i><b>5.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome---edger---females-1"><i class="fa fa-check"></i><b>5.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-9"><i class="fa fa-check"></i><b>5.2.1</b> Libraries</a></li>
<li class="chapter" data-level="5.2.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-8"><i class="fa fa-check"></i><b>5.2.2</b> Data</a></li>
<li class="chapter" data-level="5.2.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-5"><i class="fa fa-check"></i><b>5.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.2.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-5"><i class="fa fa-check"></i><b>5.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.2.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-5"><i class="fa fa-check"></i><b>5.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.2.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-5"><i class="fa fa-check"></i><b>5.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.2.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-5"><i class="fa fa-check"></i><b>5.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#comapre-t-g-approach---edger---females"><i class="fa fa-check"></i><b>5.3</b> Comapre T &amp; G Approach - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-10"><i class="fa fa-check"></i><b>5.3.1</b> Libraries</a></li>
<li class="chapter" data-level="5.3.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-9"><i class="fa fa-check"></i><b>5.3.2</b> Data</a></li>
<li class="chapter" data-level="5.3.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#number-of-up-down-reg-genes"><i class="fa fa-check"></i><b>5.3.3</b> Number of up &amp; down reg genes</a></li>
<li class="chapter" data-level="5.3.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#gene-identities"><i class="fa fa-check"></i><b>5.3.4</b> Gene identities</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.4</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-11"><i class="fa fa-check"></i><b>5.4.1</b> Libraries</a></li>
<li class="chapter" data-level="5.4.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-10"><i class="fa fa-check"></i><b>5.4.2</b> Data</a></li>
<li class="chapter" data-level="5.4.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-6"><i class="fa fa-check"></i><b>5.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-6"><i class="fa fa-check"></i><b>5.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.5</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-12"><i class="fa fa-check"></i><b>5.5.1</b> Libraries</a></li>
<li class="chapter" data-level="5.5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-11"><i class="fa fa-check"></i><b>5.5.2</b> Data</a></li>
<li class="chapter" data-level="5.5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>5.5.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-7"><i class="fa fa-check"></i><b>5.5.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-7"><i class="fa fa-check"></i><b>5.5.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#refine-gokegg-1"><i class="fa fa-check"></i><b>5.6</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-13"><i class="fa fa-check"></i><b>5.6.1</b> Libraries</a></li>
<li class="chapter" data-level="5.6.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-12"><i class="fa fa-check"></i><b>5.6.2</b> Data</a></li>
<li class="chapter" data-level="5.6.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-2"><i class="fa fa-check"></i><b>5.6.3</b> Transcriptome</a></li>
<li class="chapter" data-level="5.6.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-2"><i class="fa fa-check"></i><b>5.6.4</b> Genome</a></li>
<li class="chapter" data-level="5.6.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-term-figure-1"><i class="fa fa-check"></i><b>5.6.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="5.6.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#supplemental-table-1"><i class="fa fa-check"></i><b>5.6.6</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html"><i class="fa fa-check"></i><b>6</b> Supplemental Null Set Exploratory Statistics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#libaries"><i class="fa fa-check"></i><b>6.1</b> Libaries</a></li>
<li class="chapter" data-level="6.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-degs"><i class="fa fa-check"></i><b>6.2</b> Number of DEGs</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-13"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats"><i class="fa fa-check"></i><b>6.2.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.2.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-1"><i class="fa fa-check"></i><b>6.2.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#deg-frequency"><i class="fa fa-check"></i><b>6.3</b> DEG Frequency</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-14"><i class="fa fa-check"></i><b>6.3.1</b> Data</a></li>
<li class="chapter" data-level="6.3.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats-1"><i class="fa fa-check"></i><b>6.3.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.3.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-2"><i class="fa fa-check"></i><b>6.3.3</b> Plot</a></li>
<li class="chapter" data-level="6.3.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#combine-gene-number-frequency-plots"><i class="fa fa-check"></i><b>6.3.4</b> Combine gene number &amp; frequency plots</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-go-terms"><i class="fa fa-check"></i><b>6.4</b> Number of GO terms</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data---import-configure"><i class="fa fa-check"></i><b>6.4.1</b> Data - Import &amp; Configure</a></li>
<li class="chapter" data-level="6.4.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-3"><i class="fa fa-check"></i><b>6.4.2</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-kegg-pathways"><i class="fa fa-check"></i><b>6.5</b> Number of KEGG Pathways</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-15"><i class="fa fa-check"></i><b>6.5.1</b> Data</a></li>
<li class="chapter" data-level="6.5.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-4"><i class="fa fa-check"></i><b>6.5.2</b> Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html"><i class="fa fa-check"></i><b>7</b> Influenza DGE DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#transcriptome---deseq2---females"><i class="fa fa-check"></i><b>7.1</b> Transcriptome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-14"><i class="fa fa-check"></i><b>7.1.1</b> Libraries</a></li>
<li class="chapter" data-level="7.1.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-16"><i class="fa fa-check"></i><b>7.1.2</b> Data</a></li>
<li class="chapter" data-level="7.1.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna"><i class="fa fa-check"></i><b>7.1.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.1.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage"><i class="fa fa-check"></i><b>7.1.4</b> Influenza Stage</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#genome---deseq2---females"><i class="fa fa-check"></i><b>7.2</b> Genome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-15"><i class="fa fa-check"></i><b>7.2.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-17"><i class="fa fa-check"></i><b>7.2.2</b> Data</a></li>
<li class="chapter" data-level="7.2.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna-1"><i class="fa fa-check"></i><b>7.2.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.2.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage-1"><i class="fa fa-check"></i><b>7.2.4</b> Influenza Stage</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html"><i class="fa fa-check"></i><b>8</b> R Package Comparison - edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#libraries-16"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#data-18"><i class="fa fa-check"></i><b>8.2</b> Data</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#metadata"><i class="fa fa-check"></i><b>8.2.1</b> Metadata</a></li>
<li class="chapter" data-level="8.2.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#results-from-dge-testing"><i class="fa fa-check"></i><b>8.2.2</b> Results from DGE testing</a></li>
<li class="chapter" data-level="8.2.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#differentially-expressed-genes-only"><i class="fa fa-check"></i><b>8.2.3</b> Differentially expressed genes only</a></li>
<li class="chapter" data-level="8.2.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pre-filter-lists"><i class="fa fa-check"></i><b>8.2.4</b> Pre-filter lists</a></li>
<li class="chapter" data-level="8.2.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#normalized-gene-counts"><i class="fa fa-check"></i><b>8.2.5</b> Normalized gene counts</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#edger-vs-deseq2"><i class="fa fa-check"></i><b>8.3</b> edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#number-of-pre-vs-post-filter-genes"><i class="fa fa-check"></i><b>8.3.1</b> Number of Pre vs Post Filter Genes</a></li>
<li class="chapter" data-level="8.3.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results"><i class="fa fa-check"></i><b>8.3.2</b> Merge results</a></li>
<li class="chapter" data-level="8.3.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value"><i class="fa fa-check"></i><b>8.3.3</b> p-value</a></li>
<li class="chapter" data-level="8.3.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc"><i class="fa fa-check"></i><b>8.3.4</b> log(FC)</a></li>
<li class="chapter" data-level="8.3.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression"><i class="fa fa-check"></i><b>8.3.5</b> log mean expression</a></li>
<li class="chapter" data-level="8.3.6" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#dge-explortory-stats"><i class="fa fa-check"></i><b>8.3.6</b> DGE explortory stats</a></li>
<li class="chapter" data-level="8.3.7" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#method-gene-ids"><i class="fa fa-check"></i><b>8.3.7</b> Method Gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#transcriptome-vs-genome"><i class="fa fa-check"></i><b>8.4</b> Transcriptome vs Genome</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results-1"><i class="fa fa-check"></i><b>8.4.1</b> Merge results</a></li>
<li class="chapter" data-level="8.4.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value-1"><i class="fa fa-check"></i><b>8.4.2</b> p-value</a></li>
<li class="chapter" data-level="8.4.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc-1"><i class="fa fa-check"></i><b>8.4.3</b> log(FC)</a></li>
<li class="chapter" data-level="8.4.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression-1"><i class="fa fa-check"></i><b>8.4.4</b> log mean expression</a></li>
<li class="chapter" data-level="8.4.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pca-plots-6"><i class="fa fa-check"></i><b>8.4.5</b> PCA plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html"><i class="fa fa-check"></i><b>9</b> IAV Kegg Pathway Candidate Gene Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#libraries-17"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#data-19"><i class="fa fa-check"></i><b>9.2</b> Data</a></li>
<li class="chapter" data-level="9.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#extract-genes-from-kegg-pathway"><i class="fa fa-check"></i><b>9.3</b> Extract genes from KEGG Pathway</a></li>
<li class="chapter" data-level="9.4" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#overlap-with-gene-lists"><i class="fa fa-check"></i><b>9.4</b> Overlap with Gene Lists</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#postfilter"><i class="fa fa-check"></i><b>9.4.1</b> Postfilter</a></li>
<li class="chapter" data-level="9.4.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#differentially-expressed-genes"><i class="fa fa-check"></i><b>9.4.2</b> Differentially Expressed Genes</a></li>
<li class="chapter" data-level="9.4.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#compile-gene-ids"><i class="fa fa-check"></i><b>9.4.3</b> Compile gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#plot-pathway-map"><i class="fa fa-check"></i><b>9.5</b> Plot pathway map</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#create-legend"><i class="fa fa-check"></i><b>9.5.1</b> Create Legend</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="outlier-analysis.html"><a href="outlier-analysis.html"><i class="fa fa-check"></i><b>10</b> Outlier Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#libraries-18"><i class="fa fa-check"></i><b>10.1</b> Libraries</a></li>
<li class="chapter" data-level="10.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#data-20"><i class="fa fa-check"></i><b>10.2</b> Data</a></li>
<li class="chapter" data-level="10.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#pca-2"><i class="fa fa-check"></i><b>10.3</b> PCA</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#transcriptome-6"><i class="fa fa-check"></i><b>10.3.1</b> Transcriptome</a></li>
<li class="chapter" data-level="10.3.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#genome-6"><i class="fa fa-check"></i><b>10.3.2</b> Genome</a></li>
<li class="chapter" data-level="10.3.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#number-of-reads"><i class="fa fa-check"></i><b>10.3.3</b> Number of Reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html"><i class="fa fa-check"></i><b>11</b> Downsample Late Group to test # DEGs</a>
<ul>
<li class="chapter" data-level="11.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#libraries-19"><i class="fa fa-check"></i><b>11.1</b> Libraries</a></li>
<li class="chapter" data-level="11.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#data-21"><i class="fa fa-check"></i><b>11.2</b> Data</a></li>
<li class="chapter" data-level="11.3" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#transcriptome-7"><i class="fa fa-check"></i><b>11.3</b> Transcriptome</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop"><i class="fa fa-check"></i><b>11.3.1</b> Downsample &amp; Loop</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#genome-7"><i class="fa fa-check"></i><b>11.4</b> Genome</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop-1"><i class="fa fa-check"></i><b>11.4.1</b> Downsample &amp; Loop</a></li>
<li class="chapter" data-level="11.4.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#summarize-degs"><i class="fa fa-check"></i><b>11.4.2</b> Summarize # DEGs</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Untitled</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplemental-null-set-exploratory-statistics" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Supplemental Null Set Exploratory Statistics<a href="supplemental-null-set-exploratory-statistics.html#supplemental-null-set-exploratory-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="libaries" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Libaries<a href="supplemental-null-set-exploratory-statistics.html#libaries" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="supplemental-null-set-exploratory-statistics.html#cb345-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb345-2"><a href="supplemental-null-set-exploratory-statistics.html#cb345-2" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span></code></pre></div>
</div>
<div id="number-of-degs" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Number of DEGs<a href="supplemental-null-set-exploratory-statistics.html#number-of-degs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-13" class="section level3 hasAnchor" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Data<a href="supplemental-null-set-exploratory-statistics.html#data-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="supplemental-null-set-exploratory-statistics.html#cb346-1" tabindex="-1"></a>t_virus_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-2"><a href="supplemental-null-set-exploratory-statistics.html#cb346-2" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-3"><a href="supplemental-null-set-exploratory-statistics.html#cb346-3" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-4"><a href="supplemental-null-set-exploratory-statistics.html#cb346-4" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-5"><a href="supplemental-null-set-exploratory-statistics.html#cb346-5" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-6"><a href="supplemental-null-set-exploratory-statistics.html#cb346-6" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-7"><a href="supplemental-null-set-exploratory-statistics.html#cb346-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb346-8"><a href="supplemental-null-set-exploratory-statistics.html#cb346-8" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-9"><a href="supplemental-null-set-exploratory-statistics.html#cb346-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-10"><a href="supplemental-null-set-exploratory-statistics.html#cb346-10" tabindex="-1"></a></span>
<span id="cb346-11"><a href="supplemental-null-set-exploratory-statistics.html#cb346-11" tabindex="-1"></a>t_acute_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-12"><a href="supplemental-null-set-exploratory-statistics.html#cb346-12" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-13"><a href="supplemental-null-set-exploratory-statistics.html#cb346-13" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-14"><a href="supplemental-null-set-exploratory-statistics.html#cb346-14" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-15"><a href="supplemental-null-set-exploratory-statistics.html#cb346-15" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-16"><a href="supplemental-null-set-exploratory-statistics.html#cb346-16" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-17"><a href="supplemental-null-set-exploratory-statistics.html#cb346-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb346-18"><a href="supplemental-null-set-exploratory-statistics.html#cb346-18" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-19"><a href="supplemental-null-set-exploratory-statistics.html#cb346-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-20"><a href="supplemental-null-set-exploratory-statistics.html#cb346-20" tabindex="-1"></a></span>
<span id="cb346-21"><a href="supplemental-null-set-exploratory-statistics.html#cb346-21" tabindex="-1"></a>t_peak_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-22"><a href="supplemental-null-set-exploratory-statistics.html#cb346-22" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-23"><a href="supplemental-null-set-exploratory-statistics.html#cb346-23" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-24"><a href="supplemental-null-set-exploratory-statistics.html#cb346-24" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-25"><a href="supplemental-null-set-exploratory-statistics.html#cb346-25" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-26"><a href="supplemental-null-set-exploratory-statistics.html#cb346-26" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-27"><a href="supplemental-null-set-exploratory-statistics.html#cb346-27" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb346-28"><a href="supplemental-null-set-exploratory-statistics.html#cb346-28" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-29"><a href="supplemental-null-set-exploratory-statistics.html#cb346-29" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-30"><a href="supplemental-null-set-exploratory-statistics.html#cb346-30" tabindex="-1"></a></span>
<span id="cb346-31"><a href="supplemental-null-set-exploratory-statistics.html#cb346-31" tabindex="-1"></a>t_late_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-32"><a href="supplemental-null-set-exploratory-statistics.html#cb346-32" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-33"><a href="supplemental-null-set-exploratory-statistics.html#cb346-33" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-34"><a href="supplemental-null-set-exploratory-statistics.html#cb346-34" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-35"><a href="supplemental-null-set-exploratory-statistics.html#cb346-35" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-36"><a href="supplemental-null-set-exploratory-statistics.html#cb346-36" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-37"><a href="supplemental-null-set-exploratory-statistics.html#cb346-37" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb346-38"><a href="supplemental-null-set-exploratory-statistics.html#cb346-38" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-39"><a href="supplemental-null-set-exploratory-statistics.html#cb346-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-40"><a href="supplemental-null-set-exploratory-statistics.html#cb346-40" tabindex="-1"></a></span>
<span id="cb346-41"><a href="supplemental-null-set-exploratory-statistics.html#cb346-41" tabindex="-1"></a>g_virus_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-42"><a href="supplemental-null-set-exploratory-statistics.html#cb346-42" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-43"><a href="supplemental-null-set-exploratory-statistics.html#cb346-43" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-44"><a href="supplemental-null-set-exploratory-statistics.html#cb346-44" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-45"><a href="supplemental-null-set-exploratory-statistics.html#cb346-45" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-46"><a href="supplemental-null-set-exploratory-statistics.html#cb346-46" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-47"><a href="supplemental-null-set-exploratory-statistics.html#cb346-47" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb346-48"><a href="supplemental-null-set-exploratory-statistics.html#cb346-48" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-49"><a href="supplemental-null-set-exploratory-statistics.html#cb346-49" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-50"><a href="supplemental-null-set-exploratory-statistics.html#cb346-50" tabindex="-1"></a></span>
<span id="cb346-51"><a href="supplemental-null-set-exploratory-statistics.html#cb346-51" tabindex="-1"></a>g_acute_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-52"><a href="supplemental-null-set-exploratory-statistics.html#cb346-52" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-53"><a href="supplemental-null-set-exploratory-statistics.html#cb346-53" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-54"><a href="supplemental-null-set-exploratory-statistics.html#cb346-54" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-55"><a href="supplemental-null-set-exploratory-statistics.html#cb346-55" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-56"><a href="supplemental-null-set-exploratory-statistics.html#cb346-56" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-57"><a href="supplemental-null-set-exploratory-statistics.html#cb346-57" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb346-58"><a href="supplemental-null-set-exploratory-statistics.html#cb346-58" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-59"><a href="supplemental-null-set-exploratory-statistics.html#cb346-59" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-60"><a href="supplemental-null-set-exploratory-statistics.html#cb346-60" tabindex="-1"></a></span>
<span id="cb346-61"><a href="supplemental-null-set-exploratory-statistics.html#cb346-61" tabindex="-1"></a>g_peak_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-62"><a href="supplemental-null-set-exploratory-statistics.html#cb346-62" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-63"><a href="supplemental-null-set-exploratory-statistics.html#cb346-63" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-64"><a href="supplemental-null-set-exploratory-statistics.html#cb346-64" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-65"><a href="supplemental-null-set-exploratory-statistics.html#cb346-65" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-66"><a href="supplemental-null-set-exploratory-statistics.html#cb346-66" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-67"><a href="supplemental-null-set-exploratory-statistics.html#cb346-67" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb346-68"><a href="supplemental-null-set-exploratory-statistics.html#cb346-68" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-69"><a href="supplemental-null-set-exploratory-statistics.html#cb346-69" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-70"><a href="supplemental-null-set-exploratory-statistics.html#cb346-70" tabindex="-1"></a></span>
<span id="cb346-71"><a href="supplemental-null-set-exploratory-statistics.html#cb346-71" tabindex="-1"></a>g_late_genes <span class="ot">&lt;-</span> </span>
<span id="cb346-72"><a href="supplemental-null-set-exploratory-statistics.html#cb346-72" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-73"><a href="supplemental-null-set-exploratory-statistics.html#cb346-73" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-74"><a href="supplemental-null-set-exploratory-statistics.html#cb346-74" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb346-75"><a href="supplemental-null-set-exploratory-statistics.html#cb346-75" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-76"><a href="supplemental-null-set-exploratory-statistics.html#cb346-76" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb346-77"><a href="supplemental-null-set-exploratory-statistics.html#cb346-77" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb346-78"><a href="supplemental-null-set-exploratory-statistics.html#cb346-78" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb346-79"><a href="supplemental-null-set-exploratory-statistics.html#cb346-79" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb346-80"><a href="supplemental-null-set-exploratory-statistics.html#cb346-80" tabindex="-1"></a></span>
<span id="cb346-81"><a href="supplemental-null-set-exploratory-statistics.html#cb346-81" tabindex="-1"></a></span>
<span id="cb346-82"><a href="supplemental-null-set-exploratory-statistics.html#cb346-82" tabindex="-1"></a>genes <span class="ot">&lt;-</span> </span>
<span id="cb346-83"><a href="supplemental-null-set-exploratory-statistics.html#cb346-83" tabindex="-1"></a>  <span class="fu">rbind</span>(t_virus_genes, g_virus_genes,</span>
<span id="cb346-84"><a href="supplemental-null-set-exploratory-statistics.html#cb346-84" tabindex="-1"></a>        t_acute_genes, t_peak_genes, t_late_genes,</span>
<span id="cb346-85"><a href="supplemental-null-set-exploratory-statistics.html#cb346-85" tabindex="-1"></a>        g_acute_genes, g_peak_genes, g_late_genes) <span class="sc">%&gt;%</span> </span>
<span id="cb346-86"><a href="supplemental-null-set-exploratory-statistics.html#cb346-86" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="fu">factor</span>(approach, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Transcriptome&quot;</span>, <span class="st">&quot;Genome&quot;</span>)),</span>
<span id="cb346-87"><a href="supplemental-null-set-exploratory-statistics.html#cb346-87" tabindex="-1"></a>         <span class="at">stage=</span><span class="fu">factor</span>(stage, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;V+ vs V-&quot;</span>, <span class="st">&quot;Acute&quot;</span>, <span class="st">&quot;Peak&quot;</span>, <span class="st">&quot;Late&quot;</span>)))</span></code></pre></div>
</div>
<div id="quick-stats" class="section level3 hasAnchor" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Quick Stats<a href="supplemental-null-set-exploratory-statistics.html#quick-stats" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="supplemental-null-set-exploratory-statistics.html#cb347-1" tabindex="-1"></a><span class="fu">range</span>(t_virus_genes<span class="sc">$</span>genes)</span>
<span id="cb347-2"><a href="supplemental-null-set-exploratory-statistics.html#cb347-2" tabindex="-1"></a><span class="fu">range</span>(g_virus_genes<span class="sc">$</span>genes)</span>
<span id="cb347-3"><a href="supplemental-null-set-exploratory-statistics.html#cb347-3" tabindex="-1"></a></span>
<span id="cb347-4"><a href="supplemental-null-set-exploratory-statistics.html#cb347-4" tabindex="-1"></a><span class="fu">range</span>(t_acute_genes<span class="sc">$</span>genes)</span>
<span id="cb347-5"><a href="supplemental-null-set-exploratory-statistics.html#cb347-5" tabindex="-1"></a><span class="fu">range</span>(t_peak_genes<span class="sc">$</span>genes)</span>
<span id="cb347-6"><a href="supplemental-null-set-exploratory-statistics.html#cb347-6" tabindex="-1"></a><span class="fu">range</span>(t_late_genes<span class="sc">$</span>genes)</span>
<span id="cb347-7"><a href="supplemental-null-set-exploratory-statistics.html#cb347-7" tabindex="-1"></a></span>
<span id="cb347-8"><a href="supplemental-null-set-exploratory-statistics.html#cb347-8" tabindex="-1"></a><span class="fu">range</span>(g_acute_genes<span class="sc">$</span>genes)</span>
<span id="cb347-9"><a href="supplemental-null-set-exploratory-statistics.html#cb347-9" tabindex="-1"></a><span class="fu">range</span>(g_peak_genes<span class="sc">$</span>genes)</span>
<span id="cb347-10"><a href="supplemental-null-set-exploratory-statistics.html#cb347-10" tabindex="-1"></a><span class="fu">range</span>(g_late_genes<span class="sc">$</span>genes)</span></code></pre></div>
</div>
<div id="plot-1" class="section level3 hasAnchor" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Plot<a href="supplemental-null-set-exploratory-statistics.html#plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="supplemental-null-set-exploratory-statistics.html#cb348-1" tabindex="-1"></a>null_genes_plot <span class="ot">&lt;-</span></span>
<span id="cb348-2"><a href="supplemental-null-set-exploratory-statistics.html#cb348-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(genes, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span><span class="fu">log</span>(genes))) <span class="sc">+</span></span>
<span id="cb348-3"><a href="supplemental-null-set-exploratory-statistics.html#cb348-3" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb348-4"><a href="supplemental-null-set-exploratory-statistics.html#cb348-4" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>approach) <span class="sc">+</span></span>
<span id="cb348-5"><a href="supplemental-null-set-exploratory-statistics.html#cb348-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span> <span class="cn">NULL</span>, <span class="at">y=</span><span class="st">&quot;log(# DEGs) per Trial&quot;</span>) <span class="sc">+</span></span>
<span id="cb348-6"><a href="supplemental-null-set-exploratory-statistics.html#cb348-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb348-7"><a href="supplemental-null-set-exploratory-statistics.html#cb348-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
</div>
</div>
<div id="deg-frequency" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> DEG Frequency<a href="supplemental-null-set-exploratory-statistics.html#deg-frequency" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-14" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Data<a href="supplemental-null-set-exploratory-statistics.html#data-14" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="supplemental-null-set-exploratory-statistics.html#cb349-1" tabindex="-1"></a>t_virus_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-2"><a href="supplemental-null-set-exploratory-statistics.html#cb349-2" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-3"><a href="supplemental-null-set-exploratory-statistics.html#cb349-3" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-4"><a href="supplemental-null-set-exploratory-statistics.html#cb349-4" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-5"><a href="supplemental-null-set-exploratory-statistics.html#cb349-5" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-6"><a href="supplemental-null-set-exploratory-statistics.html#cb349-6" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-7"><a href="supplemental-null-set-exploratory-statistics.html#cb349-7" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-8"><a href="supplemental-null-set-exploratory-statistics.html#cb349-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb349-9"><a href="supplemental-null-set-exploratory-statistics.html#cb349-9" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-10"><a href="supplemental-null-set-exploratory-statistics.html#cb349-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-11"><a href="supplemental-null-set-exploratory-statistics.html#cb349-11" tabindex="-1"></a></span>
<span id="cb349-12"><a href="supplemental-null-set-exploratory-statistics.html#cb349-12" tabindex="-1"></a>t_acute_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-13"><a href="supplemental-null-set-exploratory-statistics.html#cb349-13" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-14"><a href="supplemental-null-set-exploratory-statistics.html#cb349-14" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-15"><a href="supplemental-null-set-exploratory-statistics.html#cb349-15" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-16"><a href="supplemental-null-set-exploratory-statistics.html#cb349-16" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-17"><a href="supplemental-null-set-exploratory-statistics.html#cb349-17" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-18"><a href="supplemental-null-set-exploratory-statistics.html#cb349-18" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-19"><a href="supplemental-null-set-exploratory-statistics.html#cb349-19" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb349-20"><a href="supplemental-null-set-exploratory-statistics.html#cb349-20" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-21"><a href="supplemental-null-set-exploratory-statistics.html#cb349-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-22"><a href="supplemental-null-set-exploratory-statistics.html#cb349-22" tabindex="-1"></a></span>
<span id="cb349-23"><a href="supplemental-null-set-exploratory-statistics.html#cb349-23" tabindex="-1"></a>t_peak_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-24"><a href="supplemental-null-set-exploratory-statistics.html#cb349-24" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-25"><a href="supplemental-null-set-exploratory-statistics.html#cb349-25" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-26"><a href="supplemental-null-set-exploratory-statistics.html#cb349-26" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-27"><a href="supplemental-null-set-exploratory-statistics.html#cb349-27" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-28"><a href="supplemental-null-set-exploratory-statistics.html#cb349-28" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-29"><a href="supplemental-null-set-exploratory-statistics.html#cb349-29" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-30"><a href="supplemental-null-set-exploratory-statistics.html#cb349-30" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb349-31"><a href="supplemental-null-set-exploratory-statistics.html#cb349-31" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-32"><a href="supplemental-null-set-exploratory-statistics.html#cb349-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-33"><a href="supplemental-null-set-exploratory-statistics.html#cb349-33" tabindex="-1"></a></span>
<span id="cb349-34"><a href="supplemental-null-set-exploratory-statistics.html#cb349-34" tabindex="-1"></a>t_late_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-35"><a href="supplemental-null-set-exploratory-statistics.html#cb349-35" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_DEG_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-36"><a href="supplemental-null-set-exploratory-statistics.html#cb349-36" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-37"><a href="supplemental-null-set-exploratory-statistics.html#cb349-37" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-38"><a href="supplemental-null-set-exploratory-statistics.html#cb349-38" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-39"><a href="supplemental-null-set-exploratory-statistics.html#cb349-39" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-40"><a href="supplemental-null-set-exploratory-statistics.html#cb349-40" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span>  </span>
<span id="cb349-41"><a href="supplemental-null-set-exploratory-statistics.html#cb349-41" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb349-42"><a href="supplemental-null-set-exploratory-statistics.html#cb349-42" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-43"><a href="supplemental-null-set-exploratory-statistics.html#cb349-43" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-44"><a href="supplemental-null-set-exploratory-statistics.html#cb349-44" tabindex="-1"></a></span>
<span id="cb349-45"><a href="supplemental-null-set-exploratory-statistics.html#cb349-45" tabindex="-1"></a></span>
<span id="cb349-46"><a href="supplemental-null-set-exploratory-statistics.html#cb349-46" tabindex="-1"></a>g_virus_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-47"><a href="supplemental-null-set-exploratory-statistics.html#cb349-47" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-48"><a href="supplemental-null-set-exploratory-statistics.html#cb349-48" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-49"><a href="supplemental-null-set-exploratory-statistics.html#cb349-49" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-50"><a href="supplemental-null-set-exploratory-statistics.html#cb349-50" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-51"><a href="supplemental-null-set-exploratory-statistics.html#cb349-51" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-52"><a href="supplemental-null-set-exploratory-statistics.html#cb349-52" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-53"><a href="supplemental-null-set-exploratory-statistics.html#cb349-53" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb349-54"><a href="supplemental-null-set-exploratory-statistics.html#cb349-54" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-55"><a href="supplemental-null-set-exploratory-statistics.html#cb349-55" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-56"><a href="supplemental-null-set-exploratory-statistics.html#cb349-56" tabindex="-1"></a></span>
<span id="cb349-57"><a href="supplemental-null-set-exploratory-statistics.html#cb349-57" tabindex="-1"></a>g_acute_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-58"><a href="supplemental-null-set-exploratory-statistics.html#cb349-58" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-59"><a href="supplemental-null-set-exploratory-statistics.html#cb349-59" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-60"><a href="supplemental-null-set-exploratory-statistics.html#cb349-60" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-61"><a href="supplemental-null-set-exploratory-statistics.html#cb349-61" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-62"><a href="supplemental-null-set-exploratory-statistics.html#cb349-62" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-63"><a href="supplemental-null-set-exploratory-statistics.html#cb349-63" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-64"><a href="supplemental-null-set-exploratory-statistics.html#cb349-64" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb349-65"><a href="supplemental-null-set-exploratory-statistics.html#cb349-65" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-66"><a href="supplemental-null-set-exploratory-statistics.html#cb349-66" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-67"><a href="supplemental-null-set-exploratory-statistics.html#cb349-67" tabindex="-1"></a></span>
<span id="cb349-68"><a href="supplemental-null-set-exploratory-statistics.html#cb349-68" tabindex="-1"></a>g_peak_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-69"><a href="supplemental-null-set-exploratory-statistics.html#cb349-69" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-70"><a href="supplemental-null-set-exploratory-statistics.html#cb349-70" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-71"><a href="supplemental-null-set-exploratory-statistics.html#cb349-71" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-72"><a href="supplemental-null-set-exploratory-statistics.html#cb349-72" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-73"><a href="supplemental-null-set-exploratory-statistics.html#cb349-73" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-74"><a href="supplemental-null-set-exploratory-statistics.html#cb349-74" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-75"><a href="supplemental-null-set-exploratory-statistics.html#cb349-75" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb349-76"><a href="supplemental-null-set-exploratory-statistics.html#cb349-76" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-77"><a href="supplemental-null-set-exploratory-statistics.html#cb349-77" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-78"><a href="supplemental-null-set-exploratory-statistics.html#cb349-78" tabindex="-1"></a></span>
<span id="cb349-79"><a href="supplemental-null-set-exploratory-statistics.html#cb349-79" tabindex="-1"></a>g_late_genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-80"><a href="supplemental-null-set-exploratory-statistics.html#cb349-80" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_DEG_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-81"><a href="supplemental-null-set-exploratory-statistics.html#cb349-81" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-82"><a href="supplemental-null-set-exploratory-statistics.html#cb349-82" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb349-83"><a href="supplemental-null-set-exploratory-statistics.html#cb349-83" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-84"><a href="supplemental-null-set-exploratory-statistics.html#cb349-84" tabindex="-1"></a>  <span class="fu">table</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-85"><a href="supplemental-null-set-exploratory-statistics.html#cb349-85" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb349-86"><a href="supplemental-null-set-exploratory-statistics.html#cb349-86" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb349-87"><a href="supplemental-null-set-exploratory-statistics.html#cb349-87" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb349-88"><a href="supplemental-null-set-exploratory-statistics.html#cb349-88" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;genes&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb349-89"><a href="supplemental-null-set-exploratory-statistics.html#cb349-89" tabindex="-1"></a></span>
<span id="cb349-90"><a href="supplemental-null-set-exploratory-statistics.html#cb349-90" tabindex="-1"></a>genefreq <span class="ot">&lt;-</span> </span>
<span id="cb349-91"><a href="supplemental-null-set-exploratory-statistics.html#cb349-91" tabindex="-1"></a>  <span class="fu">rbind</span>(t_virus_genefreq, g_virus_genefreq, </span>
<span id="cb349-92"><a href="supplemental-null-set-exploratory-statistics.html#cb349-92" tabindex="-1"></a>        t_acute_genefreq, t_peak_genefreq, t_late_genefreq,</span>
<span id="cb349-93"><a href="supplemental-null-set-exploratory-statistics.html#cb349-93" tabindex="-1"></a>        g_acute_genefreq, g_peak_genefreq, g_late_genefreq) <span class="sc">%&gt;%</span> </span>
<span id="cb349-94"><a href="supplemental-null-set-exploratory-statistics.html#cb349-94" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="fu">factor</span>(approach, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Transcriptome&quot;</span>, <span class="st">&quot;Genome&quot;</span>)),</span>
<span id="cb349-95"><a href="supplemental-null-set-exploratory-statistics.html#cb349-95" tabindex="-1"></a>         <span class="at">stage=</span><span class="fu">factor</span>(stage, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;V+ vs V-&quot;</span>, <span class="st">&quot;Acute&quot;</span>, <span class="st">&quot;Peak&quot;</span>, <span class="st">&quot;Late&quot;</span>)))</span></code></pre></div>
</div>
<div id="quick-stats-1" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Quick Stats<a href="supplemental-null-set-exploratory-statistics.html#quick-stats-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb350-1"><a href="supplemental-null-set-exploratory-statistics.html#cb350-1" tabindex="-1"></a><span class="fu">range</span>(t_acute_genefreq<span class="sc">$</span>Freq)</span>
<span id="cb350-2"><a href="supplemental-null-set-exploratory-statistics.html#cb350-2" tabindex="-1"></a><span class="fu">range</span>(t_peak_genefreq<span class="sc">$</span>Freq)</span>
<span id="cb350-3"><a href="supplemental-null-set-exploratory-statistics.html#cb350-3" tabindex="-1"></a><span class="fu">range</span>(t_late_genefreq<span class="sc">$</span>Freq)</span>
<span id="cb350-4"><a href="supplemental-null-set-exploratory-statistics.html#cb350-4" tabindex="-1"></a></span>
<span id="cb350-5"><a href="supplemental-null-set-exploratory-statistics.html#cb350-5" tabindex="-1"></a><span class="fu">range</span>(g_acute_genefreq<span class="sc">$</span>Freq)</span>
<span id="cb350-6"><a href="supplemental-null-set-exploratory-statistics.html#cb350-6" tabindex="-1"></a><span class="fu">range</span>(g_peak_genefreq<span class="sc">$</span>Freq)</span>
<span id="cb350-7"><a href="supplemental-null-set-exploratory-statistics.html#cb350-7" tabindex="-1"></a><span class="fu">range</span>(g_late_genefreq<span class="sc">$</span>Freq)</span></code></pre></div>
</div>
<div id="plot-2" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Plot<a href="supplemental-null-set-exploratory-statistics.html#plot-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="supplemental-null-set-exploratory-statistics.html#cb351-1" tabindex="-1"></a>null_genefreq_plot <span class="ot">&lt;-</span></span>
<span id="cb351-2"><a href="supplemental-null-set-exploratory-statistics.html#cb351-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(genefreq, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span><span class="fu">log</span>(Freq))) <span class="sc">+</span></span>
<span id="cb351-3"><a href="supplemental-null-set-exploratory-statistics.html#cb351-3" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb351-4"><a href="supplemental-null-set-exploratory-statistics.html#cb351-4" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>approach) <span class="sc">+</span></span>
<span id="cb351-5"><a href="supplemental-null-set-exploratory-statistics.html#cb351-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y=</span><span class="st">&quot;log(Gene Frequency)&quot;</span>) <span class="sc">+</span></span>
<span id="cb351-6"><a href="supplemental-null-set-exploratory-statistics.html#cb351-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb351-7"><a href="supplemental-null-set-exploratory-statistics.html#cb351-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span></code></pre></div>
</div>
<div id="combine-gene-number-frequency-plots" class="section level3 hasAnchor" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Combine gene number &amp; frequency plots<a href="supplemental-null-set-exploratory-statistics.html#combine-gene-number-frequency-plots" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="supplemental-null-set-exploratory-statistics.html#cb352-1" tabindex="-1"></a>gene_numfreq <span class="ot">&lt;-</span></span>
<span id="cb352-2"><a href="supplemental-null-set-exploratory-statistics.html#cb352-2" tabindex="-1"></a>  <span class="fu">grid.arrange</span>(null_genes_plot, null_genefreq_plot,</span>
<span id="cb352-3"><a href="supplemental-null-set-exploratory-statistics.html#cb352-3" tabindex="-1"></a>               <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">bottom=</span><span class="st">&quot;Influenza Stage&quot;</span>)</span>
<span id="cb352-4"><a href="supplemental-null-set-exploratory-statistics.html#cb352-4" tabindex="-1"></a></span>
<span id="cb352-5"><a href="supplemental-null-set-exploratory-statistics.html#cb352-5" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/supplemental_TG_F_null_genenum-freq.jpg&quot;</span>, gene_numfreq, </span>
<span id="cb352-6"><a href="supplemental-null-set-exploratory-statistics.html#cb352-6" tabindex="-1"></a>       <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="number-of-go-terms" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Number of GO terms<a href="supplemental-null-set-exploratory-statistics.html#number-of-go-terms" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data---import-configure" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Data - Import &amp; Configure<a href="supplemental-null-set-exploratory-statistics.html#data---import-configure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="supplemental-null-set-exploratory-statistics.html#cb353-1" tabindex="-1"></a>t_virus_bp <span class="ot">&lt;-</span></span>
<span id="cb353-2"><a href="supplemental-null-set-exploratory-statistics.html#cb353-2" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-3"><a href="supplemental-null-set-exploratory-statistics.html#cb353-3" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-4"><a href="supplemental-null-set-exploratory-statistics.html#cb353-4" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-5"><a href="supplemental-null-set-exploratory-statistics.html#cb353-5" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-6"><a href="supplemental-null-set-exploratory-statistics.html#cb353-6" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-7"><a href="supplemental-null-set-exploratory-statistics.html#cb353-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-8"><a href="supplemental-null-set-exploratory-statistics.html#cb353-8" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-9"><a href="supplemental-null-set-exploratory-statistics.html#cb353-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-10"><a href="supplemental-null-set-exploratory-statistics.html#cb353-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-11"><a href="supplemental-null-set-exploratory-statistics.html#cb353-11" tabindex="-1"></a></span>
<span id="cb353-12"><a href="supplemental-null-set-exploratory-statistics.html#cb353-12" tabindex="-1"></a>t_virus_cc <span class="ot">&lt;-</span></span>
<span id="cb353-13"><a href="supplemental-null-set-exploratory-statistics.html#cb353-13" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-14"><a href="supplemental-null-set-exploratory-statistics.html#cb353-14" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-15"><a href="supplemental-null-set-exploratory-statistics.html#cb353-15" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-16"><a href="supplemental-null-set-exploratory-statistics.html#cb353-16" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-17"><a href="supplemental-null-set-exploratory-statistics.html#cb353-17" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-18"><a href="supplemental-null-set-exploratory-statistics.html#cb353-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-19"><a href="supplemental-null-set-exploratory-statistics.html#cb353-19" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-20"><a href="supplemental-null-set-exploratory-statistics.html#cb353-20" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-21"><a href="supplemental-null-set-exploratory-statistics.html#cb353-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-22"><a href="supplemental-null-set-exploratory-statistics.html#cb353-22" tabindex="-1"></a></span>
<span id="cb353-23"><a href="supplemental-null-set-exploratory-statistics.html#cb353-23" tabindex="-1"></a>t_virus_mf <span class="ot">&lt;-</span></span>
<span id="cb353-24"><a href="supplemental-null-set-exploratory-statistics.html#cb353-24" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-25"><a href="supplemental-null-set-exploratory-statistics.html#cb353-25" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-26"><a href="supplemental-null-set-exploratory-statistics.html#cb353-26" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-27"><a href="supplemental-null-set-exploratory-statistics.html#cb353-27" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-28"><a href="supplemental-null-set-exploratory-statistics.html#cb353-28" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-29"><a href="supplemental-null-set-exploratory-statistics.html#cb353-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-30"><a href="supplemental-null-set-exploratory-statistics.html#cb353-30" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-31"><a href="supplemental-null-set-exploratory-statistics.html#cb353-31" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-32"><a href="supplemental-null-set-exploratory-statistics.html#cb353-32" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-33"><a href="supplemental-null-set-exploratory-statistics.html#cb353-33" tabindex="-1"></a></span>
<span id="cb353-34"><a href="supplemental-null-set-exploratory-statistics.html#cb353-34" tabindex="-1"></a>t_acute_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-35"><a href="supplemental-null-set-exploratory-statistics.html#cb353-35" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_bp_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-36"><a href="supplemental-null-set-exploratory-statistics.html#cb353-36" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-37"><a href="supplemental-null-set-exploratory-statistics.html#cb353-37" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-38"><a href="supplemental-null-set-exploratory-statistics.html#cb353-38" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-39"><a href="supplemental-null-set-exploratory-statistics.html#cb353-39" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-40"><a href="supplemental-null-set-exploratory-statistics.html#cb353-40" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-41"><a href="supplemental-null-set-exploratory-statistics.html#cb353-41" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-42"><a href="supplemental-null-set-exploratory-statistics.html#cb353-42" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-43"><a href="supplemental-null-set-exploratory-statistics.html#cb353-43" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-44"><a href="supplemental-null-set-exploratory-statistics.html#cb353-44" tabindex="-1"></a></span>
<span id="cb353-45"><a href="supplemental-null-set-exploratory-statistics.html#cb353-45" tabindex="-1"></a>t_acute_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-46"><a href="supplemental-null-set-exploratory-statistics.html#cb353-46" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_cc_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-47"><a href="supplemental-null-set-exploratory-statistics.html#cb353-47" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-48"><a href="supplemental-null-set-exploratory-statistics.html#cb353-48" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-49"><a href="supplemental-null-set-exploratory-statistics.html#cb353-49" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-50"><a href="supplemental-null-set-exploratory-statistics.html#cb353-50" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-51"><a href="supplemental-null-set-exploratory-statistics.html#cb353-51" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-52"><a href="supplemental-null-set-exploratory-statistics.html#cb353-52" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-53"><a href="supplemental-null-set-exploratory-statistics.html#cb353-53" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-54"><a href="supplemental-null-set-exploratory-statistics.html#cb353-54" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-55"><a href="supplemental-null-set-exploratory-statistics.html#cb353-55" tabindex="-1"></a></span>
<span id="cb353-56"><a href="supplemental-null-set-exploratory-statistics.html#cb353-56" tabindex="-1"></a>t_acute_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-57"><a href="supplemental-null-set-exploratory-statistics.html#cb353-57" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_mf_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-58"><a href="supplemental-null-set-exploratory-statistics.html#cb353-58" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-59"><a href="supplemental-null-set-exploratory-statistics.html#cb353-59" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-60"><a href="supplemental-null-set-exploratory-statistics.html#cb353-60" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-61"><a href="supplemental-null-set-exploratory-statistics.html#cb353-61" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-62"><a href="supplemental-null-set-exploratory-statistics.html#cb353-62" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-63"><a href="supplemental-null-set-exploratory-statistics.html#cb353-63" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-64"><a href="supplemental-null-set-exploratory-statistics.html#cb353-64" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-65"><a href="supplemental-null-set-exploratory-statistics.html#cb353-65" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-66"><a href="supplemental-null-set-exploratory-statistics.html#cb353-66" tabindex="-1"></a></span>
<span id="cb353-67"><a href="supplemental-null-set-exploratory-statistics.html#cb353-67" tabindex="-1"></a>t_peak_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-68"><a href="supplemental-null-set-exploratory-statistics.html#cb353-68" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_bp_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-69"><a href="supplemental-null-set-exploratory-statistics.html#cb353-69" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-70"><a href="supplemental-null-set-exploratory-statistics.html#cb353-70" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-71"><a href="supplemental-null-set-exploratory-statistics.html#cb353-71" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-72"><a href="supplemental-null-set-exploratory-statistics.html#cb353-72" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-73"><a href="supplemental-null-set-exploratory-statistics.html#cb353-73" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-74"><a href="supplemental-null-set-exploratory-statistics.html#cb353-74" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-75"><a href="supplemental-null-set-exploratory-statistics.html#cb353-75" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-76"><a href="supplemental-null-set-exploratory-statistics.html#cb353-76" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-77"><a href="supplemental-null-set-exploratory-statistics.html#cb353-77" tabindex="-1"></a></span>
<span id="cb353-78"><a href="supplemental-null-set-exploratory-statistics.html#cb353-78" tabindex="-1"></a>t_peak_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-79"><a href="supplemental-null-set-exploratory-statistics.html#cb353-79" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_cc_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-80"><a href="supplemental-null-set-exploratory-statistics.html#cb353-80" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-81"><a href="supplemental-null-set-exploratory-statistics.html#cb353-81" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-82"><a href="supplemental-null-set-exploratory-statistics.html#cb353-82" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-83"><a href="supplemental-null-set-exploratory-statistics.html#cb353-83" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-84"><a href="supplemental-null-set-exploratory-statistics.html#cb353-84" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-85"><a href="supplemental-null-set-exploratory-statistics.html#cb353-85" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-86"><a href="supplemental-null-set-exploratory-statistics.html#cb353-86" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-87"><a href="supplemental-null-set-exploratory-statistics.html#cb353-87" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-88"><a href="supplemental-null-set-exploratory-statistics.html#cb353-88" tabindex="-1"></a></span>
<span id="cb353-89"><a href="supplemental-null-set-exploratory-statistics.html#cb353-89" tabindex="-1"></a>t_peak_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-90"><a href="supplemental-null-set-exploratory-statistics.html#cb353-90" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_mf_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-91"><a href="supplemental-null-set-exploratory-statistics.html#cb353-91" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-92"><a href="supplemental-null-set-exploratory-statistics.html#cb353-92" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-93"><a href="supplemental-null-set-exploratory-statistics.html#cb353-93" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-94"><a href="supplemental-null-set-exploratory-statistics.html#cb353-94" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-95"><a href="supplemental-null-set-exploratory-statistics.html#cb353-95" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-96"><a href="supplemental-null-set-exploratory-statistics.html#cb353-96" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-97"><a href="supplemental-null-set-exploratory-statistics.html#cb353-97" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-98"><a href="supplemental-null-set-exploratory-statistics.html#cb353-98" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-99"><a href="supplemental-null-set-exploratory-statistics.html#cb353-99" tabindex="-1"></a></span>
<span id="cb353-100"><a href="supplemental-null-set-exploratory-statistics.html#cb353-100" tabindex="-1"></a>t_late_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-101"><a href="supplemental-null-set-exploratory-statistics.html#cb353-101" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_bp_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-102"><a href="supplemental-null-set-exploratory-statistics.html#cb353-102" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-103"><a href="supplemental-null-set-exploratory-statistics.html#cb353-103" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-104"><a href="supplemental-null-set-exploratory-statistics.html#cb353-104" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-105"><a href="supplemental-null-set-exploratory-statistics.html#cb353-105" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-106"><a href="supplemental-null-set-exploratory-statistics.html#cb353-106" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-107"><a href="supplemental-null-set-exploratory-statistics.html#cb353-107" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-108"><a href="supplemental-null-set-exploratory-statistics.html#cb353-108" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-109"><a href="supplemental-null-set-exploratory-statistics.html#cb353-109" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-110"><a href="supplemental-null-set-exploratory-statistics.html#cb353-110" tabindex="-1"></a></span>
<span id="cb353-111"><a href="supplemental-null-set-exploratory-statistics.html#cb353-111" tabindex="-1"></a>t_late_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-112"><a href="supplemental-null-set-exploratory-statistics.html#cb353-112" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_cc_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-113"><a href="supplemental-null-set-exploratory-statistics.html#cb353-113" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-114"><a href="supplemental-null-set-exploratory-statistics.html#cb353-114" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-115"><a href="supplemental-null-set-exploratory-statistics.html#cb353-115" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-116"><a href="supplemental-null-set-exploratory-statistics.html#cb353-116" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-117"><a href="supplemental-null-set-exploratory-statistics.html#cb353-117" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-118"><a href="supplemental-null-set-exploratory-statistics.html#cb353-118" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-119"><a href="supplemental-null-set-exploratory-statistics.html#cb353-119" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-120"><a href="supplemental-null-set-exploratory-statistics.html#cb353-120" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-121"><a href="supplemental-null-set-exploratory-statistics.html#cb353-121" tabindex="-1"></a></span>
<span id="cb353-122"><a href="supplemental-null-set-exploratory-statistics.html#cb353-122" tabindex="-1"></a>t_late_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-123"><a href="supplemental-null-set-exploratory-statistics.html#cb353-123" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_GO_mf_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-124"><a href="supplemental-null-set-exploratory-statistics.html#cb353-124" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-125"><a href="supplemental-null-set-exploratory-statistics.html#cb353-125" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-126"><a href="supplemental-null-set-exploratory-statistics.html#cb353-126" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-127"><a href="supplemental-null-set-exploratory-statistics.html#cb353-127" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-128"><a href="supplemental-null-set-exploratory-statistics.html#cb353-128" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb353-129"><a href="supplemental-null-set-exploratory-statistics.html#cb353-129" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-130"><a href="supplemental-null-set-exploratory-statistics.html#cb353-130" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-131"><a href="supplemental-null-set-exploratory-statistics.html#cb353-131" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-132"><a href="supplemental-null-set-exploratory-statistics.html#cb353-132" tabindex="-1"></a></span>
<span id="cb353-133"><a href="supplemental-null-set-exploratory-statistics.html#cb353-133" tabindex="-1"></a></span>
<span id="cb353-134"><a href="supplemental-null-set-exploratory-statistics.html#cb353-134" tabindex="-1"></a>g_virus_bp <span class="ot">&lt;-</span></span>
<span id="cb353-135"><a href="supplemental-null-set-exploratory-statistics.html#cb353-135" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-136"><a href="supplemental-null-set-exploratory-statistics.html#cb353-136" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-137"><a href="supplemental-null-set-exploratory-statistics.html#cb353-137" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-138"><a href="supplemental-null-set-exploratory-statistics.html#cb353-138" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-139"><a href="supplemental-null-set-exploratory-statistics.html#cb353-139" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-140"><a href="supplemental-null-set-exploratory-statistics.html#cb353-140" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-141"><a href="supplemental-null-set-exploratory-statistics.html#cb353-141" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-142"><a href="supplemental-null-set-exploratory-statistics.html#cb353-142" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-143"><a href="supplemental-null-set-exploratory-statistics.html#cb353-143" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-144"><a href="supplemental-null-set-exploratory-statistics.html#cb353-144" tabindex="-1"></a></span>
<span id="cb353-145"><a href="supplemental-null-set-exploratory-statistics.html#cb353-145" tabindex="-1"></a>g_virus_cc <span class="ot">&lt;-</span></span>
<span id="cb353-146"><a href="supplemental-null-set-exploratory-statistics.html#cb353-146" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-147"><a href="supplemental-null-set-exploratory-statistics.html#cb353-147" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-148"><a href="supplemental-null-set-exploratory-statistics.html#cb353-148" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-149"><a href="supplemental-null-set-exploratory-statistics.html#cb353-149" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-150"><a href="supplemental-null-set-exploratory-statistics.html#cb353-150" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-151"><a href="supplemental-null-set-exploratory-statistics.html#cb353-151" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-152"><a href="supplemental-null-set-exploratory-statistics.html#cb353-152" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-153"><a href="supplemental-null-set-exploratory-statistics.html#cb353-153" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-154"><a href="supplemental-null-set-exploratory-statistics.html#cb353-154" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-155"><a href="supplemental-null-set-exploratory-statistics.html#cb353-155" tabindex="-1"></a></span>
<span id="cb353-156"><a href="supplemental-null-set-exploratory-statistics.html#cb353-156" tabindex="-1"></a>g_virus_mf <span class="ot">&lt;-</span></span>
<span id="cb353-157"><a href="supplemental-null-set-exploratory-statistics.html#cb353-157" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-158"><a href="supplemental-null-set-exploratory-statistics.html#cb353-158" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-159"><a href="supplemental-null-set-exploratory-statistics.html#cb353-159" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-160"><a href="supplemental-null-set-exploratory-statistics.html#cb353-160" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-161"><a href="supplemental-null-set-exploratory-statistics.html#cb353-161" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-162"><a href="supplemental-null-set-exploratory-statistics.html#cb353-162" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-163"><a href="supplemental-null-set-exploratory-statistics.html#cb353-163" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>,</span>
<span id="cb353-164"><a href="supplemental-null-set-exploratory-statistics.html#cb353-164" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-165"><a href="supplemental-null-set-exploratory-statistics.html#cb353-165" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-166"><a href="supplemental-null-set-exploratory-statistics.html#cb353-166" tabindex="-1"></a></span>
<span id="cb353-167"><a href="supplemental-null-set-exploratory-statistics.html#cb353-167" tabindex="-1"></a>g_acute_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-168"><a href="supplemental-null-set-exploratory-statistics.html#cb353-168" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_bp_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-169"><a href="supplemental-null-set-exploratory-statistics.html#cb353-169" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-170"><a href="supplemental-null-set-exploratory-statistics.html#cb353-170" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-171"><a href="supplemental-null-set-exploratory-statistics.html#cb353-171" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-172"><a href="supplemental-null-set-exploratory-statistics.html#cb353-172" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-173"><a href="supplemental-null-set-exploratory-statistics.html#cb353-173" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-174"><a href="supplemental-null-set-exploratory-statistics.html#cb353-174" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-175"><a href="supplemental-null-set-exploratory-statistics.html#cb353-175" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-176"><a href="supplemental-null-set-exploratory-statistics.html#cb353-176" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-177"><a href="supplemental-null-set-exploratory-statistics.html#cb353-177" tabindex="-1"></a></span>
<span id="cb353-178"><a href="supplemental-null-set-exploratory-statistics.html#cb353-178" tabindex="-1"></a>g_acute_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-179"><a href="supplemental-null-set-exploratory-statistics.html#cb353-179" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_cc_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-180"><a href="supplemental-null-set-exploratory-statistics.html#cb353-180" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-181"><a href="supplemental-null-set-exploratory-statistics.html#cb353-181" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-182"><a href="supplemental-null-set-exploratory-statistics.html#cb353-182" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-183"><a href="supplemental-null-set-exploratory-statistics.html#cb353-183" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-184"><a href="supplemental-null-set-exploratory-statistics.html#cb353-184" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-185"><a href="supplemental-null-set-exploratory-statistics.html#cb353-185" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-186"><a href="supplemental-null-set-exploratory-statistics.html#cb353-186" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-187"><a href="supplemental-null-set-exploratory-statistics.html#cb353-187" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-188"><a href="supplemental-null-set-exploratory-statistics.html#cb353-188" tabindex="-1"></a></span>
<span id="cb353-189"><a href="supplemental-null-set-exploratory-statistics.html#cb353-189" tabindex="-1"></a>g_acute_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-190"><a href="supplemental-null-set-exploratory-statistics.html#cb353-190" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_mf_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-191"><a href="supplemental-null-set-exploratory-statistics.html#cb353-191" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-192"><a href="supplemental-null-set-exploratory-statistics.html#cb353-192" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-193"><a href="supplemental-null-set-exploratory-statistics.html#cb353-193" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-194"><a href="supplemental-null-set-exploratory-statistics.html#cb353-194" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-195"><a href="supplemental-null-set-exploratory-statistics.html#cb353-195" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-196"><a href="supplemental-null-set-exploratory-statistics.html#cb353-196" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>,</span>
<span id="cb353-197"><a href="supplemental-null-set-exploratory-statistics.html#cb353-197" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-198"><a href="supplemental-null-set-exploratory-statistics.html#cb353-198" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-199"><a href="supplemental-null-set-exploratory-statistics.html#cb353-199" tabindex="-1"></a></span>
<span id="cb353-200"><a href="supplemental-null-set-exploratory-statistics.html#cb353-200" tabindex="-1"></a>g_peak_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-201"><a href="supplemental-null-set-exploratory-statistics.html#cb353-201" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_bp_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-202"><a href="supplemental-null-set-exploratory-statistics.html#cb353-202" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-203"><a href="supplemental-null-set-exploratory-statistics.html#cb353-203" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-204"><a href="supplemental-null-set-exploratory-statistics.html#cb353-204" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-205"><a href="supplemental-null-set-exploratory-statistics.html#cb353-205" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-206"><a href="supplemental-null-set-exploratory-statistics.html#cb353-206" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-207"><a href="supplemental-null-set-exploratory-statistics.html#cb353-207" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-208"><a href="supplemental-null-set-exploratory-statistics.html#cb353-208" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-209"><a href="supplemental-null-set-exploratory-statistics.html#cb353-209" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-210"><a href="supplemental-null-set-exploratory-statistics.html#cb353-210" tabindex="-1"></a></span>
<span id="cb353-211"><a href="supplemental-null-set-exploratory-statistics.html#cb353-211" tabindex="-1"></a>g_peak_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-212"><a href="supplemental-null-set-exploratory-statistics.html#cb353-212" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_cc_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-213"><a href="supplemental-null-set-exploratory-statistics.html#cb353-213" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-214"><a href="supplemental-null-set-exploratory-statistics.html#cb353-214" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-215"><a href="supplemental-null-set-exploratory-statistics.html#cb353-215" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-216"><a href="supplemental-null-set-exploratory-statistics.html#cb353-216" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-217"><a href="supplemental-null-set-exploratory-statistics.html#cb353-217" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-218"><a href="supplemental-null-set-exploratory-statistics.html#cb353-218" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-219"><a href="supplemental-null-set-exploratory-statistics.html#cb353-219" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-220"><a href="supplemental-null-set-exploratory-statistics.html#cb353-220" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-221"><a href="supplemental-null-set-exploratory-statistics.html#cb353-221" tabindex="-1"></a></span>
<span id="cb353-222"><a href="supplemental-null-set-exploratory-statistics.html#cb353-222" tabindex="-1"></a>g_peak_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-223"><a href="supplemental-null-set-exploratory-statistics.html#cb353-223" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_mf_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-224"><a href="supplemental-null-set-exploratory-statistics.html#cb353-224" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-225"><a href="supplemental-null-set-exploratory-statistics.html#cb353-225" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-226"><a href="supplemental-null-set-exploratory-statistics.html#cb353-226" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-227"><a href="supplemental-null-set-exploratory-statistics.html#cb353-227" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-228"><a href="supplemental-null-set-exploratory-statistics.html#cb353-228" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-229"><a href="supplemental-null-set-exploratory-statistics.html#cb353-229" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>,</span>
<span id="cb353-230"><a href="supplemental-null-set-exploratory-statistics.html#cb353-230" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-231"><a href="supplemental-null-set-exploratory-statistics.html#cb353-231" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-232"><a href="supplemental-null-set-exploratory-statistics.html#cb353-232" tabindex="-1"></a></span>
<span id="cb353-233"><a href="supplemental-null-set-exploratory-statistics.html#cb353-233" tabindex="-1"></a>g_late_bp <span class="ot">&lt;-</span> </span>
<span id="cb353-234"><a href="supplemental-null-set-exploratory-statistics.html#cb353-234" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_bp_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-235"><a href="supplemental-null-set-exploratory-statistics.html#cb353-235" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-236"><a href="supplemental-null-set-exploratory-statistics.html#cb353-236" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-237"><a href="supplemental-null-set-exploratory-statistics.html#cb353-237" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-238"><a href="supplemental-null-set-exploratory-statistics.html#cb353-238" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-239"><a href="supplemental-null-set-exploratory-statistics.html#cb353-239" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-240"><a href="supplemental-null-set-exploratory-statistics.html#cb353-240" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-241"><a href="supplemental-null-set-exploratory-statistics.html#cb353-241" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Biological Process&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-242"><a href="supplemental-null-set-exploratory-statistics.html#cb353-242" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-243"><a href="supplemental-null-set-exploratory-statistics.html#cb353-243" tabindex="-1"></a></span>
<span id="cb353-244"><a href="supplemental-null-set-exploratory-statistics.html#cb353-244" tabindex="-1"></a>g_late_cc <span class="ot">&lt;-</span> </span>
<span id="cb353-245"><a href="supplemental-null-set-exploratory-statistics.html#cb353-245" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_cc_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-246"><a href="supplemental-null-set-exploratory-statistics.html#cb353-246" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-247"><a href="supplemental-null-set-exploratory-statistics.html#cb353-247" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-248"><a href="supplemental-null-set-exploratory-statistics.html#cb353-248" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-249"><a href="supplemental-null-set-exploratory-statistics.html#cb353-249" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-250"><a href="supplemental-null-set-exploratory-statistics.html#cb353-250" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-251"><a href="supplemental-null-set-exploratory-statistics.html#cb353-251" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-252"><a href="supplemental-null-set-exploratory-statistics.html#cb353-252" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Cellular Component&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-253"><a href="supplemental-null-set-exploratory-statistics.html#cb353-253" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-254"><a href="supplemental-null-set-exploratory-statistics.html#cb353-254" tabindex="-1"></a></span>
<span id="cb353-255"><a href="supplemental-null-set-exploratory-statistics.html#cb353-255" tabindex="-1"></a>g_late_mf <span class="ot">&lt;-</span> </span>
<span id="cb353-256"><a href="supplemental-null-set-exploratory-statistics.html#cb353-256" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_GO_mf_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-257"><a href="supplemental-null-set-exploratory-statistics.html#cb353-257" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-258"><a href="supplemental-null-set-exploratory-statistics.html#cb353-258" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb353-259"><a href="supplemental-null-set-exploratory-statistics.html#cb353-259" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-260"><a href="supplemental-null-set-exploratory-statistics.html#cb353-260" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb353-261"><a href="supplemental-null-set-exploratory-statistics.html#cb353-261" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb353-262"><a href="supplemental-null-set-exploratory-statistics.html#cb353-262" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>,</span>
<span id="cb353-263"><a href="supplemental-null-set-exploratory-statistics.html#cb353-263" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;Molecular Function&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb353-264"><a href="supplemental-null-set-exploratory-statistics.html#cb353-264" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb353-265"><a href="supplemental-null-set-exploratory-statistics.html#cb353-265" tabindex="-1"></a></span>
<span id="cb353-266"><a href="supplemental-null-set-exploratory-statistics.html#cb353-266" tabindex="-1"></a></span>
<span id="cb353-267"><a href="supplemental-null-set-exploratory-statistics.html#cb353-267" tabindex="-1"></a>goterms <span class="ot">&lt;-</span> <span class="fu">rbind</span>(t_virus_bp, t_virus_cc, t_virus_mf,</span>
<span id="cb353-268"><a href="supplemental-null-set-exploratory-statistics.html#cb353-268" tabindex="-1"></a>                 t_acute_bp, t_acute_cc, t_acute_mf,</span>
<span id="cb353-269"><a href="supplemental-null-set-exploratory-statistics.html#cb353-269" tabindex="-1"></a>                 t_peak_bp, t_peak_cc, t_peak_mf, </span>
<span id="cb353-270"><a href="supplemental-null-set-exploratory-statistics.html#cb353-270" tabindex="-1"></a>                 t_late_bp, t_late_cc, t_late_mf,</span>
<span id="cb353-271"><a href="supplemental-null-set-exploratory-statistics.html#cb353-271" tabindex="-1"></a>                 g_virus_bp, g_virus_cc, g_virus_mf,</span>
<span id="cb353-272"><a href="supplemental-null-set-exploratory-statistics.html#cb353-272" tabindex="-1"></a>                 g_acute_bp, g_acute_cc, g_acute_mf,</span>
<span id="cb353-273"><a href="supplemental-null-set-exploratory-statistics.html#cb353-273" tabindex="-1"></a>                 g_peak_bp, g_peak_cc, g_peak_mf, </span>
<span id="cb353-274"><a href="supplemental-null-set-exploratory-statistics.html#cb353-274" tabindex="-1"></a>                 g_late_bp, g_late_cc, g_late_mf) <span class="sc">%&gt;%</span> </span>
<span id="cb353-275"><a href="supplemental-null-set-exploratory-statistics.html#cb353-275" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="fu">factor</span>(approach, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Transcriptome&quot;</span>, <span class="st">&quot;Genome&quot;</span>)),</span>
<span id="cb353-276"><a href="supplemental-null-set-exploratory-statistics.html#cb353-276" tabindex="-1"></a>         <span class="at">stage=</span><span class="fu">factor</span>(stage, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;V+ vs V-&quot;</span>, <span class="st">&quot;Acute&quot;</span>, <span class="st">&quot;Peak&quot;</span>, <span class="st">&quot;Late&quot;</span>)))</span></code></pre></div>
</div>
<div id="plot-3" class="section level3 hasAnchor" number="6.4.2">
<h3><span class="header-section-number">6.4.2</span> Plot<a href="supplemental-null-set-exploratory-statistics.html#plot-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="supplemental-null-set-exploratory-statistics.html#cb354-1" tabindex="-1"></a>null_go_plot <span class="ot">&lt;-</span></span>
<span id="cb354-2"><a href="supplemental-null-set-exploratory-statistics.html#cb354-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(goterms, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span><span class="fu">log</span>(terms))) <span class="sc">+</span></span>
<span id="cb354-3"><a href="supplemental-null-set-exploratory-statistics.html#cb354-3" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb354-4"><a href="supplemental-null-set-exploratory-statistics.html#cb354-4" tabindex="-1"></a>  <span class="fu">facet_nested_wrap</span>(<span class="fu">vars</span>(approach, go), <span class="at">dir=</span><span class="st">&quot;h&quot;</span>, </span>
<span id="cb354-5"><a href="supplemental-null-set-exploratory-statistics.html#cb354-5" tabindex="-1"></a>                    <span class="at">strip.position=</span><span class="st">&quot;top&quot;</span>, <span class="at">ncol=</span><span class="dv">3</span>, <span class="at">drop=</span><span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb354-6"><a href="supplemental-null-set-exploratory-statistics.html#cb354-6" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Influenza Stage&quot;</span>, <span class="at">y=</span><span class="st">&quot;log(# GO Terms) per Trial&quot;</span>) <span class="sc">+</span></span>
<span id="cb354-7"><a href="supplemental-null-set-exploratory-statistics.html#cb354-7" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb354-8"><a href="supplemental-null-set-exploratory-statistics.html#cb354-8" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb354-9"><a href="supplemental-null-set-exploratory-statistics.html#cb354-9" tabindex="-1"></a></span>
<span id="cb354-10"><a href="supplemental-null-set-exploratory-statistics.html#cb354-10" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/supplemental_TG_F_null_#gotermsplot.jpg&quot;</span>, null_go_plot, </span>
<span id="cb354-11"><a href="supplemental-null-set-exploratory-statistics.html#cb354-11" tabindex="-1"></a>       <span class="at">width=</span><span class="dv">5</span>, <span class="at">height=</span><span class="dv">7</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="number-of-kegg-pathways" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Number of KEGG Pathways<a href="supplemental-null-set-exploratory-statistics.html#number-of-kegg-pathways" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-15" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> Data<a href="supplemental-null-set-exploratory-statistics.html#data-15" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="supplemental-null-set-exploratory-statistics.html#cb355-1" tabindex="-1"></a>t_virus_kegg <span class="ot">&lt;-</span></span>
<span id="cb355-2"><a href="supplemental-null-set-exploratory-statistics.html#cb355-2" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-3"><a href="supplemental-null-set-exploratory-statistics.html#cb355-3" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-4"><a href="supplemental-null-set-exploratory-statistics.html#cb355-4" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-5"><a href="supplemental-null-set-exploratory-statistics.html#cb355-5" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-6"><a href="supplemental-null-set-exploratory-statistics.html#cb355-6" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-7"><a href="supplemental-null-set-exploratory-statistics.html#cb355-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb355-8"><a href="supplemental-null-set-exploratory-statistics.html#cb355-8" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-9"><a href="supplemental-null-set-exploratory-statistics.html#cb355-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-10"><a href="supplemental-null-set-exploratory-statistics.html#cb355-10" tabindex="-1"></a></span>
<span id="cb355-11"><a href="supplemental-null-set-exploratory-statistics.html#cb355-11" tabindex="-1"></a>t_acute_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-12"><a href="supplemental-null-set-exploratory-statistics.html#cb355-12" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_kegg_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-13"><a href="supplemental-null-set-exploratory-statistics.html#cb355-13" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-14"><a href="supplemental-null-set-exploratory-statistics.html#cb355-14" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-15"><a href="supplemental-null-set-exploratory-statistics.html#cb355-15" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-16"><a href="supplemental-null-set-exploratory-statistics.html#cb355-16" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-17"><a href="supplemental-null-set-exploratory-statistics.html#cb355-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb355-18"><a href="supplemental-null-set-exploratory-statistics.html#cb355-18" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-19"><a href="supplemental-null-set-exploratory-statistics.html#cb355-19" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-20"><a href="supplemental-null-set-exploratory-statistics.html#cb355-20" tabindex="-1"></a></span>
<span id="cb355-21"><a href="supplemental-null-set-exploratory-statistics.html#cb355-21" tabindex="-1"></a>t_peak_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-22"><a href="supplemental-null-set-exploratory-statistics.html#cb355-22" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_kegg_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-23"><a href="supplemental-null-set-exploratory-statistics.html#cb355-23" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-24"><a href="supplemental-null-set-exploratory-statistics.html#cb355-24" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-25"><a href="supplemental-null-set-exploratory-statistics.html#cb355-25" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-26"><a href="supplemental-null-set-exploratory-statistics.html#cb355-26" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-27"><a href="supplemental-null-set-exploratory-statistics.html#cb355-27" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb355-28"><a href="supplemental-null-set-exploratory-statistics.html#cb355-28" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-29"><a href="supplemental-null-set-exploratory-statistics.html#cb355-29" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-30"><a href="supplemental-null-set-exploratory-statistics.html#cb355-30" tabindex="-1"></a></span>
<span id="cb355-31"><a href="supplemental-null-set-exploratory-statistics.html#cb355-31" tabindex="-1"></a>t_late_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-32"><a href="supplemental-null-set-exploratory-statistics.html#cb355-32" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_null_kegg_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-33"><a href="supplemental-null-set-exploratory-statistics.html#cb355-33" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-34"><a href="supplemental-null-set-exploratory-statistics.html#cb355-34" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-35"><a href="supplemental-null-set-exploratory-statistics.html#cb355-35" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-36"><a href="supplemental-null-set-exploratory-statistics.html#cb355-36" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-37"><a href="supplemental-null-set-exploratory-statistics.html#cb355-37" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>, </span>
<span id="cb355-38"><a href="supplemental-null-set-exploratory-statistics.html#cb355-38" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-39"><a href="supplemental-null-set-exploratory-statistics.html#cb355-39" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-40"><a href="supplemental-null-set-exploratory-statistics.html#cb355-40" tabindex="-1"></a></span>
<span id="cb355-41"><a href="supplemental-null-set-exploratory-statistics.html#cb355-41" tabindex="-1"></a></span>
<span id="cb355-42"><a href="supplemental-null-set-exploratory-statistics.html#cb355-42" tabindex="-1"></a>g_virus_kegg <span class="ot">&lt;-</span></span>
<span id="cb355-43"><a href="supplemental-null-set-exploratory-statistics.html#cb355-43" tabindex="-1"></a>   <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-44"><a href="supplemental-null-set-exploratory-statistics.html#cb355-44" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-45"><a href="supplemental-null-set-exploratory-statistics.html#cb355-45" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-46"><a href="supplemental-null-set-exploratory-statistics.html#cb355-46" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-47"><a href="supplemental-null-set-exploratory-statistics.html#cb355-47" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-48"><a href="supplemental-null-set-exploratory-statistics.html#cb355-48" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb355-49"><a href="supplemental-null-set-exploratory-statistics.html#cb355-49" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;V+ vs V-&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-50"><a href="supplemental-null-set-exploratory-statistics.html#cb355-50" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-51"><a href="supplemental-null-set-exploratory-statistics.html#cb355-51" tabindex="-1"></a></span>
<span id="cb355-52"><a href="supplemental-null-set-exploratory-statistics.html#cb355-52" tabindex="-1"></a>g_acute_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-53"><a href="supplemental-null-set-exploratory-statistics.html#cb355-53" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_kegg_acute.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-54"><a href="supplemental-null-set-exploratory-statistics.html#cb355-54" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-55"><a href="supplemental-null-set-exploratory-statistics.html#cb355-55" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-56"><a href="supplemental-null-set-exploratory-statistics.html#cb355-56" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-57"><a href="supplemental-null-set-exploratory-statistics.html#cb355-57" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-58"><a href="supplemental-null-set-exploratory-statistics.html#cb355-58" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb355-59"><a href="supplemental-null-set-exploratory-statistics.html#cb355-59" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Acute&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-60"><a href="supplemental-null-set-exploratory-statistics.html#cb355-60" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-61"><a href="supplemental-null-set-exploratory-statistics.html#cb355-61" tabindex="-1"></a></span>
<span id="cb355-62"><a href="supplemental-null-set-exploratory-statistics.html#cb355-62" tabindex="-1"></a>g_peak_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-63"><a href="supplemental-null-set-exploratory-statistics.html#cb355-63" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_kegg_peak.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-64"><a href="supplemental-null-set-exploratory-statistics.html#cb355-64" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-65"><a href="supplemental-null-set-exploratory-statistics.html#cb355-65" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-66"><a href="supplemental-null-set-exploratory-statistics.html#cb355-66" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-67"><a href="supplemental-null-set-exploratory-statistics.html#cb355-67" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-68"><a href="supplemental-null-set-exploratory-statistics.html#cb355-68" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb355-69"><a href="supplemental-null-set-exploratory-statistics.html#cb355-69" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Peak&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-70"><a href="supplemental-null-set-exploratory-statistics.html#cb355-70" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-71"><a href="supplemental-null-set-exploratory-statistics.html#cb355-71" tabindex="-1"></a></span>
<span id="cb355-72"><a href="supplemental-null-set-exploratory-statistics.html#cb355-72" tabindex="-1"></a>g_late_kegg <span class="ot">&lt;-</span> </span>
<span id="cb355-73"><a href="supplemental-null-set-exploratory-statistics.html#cb355-73" tabindex="-1"></a>  <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_null_kegg_late.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-74"><a href="supplemental-null-set-exploratory-statistics.html#cb355-74" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-75"><a href="supplemental-null-set-exploratory-statistics.html#cb355-75" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)]) <span class="sc">%&gt;%</span> </span>
<span id="cb355-76"><a href="supplemental-null-set-exploratory-statistics.html#cb355-76" tabindex="-1"></a>  <span class="fu">lengths</span>(., <span class="at">use.names=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-77"><a href="supplemental-null-set-exploratory-statistics.html#cb355-77" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb355-78"><a href="supplemental-null-set-exploratory-statistics.html#cb355-78" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Genome&quot;</span>, </span>
<span id="cb355-79"><a href="supplemental-null-set-exploratory-statistics.html#cb355-79" tabindex="-1"></a>         <span class="at">stage=</span><span class="st">&quot;Late&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb355-80"><a href="supplemental-null-set-exploratory-statistics.html#cb355-80" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;terms&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>)</span>
<span id="cb355-81"><a href="supplemental-null-set-exploratory-statistics.html#cb355-81" tabindex="-1"></a></span>
<span id="cb355-82"><a href="supplemental-null-set-exploratory-statistics.html#cb355-82" tabindex="-1"></a></span>
<span id="cb355-83"><a href="supplemental-null-set-exploratory-statistics.html#cb355-83" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">rbind</span>(t_virus_kegg,</span>
<span id="cb355-84"><a href="supplemental-null-set-exploratory-statistics.html#cb355-84" tabindex="-1"></a>              t_acute_kegg, </span>
<span id="cb355-85"><a href="supplemental-null-set-exploratory-statistics.html#cb355-85" tabindex="-1"></a>              t_peak_kegg, </span>
<span id="cb355-86"><a href="supplemental-null-set-exploratory-statistics.html#cb355-86" tabindex="-1"></a>              t_late_kegg,</span>
<span id="cb355-87"><a href="supplemental-null-set-exploratory-statistics.html#cb355-87" tabindex="-1"></a>              g_virus_kegg,</span>
<span id="cb355-88"><a href="supplemental-null-set-exploratory-statistics.html#cb355-88" tabindex="-1"></a>              g_acute_kegg, </span>
<span id="cb355-89"><a href="supplemental-null-set-exploratory-statistics.html#cb355-89" tabindex="-1"></a>              g_peak_kegg,</span>
<span id="cb355-90"><a href="supplemental-null-set-exploratory-statistics.html#cb355-90" tabindex="-1"></a>              g_late_kegg) <span class="sc">%&gt;%</span> </span>
<span id="cb355-91"><a href="supplemental-null-set-exploratory-statistics.html#cb355-91" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="fu">factor</span>(approach, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Transcriptome&quot;</span>, <span class="st">&quot;Genome&quot;</span>)),</span>
<span id="cb355-92"><a href="supplemental-null-set-exploratory-statistics.html#cb355-92" tabindex="-1"></a>         <span class="at">stage=</span><span class="fu">factor</span>(stage, <span class="at">levels=</span><span class="fu">c</span>( <span class="st">&quot;V+ vs V-&quot;</span>, <span class="st">&quot;Acute&quot;</span>, <span class="st">&quot;Peak&quot;</span>, <span class="st">&quot;Late&quot;</span>)))</span></code></pre></div>
</div>
<div id="plot-4" class="section level3 hasAnchor" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> Plot<a href="supplemental-null-set-exploratory-statistics.html#plot-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="supplemental-null-set-exploratory-statistics.html#cb356-1" tabindex="-1"></a>null_kegg_plot <span class="ot">&lt;-</span></span>
<span id="cb356-2"><a href="supplemental-null-set-exploratory-statistics.html#cb356-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(kegg, <span class="fu">aes</span>(<span class="at">x=</span>stage, <span class="at">y=</span><span class="fu">log</span>(terms))) <span class="sc">+</span></span>
<span id="cb356-3"><a href="supplemental-null-set-exploratory-statistics.html#cb356-3" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb356-4"><a href="supplemental-null-set-exploratory-statistics.html#cb356-4" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>approach) <span class="sc">+</span></span>
<span id="cb356-5"><a href="supplemental-null-set-exploratory-statistics.html#cb356-5" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Influenza Stage&quot;</span>, <span class="at">y=</span><span class="st">&quot;log(# KEGG Pathways) per Trial&quot;</span>) <span class="sc">+</span></span>
<span id="cb356-6"><a href="supplemental-null-set-exploratory-statistics.html#cb356-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb356-7"><a href="supplemental-null-set-exploratory-statistics.html#cb356-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb356-8"><a href="supplemental-null-set-exploratory-statistics.html#cb356-8" tabindex="-1"></a></span>
<span id="cb356-9"><a href="supplemental-null-set-exploratory-statistics.html#cb356-9" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/supplemental_TG_F_null_#keggplot.jpg&quot;</span>, null_kegg_plot, </span>
<span id="cb356-10"><a href="supplemental-null-set-exploratory-statistics.html#cb356-10" tabindex="-1"></a>       <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="dv">3</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="influenza-stage-dge.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="influenza-dge-deseq2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": false
});
});
</script>

</body>

</html>
