<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Virus+ vs Virus- DGE | Untitled</title>
  <meta name="description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Virus+ vs Virus- DGE | Untitled" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Virus+ vs Virus- DGE | Untitled" />
  
  <meta name="twitter:description" content="McCosker Dissertation Chapter 1 Statistical Analyses" />
  

<meta name="author" content="Christina McCosker" />


<meta name="date" content="2025-01-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="male-v-female-dge.html"/>
<link rel="next" href="influenza-stage-dge.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Hg-RNA Stats</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Publication Information</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#authors"><i class="fa fa-check"></i><b>1.1</b> Authors</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#abstract"><i class="fa fa-check"></i><b>1.2</b> Abstract</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html"><i class="fa fa-check"></i><b>2</b> LOC Gene Identification</a>
<ul>
<li class="chapter" data-level="2.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#libraries"><i class="fa fa-check"></i><b>2.1</b> Libraries</a></li>
<li class="chapter" data-level="2.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#load-data"><i class="fa fa-check"></i><b>2.2</b> Load data</a></li>
<li class="chapter" data-level="2.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-gene-sequences"><i class="fa fa-check"></i><b>2.3</b> Extract gene sequences</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-loc-genes-from-gene-count-matrices"><i class="fa fa-check"></i><b>2.3.1</b> Extract LOC genes from gene count matrices</a></li>
<li class="chapter" data-level="2.3.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-ids-for-all-sequence-ids"><i class="fa fa-check"></i><b>2.3.2</b> Extract transcript IDs for all sequence IDs</a></li>
<li class="chapter" data-level="2.3.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-transcript-sequences-for-blast"><i class="fa fa-check"></i><b>2.3.3</b> Extract transcript sequences for blast</a></li>
<li class="chapter" data-level="2.3.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-sequences"><i class="fa fa-check"></i><b>2.3.4</b> Export sequences</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#blast-sequences"><i class="fa fa-check"></i><b>2.4</b> Blast sequences</a></li>
<li class="chapter" data-level="2.5" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#process-blastn-results"><i class="fa fa-check"></i><b>2.5</b> Process BLASTn Results</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#transcriptome"><i class="fa fa-check"></i><b>2.5.1</b> Transcriptome</a></li>
<li class="chapter" data-level="2.5.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#genome"><i class="fa fa-check"></i><b>2.5.2</b> Genome</a></li>
<li class="chapter" data-level="2.5.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-transcriptome-genome-blast-results"><i class="fa fa-check"></i><b>2.5.3</b> Combine Transcriptome &amp; Genome Blast Results</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#combine-with-prior-loc-id-master-list"><i class="fa fa-check"></i><b>2.6</b> Combine with prior LOC ID master list</a></li>
<li class="chapter" data-level="2.7" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#update-tidy-gene-count-matrices"><i class="fa fa-check"></i><b>2.7</b> Update &amp; tidy gene count matrices</a></li>
<li class="chapter" data-level="2.8" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#hemoglobin"><i class="fa fa-check"></i><b>2.8</b> Hemoglobin</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#extract-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.1</b> Extract hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.2" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#prop-of-total-gene-counts"><i class="fa fa-check"></i><b>2.8.2</b> Prop of total gene counts</a></li>
<li class="chapter" data-level="2.8.3" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#remove-hemoglobin-genes"><i class="fa fa-check"></i><b>2.8.3</b> Remove hemoglobin genes</a></li>
<li class="chapter" data-level="2.8.4" data-path="loc-gene-identification.html"><a href="loc-gene-identification.html#export-new-gene-count-matrices"><i class="fa fa-check"></i><b>2.8.4</b> Export new gene count matrices</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html"><i class="fa fa-check"></i><b>3</b> Male v Female DGE</a>
<ul>
<li class="chapter" data-level="3.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#transcriptome---edger"><i class="fa fa-check"></i><b>3.1</b> Transcriptome - edgeR</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-1"><i class="fa fa-check"></i><b>3.1.1</b> Libraries</a></li>
<li class="chapter" data-level="3.1.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data"><i class="fa fa-check"></i><b>3.1.2</b> Data</a></li>
<li class="chapter" data-level="3.1.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis"><i class="fa fa-check"></i><b>3.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.1.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots"><i class="fa fa-check"></i><b>3.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.1.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs"><i class="fa fa-check"></i><b>3.1.5</b> Test for DEGs</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#genome---edger"><i class="fa fa-check"></i><b>3.2</b> Genome - edgeR</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#libraries-2"><i class="fa fa-check"></i><b>3.2.1</b> Libraries</a></li>
<li class="chapter" data-level="3.2.2" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#data-1"><i class="fa fa-check"></i><b>3.2.2</b> Data</a></li>
<li class="chapter" data-level="3.2.3" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>3.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="3.2.4" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#exploratory-plots-1"><i class="fa fa-check"></i><b>3.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="3.2.5" data-path="male-v-female-dge.html"><a href="male-v-female-dge.html#test-for-degs-1"><i class="fa fa-check"></i><b>3.2.5</b> Test for DEGs</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html"><i class="fa fa-check"></i><b>4</b> Virus+ vs Virus- DGE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome---edger---females"><i class="fa fa-check"></i><b>4.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-3"><i class="fa fa-check"></i><b>4.1.1</b> Libraries</a></li>
<li class="chapter" data-level="4.1.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-2"><i class="fa fa-check"></i><b>4.1.2</b> Data</a></li>
<li class="chapter" data-level="4.1.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>4.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.1.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-2"><i class="fa fa-check"></i><b>4.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.1.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-2"><i class="fa fa-check"></i><b>4.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.1.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis"><i class="fa fa-check"></i><b>4.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.1.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis"><i class="fa fa-check"></i><b>4.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome---edger---females"><i class="fa fa-check"></i><b>4.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-4"><i class="fa fa-check"></i><b>4.2.1</b> Libraries</a></li>
<li class="chapter" data-level="4.2.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-3"><i class="fa fa-check"></i><b>4.2.2</b> Data</a></li>
<li class="chapter" data-level="4.2.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>4.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.2.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#exploratory-plots-3"><i class="fa fa-check"></i><b>4.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="4.2.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#test-for-degs-3"><i class="fa fa-check"></i><b>4.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="4.2.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-1"><i class="fa fa-check"></i><b>4.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.2.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-1"><i class="fa fa-check"></i><b>4.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-null-set---edger---females"><i class="fa fa-check"></i><b>4.3</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-5"><i class="fa fa-check"></i><b>4.3.1</b> Libraries</a></li>
<li class="chapter" data-level="4.3.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-4"><i class="fa fa-check"></i><b>4.3.2</b> Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-2"><i class="fa fa-check"></i><b>4.3.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.3.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-2"><i class="fa fa-check"></i><b>4.3.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-null-set---edger---females"><i class="fa fa-check"></i><b>4.4</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-6"><i class="fa fa-check"></i><b>4.4.1</b> Libraries</a></li>
<li class="chapter" data-level="4.4.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-5"><i class="fa fa-check"></i><b>4.4.2</b> Data</a></li>
<li class="chapter" data-level="4.4.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis-1"><i class="fa fa-check"></i><b>4.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="4.4.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-enrichment-analysis-3"><i class="fa fa-check"></i><b>4.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="4.4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-analysis-3"><i class="fa fa-check"></i><b>4.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#refine-gokegg"><i class="fa fa-check"></i><b>4.5</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#libraries-7"><i class="fa fa-check"></i><b>4.5.1</b> Libraries</a></li>
<li class="chapter" data-level="4.5.2" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#data-6"><i class="fa fa-check"></i><b>4.5.2</b> Data</a></li>
<li class="chapter" data-level="4.5.3" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#transcriptome-1"><i class="fa fa-check"></i><b>4.5.3</b> Transcriptome</a></li>
<li class="chapter" data-level="4.5.4" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#genome-1"><i class="fa fa-check"></i><b>4.5.4</b> Genome</a></li>
<li class="chapter" data-level="4.5.5" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#go-term-figure"><i class="fa fa-check"></i><b>4.5.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="4.5.6" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#immunne-go-term-genes"><i class="fa fa-check"></i><b>4.5.6</b> Immunne GO term Genes</a></li>
<li class="chapter" data-level="4.5.7" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#kegg-table---manuscript"><i class="fa fa-check"></i><b>4.5.7</b> KEGG Table - Manuscript</a></li>
<li class="chapter" data-level="4.5.8" data-path="virus-vs-virus--dge.html"><a href="virus-vs-virus--dge.html#supplemental-table"><i class="fa fa-check"></i><b>4.5.8</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html"><i class="fa fa-check"></i><b>5</b> Influenza Stage DGE</a>
<ul>
<li class="chapter" data-level="5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome---edger---females-1"><i class="fa fa-check"></i><b>5.1</b> Transcriptome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-8"><i class="fa fa-check"></i><b>5.1.1</b> Libraries</a></li>
<li class="chapter" data-level="5.1.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-7"><i class="fa fa-check"></i><b>5.1.2</b> Data</a></li>
<li class="chapter" data-level="5.1.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-4"><i class="fa fa-check"></i><b>5.1.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-4"><i class="fa fa-check"></i><b>5.1.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.1.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-4"><i class="fa fa-check"></i><b>5.1.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.1.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-4"><i class="fa fa-check"></i><b>5.1.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.1.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-4"><i class="fa fa-check"></i><b>5.1.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome---edger---females-1"><i class="fa fa-check"></i><b>5.2</b> Genome - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-9"><i class="fa fa-check"></i><b>5.2.1</b> Libraries</a></li>
<li class="chapter" data-level="5.2.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-8"><i class="fa fa-check"></i><b>5.2.2</b> Data</a></li>
<li class="chapter" data-level="5.2.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#differential-gene-expression-analysis-5"><i class="fa fa-check"></i><b>5.2.3</b> Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.2.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#exploratory-plots-5"><i class="fa fa-check"></i><b>5.2.4</b> Exploratory Plots</a></li>
<li class="chapter" data-level="5.2.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#test-for-degs-5"><i class="fa fa-check"></i><b>5.2.5</b> Test for DEGs</a></li>
<li class="chapter" data-level="5.2.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-5"><i class="fa fa-check"></i><b>5.2.6</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.2.7" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-5"><i class="fa fa-check"></i><b>5.2.7</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#comapre-t-g-approach---edger---females"><i class="fa fa-check"></i><b>5.3</b> Comapre T &amp; G Approach - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-10"><i class="fa fa-check"></i><b>5.3.1</b> Libraries</a></li>
<li class="chapter" data-level="5.3.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-9"><i class="fa fa-check"></i><b>5.3.2</b> Data</a></li>
<li class="chapter" data-level="5.3.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#number-of-up-down-reg-genes"><i class="fa fa-check"></i><b>5.3.3</b> Number of up &amp; down reg genes</a></li>
<li class="chapter" data-level="5.3.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#gene-identities"><i class="fa fa-check"></i><b>5.3.4</b> Gene identities</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.4</b> Transcriptome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-11"><i class="fa fa-check"></i><b>5.4.1</b> Libraries</a></li>
<li class="chapter" data-level="5.4.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-10"><i class="fa fa-check"></i><b>5.4.2</b> Data</a></li>
<li class="chapter" data-level="5.4.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-6"><i class="fa fa-check"></i><b>5.4.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.4.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-6"><i class="fa fa-check"></i><b>5.4.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-null-set---edger---females-1"><i class="fa fa-check"></i><b>5.5</b> Genome Null Set - edgeR - Females</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-12"><i class="fa fa-check"></i><b>5.5.1</b> Libraries</a></li>
<li class="chapter" data-level="5.5.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-11"><i class="fa fa-check"></i><b>5.5.2</b> Data</a></li>
<li class="chapter" data-level="5.5.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#loop-differential-gene-expression-analysis-3"><i class="fa fa-check"></i><b>5.5.3</b> Loop Differential Gene Expression Analysis</a></li>
<li class="chapter" data-level="5.5.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-enrichment-analysis-7"><i class="fa fa-check"></i><b>5.5.4</b> GO Enrichment Analysis</a></li>
<li class="chapter" data-level="5.5.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#kegg-analysis-7"><i class="fa fa-check"></i><b>5.5.5</b> KEGG Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#refine-gokegg-1"><i class="fa fa-check"></i><b>5.6</b> Refine GO/KEGG</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#libraries-13"><i class="fa fa-check"></i><b>5.6.1</b> Libraries</a></li>
<li class="chapter" data-level="5.6.2" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#data-12"><i class="fa fa-check"></i><b>5.6.2</b> Data</a></li>
<li class="chapter" data-level="5.6.3" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#transcriptome-2"><i class="fa fa-check"></i><b>5.6.3</b> Transcriptome</a></li>
<li class="chapter" data-level="5.6.4" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#genome-2"><i class="fa fa-check"></i><b>5.6.4</b> Genome</a></li>
<li class="chapter" data-level="5.6.5" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#go-term-figure-1"><i class="fa fa-check"></i><b>5.6.5</b> GO Term Figure</a></li>
<li class="chapter" data-level="5.6.6" data-path="influenza-stage-dge.html"><a href="influenza-stage-dge.html#supplemental-table-1"><i class="fa fa-check"></i><b>5.6.6</b> Supplemental Table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html"><i class="fa fa-check"></i><b>6</b> Supplemental Null Set Exploratory Statistics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#libaries"><i class="fa fa-check"></i><b>6.1</b> Libaries</a></li>
<li class="chapter" data-level="6.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-degs"><i class="fa fa-check"></i><b>6.2</b> Number of DEGs</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-13"><i class="fa fa-check"></i><b>6.2.1</b> Data</a></li>
<li class="chapter" data-level="6.2.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats"><i class="fa fa-check"></i><b>6.2.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.2.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-1"><i class="fa fa-check"></i><b>6.2.3</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#deg-frequency"><i class="fa fa-check"></i><b>6.3</b> DEG Frequency</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-14"><i class="fa fa-check"></i><b>6.3.1</b> Data</a></li>
<li class="chapter" data-level="6.3.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#quick-stats-1"><i class="fa fa-check"></i><b>6.3.2</b> Quick Stats</a></li>
<li class="chapter" data-level="6.3.3" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-2"><i class="fa fa-check"></i><b>6.3.3</b> Plot</a></li>
<li class="chapter" data-level="6.3.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#combine-gene-number-frequency-plots"><i class="fa fa-check"></i><b>6.3.4</b> Combine gene number &amp; frequency plots</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-go-terms"><i class="fa fa-check"></i><b>6.4</b> Number of GO terms</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data---import-configure"><i class="fa fa-check"></i><b>6.4.1</b> Data - Import &amp; Configure</a></li>
<li class="chapter" data-level="6.4.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-3"><i class="fa fa-check"></i><b>6.4.2</b> Plot</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#number-of-kegg-pathways"><i class="fa fa-check"></i><b>6.5</b> Number of KEGG Pathways</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#data-15"><i class="fa fa-check"></i><b>6.5.1</b> Data</a></li>
<li class="chapter" data-level="6.5.2" data-path="supplemental-null-set-exploratory-statistics.html"><a href="supplemental-null-set-exploratory-statistics.html#plot-4"><i class="fa fa-check"></i><b>6.5.2</b> Plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html"><i class="fa fa-check"></i><b>7</b> Influenza DGE DESeq2</a>
<ul>
<li class="chapter" data-level="7.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#transcriptome---deseq2---females"><i class="fa fa-check"></i><b>7.1</b> Transcriptome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-14"><i class="fa fa-check"></i><b>7.1.1</b> Libraries</a></li>
<li class="chapter" data-level="7.1.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-16"><i class="fa fa-check"></i><b>7.1.2</b> Data</a></li>
<li class="chapter" data-level="7.1.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna"><i class="fa fa-check"></i><b>7.1.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.1.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage"><i class="fa fa-check"></i><b>7.1.4</b> Influenza Stage</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#genome---deseq2---females"><i class="fa fa-check"></i><b>7.2</b> Genome - DESeq2 - Females</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#libraries-15"><i class="fa fa-check"></i><b>7.2.1</b> Libraries</a></li>
<li class="chapter" data-level="7.2.2" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#data-17"><i class="fa fa-check"></i><b>7.2.2</b> Data</a></li>
<li class="chapter" data-level="7.2.3" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#viral-rna-1"><i class="fa fa-check"></i><b>7.2.3</b> Viral RNA</a></li>
<li class="chapter" data-level="7.2.4" data-path="influenza-dge-deseq2.html"><a href="influenza-dge-deseq2.html#influenza-stage-1"><i class="fa fa-check"></i><b>7.2.4</b> Influenza Stage</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html"><i class="fa fa-check"></i><b>8</b> R Package Comparison - edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#libraries-16"><i class="fa fa-check"></i><b>8.1</b> Libraries</a></li>
<li class="chapter" data-level="8.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#data-18"><i class="fa fa-check"></i><b>8.2</b> Data</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#metadata"><i class="fa fa-check"></i><b>8.2.1</b> Metadata</a></li>
<li class="chapter" data-level="8.2.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#results-from-dge-testing"><i class="fa fa-check"></i><b>8.2.2</b> Results from DGE testing</a></li>
<li class="chapter" data-level="8.2.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#differentially-expressed-genes-only"><i class="fa fa-check"></i><b>8.2.3</b> Differentially expressed genes only</a></li>
<li class="chapter" data-level="8.2.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pre-filter-lists"><i class="fa fa-check"></i><b>8.2.4</b> Pre-filter lists</a></li>
<li class="chapter" data-level="8.2.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#normalized-gene-counts"><i class="fa fa-check"></i><b>8.2.5</b> Normalized gene counts</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#edger-vs-deseq2"><i class="fa fa-check"></i><b>8.3</b> edgeR vs DESeq2</a>
<ul>
<li class="chapter" data-level="8.3.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#number-of-pre-vs-post-filter-genes"><i class="fa fa-check"></i><b>8.3.1</b> Number of Pre vs Post Filter Genes</a></li>
<li class="chapter" data-level="8.3.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results"><i class="fa fa-check"></i><b>8.3.2</b> Merge results</a></li>
<li class="chapter" data-level="8.3.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value"><i class="fa fa-check"></i><b>8.3.3</b> p-value</a></li>
<li class="chapter" data-level="8.3.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc"><i class="fa fa-check"></i><b>8.3.4</b> log(FC)</a></li>
<li class="chapter" data-level="8.3.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression"><i class="fa fa-check"></i><b>8.3.5</b> log mean expression</a></li>
<li class="chapter" data-level="8.3.6" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#dge-explortory-stats"><i class="fa fa-check"></i><b>8.3.6</b> DGE explortory stats</a></li>
<li class="chapter" data-level="8.3.7" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#method-gene-ids"><i class="fa fa-check"></i><b>8.3.7</b> Method Gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#transcriptome-vs-genome"><i class="fa fa-check"></i><b>8.4</b> Transcriptome vs Genome</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#merge-results-1"><i class="fa fa-check"></i><b>8.4.1</b> Merge results</a></li>
<li class="chapter" data-level="8.4.2" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#p-value-1"><i class="fa fa-check"></i><b>8.4.2</b> p-value</a></li>
<li class="chapter" data-level="8.4.3" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#logfc-1"><i class="fa fa-check"></i><b>8.4.3</b> log(FC)</a></li>
<li class="chapter" data-level="8.4.4" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#log-mean-expression-1"><i class="fa fa-check"></i><b>8.4.4</b> log mean expression</a></li>
<li class="chapter" data-level="8.4.5" data-path="r-package-comparison---edger-vs-deseq2.html"><a href="r-package-comparison---edger-vs-deseq2.html#pca-plots-6"><i class="fa fa-check"></i><b>8.4.5</b> PCA plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html"><i class="fa fa-check"></i><b>9</b> IAV Kegg Pathway Candidate Gene Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#libraries-17"><i class="fa fa-check"></i><b>9.1</b> Libraries</a></li>
<li class="chapter" data-level="9.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#data-19"><i class="fa fa-check"></i><b>9.2</b> Data</a></li>
<li class="chapter" data-level="9.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#extract-genes-from-kegg-pathway"><i class="fa fa-check"></i><b>9.3</b> Extract genes from KEGG Pathway</a></li>
<li class="chapter" data-level="9.4" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#overlap-with-gene-lists"><i class="fa fa-check"></i><b>9.4</b> Overlap with Gene Lists</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#postfilter"><i class="fa fa-check"></i><b>9.4.1</b> Postfilter</a></li>
<li class="chapter" data-level="9.4.2" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#differentially-expressed-genes"><i class="fa fa-check"></i><b>9.4.2</b> Differentially Expressed Genes</a></li>
<li class="chapter" data-level="9.4.3" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#compile-gene-ids"><i class="fa fa-check"></i><b>9.4.3</b> Compile gene IDs</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#plot-pathway-map"><i class="fa fa-check"></i><b>9.5</b> Plot pathway map</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="iav-kegg-pathway-candidate-gene-analysis.html"><a href="iav-kegg-pathway-candidate-gene-analysis.html#create-legend"><i class="fa fa-check"></i><b>9.5.1</b> Create Legend</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="outlier-analysis.html"><a href="outlier-analysis.html"><i class="fa fa-check"></i><b>10</b> Outlier Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#libraries-18"><i class="fa fa-check"></i><b>10.1</b> Libraries</a></li>
<li class="chapter" data-level="10.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#data-20"><i class="fa fa-check"></i><b>10.2</b> Data</a></li>
<li class="chapter" data-level="10.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#pca-2"><i class="fa fa-check"></i><b>10.3</b> PCA</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="outlier-analysis.html"><a href="outlier-analysis.html#transcriptome-6"><i class="fa fa-check"></i><b>10.3.1</b> Transcriptome</a></li>
<li class="chapter" data-level="10.3.2" data-path="outlier-analysis.html"><a href="outlier-analysis.html#genome-6"><i class="fa fa-check"></i><b>10.3.2</b> Genome</a></li>
<li class="chapter" data-level="10.3.3" data-path="outlier-analysis.html"><a href="outlier-analysis.html#number-of-reads"><i class="fa fa-check"></i><b>10.3.3</b> Number of Reads</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html"><i class="fa fa-check"></i><b>11</b> Downsample Late Group to test # DEGs</a>
<ul>
<li class="chapter" data-level="11.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#libraries-19"><i class="fa fa-check"></i><b>11.1</b> Libraries</a></li>
<li class="chapter" data-level="11.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#data-21"><i class="fa fa-check"></i><b>11.2</b> Data</a></li>
<li class="chapter" data-level="11.3" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#transcriptome-7"><i class="fa fa-check"></i><b>11.3</b> Transcriptome</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop"><i class="fa fa-check"></i><b>11.3.1</b> Downsample &amp; Loop</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#genome-7"><i class="fa fa-check"></i><b>11.4</b> Genome</a>
<ul>
<li class="chapter" data-level="11.4.1" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#downsample-loop-1"><i class="fa fa-check"></i><b>11.4.1</b> Downsample &amp; Loop</a></li>
<li class="chapter" data-level="11.4.2" data-path="downsample-late-group-to-test-degs.html"><a href="downsample-late-group-to-test-degs.html#summarize-degs"><i class="fa fa-check"></i><b>11.4.2</b> Summarize # DEGs</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Untitled</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="virus-vs-virus--dge" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Virus+ vs Virus- DGE<a href="virus-vs-virus--dge.html#virus-vs-virus--dge" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="transcriptome---edger---females" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Transcriptome - edgeR - Females<a href="virus-vs-virus--dge.html#transcriptome---edger---females" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-3" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> Libraries<a href="virus-vs-virus--dge.html#libraries-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="virus-vs-virus--dge.html#cb77-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb77-2"><a href="virus-vs-virus--dge.html#cb77-2" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb77-3"><a href="virus-vs-virus--dge.html#cb77-3" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb77-4"><a href="virus-vs-virus--dge.html#cb77-4" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb77-5"><a href="virus-vs-virus--dge.html#cb77-5" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb77-6"><a href="virus-vs-virus--dge.html#cb77-6" tabindex="-1"></a><span class="fu">library</span>(WebGestaltR)</span></code></pre></div>
</div>
<div id="data-2" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> Data<a href="virus-vs-virus--dge.html#data-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="virus-vs-virus--dge.html#cb78-1" tabindex="-1"></a>SampleInfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/metadata.csv&quot;</span>, </span>
<span id="cb78-2"><a href="virus-vs-virus--dge.html#cb78-2" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb78-3"><a href="virus-vs-virus--dge.html#cb78-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(disease_stage<span class="sc">==</span><span class="st">&quot;acute&quot;</span> <span class="sc">|</span> disease_stage<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;virus+&quot;</span>, <span class="st">&quot;virus-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb78-4"><a href="virus-vs-virus--dge.html#cb78-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sex<span class="sc">==</span><span class="st">&quot;male&quot;</span>)</span>
<span id="cb78-5"><a href="virus-vs-virus--dge.html#cb78-5" tabindex="-1"></a>  </span>
<span id="cb78-6"><a href="virus-vs-virus--dge.html#cb78-6" tabindex="-1"></a><span class="co">#Raw RSEM Gene Counts</span></span>
<span id="cb78-7"><a href="virus-vs-virus--dge.html#cb78-7" tabindex="-1"></a>Raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;Output Files/txome_genecounts_locid_nohemo.csv&quot;</span>, </span>
<span id="cb78-8"><a href="virus-vs-virus--dge.html#cb78-8" tabindex="-1"></a>                <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>,</span>
<span id="cb78-9"><a href="virus-vs-virus--dge.html#cb78-9" tabindex="-1"></a>                <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb78-10"><a href="virus-vs-virus--dge.html#cb78-10" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb78-11"><a href="virus-vs-virus--dge.html#cb78-11" tabindex="-1"></a>  <span class="fu">select</span>(SampleInfo<span class="sc">$</span>sample)</span></code></pre></div>
</div>
<div id="differential-gene-expression-analysis-2" class="section level3 hasAnchor" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> Differential Gene Expression Analysis<a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="dgelist-2" class="section level4 hasAnchor" number="4.1.3.1">
<h4><span class="header-section-number">4.1.3.1</span> DGElist<a href="virus-vs-virus--dge.html#dgelist-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="virus-vs-virus--dge.html#cb79-1" tabindex="-1"></a><span class="co">#Create DGE Object with counts &amp; sample groups</span></span>
<span id="cb79-2"><a href="virus-vs-virus--dge.html#cb79-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> Raw, <span class="at">group =</span> SampleInfo<span class="sc">$</span>RNA)</span>
<span id="cb79-3"><a href="virus-vs-virus--dge.html#cb79-3" tabindex="-1"></a></span>
<span id="cb79-4"><a href="virus-vs-virus--dge.html#cb79-4" tabindex="-1"></a><span class="co">#Check DGE Object Samples, Groups, Lib Size, and Norm Factor</span></span>
<span id="cb79-5"><a href="virus-vs-virus--dge.html#cb79-5" tabindex="-1"></a>DGE<span class="sc">$</span>samples </span></code></pre></div>
<pre><code>##          group lib.size norm.factors
## Hg11513 virus+  2842670            1
## Hg1354  virus+  3618804            1
## Hg1403  virus+  6100522            1
## Hg1404  virus+  3533565            1
## Hg908b  virus+  5871501            1
## Hg1256  virus+  4573826            1
## Hg1265  virus+  2979315            1
## Hg1288  virus+  2823770            1
## Hg266   virus+  3543715            1
## Hg1261  virus-  1943267            1
## Hg1272  virus-  4708624            1
## Hg1380  virus-  3204059            1
## Hg1387  virus-  4158987            1
## Hg1397  virus-  4343361            1
## Hg1421  virus-  4508013            1
## Hg1528  virus-  2894695            1
## Hg861a  virus-  5407908            1
## Hg862a  virus-  4075756            1
## Hg934   virus-  3956989            1
## Hg1262  virus-  4653972            1
## Hg1273  virus-  5011031            1
## Hg1282  virus-  2951861            1
## Hg1283  virus-  4658064            1
## Hg1286  virus-  4284639            1
## Hg1351  virus-  1430600            1
## Hg1516  virus-  4688415            1
## Hg1527  virus-  3763080            1</code></pre>
</div>
<div id="filter-out-lowly-expressed-genes-2" class="section level4 hasAnchor" number="4.1.3.2">
<h4><span class="header-section-number">4.1.3.2</span> Filter out lowly expressed genes<a href="virus-vs-virus--dge.html#filter-out-lowly-expressed-genes-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="virus-vs-virus--dge.html#cb81-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(DGE)</span>
<span id="cb81-2"><a href="virus-vs-virus--dge.html#cb81-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> DGE[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb81-3"><a href="virus-vs-virus--dge.html#cb81-3" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">rownames</span>(DGE<span class="sc">$</span>counts))</span></code></pre></div>
<pre><code>## [1] 11387</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="virus-vs-virus--dge.html#cb83-1" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">rownames</span>(DGE<span class="sc">$</span>counts)</span>
<span id="cb83-2"><a href="virus-vs-virus--dge.html#cb83-2" tabindex="-1"></a><span class="fu">write.table</span>(postfilter, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_postfilter.txt&quot;</span>,</span>
<span id="cb83-3"><a href="virus-vs-virus--dge.html#cb83-3" tabindex="-1"></a>            <span class="at">row.name=</span><span class="cn">FALSE</span>, <span class="at">col.name=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="normalize-based-on-library-sizes-2" class="section level4 hasAnchor" number="4.1.3.3">
<h4><span class="header-section-number">4.1.3.3</span> Normalize based on library sizes<a href="virus-vs-virus--dge.html#normalize-based-on-library-sizes-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="virus-vs-virus--dge.html#cb84-1" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(DGE)</span></code></pre></div>
</div>
<div id="estimate-dispersion-2" class="section level4 hasAnchor" number="4.1.3.4">
<h4><span class="header-section-number">4.1.3.4</span> Estimate Dispersion<a href="virus-vs-virus--dge.html#estimate-dispersion-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="virus-vs-virus--dge.html#cb85-1" tabindex="-1"></a><span class="co"># Overall common dispersion</span></span>
<span id="cb85-2"><a href="virus-vs-virus--dge.html#cb85-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateCommonDisp</span>(DGE, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Disp = 0.16755 , BCV = 0.4093</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="virus-vs-virus--dge.html#cb87-1" tabindex="-1"></a><span class="co"># Dispersion trend based on gene abundance</span></span>
<span id="cb87-2"><a href="virus-vs-virus--dge.html#cb87-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTrendedDisp</span>(DGE)</span>
<span id="cb87-3"><a href="virus-vs-virus--dge.html#cb87-3" tabindex="-1"></a></span>
<span id="cb87-4"><a href="virus-vs-virus--dge.html#cb87-4" tabindex="-1"></a><span class="co"># Tagwise dispersion - allows different dispersion for each gene</span></span>
<span id="cb87-5"><a href="virus-vs-virus--dge.html#cb87-5" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTagwiseDisp</span>(DGE, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Using interpolation to estimate tagwise dispersion.</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="virus-vs-virus--dge.html#cb89-1" tabindex="-1"></a><span class="fu">plotBCV</span>(DGE)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
</div>
</div>
<div id="exploratory-plots-2" class="section level3 hasAnchor" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> Exploratory Plots<a href="virus-vs-virus--dge.html#exploratory-plots-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="normalized-counts-2" class="section level4 hasAnchor" number="4.1.4.1">
<h4><span class="header-section-number">4.1.4.1</span> Normalized counts<a href="virus-vs-virus--dge.html#normalized-counts-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="virus-vs-virus--dge.html#cb90-1" tabindex="-1"></a>CPM <span class="ot">&lt;-</span> </span>
<span id="cb90-2"><a href="virus-vs-virus--dge.html#cb90-2" tabindex="-1"></a>  <span class="fu">cpm</span>(DGE, <span class="at">normalized.lib.sizes =</span> <span class="cn">TRUE</span>, <span class="at">log =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-3"><a href="virus-vs-virus--dge.html#cb90-3" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb90-4"><a href="virus-vs-virus--dge.html#cb90-4" tabindex="-1"></a><span class="fu">boxplot</span>(CPM, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">ylab =</span> <span class="st">&quot;log2 CPM&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Normalized Data&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-50-1.png" width="672" /></p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="virus-vs-virus--dge.html#cb91-1" tabindex="-1"></a><span class="fu">write.csv</span>(CPM, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_normcounts.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="cluster-dendrogram-2" class="section level4 hasAnchor" number="4.1.4.2">
<h4><span class="header-section-number">4.1.4.2</span> Cluster dendrogram<a href="virus-vs-virus--dge.html#cluster-dendrogram-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="virus-vs-virus--dge.html#cb92-1" tabindex="-1"></a>RawDist <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(CPM), <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb92-2"><a href="virus-vs-virus--dge.html#cb92-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">hclust</span>(RawDist, <span class="at">method =</span> <span class="st">&quot;average&quot;</span>), <span class="at">xlab=</span><span class="st">&quot;Average Euclidean Distance&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-51-1.png" width="672" /></p>
</div>
<div id="pca-plots-2" class="section level4 hasAnchor" number="4.1.4.3">
<h4><span class="header-section-number">4.1.4.3</span> PCA Plots<a href="virus-vs-virus--dge.html#pca-plots-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="virus-vs-virus--dge.html#cb93-1" tabindex="-1"></a>PCA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(RawDist))</span>
<span id="cb93-2"><a href="virus-vs-virus--dge.html#cb93-2" tabindex="-1"></a></span>
<span id="cb93-3"><a href="virus-vs-virus--dge.html#cb93-3" tabindex="-1"></a><span class="co"># PCs 1-2</span></span>
<span id="cb93-4"><a href="virus-vs-virus--dge.html#cb93-4" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb93-5"><a href="virus-vs-virus--dge.html#cb93-5" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-1.png" width="672" /></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="virus-vs-virus--dge.html#cb94-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb94-2"><a href="virus-vs-virus--dge.html#cb94-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-2.png" width="672" /></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="virus-vs-virus--dge.html#cb95-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;year&quot;</span>) <span class="sc">+</span></span>
<span id="cb95-2"><a href="virus-vs-virus--dge.html#cb95-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-3.png" width="672" /></p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="virus-vs-virus--dge.html#cb96-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;molt&quot;</span>) <span class="sc">+</span></span>
<span id="cb96-2"><a href="virus-vs-virus--dge.html#cb96-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-4.png" width="672" /></p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="virus-vs-virus--dge.html#cb97-1" tabindex="-1"></a><span class="co"># PCs 3-4</span></span>
<span id="cb97-2"><a href="virus-vs-virus--dge.html#cb97-2" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb97-3"><a href="virus-vs-virus--dge.html#cb97-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-5.png" width="672" /></p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="virus-vs-virus--dge.html#cb98-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb98-2"><a href="virus-vs-virus--dge.html#cb98-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-6.png" width="672" /></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="virus-vs-virus--dge.html#cb99-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;year&quot;</span>) <span class="sc">+</span></span>
<span id="cb99-2"><a href="virus-vs-virus--dge.html#cb99-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-7.png" width="672" /></p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="virus-vs-virus--dge.html#cb100-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;molt&quot;</span>) <span class="sc">+</span></span>
<span id="cb100-2"><a href="virus-vs-virus--dge.html#cb100-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-8.png" width="672" /></p>
</div>
</div>
<div id="test-for-degs-2" class="section level3 hasAnchor" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> Test for DEGs<a href="virus-vs-virus--dge.html#test-for-degs-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="virus-vs-virus--dge.html#cb101-1" tabindex="-1"></a>et <span class="ot">&lt;-</span> <span class="fu">exactTest</span>(DGE, <span class="at">pair=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))</span>
<span id="cb101-2"><a href="virus-vs-virus--dge.html#cb101-2" tabindex="-1"></a></span>
<span id="cb101-3"><a href="virus-vs-virus--dge.html#cb101-3" tabindex="-1"></a><span class="fu">write.csv</span>(et, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_DGE_results.csv&quot;</span>)</span></code></pre></div>
<div id="select-degs-by-fold-change-pvalue-2" class="section level4 hasAnchor" number="4.1.5.1">
<h4><span class="header-section-number">4.1.5.1</span> Select DEGs by fold change &amp; pvalue<a href="virus-vs-virus--dge.html#select-degs-by-fold-change-pvalue-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Uncorrected p &lt; 0.05
log2FC &gt; 1 (2-fold change)</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="virus-vs-virus--dge.html#cb102-1" tabindex="-1"></a>results <span class="ot">&lt;-</span> </span>
<span id="cb102-2"><a href="virus-vs-virus--dge.html#cb102-2" tabindex="-1"></a>  <span class="fu">topTags</span>(et, <span class="at">n =</span> <span class="fu">dim</span>(DGE)[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb102-3"><a href="virus-vs-virus--dge.html#cb102-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb102-4"><a href="virus-vs-virus--dge.html#cb102-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb102-5"><a href="virus-vs-virus--dge.html#cb102-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb102-6"><a href="virus-vs-virus--dge.html#cb102-6" tabindex="-1"></a><span class="fu">nrow</span>(results)</span></code></pre></div>
<pre><code>## [1] 136</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="virus-vs-virus--dge.html#cb104-1" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_DEG.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="make-sure-genes-are-updown-regulated-in-virus-vs-virus-" class="section level4 hasAnchor" number="4.1.5.2">
<h4><span class="header-section-number">4.1.5.2</span> Make sure genes are up/down regulated in Virus+ vs Virus-<a href="virus-vs-virus--dge.html#make-sure-genes-are-updown-regulated-in-virus-vs-virus-" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Visually compare gene expression across viral RNA status to ensure directional change is correct.</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="virus-vs-virus--dge.html#cb105-1" tabindex="-1"></a>CPM <span class="sc">%&gt;%</span> </span>
<span id="cb105-2"><a href="virus-vs-virus--dge.html#cb105-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb105-3"><a href="virus-vs-virus--dge.html#cb105-3" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> results<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> </span>
<span id="cb105-4"><a href="virus-vs-virus--dge.html#cb105-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">!</span>gene, <span class="at">names_to=</span><span class="st">&quot;sample&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb105-5"><a href="virus-vs-virus--dge.html#cb105-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> SampleInfo<span class="sc">$</span>sample, </span>
<span id="cb105-6"><a href="virus-vs-virus--dge.html#cb105-6" tabindex="-1"></a>                              SampleInfo[<span class="fu">match</span>(sample, SampleInfo<span class="sc">$</span>sample), <span class="dv">8</span>],</span>
<span id="cb105-7"><a href="virus-vs-virus--dge.html#cb105-7" tabindex="-1"></a>                              <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb105-8"><a href="virus-vs-virus--dge.html#cb105-8" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> results<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb105-9"><a href="virus-vs-virus--dge.html#cb105-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">factor</span>(RNA, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb105-10"><a href="virus-vs-virus--dge.html#cb105-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>RNA, <span class="at">y=</span>expression)) <span class="sc">+</span></span>
<span id="cb105-11"><a href="virus-vs-virus--dge.html#cb105-11" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb105-12"><a href="virus-vs-virus--dge.html#cb105-12" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-55-1.png" width="672" /></p>
</div>
</div>
<div id="go-enrichment-analysis" class="section level3 hasAnchor" number="4.1.6">
<h3><span class="header-section-number">4.1.6</span> GO Enrichment Analysis<a href="virus-vs-virus--dge.html#go-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="configure-gene-lists" class="section level4 hasAnchor" number="4.1.6.1">
<h4><span class="header-section-number">4.1.6.1</span> Configure Gene Lists<a href="virus-vs-virus--dge.html#configure-gene-lists" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="virus-vs-virus--dge.html#cb106-1" tabindex="-1"></a>background <span class="ot">&lt;-</span> <span class="fu">rownames</span>(CPM)</span>
<span id="cb106-2"><a href="virus-vs-virus--dge.html#cb106-2" tabindex="-1"></a></span>
<span id="cb106-3"><a href="virus-vs-virus--dge.html#cb106-3" tabindex="-1"></a>up <span class="ot">&lt;-</span> </span>
<span id="cb106-4"><a href="virus-vs-virus--dge.html#cb106-4" tabindex="-1"></a>  results <span class="sc">%&gt;%</span> </span>
<span id="cb106-5"><a href="virus-vs-virus--dge.html#cb106-5" tabindex="-1"></a>  <span class="fu">filter</span>(logFC<span class="sc">&gt;=</span><span class="dv">1</span>)</span>
<span id="cb106-6"><a href="virus-vs-virus--dge.html#cb106-6" tabindex="-1"></a></span>
<span id="cb106-7"><a href="virus-vs-virus--dge.html#cb106-7" tabindex="-1"></a>down <span class="ot">&lt;-</span></span>
<span id="cb106-8"><a href="virus-vs-virus--dge.html#cb106-8" tabindex="-1"></a>  results <span class="sc">%&gt;%</span> </span>
<span id="cb106-9"><a href="virus-vs-virus--dge.html#cb106-9" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="up-regulated-genes" class="section level4 hasAnchor" number="4.1.6.2">
<h4><span class="header-section-number">4.1.6.2</span> Up-regulated genes<a href="virus-vs-virus--dge.html#up-regulated-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="biological-process" class="section level5 hasAnchor" number="4.1.6.2.1">
<h5><span class="header-section-number">4.1.6.2.1</span> Biological Process<a href="virus-vs-virus--dge.html#biological-process" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="virus-vs-virus--dge.html#cb107-1" tabindex="-1"></a>go_bp_up <span class="ot">&lt;-</span></span>
<span id="cb107-2"><a href="virus-vs-virus--dge.html#cb107-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb107-3"><a href="virus-vs-virus--dge.html#cb107-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb107-4"><a href="virus-vs-virus--dge.html#cb107-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb107-5"><a href="virus-vs-virus--dge.html#cb107-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb107-6"><a href="virus-vs-virus--dge.html#cb107-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb107-7"><a href="virus-vs-virus--dge.html#cb107-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb107-8"><a href="virus-vs-virus--dge.html#cb107-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb107-9"><a href="virus-vs-virus--dge.html#cb107-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;bp&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="molecular-function" class="section level5 hasAnchor" number="4.1.6.2.2">
<h5><span class="header-section-number">4.1.6.2.2</span> Molecular Function<a href="virus-vs-virus--dge.html#molecular-function" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="virus-vs-virus--dge.html#cb109-1" tabindex="-1"></a>go_mf_up <span class="ot">&lt;-</span> </span>
<span id="cb109-2"><a href="virus-vs-virus--dge.html#cb109-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb109-3"><a href="virus-vs-virus--dge.html#cb109-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb109-4"><a href="virus-vs-virus--dge.html#cb109-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb109-5"><a href="virus-vs-virus--dge.html#cb109-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb109-6"><a href="virus-vs-virus--dge.html#cb109-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb109-7"><a href="virus-vs-virus--dge.html#cb109-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
</div>
<div id="cellular-component" class="section level5 hasAnchor" number="4.1.6.2.3">
<h5><span class="header-section-number">4.1.6.2.3</span> Cellular Component<a href="virus-vs-virus--dge.html#cellular-component" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="virus-vs-virus--dge.html#cb112-1" tabindex="-1"></a>go_cc_up <span class="ot">&lt;-</span> </span>
<span id="cb112-2"><a href="virus-vs-virus--dge.html#cb112-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb112-3"><a href="virus-vs-virus--dge.html#cb112-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb112-4"><a href="virus-vs-virus--dge.html#cb112-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb112-5"><a href="virus-vs-virus--dge.html#cb112-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb112-6"><a href="virus-vs-virus--dge.html#cb112-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb112-7"><a href="virus-vs-virus--dge.html#cb112-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
</div>
</div>
<div id="down-regulated-genes" class="section level4 hasAnchor" number="4.1.6.3">
<h4><span class="header-section-number">4.1.6.3</span> Down-regulated Genes<a href="virus-vs-virus--dge.html#down-regulated-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="biological-process-1" class="section level5 hasAnchor" number="4.1.6.3.1">
<h5><span class="header-section-number">4.1.6.3.1</span> Biological Process<a href="virus-vs-virus--dge.html#biological-process-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="virus-vs-virus--dge.html#cb115-1" tabindex="-1"></a>go_bp_down <span class="ot">&lt;-</span> </span>
<span id="cb115-2"><a href="virus-vs-virus--dge.html#cb115-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb115-3"><a href="virus-vs-virus--dge.html#cb115-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb115-4"><a href="virus-vs-virus--dge.html#cb115-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb115-5"><a href="virus-vs-virus--dge.html#cb115-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb115-6"><a href="virus-vs-virus--dge.html#cb115-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb115-7"><a href="virus-vs-virus--dge.html#cb115-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb115-8"><a href="virus-vs-virus--dge.html#cb115-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>,</span>
<span id="cb115-9"><a href="virus-vs-virus--dge.html#cb115-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;bp&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="molecular-function-1" class="section level5 hasAnchor" number="4.1.6.3.2">
<h5><span class="header-section-number">4.1.6.3.2</span> Molecular Function<a href="virus-vs-virus--dge.html#molecular-function-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="virus-vs-virus--dge.html#cb117-1" tabindex="-1"></a>go_mf_down <span class="ot">&lt;-</span> </span>
<span id="cb117-2"><a href="virus-vs-virus--dge.html#cb117-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb117-3"><a href="virus-vs-virus--dge.html#cb117-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb117-4"><a href="virus-vs-virus--dge.html#cb117-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb117-5"><a href="virus-vs-virus--dge.html#cb117-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb117-6"><a href="virus-vs-virus--dge.html#cb117-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb117-7"><a href="virus-vs-virus--dge.html#cb117-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb117-8"><a href="virus-vs-virus--dge.html#cb117-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>, </span>
<span id="cb117-9"><a href="virus-vs-virus--dge.html#cb117-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;mf&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="cellular-component-1" class="section level5 hasAnchor" number="4.1.6.3.3">
<h5><span class="header-section-number">4.1.6.3.3</span> Cellular Component<a href="virus-vs-virus--dge.html#cellular-component-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="virus-vs-virus--dge.html#cb119-1" tabindex="-1"></a>go_cc_down <span class="ot">&lt;-</span> </span>
<span id="cb119-2"><a href="virus-vs-virus--dge.html#cb119-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb119-3"><a href="virus-vs-virus--dge.html#cb119-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb119-4"><a href="virus-vs-virus--dge.html#cb119-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb119-5"><a href="virus-vs-virus--dge.html#cb119-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb119-6"><a href="virus-vs-virus--dge.html#cb119-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb119-7"><a href="virus-vs-virus--dge.html#cb119-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb119-8"><a href="virus-vs-virus--dge.html#cb119-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>, </span>
<span id="cb119-9"><a href="virus-vs-virus--dge.html#cb119-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;cc&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
</div>
<div id="save-results" class="section level4 hasAnchor" number="4.1.6.4">
<h4><span class="header-section-number">4.1.6.4</span> Save Results<a href="virus-vs-virus--dge.html#save-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="virus-vs-virus--dge.html#cb121-1" tabindex="-1"></a>go <span class="ot">&lt;-</span></span>
<span id="cb121-2"><a href="virus-vs-virus--dge.html#cb121-2" tabindex="-1"></a>  <span class="fu">bind_rows</span>(go_bp_down, go_bp_up,</span>
<span id="cb121-3"><a href="virus-vs-virus--dge.html#cb121-3" tabindex="-1"></a>            go_cc_down, </span>
<span id="cb121-4"><a href="virus-vs-virus--dge.html#cb121-4" tabindex="-1"></a>            go_mf_down)</span>
<span id="cb121-5"><a href="virus-vs-virus--dge.html#cb121-5" tabindex="-1"></a></span>
<span id="cb121-6"><a href="virus-vs-virus--dge.html#cb121-6" tabindex="-1"></a>go <span class="ot">&lt;-</span> </span>
<span id="cb121-7"><a href="virus-vs-virus--dge.html#cb121-7" tabindex="-1"></a>  go <span class="sc">%&gt;%</span> </span>
<span id="cb121-8"><a href="virus-vs-virus--dge.html#cb121-8" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>link <span class="sc">&amp;</span> <span class="sc">!</span>overlapId) </span>
<span id="cb121-9"><a href="virus-vs-virus--dge.html#cb121-9" tabindex="-1"></a></span>
<span id="cb121-10"><a href="virus-vs-virus--dge.html#cb121-10" tabindex="-1"></a>go <span class="sc">%&gt;%</span> </span>
<span id="cb121-11"><a href="virus-vs-virus--dge.html#cb121-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), \(x) <span class="fu">round</span>(x, <span class="at">digits=</span><span class="dv">5</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb121-12"><a href="virus-vs-virus--dge.html#cb121-12" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb121-13"><a href="virus-vs-virus--dge.html#cb121-13" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>, <span class="at">font_size=</span><span class="dv">9</span>)</span></code></pre></div>
<table class="table" style="font-size: 9px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:right;">
size
</th>
<th style="text-align:right;">
overlap
</th>
<th style="text-align:right;">
expect
</th>
<th style="text-align:right;">
enrichmentRatio
</th>
<th style="text-align:right;">
pValue
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:left;">
userId
</th>
<th style="text-align:left;">
direction
</th>
<th style="text-align:left;">
go
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0045087" class="uri">GO:0045087</a>
</td>
<td style="text-align:left;">
innate immune response
</td>
<td style="text-align:right;">
491
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
5.07520
</td>
<td style="text-align:right;">
5.12295
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:left;">
IFI6;C1QB;NR1H3;ACTG1;C1QC;C2;C1QA;NECTIN2;CFB;CFD;HLA-DQA1;MNDA;UBC;APOBEC3G;DHX58;RPL13A;APOE;MX2;AXL;SOCS1;UBA52;ISG15;IFIT3;AIF1;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002253" class="uri">GO:0002253</a>
</td>
<td style="text-align:left;">
activation of immune response
</td>
<td style="text-align:right;">
370
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
3.82449
</td>
<td style="text-align:right;">
4.70651
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00009
</td>
<td style="text-align:left;">
C1QB;NR1H3;ACTG1;C1QC;C2;C1QA;RPS3;NECTIN2;CD79A;CFB;ACTB;CFD;HLA-DQA1;MNDA;UBC;DHX58;UBA52;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050778" class="uri">GO:0050778</a>
</td>
<td style="text-align:left;">
positive regulation of immune response
</td>
<td style="text-align:right;">
476
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
4.92016
</td>
<td style="text-align:right;">
4.06491
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:left;">
C1QB;NR1H3;ACTG1;C1QC;C2;C1QA;RPS3;NECTIN2;HLX;CD79A;CFB;IDO1;ACTB;CFD;HLA-DQA1;MNDA;UBC;DHX58;UBA52;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072376" class="uri">GO:0072376</a>
</td>
<td style="text-align:left;">
protein activation cascade
</td>
<td style="text-align:right;">
38
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.39279
</td>
<td style="text-align:right;">
17.82143
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00012
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;GP1BB;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006956" class="uri">GO:0006956</a>
</td>
<td style="text-align:left;">
complement activation
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.27908
</td>
<td style="text-align:right;">
21.49887
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00026
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019221" class="uri">GO:0019221</a>
</td>
<td style="text-align:left;">
cytokine-mediated signaling pathway
</td>
<td style="text-align:right;">
458
</td>
<td style="text-align:right;">
18
</td>
<td style="text-align:right;">
4.73410
</td>
<td style="text-align:right;">
3.80220
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00070
</td>
<td style="text-align:left;">
IL32;IFI6;NR1H3;RACK1;HLA-DQA1;UBC;MPL;HSPA8;CARD14;MX2;AXL;SOCS1;UBA52;ISG15;COL1A2;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019058" class="uri">GO:0019058</a>
</td>
<td style="text-align:left;">
viral life cycle
</td>
<td style="text-align:right;">
208
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
2.14998
</td>
<td style="text-align:right;">
5.58144
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00101
</td>
<td style="text-align:left;">
ITGB5;NECTIN2;UBC;HSPA8;APOBEC3G;APOE;AXL;UBA52;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006614" class="uri">GO:0006614</a>
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:right;">
83
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.85793
</td>
<td style="text-align:right;">
9.32481
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00117
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031347" class="uri">GO:0031347</a>
</td>
<td style="text-align:left;">
regulation of defense response
</td>
<td style="text-align:right;">
438
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
4.52737
</td>
<td style="text-align:right;">
3.75494
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00117
</td>
<td style="text-align:left;">
C1QB;NR1H3;C1QC;C2;C1QA;NECTIN2;GPSM3;CFB;IDO1;MNDA;UBC;APOBEC3G;DHX58;APOE;SOCS1;UBA52;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030449" class="uri">GO:0030449</a>
</td>
<td style="text-align:left;">
regulation of complement activation
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.22740
</td>
<td style="text-align:right;">
21.98748
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00123
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006613" class="uri">GO:0006613</a>
</td>
<td style="text-align:left;">
cotranslational protein targeting to membrane
</td>
<td style="text-align:right;">
87
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.89927
</td>
<td style="text-align:right;">
8.89608
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00129
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000257" class="uri">GO:2000257</a>
</td>
<td style="text-align:left;">
regulation of protein activation cascade
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.23774
</td>
<td style="text-align:right;">
21.03150
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00129
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045047" class="uri">GO:0045047</a>
</td>
<td style="text-align:left;">
protein targeting to ER
</td>
<td style="text-align:right;">
93
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.96129
</td>
<td style="text-align:right;">
8.32214
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00187
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072599" class="uri">GO:0072599</a>
</td>
<td style="text-align:left;">
establishment of protein localization to endoplasmic reticulum
</td>
<td style="text-align:right;">
96
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.99230
</td>
<td style="text-align:right;">
8.06207
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00220
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.13437
</td>
<td style="text-align:right;">
29.76766
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00239
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006413" class="uri">GO:0006413</a>
</td>
<td style="text-align:left;">
translational initiation
</td>
<td style="text-align:right;">
167
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1.72619
</td>
<td style="text-align:right;">
5.79311
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00254
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;ATF4;RPL13A;EIF4G2;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002920" class="uri">GO:0002920</a>
</td>
<td style="text-align:left;">
regulation of humoral immune response
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.32043
</td>
<td style="text-align:right;">
15.60402
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00407
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000184" class="uri">GO:0000184</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process, nonsense-mediated decay
</td>
<td style="text-align:right;">
108
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.11634
</td>
<td style="text-align:right;">
7.16629
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00407
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0060337" class="uri">GO:0060337</a>
</td>
<td style="text-align:left;">
type I interferon signaling pathway
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.54783
</td>
<td style="text-align:right;">
10.95225
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00407
</td>
<td style="text-align:left;">
IFI6;MX2;ISG15;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071357" class="uri">GO:0071357</a>
</td>
<td style="text-align:left;">
cellular response to type I interferon
</td>
<td style="text-align:right;">
53
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.54783
</td>
<td style="text-align:right;">
10.95225
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00407
</td>
<td style="text-align:left;">
IFI6;MX2;ISG15;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006612" class="uri">GO:0006612</a>
</td>
<td style="text-align:left;">
protein targeting to membrane
</td>
<td style="text-align:right;">
147
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1.51946
</td>
<td style="text-align:right;">
5.92316
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00472
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12;RTP5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034340" class="uri">GO:0034340</a>
</td>
<td style="text-align:left;">
response to type I interferon
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.58918
</td>
<td style="text-align:right;">
10.18367
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00566
</td>
<td style="text-align:left;">
IFI6;MX2;ISG15;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050866" class="uri">GO:0050866</a>
</td>
<td style="text-align:left;">
negative regulation of cell activation
</td>
<td style="text-align:right;">
118
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.21970
</td>
<td style="text-align:right;">
6.55898
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00613
</td>
<td style="text-align:left;">
BPI;NR1H3;HLX;IDO1;MNDA;APOE;AXL;SOCS1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070972" class="uri">GO:0070972</a>
</td>
<td style="text-align:left;">
protein localization to endoplasmic reticulum
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.23004
</td>
<td style="text-align:right;">
6.50386
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00625
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006959" class="uri">GO:0006959</a>
</td>
<td style="text-align:left;">
humoral immune response
</td>
<td style="text-align:right;">
93
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.96129
</td>
<td style="text-align:right;">
7.28187
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:right;">
0.00936
</td>
<td style="text-align:left;">
BPI;C1QB;C1QC;C2;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045069" class="uri">GO:0045069</a>
</td>
<td style="text-align:left;">
regulation of viral genome replication
</td>
<td style="text-align:right;">
65
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.67187
</td>
<td style="text-align:right;">
8.93030
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:right;">
0.01010
</td>
<td style="text-align:left;">
HSPA8;APOBEC3G;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098542" class="uri">GO:0098542</a>
</td>
<td style="text-align:left;">
defense response to other organism
</td>
<td style="text-align:right;">
252
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
2.60479
</td>
<td style="text-align:right;">
4.22299
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:right;">
0.01010
</td>
<td style="text-align:left;">
BPI;IFI6;BATF2;CEBPE;APOBEC3G;DHX58;MX2;ISG15;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050792" class="uri">GO:0050792</a>
</td>
<td style="text-align:left;">
regulation of viral process
</td>
<td style="text-align:right;">
130
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.34374
</td>
<td style="text-align:right;">
5.95353
</td>
<td style="text-align:right;">
0.00006
</td>
<td style="text-align:right;">
0.01010
</td>
<td style="text-align:left;">
NECTIN2;HSPA8;APOBEC3G;APOE;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903900" class="uri">GO:1903900</a>
</td>
<td style="text-align:left;">
regulation of viral life cycle
</td>
<td style="text-align:right;">
100
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1.03365
</td>
<td style="text-align:right;">
6.77214
</td>
<td style="text-align:right;">
0.00007
</td>
<td style="text-align:right;">
0.01285
</td>
<td style="text-align:left;">
NECTIN2;HSPA8;APOBEC3G;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002455" class="uri">GO:0002455</a>
</td>
<td style="text-align:left;">
humoral immune response mediated by circulating immunoglobulin
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.23774
</td>
<td style="text-align:right;">
16.82520
</td>
<td style="text-align:right;">
0.00008
</td>
<td style="text-align:right;">
0.01368
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006402" class="uri">GO:0006402</a>
</td>
<td style="text-align:left;">
mRNA catabolic process
</td>
<td style="text-align:right;">
270
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
2.79084
</td>
<td style="text-align:right;">
3.94146
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:right;">
0.01646
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;UBC;HSPA8;RPL13A;RPL14;UBA52;RPS12;DDX5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002695" class="uri">GO:0002695</a>
</td>
<td style="text-align:left;">
negative regulation of leukocyte activation
</td>
<td style="text-align:right;">
106
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1.09567
</td>
<td style="text-align:right;">
6.38881
</td>
<td style="text-align:right;">
0.00011
</td>
<td style="text-align:right;">
0.01670
</td>
<td style="text-align:left;">
BPI;NR1H3;HLX;IDO1;MNDA;AXL;SOCS1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0000956" class="uri">GO:0000956</a>
</td>
<td style="text-align:left;">
nuclear-transcribed mRNA catabolic process
</td>
<td style="text-align:right;">
183
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1.89157
</td>
<td style="text-align:right;">
4.75795
</td>
<td style="text-align:right;">
0.00011
</td>
<td style="text-align:right;">
0.01670
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12;DDX5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002673" class="uri">GO:0002673</a>
</td>
<td style="text-align:left;">
regulation of acute inflammatory response
</td>
<td style="text-align:right;">
47
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.48581
</td>
<td style="text-align:right;">
10.29201
</td>
<td style="text-align:right;">
0.00012
</td>
<td style="text-align:right;">
0.01711
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043903" class="uri">GO:0043903</a>
</td>
<td style="text-align:left;">
regulation of symbiosis, encompassing mutualism through parasitism
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.50912
</td>
<td style="text-align:right;">
5.30109
</td>
<td style="text-align:right;">
0.00013
</td>
<td style="text-align:right;">
0.01831
</td>
<td style="text-align:left;">
NECTIN2;HSPA8;APOBEC3G;APOE;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030168" class="uri">GO:0030168</a>
</td>
<td style="text-align:left;">
platelet activation
</td>
<td style="text-align:right;">
112
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1.15768
</td>
<td style="text-align:right;">
6.04656
</td>
<td style="text-align:right;">
0.00015
</td>
<td style="text-align:right;">
0.02091
</td>
<td style="text-align:left;">
ACTG1;GP1BB;ACTB;MPL;APOE;AXL;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009636" class="uri">GO:0009636</a>
</td>
<td style="text-align:left;">
response to toxic substance
</td>
<td style="text-align:right;">
283
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
2.92522
</td>
<td style="text-align:right;">
3.76040
</td>
<td style="text-align:right;">
0.00015
</td>
<td style="text-align:right;">
0.02091
</td>
<td style="text-align:left;">
G6PD;RPS3;RACK1;EEF2;ACTB;PCGF2;ATF4;APOE;AXL;UBA52;TXNIP
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051607" class="uri">GO:0051607</a>
</td>
<td style="text-align:left;">
defense response to virus
</td>
<td style="text-align:right;">
151
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.56081
</td>
<td style="text-align:right;">
5.12556
</td>
<td style="text-align:right;">
0.00016
</td>
<td style="text-align:right;">
0.02130
</td>
<td style="text-align:left;">
IFI6;APOBEC3G;DHX58;MX2;ISG15;IFIT3;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002757" class="uri">GO:0002757</a>
</td>
<td style="text-align:left;">
immune response-activating signal transduction
</td>
<td style="text-align:right;">
338
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
3.49372
</td>
<td style="text-align:right;">
3.43473
</td>
<td style="text-align:right;">
0.00018
</td>
<td style="text-align:right;">
0.02301
</td>
<td style="text-align:left;">
NR1H3;ACTG1;RPS3;NECTIN2;CD79A;ACTB;HLA-DQA1;MNDA;UBC;DHX58;UBA52;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0010608" class="uri">GO:0010608</a>
</td>
<td style="text-align:left;">
posttranscriptional regulation of gene expression
</td>
<td style="text-align:right;">
392
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4.05189
</td>
<td style="text-align:right;">
3.20838
</td>
<td style="text-align:right;">
0.00019
</td>
<td style="text-align:right;">
0.02336
</td>
<td style="text-align:left;">
RPS3;RACK1;EEF2;RPS14;EIF5A;UBC;NCL;HSPA8;ATF4;RPL13A;EIF4G2;UBA52;DDX5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006401" class="uri">GO:0006401</a>
</td>
<td style="text-align:left;">
RNA catabolic process
</td>
<td style="text-align:right;">
292
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
3.01825
</td>
<td style="text-align:right;">
3.64450
</td>
<td style="text-align:right;">
0.00020
</td>
<td style="text-align:right;">
0.02481
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;UBC;HSPA8;RPL13A;RPL14;UBA52;RPS12;DDX5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0090150" class="uri">GO:0090150</a>
</td>
<td style="text-align:left;">
establishment of protein localization to membrane
</td>
<td style="text-align:right;">
245
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2.53243
</td>
<td style="text-align:right;">
3.94877
</td>
<td style="text-align:right;">
0.00021
</td>
<td style="text-align:right;">
0.02517
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;RPS14;RPL13A;RPL14;UBA52;RPS12;RTP5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019079" class="uri">GO:0019079</a>
</td>
<td style="text-align:left;">
viral genome replication
</td>
<td style="text-align:right;">
85
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.87860
</td>
<td style="text-align:right;">
6.82905
</td>
<td style="text-align:right;">
0.00024
</td>
<td style="text-align:right;">
0.02765
</td>
<td style="text-align:left;">
HSPA8;APOBEC3G;ISG15;DDX5;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070613" class="uri">GO:0070613</a>
</td>
<td style="text-align:left;">
regulation of protein processing
</td>
<td style="text-align:right;">
57
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.58918
</td>
<td style="text-align:right;">
8.48639
</td>
<td style="text-align:right;">
0.00029
</td>
<td style="text-align:right;">
0.03326
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002697" class="uri">GO:0002697</a>
</td>
<td style="text-align:left;">
regulation of immune effector process
</td>
<td style="text-align:right;">
256
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2.64613
</td>
<td style="text-align:right;">
3.77910
</td>
<td style="text-align:right;">
0.00030
</td>
<td style="text-align:right;">
0.03346
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;NECTIN2;HLX;CFB;APOBEC3G;DHX58;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050727" class="uri">GO:0050727</a>
</td>
<td style="text-align:left;">
regulation of inflammatory response
</td>
<td style="text-align:right;">
210
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
2.17066
</td>
<td style="text-align:right;">
4.14621
</td>
<td style="text-align:right;">
0.00031
</td>
<td style="text-align:right;">
0.03374
</td>
<td style="text-align:left;">
C1QB;NR1H3;C1QC;C2;C1QA;GPSM3;CFB;IDO1;APOE
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002764" class="uri">GO:0002764</a>
</td>
<td style="text-align:left;">
immune response-regulating signaling pathway
</td>
<td style="text-align:right;">
360
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
3.72113
</td>
<td style="text-align:right;">
3.22483
</td>
<td style="text-align:right;">
0.00032
</td>
<td style="text-align:right;">
0.03415
</td>
<td style="text-align:left;">
NR1H3;ACTG1;RPS3;NECTIN2;CD79A;ACTB;HLA-DQA1;MNDA;UBC;DHX58;UBA52;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903317" class="uri">GO:1903317</a>
</td>
<td style="text-align:left;">
regulation of protein maturation
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.60985
</td>
<td style="text-align:right;">
8.19872
</td>
<td style="text-align:right;">
0.00034
</td>
<td style="text-align:right;">
0.03466
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050865" class="uri">GO:0050865</a>
</td>
<td style="text-align:left;">
regulation of cell activation
</td>
<td style="text-align:right;">
363
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
3.75214
</td>
<td style="text-align:right;">
3.19818
</td>
<td style="text-align:right;">
0.00034
</td>
<td style="text-align:right;">
0.03466
</td>
<td style="text-align:left;">
BPI;NR1H3;RPS3;NECTIN2;HLX;IDO1;MNDA;MPL;APOE;AXL;SOCS1;AIF1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045071" class="uri">GO:0045071</a>
</td>
<td style="text-align:left;">
negative regulation of viral genome replication
</td>
<td style="text-align:right;">
33
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.34110
</td>
<td style="text-align:right;">
11.72665
</td>
<td style="text-align:right;">
0.00035
</td>
<td style="text-align:right;">
0.03466
</td>
<td style="text-align:left;">
APOBEC3G;ISG15;MX1;RSAD2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007596" class="uri">GO:0007596</a>
</td>
<td style="text-align:left;">
blood coagulation
</td>
<td style="text-align:right;">
214
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
2.21200
</td>
<td style="text-align:right;">
4.06871
</td>
<td style="text-align:right;">
0.00036
</td>
<td style="text-align:right;">
0.03466
</td>
<td style="text-align:left;">
ACTG1;GP1BB;USF1;ACTB;MPL;HMG20B;APOE;AXL;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0035455" class="uri">GO:0035455</a>
</td>
<td style="text-align:left;">
response to interferon-alpha
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.14471
</td>
<td style="text-align:right;">
20.73105
</td>
<td style="text-align:right;">
0.00036
</td>
<td style="text-align:right;">
0.03466
</td>
<td style="text-align:left;">
MX2;AXL;IFIT3
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050817" class="uri">GO:0050817</a>
</td>
<td style="text-align:left;">
coagulation
</td>
<td style="text-align:right;">
216
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
2.23268
</td>
<td style="text-align:right;">
4.03104
</td>
<td style="text-align:right;">
0.00038
</td>
<td style="text-align:right;">
0.03604
</td>
<td style="text-align:left;">
ACTG1;GP1BB;USF1;ACTB;MPL;HMG20B;APOE;AXL;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002683" class="uri">GO:0002683</a>
</td>
<td style="text-align:left;">
negative regulation of immune system process
</td>
<td style="text-align:right;">
265
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
2.73916
</td>
<td style="text-align:right;">
3.65075
</td>
<td style="text-align:right;">
0.00039
</td>
<td style="text-align:right;">
0.03671
</td>
<td style="text-align:left;">
BPI;NR1H3;C1QC;HLX;IDO1;MNDA;DHX58;TSC22D3;AXL;SOCS1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0007599" class="uri">GO:0007599</a>
</td>
<td style="text-align:left;">
hemostasis
</td>
<td style="text-align:right;">
218
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
2.25335
</td>
<td style="text-align:right;">
3.99406
</td>
<td style="text-align:right;">
0.00041
</td>
<td style="text-align:right;">
0.03716
</td>
<td style="text-align:left;">
ACTG1;GP1BB;USF1;ACTB;MPL;HMG20B;APOE;AXL;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1901361" class="uri">GO:1901361</a>
</td>
<td style="text-align:left;">
organic cyclic compound catabolic process
</td>
<td style="text-align:right;">
484
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
5.00285
</td>
<td style="text-align:right;">
2.79841
</td>
<td style="text-align:right;">
0.00042
</td>
<td style="text-align:right;">
0.03808
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;IDO1;UBC;HSPA8;APOBEC3G;RPL13A;APOE;RPL14;UBA52;RPS12;DDX5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006412" class="uri">GO:0006412</a>
</td>
<td style="text-align:left;">
translation
</td>
<td style="text-align:right;">
489
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
5.05453
</td>
<td style="text-align:right;">
2.76979
</td>
<td style="text-align:right;">
0.00047
</td>
<td style="text-align:right;">
0.04148
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;EEF2;RPS14;EIF5A;NCL;ATF4;RPL13A;EIF4G2;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006605" class="uri">GO:0006605</a>
</td>
<td style="text-align:left;">
protein targeting
</td>
<td style="text-align:right;">
324
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
3.34901
</td>
<td style="text-align:right;">
3.28455
</td>
<td style="text-align:right;">
0.00049
</td>
<td style="text-align:right;">
0.04273
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;UBC;HSPA8;RPL13A;RPL14;UBA52;RPS12;RTP5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015701" class="uri">GO:0015701</a>
</td>
<td style="text-align:left;">
bicarbonate transport
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.03228
</td>
<td style="text-align:right;">
92.95098
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.01930
</td>
<td style="text-align:left;">
CA2;SLC4A1;RHAG
</td>
<td style="text-align:left;">
up
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072562" class="uri">GO:0072562</a>
</td>
<td style="text-align:left;">
blood microparticle
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.46867
</td>
<td style="text-align:right;">
14.93587
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00014
</td>
<td style="text-align:left;">
C1QB;ACTG1;C1QC;CFB;ACTB;HSPA8;APOE
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0022626" class="uri">GO:0022626</a>
</td>
<td style="text-align:left;">
cytosolic ribosome
</td>
<td style="text-align:right;">
94
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.95772
</td>
<td style="text-align:right;">
9.39734
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00014
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0022627" class="uri">GO:0022627</a>
</td>
<td style="text-align:left;">
cytosolic small ribosomal subunit
</td>
<td style="text-align:right;">
35
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.35660
</td>
<td style="text-align:right;">
16.82571
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00027
</td>
<td style="text-align:left;">
RPS26;RPS3;RACK1;RPS14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005840" class="uri">GO:0005840</a>
</td>
<td style="text-align:left;">
ribosome
</td>
<td style="text-align:right;">
192
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1.95619
</td>
<td style="text-align:right;">
5.11198
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00279
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;EEF2;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015935" class="uri">GO:0015935</a>
</td>
<td style="text-align:left;">
small ribosomal subunit
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.59093
</td>
<td style="text-align:right;">
10.15345
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00279
</td>
<td style="text-align:left;">
RPS26;RPS3;RACK1;RPS14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044445" class="uri">GO:0044445</a>
</td>
<td style="text-align:left;">
cytosolic part
</td>
<td style="text-align:right;">
193
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1.96638
</td>
<td style="text-align:right;">
5.08549
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00279
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;RPS14;HSPA8;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0044391" class="uri">GO:0044391</a>
</td>
<td style="text-align:left;">
ribosomal subunit
</td>
<td style="text-align:right;">
157
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1.59959
</td>
<td style="text-align:right;">
5.62643
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00281
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RACK1;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005581" class="uri">GO:0005581</a>
</td>
<td style="text-align:left;">
collagen trimer
</td>
<td style="text-align:right;">
28
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.28528
</td>
<td style="text-align:right;">
14.02143
</td>
<td style="text-align:right;">
0.00017
</td>
<td style="text-align:right;">
0.01412
</td>
<td style="text-align:left;">
C1QB;C1QC;C1QA;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0005198" class="uri">GO:0005198</a>
</td>
<td style="text-align:left;">
structural molecule activity
</td>
<td style="text-align:right;">
407
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
4.17174
</td>
<td style="text-align:right;">
4.07504
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00060
</td>
<td style="text-align:left;">
RPS26;ACTG1;SORBS3;RPL36;RPS3;RACK1;DES;RPS14;ACTB;HSPB6;RPL13A;APOE;TPM2;RPL14;UBA52;RPS12;COL1A2
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0003735" class="uri">GO:0003735</a>
</td>
<td style="text-align:left;">
structural constituent of ribosome
</td>
<td style="text-align:right;">
133
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
1.36325
</td>
<td style="text-align:right;">
5.86834
</td>
<td style="text-align:right;">
0.00006
</td>
<td style="text-align:right;">
0.02771
</td>
<td style="text-align:left;">
RPS26;RPL36;RPS3;RPS14;RPL13A;RPL14;UBA52;RPS12
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0004252" class="uri">GO:0004252</a>
</td>
<td style="text-align:left;">
serine-type endopeptidase activity
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.75850
</td>
<td style="text-align:right;">
7.91037
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:right;">
0.03103
</td>
<td style="text-align:left;">
C1QB;C1QC;C2;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="virus-vs-virus--dge.html#cb122-1" tabindex="-1"></a><span class="fu">write.csv</span>(go, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_go.csv&quot;</span>,</span>
<span id="cb122-2"><a href="virus-vs-virus--dge.html#cb122-2" tabindex="-1"></a>          <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="kegg-analysis" class="section level3 hasAnchor" number="4.1.7">
<h3><span class="header-section-number">4.1.7</span> KEGG Analysis<a href="virus-vs-virus--dge.html#kegg-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="virus-vs-virus--dge.html#cb123-1" tabindex="-1"></a>kegg <span class="ot">&lt;-</span></span>
<span id="cb123-2"><a href="virus-vs-virus--dge.html#cb123-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">enrichDatabase =</span> <span class="st">&quot;pathway_KEGG&quot;</span>,</span>
<span id="cb123-3"><a href="virus-vs-virus--dge.html#cb123-3" tabindex="-1"></a>              <span class="at">interestGene =</span> results<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb123-4"><a href="virus-vs-virus--dge.html#cb123-4" tabindex="-1"></a>              <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb123-5"><a href="virus-vs-virus--dge.html#cb123-5" tabindex="-1"></a>              <span class="at">minNum =</span> <span class="dv">5</span>, <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">isOutput =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="virus-vs-virus--dge.html#cb125-1" tabindex="-1"></a><span class="fu">write.csv</span>(kegg, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_kegg.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
<div id="genome---edger---females" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Genome - edgeR - Females<a href="virus-vs-virus--dge.html#genome---edger---females" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-4" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Libraries<a href="virus-vs-virus--dge.html#libraries-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="virus-vs-virus--dge.html#cb126-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb126-2"><a href="virus-vs-virus--dge.html#cb126-2" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb126-3"><a href="virus-vs-virus--dge.html#cb126-3" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb126-4"><a href="virus-vs-virus--dge.html#cb126-4" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb126-5"><a href="virus-vs-virus--dge.html#cb126-5" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb126-6"><a href="virus-vs-virus--dge.html#cb126-6" tabindex="-1"></a><span class="fu">library</span>(WebGestaltR)</span></code></pre></div>
</div>
<div id="data-3" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Data<a href="virus-vs-virus--dge.html#data-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="virus-vs-virus--dge.html#cb127-1" tabindex="-1"></a>SampleInfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/metadata.csv&quot;</span>, </span>
<span id="cb127-2"><a href="virus-vs-virus--dge.html#cb127-2" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-3"><a href="virus-vs-virus--dge.html#cb127-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(disease_stage<span class="sc">==</span><span class="st">&quot;acute&quot;</span> <span class="sc">|</span> disease_stage<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;virus+&quot;</span>, <span class="st">&quot;virus-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb127-4"><a href="virus-vs-virus--dge.html#cb127-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sex<span class="sc">==</span><span class="st">&quot;male&quot;</span>)</span>
<span id="cb127-5"><a href="virus-vs-virus--dge.html#cb127-5" tabindex="-1"></a>  </span>
<span id="cb127-6"><a href="virus-vs-virus--dge.html#cb127-6" tabindex="-1"></a><span class="co">#Raw RSEM Gene Counts</span></span>
<span id="cb127-7"><a href="virus-vs-virus--dge.html#cb127-7" tabindex="-1"></a>Raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;Output Files/gnome_genecounts_locid_nohemo.csv&quot;</span>, </span>
<span id="cb127-8"><a href="virus-vs-virus--dge.html#cb127-8" tabindex="-1"></a>                <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>,</span>
<span id="cb127-9"><a href="virus-vs-virus--dge.html#cb127-9" tabindex="-1"></a>                <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb127-10"><a href="virus-vs-virus--dge.html#cb127-10" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-11"><a href="virus-vs-virus--dge.html#cb127-11" tabindex="-1"></a>  <span class="fu">select</span>(SampleInfo<span class="sc">$</span>sample)</span></code></pre></div>
</div>
<div id="differential-gene-expression-analysis-3" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Differential Gene Expression Analysis<a href="virus-vs-virus--dge.html#differential-gene-expression-analysis-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="dgelist-3" class="section level4 hasAnchor" number="4.2.3.1">
<h4><span class="header-section-number">4.2.3.1</span> DGElist<a href="virus-vs-virus--dge.html#dgelist-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="virus-vs-virus--dge.html#cb128-1" tabindex="-1"></a><span class="co">#Create DGE Object with counts &amp; sample groups</span></span>
<span id="cb128-2"><a href="virus-vs-virus--dge.html#cb128-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> Raw, <span class="at">group =</span> SampleInfo<span class="sc">$</span>RNA)</span>
<span id="cb128-3"><a href="virus-vs-virus--dge.html#cb128-3" tabindex="-1"></a></span>
<span id="cb128-4"><a href="virus-vs-virus--dge.html#cb128-4" tabindex="-1"></a><span class="co">#Check DGE Object Samples, Groups, Lib Size, and Norm Factor</span></span>
<span id="cb128-5"><a href="virus-vs-virus--dge.html#cb128-5" tabindex="-1"></a>DGE<span class="sc">$</span>samples </span></code></pre></div>
<pre><code>##          group lib.size norm.factors
## Hg11513 virus+  3894792            1
## Hg1354  virus+  4928497            1
## Hg1403  virus+  7571672            1
## Hg1404  virus+  4696012            1
## Hg908b  virus+  7291936            1
## Hg1256  virus+  5728747            1
## Hg1265  virus+  3851318            1
## Hg1288  virus+  3740785            1
## Hg266   virus+  4414778            1
## Hg1261  virus-  2938200            1
## Hg1272  virus-  6665010            1
## Hg1380  virus-  4719984            1
## Hg1387  virus-  5562169            1
## Hg1397  virus-  5747795            1
## Hg1421  virus-  5958925            1
## Hg1528  virus-  3655704            1
## Hg861a  virus-  6717503            1
## Hg862a  virus-  5273447            1
## Hg934   virus-  5089019            1
## Hg1262  virus-  5954066            1
## Hg1273  virus-  6511255            1
## Hg1282  virus-  3997507            1
## Hg1283  virus-  5907705            1
## Hg1286  virus-  5521600            1
## Hg1351  virus-  2230773            1
## Hg1516  virus-  6012343            1
## Hg1527  virus-  4895128            1</code></pre>
</div>
<div id="filter-out-lowly-expressed-genes-3" class="section level4 hasAnchor" number="4.2.3.2">
<h4><span class="header-section-number">4.2.3.2</span> Filter out lowly expressed genes<a href="virus-vs-virus--dge.html#filter-out-lowly-expressed-genes-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="virus-vs-virus--dge.html#cb130-1" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(DGE)</span>
<span id="cb130-2"><a href="virus-vs-virus--dge.html#cb130-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> DGE[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb130-3"><a href="virus-vs-virus--dge.html#cb130-3" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">rownames</span>(DGE<span class="sc">$</span>counts))</span></code></pre></div>
<pre><code>## [1] 11969</code></pre>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="virus-vs-virus--dge.html#cb132-1" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">rownames</span>(DGE<span class="sc">$</span>counts)</span>
<span id="cb132-2"><a href="virus-vs-virus--dge.html#cb132-2" tabindex="-1"></a><span class="fu">write.table</span>(postfilter, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_postfilter.txt&quot;</span>,</span>
<span id="cb132-3"><a href="virus-vs-virus--dge.html#cb132-3" tabindex="-1"></a>            <span class="at">row.name=</span><span class="cn">FALSE</span>, <span class="at">col.name=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="normalize-based-on-library-sizes-3" class="section level4 hasAnchor" number="4.2.3.3">
<h4><span class="header-section-number">4.2.3.3</span> Normalize based on library sizes<a href="virus-vs-virus--dge.html#normalize-based-on-library-sizes-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="virus-vs-virus--dge.html#cb133-1" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(DGE)</span></code></pre></div>
</div>
<div id="estimate-dispersion-3" class="section level4 hasAnchor" number="4.2.3.4">
<h4><span class="header-section-number">4.2.3.4</span> Estimate Dispersion<a href="virus-vs-virus--dge.html#estimate-dispersion-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>On average, the true abundance for each gene can vary up or down by xx% (BCV) between replicates
Differentially expressed = needs to vary by more than xx%</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="virus-vs-virus--dge.html#cb134-1" tabindex="-1"></a><span class="co"># Overall common dispersion</span></span>
<span id="cb134-2"><a href="virus-vs-virus--dge.html#cb134-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateCommonDisp</span>(DGE, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Disp = 0.20107 , BCV = 0.4484</code></pre>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="virus-vs-virus--dge.html#cb136-1" tabindex="-1"></a><span class="co"># Dispersion trend based on gene abundance</span></span>
<span id="cb136-2"><a href="virus-vs-virus--dge.html#cb136-2" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTrendedDisp</span>(DGE)</span>
<span id="cb136-3"><a href="virus-vs-virus--dge.html#cb136-3" tabindex="-1"></a></span>
<span id="cb136-4"><a href="virus-vs-virus--dge.html#cb136-4" tabindex="-1"></a><span class="co"># Tagwise dispersion - allows different dispersion for each gene</span></span>
<span id="cb136-5"><a href="virus-vs-virus--dge.html#cb136-5" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTagwiseDisp</span>(DGE, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Using interpolation to estimate tagwise dispersion.</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="virus-vs-virus--dge.html#cb138-1" tabindex="-1"></a><span class="fu">plotBCV</span>(DGE)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-70-1.png" width="672" /></p>
</div>
</div>
<div id="exploratory-plots-3" class="section level3 hasAnchor" number="4.2.4">
<h3><span class="header-section-number">4.2.4</span> Exploratory Plots<a href="virus-vs-virus--dge.html#exploratory-plots-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="normalized-counts-3" class="section level4 hasAnchor" number="4.2.4.1">
<h4><span class="header-section-number">4.2.4.1</span> Normalized counts<a href="virus-vs-virus--dge.html#normalized-counts-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="virus-vs-virus--dge.html#cb139-1" tabindex="-1"></a>CPM <span class="ot">&lt;-</span> </span>
<span id="cb139-2"><a href="virus-vs-virus--dge.html#cb139-2" tabindex="-1"></a>  <span class="fu">cpm</span>(DGE, <span class="at">normalized.lib.sizes =</span> <span class="cn">TRUE</span>, <span class="at">log =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb139-3"><a href="virus-vs-virus--dge.html#cb139-3" tabindex="-1"></a>  <span class="fu">data.frame</span>()</span>
<span id="cb139-4"><a href="virus-vs-virus--dge.html#cb139-4" tabindex="-1"></a><span class="fu">boxplot</span>(CPM, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">ylab =</span> <span class="st">&quot;log2 CPM&quot;</span>, <span class="at">main =</span> <span class="st">&quot;Normalized Data&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-71-1.png" width="672" /></p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="virus-vs-virus--dge.html#cb140-1" tabindex="-1"></a><span class="fu">write.csv</span>(CPM, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_normcounts.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="cluster-dendrogram-3" class="section level4 hasAnchor" number="4.2.4.2">
<h4><span class="header-section-number">4.2.4.2</span> Cluster dendrogram<a href="virus-vs-virus--dge.html#cluster-dendrogram-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="virus-vs-virus--dge.html#cb141-1" tabindex="-1"></a>RawDist <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">t</span>(CPM), <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb141-2"><a href="virus-vs-virus--dge.html#cb141-2" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">hclust</span>(RawDist, <span class="at">method =</span> <span class="st">&quot;average&quot;</span>), <span class="at">xlab=</span><span class="st">&quot;Average Euclidean Distance&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
</div>
<div id="pca-plots-3" class="section level4 hasAnchor" number="4.2.4.3">
<h4><span class="header-section-number">4.2.4.3</span> PCA Plots<a href="virus-vs-virus--dge.html#pca-plots-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="virus-vs-virus--dge.html#cb142-1" tabindex="-1"></a>PCA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(RawDist))</span>
<span id="cb142-2"><a href="virus-vs-virus--dge.html#cb142-2" tabindex="-1"></a></span>
<span id="cb142-3"><a href="virus-vs-virus--dge.html#cb142-3" tabindex="-1"></a><span class="co"># PCs 1-2</span></span>
<span id="cb142-4"><a href="virus-vs-virus--dge.html#cb142-4" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb142-5"><a href="virus-vs-virus--dge.html#cb142-5" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-1.png" width="672" /></p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="virus-vs-virus--dge.html#cb143-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb143-2"><a href="virus-vs-virus--dge.html#cb143-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-2.png" width="672" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="virus-vs-virus--dge.html#cb144-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;year&quot;</span>) <span class="sc">+</span></span>
<span id="cb144-2"><a href="virus-vs-virus--dge.html#cb144-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-3.png" width="672" /></p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="virus-vs-virus--dge.html#cb145-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">colour=</span><span class="st">&quot;molt&quot;</span>) <span class="sc">+</span></span>
<span id="cb145-2"><a href="virus-vs-virus--dge.html#cb145-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-4.png" width="672" /></p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="virus-vs-virus--dge.html#cb146-1" tabindex="-1"></a><span class="co"># PCs 3-4</span></span>
<span id="cb146-2"><a href="virus-vs-virus--dge.html#cb146-2" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;RNA&quot;</span>) <span class="sc">+</span></span>
<span id="cb146-3"><a href="virus-vs-virus--dge.html#cb146-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-5.png" width="672" /></p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="virus-vs-virus--dge.html#cb147-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;sex&quot;</span>) <span class="sc">+</span></span>
<span id="cb147-2"><a href="virus-vs-virus--dge.html#cb147-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-6.png" width="672" /></p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="virus-vs-virus--dge.html#cb148-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;year&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-2"><a href="virus-vs-virus--dge.html#cb148-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-7.png" width="672" /></p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="virus-vs-virus--dge.html#cb149-1" tabindex="-1"></a><span class="fu">autoplot</span>(PCA, <span class="at">data=</span>SampleInfo, <span class="at">x=</span><span class="dv">3</span>, <span class="at">y=</span><span class="dv">4</span>, <span class="at">colour=</span><span class="st">&quot;molt&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-2"><a href="virus-vs-virus--dge.html#cb149-2" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-73-8.png" width="672" /></p>
</div>
</div>
<div id="test-for-degs-3" class="section level3 hasAnchor" number="4.2.5">
<h3><span class="header-section-number">4.2.5</span> Test for DEGs<a href="virus-vs-virus--dge.html#test-for-degs-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="virus-vs-virus--dge.html#cb150-1" tabindex="-1"></a>et <span class="ot">&lt;-</span> <span class="fu">exactTest</span>(DGE, <span class="at">pair=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))</span>
<span id="cb150-2"><a href="virus-vs-virus--dge.html#cb150-2" tabindex="-1"></a></span>
<span id="cb150-3"><a href="virus-vs-virus--dge.html#cb150-3" tabindex="-1"></a><span class="fu">write.csv</span>(et, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_DGE_results.csv&quot;</span>)</span></code></pre></div>
<div id="select-degs-by-fold-change-pvalue-3" class="section level4 hasAnchor" number="4.2.5.1">
<h4><span class="header-section-number">4.2.5.1</span> Select DEGs by fold change &amp; pvalue<a href="virus-vs-virus--dge.html#select-degs-by-fold-change-pvalue-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Uncorrected p &lt; 0.05
log2FC &gt; 1 (2-fold change)</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="virus-vs-virus--dge.html#cb151-1" tabindex="-1"></a>results <span class="ot">&lt;-</span> </span>
<span id="cb151-2"><a href="virus-vs-virus--dge.html#cb151-2" tabindex="-1"></a>  <span class="fu">topTags</span>(et, <span class="at">n =</span> <span class="fu">dim</span>(DGE)[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb151-3"><a href="virus-vs-virus--dge.html#cb151-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb151-4"><a href="virus-vs-virus--dge.html#cb151-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb151-5"><a href="virus-vs-virus--dge.html#cb151-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb151-6"><a href="virus-vs-virus--dge.html#cb151-6" tabindex="-1"></a><span class="fu">nrow</span>(results)</span></code></pre></div>
<pre><code>## [1] 132</code></pre>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="virus-vs-virus--dge.html#cb153-1" tabindex="-1"></a><span class="fu">write.csv</span>(results, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_DEG.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="make-sure-genes-are-updown-regulated-in-virus-vs-virus--1" class="section level4 hasAnchor" number="4.2.5.2">
<h4><span class="header-section-number">4.2.5.2</span> Make sure genes are up/down regulated in Virus+ vs Virus-<a href="virus-vs-virus--dge.html#make-sure-genes-are-updown-regulated-in-virus-vs-virus--1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Visually compare gene expression across viral RNA status to ensure directional change is correct.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="virus-vs-virus--dge.html#cb154-1" tabindex="-1"></a>CPM <span class="sc">%&gt;%</span> </span>
<span id="cb154-2"><a href="virus-vs-virus--dge.html#cb154-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb154-3"><a href="virus-vs-virus--dge.html#cb154-3" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> results<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> </span>
<span id="cb154-4"><a href="virus-vs-virus--dge.html#cb154-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">!</span>gene, <span class="at">names_to=</span><span class="st">&quot;sample&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb154-5"><a href="virus-vs-virus--dge.html#cb154-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> SampleInfo<span class="sc">$</span>sample, </span>
<span id="cb154-6"><a href="virus-vs-virus--dge.html#cb154-6" tabindex="-1"></a>                              SampleInfo[<span class="fu">match</span>(sample, SampleInfo<span class="sc">$</span>sample), <span class="dv">8</span>],</span>
<span id="cb154-7"><a href="virus-vs-virus--dge.html#cb154-7" tabindex="-1"></a>                              <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb154-8"><a href="virus-vs-virus--dge.html#cb154-8" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> results<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb154-9"><a href="virus-vs-virus--dge.html#cb154-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">factor</span>(RNA, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb154-10"><a href="virus-vs-virus--dge.html#cb154-10" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>RNA, <span class="at">y=</span>expression)) <span class="sc">+</span></span>
<span id="cb154-11"><a href="virus-vs-virus--dge.html#cb154-11" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb154-12"><a href="virus-vs-virus--dge.html#cb154-12" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
</div>
</div>
<div id="go-enrichment-analysis-1" class="section level3 hasAnchor" number="4.2.6">
<h3><span class="header-section-number">4.2.6</span> GO Enrichment Analysis<a href="virus-vs-virus--dge.html#go-enrichment-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="configure-gene-lists-1" class="section level4 hasAnchor" number="4.2.6.1">
<h4><span class="header-section-number">4.2.6.1</span> Configure Gene Lists<a href="virus-vs-virus--dge.html#configure-gene-lists-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="virus-vs-virus--dge.html#cb155-1" tabindex="-1"></a>background <span class="ot">&lt;-</span> <span class="fu">rownames</span>(CPM)</span>
<span id="cb155-2"><a href="virus-vs-virus--dge.html#cb155-2" tabindex="-1"></a></span>
<span id="cb155-3"><a href="virus-vs-virus--dge.html#cb155-3" tabindex="-1"></a>up <span class="ot">&lt;-</span> </span>
<span id="cb155-4"><a href="virus-vs-virus--dge.html#cb155-4" tabindex="-1"></a>  results <span class="sc">%&gt;%</span> </span>
<span id="cb155-5"><a href="virus-vs-virus--dge.html#cb155-5" tabindex="-1"></a>  <span class="fu">filter</span>(logFC<span class="sc">&gt;=</span><span class="dv">1</span>)</span>
<span id="cb155-6"><a href="virus-vs-virus--dge.html#cb155-6" tabindex="-1"></a>down <span class="ot">&lt;-</span></span>
<span id="cb155-7"><a href="virus-vs-virus--dge.html#cb155-7" tabindex="-1"></a>  results <span class="sc">%&gt;%</span> </span>
<span id="cb155-8"><a href="virus-vs-virus--dge.html#cb155-8" tabindex="-1"></a>  <span class="fu">filter</span>(logFC <span class="sc">&lt;</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="up-regulated-genes-1" class="section level4 hasAnchor" number="4.2.6.2">
<h4><span class="header-section-number">4.2.6.2</span> Up-regulated genes<a href="virus-vs-virus--dge.html#up-regulated-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="biological-process-2" class="section level5 hasAnchor" number="4.2.6.2.1">
<h5><span class="header-section-number">4.2.6.2.1</span> Biological Process<a href="virus-vs-virus--dge.html#biological-process-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="virus-vs-virus--dge.html#cb156-1" tabindex="-1"></a>go_bp_up <span class="ot">&lt;-</span></span>
<span id="cb156-2"><a href="virus-vs-virus--dge.html#cb156-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb156-3"><a href="virus-vs-virus--dge.html#cb156-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb156-4"><a href="virus-vs-virus--dge.html#cb156-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb156-5"><a href="virus-vs-virus--dge.html#cb156-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb156-6"><a href="virus-vs-virus--dge.html#cb156-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb156-7"><a href="virus-vs-virus--dge.html#cb156-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb156-8"><a href="virus-vs-virus--dge.html#cb156-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;up&quot;</span>,</span>
<span id="cb156-9"><a href="virus-vs-virus--dge.html#cb156-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;bp&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="molecular-function-2" class="section level5 hasAnchor" number="4.2.6.2.2">
<h5><span class="header-section-number">4.2.6.2.2</span> Molecular Function<a href="virus-vs-virus--dge.html#molecular-function-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="virus-vs-virus--dge.html#cb158-1" tabindex="-1"></a>go_mf_up <span class="ot">&lt;-</span> </span>
<span id="cb158-2"><a href="virus-vs-virus--dge.html#cb158-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb158-3"><a href="virus-vs-virus--dge.html#cb158-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb158-4"><a href="virus-vs-virus--dge.html#cb158-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb158-5"><a href="virus-vs-virus--dge.html#cb158-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb158-6"><a href="virus-vs-virus--dge.html#cb158-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb158-7"><a href="virus-vs-virus--dge.html#cb158-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
</div>
<div id="cellular-component-2" class="section level5 hasAnchor" number="4.2.6.2.3">
<h5><span class="header-section-number">4.2.6.2.3</span> Cellular Component<a href="virus-vs-virus--dge.html#cellular-component-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="virus-vs-virus--dge.html#cb161-1" tabindex="-1"></a>go_cc_up <span class="ot">&lt;-</span> </span>
<span id="cb161-2"><a href="virus-vs-virus--dge.html#cb161-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb161-3"><a href="virus-vs-virus--dge.html#cb161-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb161-4"><a href="virus-vs-virus--dge.html#cb161-4" tabindex="-1"></a>                <span class="at">interestGene =</span> up<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb161-5"><a href="virus-vs-virus--dge.html#cb161-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb161-6"><a href="virus-vs-virus--dge.html#cb161-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb161-7"><a href="virus-vs-virus--dge.html#cb161-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<pre><code>## Warning in oraEnrichment(interestGeneList, referenceGeneList, geneSet, minNum =
## minNum, : No significant gene set is identified based on FDR 0.05!</code></pre>
</div>
</div>
<div id="down-regulated-genes-1" class="section level4 hasAnchor" number="4.2.6.3">
<h4><span class="header-section-number">4.2.6.3</span> Down-regulated Genes<a href="virus-vs-virus--dge.html#down-regulated-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="biological-process-3" class="section level5 hasAnchor" number="4.2.6.3.1">
<h5><span class="header-section-number">4.2.6.3.1</span> Biological Process<a href="virus-vs-virus--dge.html#biological-process-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="virus-vs-virus--dge.html#cb164-1" tabindex="-1"></a>go_bp_down <span class="ot">&lt;-</span> </span>
<span id="cb164-2"><a href="virus-vs-virus--dge.html#cb164-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb164-3"><a href="virus-vs-virus--dge.html#cb164-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb164-4"><a href="virus-vs-virus--dge.html#cb164-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb164-5"><a href="virus-vs-virus--dge.html#cb164-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb164-6"><a href="virus-vs-virus--dge.html#cb164-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb164-7"><a href="virus-vs-virus--dge.html#cb164-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb164-8"><a href="virus-vs-virus--dge.html#cb164-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>,</span>
<span id="cb164-9"><a href="virus-vs-virus--dge.html#cb164-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;bp&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="molecular-function-3" class="section level5 hasAnchor" number="4.2.6.3.2">
<h5><span class="header-section-number">4.2.6.3.2</span> Molecular Function<a href="virus-vs-virus--dge.html#molecular-function-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="virus-vs-virus--dge.html#cb166-1" tabindex="-1"></a>go_mf_down <span class="ot">&lt;-</span> </span>
<span id="cb166-2"><a href="virus-vs-virus--dge.html#cb166-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb166-3"><a href="virus-vs-virus--dge.html#cb166-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb166-4"><a href="virus-vs-virus--dge.html#cb166-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb166-5"><a href="virus-vs-virus--dge.html#cb166-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb166-6"><a href="virus-vs-virus--dge.html#cb166-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb166-7"><a href="virus-vs-virus--dge.html#cb166-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb166-8"><a href="virus-vs-virus--dge.html#cb166-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>, </span>
<span id="cb166-9"><a href="virus-vs-virus--dge.html#cb166-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;mf&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
<div id="cellular-component-3" class="section level5 hasAnchor" number="4.2.6.3.3">
<h5><span class="header-section-number">4.2.6.3.3</span> Cellular Component<a href="virus-vs-virus--dge.html#cellular-component-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="virus-vs-virus--dge.html#cb168-1" tabindex="-1"></a>go_cc_down <span class="ot">&lt;-</span> </span>
<span id="cb168-2"><a href="virus-vs-virus--dge.html#cb168-2" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb168-3"><a href="virus-vs-virus--dge.html#cb168-3" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb168-4"><a href="virus-vs-virus--dge.html#cb168-4" tabindex="-1"></a>                <span class="at">interestGene =</span> down<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb168-5"><a href="virus-vs-virus--dge.html#cb168-5" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb168-6"><a href="virus-vs-virus--dge.html#cb168-6" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb168-7"><a href="virus-vs-virus--dge.html#cb168-7" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb168-8"><a href="virus-vs-virus--dge.html#cb168-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction=</span><span class="st">&quot;down&quot;</span>, </span>
<span id="cb168-9"><a href="virus-vs-virus--dge.html#cb168-9" tabindex="-1"></a>         <span class="at">go=</span><span class="st">&quot;cc&quot;</span>)</span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
</div>
</div>
<div id="save-results-1" class="section level4 hasAnchor" number="4.2.6.4">
<h4><span class="header-section-number">4.2.6.4</span> Save Results<a href="virus-vs-virus--dge.html#save-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="virus-vs-virus--dge.html#cb170-1" tabindex="-1"></a>go <span class="ot">&lt;-</span></span>
<span id="cb170-2"><a href="virus-vs-virus--dge.html#cb170-2" tabindex="-1"></a>  <span class="fu">bind_rows</span>(go_bp_down, go_bp_up,</span>
<span id="cb170-3"><a href="virus-vs-virus--dge.html#cb170-3" tabindex="-1"></a>            go_cc_down,</span>
<span id="cb170-4"><a href="virus-vs-virus--dge.html#cb170-4" tabindex="-1"></a>            go_mf_down)</span>
<span id="cb170-5"><a href="virus-vs-virus--dge.html#cb170-5" tabindex="-1"></a></span>
<span id="cb170-6"><a href="virus-vs-virus--dge.html#cb170-6" tabindex="-1"></a>go <span class="ot">&lt;-</span> </span>
<span id="cb170-7"><a href="virus-vs-virus--dge.html#cb170-7" tabindex="-1"></a>  go <span class="sc">%&gt;%</span> </span>
<span id="cb170-8"><a href="virus-vs-virus--dge.html#cb170-8" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>link <span class="sc">&amp;</span> <span class="sc">!</span>overlapId) </span>
<span id="cb170-9"><a href="virus-vs-virus--dge.html#cb170-9" tabindex="-1"></a></span>
<span id="cb170-10"><a href="virus-vs-virus--dge.html#cb170-10" tabindex="-1"></a>go <span class="sc">%&gt;%</span> </span>
<span id="cb170-11"><a href="virus-vs-virus--dge.html#cb170-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), \(x) <span class="fu">round</span>(x, <span class="at">digits=</span><span class="dv">5</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb170-12"><a href="virus-vs-virus--dge.html#cb170-12" tabindex="-1"></a>  <span class="fu">kable</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb170-13"><a href="virus-vs-virus--dge.html#cb170-13" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="st">&quot;basic&quot;</span>, <span class="at">font_size=</span><span class="dv">9</span>)</span></code></pre></div>
<table class="table" style="font-size: 9px; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:right;">
size
</th>
<th style="text-align:right;">
overlap
</th>
<th style="text-align:right;">
expect
</th>
<th style="text-align:right;">
enrichmentRatio
</th>
<th style="text-align:right;">
pValue
</th>
<th style="text-align:right;">
FDR
</th>
<th style="text-align:left;">
userId
</th>
<th style="text-align:left;">
direction
</th>
<th style="text-align:left;">
go
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="GO:0060337" class="uri">GO:0060337</a>
</td>
<td style="text-align:left;">
type I interferon signaling pathway
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.50066
</td>
<td style="text-align:right;">
15.97902
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:left;">
IFI6;IFITM3;OASL;MX1;MX2;IRF7;ZBP1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0071357" class="uri">GO:0071357</a>
</td>
<td style="text-align:left;">
cellular response to type I interferon
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.50066
</td>
<td style="text-align:right;">
15.97902
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:left;">
IFI6;IFITM3;OASL;MX1;MX2;IRF7;ZBP1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072376" class="uri">GO:0072376</a>
</td>
<td style="text-align:left;">
protein activation cascade
</td>
<td style="text-align:right;">
39
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.33094
</td>
<td style="text-align:right;">
21.15171
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;GP1BB;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0034340" class="uri">GO:0034340</a>
</td>
<td style="text-align:left;">
response to type I interferon
</td>
<td style="text-align:right;">
63
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
0.53460
</td>
<td style="text-align:right;">
14.96447
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00006
</td>
<td style="text-align:left;">
IFI6;IFITM3;OASL;MX1;MX2;IRF7;ZBP1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006956" class="uri">GO:0006956</a>
</td>
<td style="text-align:left;">
complement activation
</td>
<td style="text-align:right;">
27
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.22911
</td>
<td style="text-align:right;">
26.18783
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00008
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006959" class="uri">GO:0006959</a>
</td>
<td style="text-align:left;">
humoral immune response
</td>
<td style="text-align:right;">
97
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
0.82311
</td>
<td style="text-align:right;">
10.93409
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB;CFD;LCN2;CXCL10;CAMP
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0098542" class="uri">GO:0098542</a>
</td>
<td style="text-align:left;">
defense response to other organism
</td>
<td style="text-align:right;">
270
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
2.29114
</td>
<td style="text-align:right;">
5.67403
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00028
</td>
<td style="text-align:left;">
IFI6;BATF2;IFITM3;OASL;APOBEC3G;MX1;MX2;DHX58;IRF7;LCN2;ISG15;CXCL10;CAMP
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0030449" class="uri">GO:0030449</a>
</td>
<td style="text-align:left;">
regulation of complement activation
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.18669
</td>
<td style="text-align:right;">
26.78301
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00058
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0051607" class="uri">GO:0051607</a>
</td>
<td style="text-align:left;">
defense response to virus
</td>
<td style="text-align:right;">
163
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
1.38317
</td>
<td style="text-align:right;">
7.22977
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00059
</td>
<td style="text-align:left;">
IFI6;IFITM3;OASL;APOBEC3G;MX1;MX2;DHX58;IRF7;ISG15;CXCL10
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:2000257" class="uri">GO:2000257</a>
</td>
<td style="text-align:left;">
regulation of protein activation cascade
</td>
<td style="text-align:right;">
23
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.19517
</td>
<td style="text-align:right;">
25.61853
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00059
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0009615" class="uri">GO:0009615</a>
</td>
<td style="text-align:left;">
response to virus
</td>
<td style="text-align:right;">
219
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
1.85837
</td>
<td style="text-align:right;">
5.91917
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00101
</td>
<td style="text-align:left;">
IFI6;IFITM3;OASL;APOBEC3G;MX1;MX2;DHX58;IRF7;LCN2;ISG15;CXCL10
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006958" class="uri">GO:0006958</a>
</td>
<td style="text-align:left;">
complement activation, classical pathway
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.11031
</td>
<td style="text-align:right;">
36.26007
</td>
<td style="text-align:right;">
0.00000
</td>
<td style="text-align:right;">
0.00138
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002920" class="uri">GO:0002920</a>
</td>
<td style="text-align:left;">
regulation of humoral immune response
</td>
<td style="text-align:right;">
31
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.26306
</td>
<td style="text-align:right;">
19.00730
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00215
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0031347" class="uri">GO:0031347</a>
</td>
<td style="text-align:left;">
regulation of defense response
</td>
<td style="text-align:right;">
458
</td>
<td style="text-align:right;">
15
</td>
<td style="text-align:right;">
3.88645
</td>
<td style="text-align:right;">
3.85956
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00215
</td>
<td style="text-align:left;">
C1QB;NR1H3;C2;C1QC;C1QA;CFB;IDO1;NCR3;APOBEC3G;DHX58;IRF7;APOE;ZBP1;LILRA5;SOCS1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045071" class="uri">GO:0045071</a>
</td>
<td style="text-align:left;">
negative regulation of viral genome replication
</td>
<td style="text-align:right;">
34
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.28851
</td>
<td style="text-align:right;">
17.33018
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.00304
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0006954" class="uri">GO:0006954</a>
</td>
<td style="text-align:left;">
inflammatory response
</td>
<td style="text-align:right;">
441
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
3.74220
</td>
<td style="text-align:right;">
3.74112
</td>
<td style="text-align:right;">
0.00002
</td>
<td style="text-align:right;">
0.00571
</td>
<td style="text-align:left;">
C1QB;NR1H3;C2;C1QC;C1QA;CFB;IDO1;SIGLEC1;NCR3;GAL;APOE;APOL3;LILRA5;CXCL10
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002455" class="uri">GO:0002455</a>
</td>
<td style="text-align:left;">
humoral immune response mediated by circulating immunoglobulin
</td>
<td style="text-align:right;">
22
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
0.18669
</td>
<td style="text-align:right;">
21.42641
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00917
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043901" class="uri">GO:0043901</a>
</td>
<td style="text-align:left;">
negative regulation of multi-organism process
</td>
<td style="text-align:right;">
107
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
0.90797
</td>
<td style="text-align:right;">
7.70950
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.00917
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;DHX58;ISG15;CAMP
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0002673" class="uri">GO:0002673</a>
</td>
<td style="text-align:left;">
regulation of acute inflammatory response
</td>
<td style="text-align:right;">
46
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.39034
</td>
<td style="text-align:right;">
12.80927
</td>
<td style="text-align:right;">
0.00004
</td>
<td style="text-align:right;">
0.01092
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019221" class="uri">GO:0019221</a>
</td>
<td style="text-align:left;">
cytokine-mediated signaling pathway
</td>
<td style="text-align:right;">
485
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
4.11557
</td>
<td style="text-align:right;">
3.40172
</td>
<td style="text-align:right;">
0.00005
</td>
<td style="text-align:right;">
0.01306
</td>
<td style="text-align:left;">
IFI6;NR1H3;IL15RA;IFITM3;OASL;MPL;MX1;MX2;IRF7;ZBP1;LCN2;SOCS1;ISG15;CXCL10
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903901" class="uri">GO:1903901</a>
</td>
<td style="text-align:left;">
negative regulation of viral life cycle
</td>
<td style="text-align:right;">
49
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.41580
</td>
<td style="text-align:right;">
12.02502
</td>
<td style="text-align:right;">
0.00006
</td>
<td style="text-align:right;">
0.01348
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0019079" class="uri">GO:0019079</a>
</td>
<td style="text-align:left;">
viral genome replication
</td>
<td style="text-align:right;">
88
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.74674
</td>
<td style="text-align:right;">
8.03490
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:right;">
0.02159
</td>
<td style="text-align:left;">
ATG16L2;IFITM3;OASL;APOBEC3G;MX1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050727" class="uri">GO:0050727</a>
</td>
<td style="text-align:left;">
regulation of inflammatory response
</td>
<td style="text-align:right;">
220
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
1.86686
</td>
<td style="text-align:right;">
4.82094
</td>
<td style="text-align:right;">
0.00010
</td>
<td style="text-align:right;">
0.02159
</td>
<td style="text-align:left;">
C1QB;NR1H3;C2;C1QC;C1QA;CFB;IDO1;APOE;LILRA5
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0070613" class="uri">GO:0070613</a>
</td>
<td style="text-align:left;">
regulation of protein processing
</td>
<td style="text-align:right;">
56
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.47520
</td>
<td style="text-align:right;">
10.52190
</td>
<td style="text-align:right;">
0.00011
</td>
<td style="text-align:right;">
0.02255
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:1903317" class="uri">GO:1903317</a>
</td>
<td style="text-align:left;">
regulation of protein maturation
</td>
<td style="text-align:right;">
58
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.49217
</td>
<td style="text-align:right;">
10.15907
</td>
<td style="text-align:right;">
0.00013
</td>
<td style="text-align:right;">
0.02562
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0048525" class="uri">GO:0048525</a>
</td>
<td style="text-align:left;">
negative regulation of viral process
</td>
<td style="text-align:right;">
59
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.50066
</td>
<td style="text-align:right;">
9.98688
</td>
<td style="text-align:right;">
0.00014
</td>
<td style="text-align:right;">
0.02674
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0050778" class="uri">GO:0050778</a>
</td>
<td style="text-align:left;">
positive regulation of immune response
</td>
<td style="text-align:right;">
490
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
4.15800
</td>
<td style="text-align:right;">
3.12651
</td>
<td style="text-align:right;">
0.00023
</td>
<td style="text-align:right;">
0.04085
</td>
<td style="text-align:left;">
C1QB;NR1H3;C2;C1QC;C1QA;HLX;CFB;IDO1;CFD;NCR3;DHX58;IRF7;ZBP1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0043903" class="uri">GO:0043903</a>
</td>
<td style="text-align:left;">
regulation of symbiosis, encompassing mutualism through parasitism
</td>
<td style="text-align:right;">
146
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
1.23891
</td>
<td style="text-align:right;">
5.65011
</td>
<td style="text-align:right;">
0.00023
</td>
<td style="text-align:right;">
0.04085
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;APOE;ISG15;CAMP
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0045069" class="uri">GO:0045069</a>
</td>
<td style="text-align:left;">
regulation of viral genome replication
</td>
<td style="text-align:right;">
66
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.56006
</td>
<td style="text-align:right;">
8.92767
</td>
<td style="text-align:right;">
0.00023
</td>
<td style="text-align:right;">
0.04085
</td>
<td style="text-align:left;">
IFITM3;OASL;APOBEC3G;MX1;ISG15
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0015701" class="uri">GO:0015701</a>
</td>
<td style="text-align:left;">
bicarbonate transport
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
3
</td>
<td style="text-align:right;">
0.04202
</td>
<td style="text-align:right;">
71.38702
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.04491
</td>
<td style="text-align:left;">
CA2;SLC4A1;RHAG
</td>
<td style="text-align:left;">
up
</td>
<td style="text-align:left;">
bp
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0072562" class="uri">GO:0072562</a>
</td>
<td style="text-align:left;">
blood microparticle
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
0.38203
</td>
<td style="text-align:right;">
13.08812
</td>
<td style="text-align:right;">
0.00004
</td>
<td style="text-align:right;">
0.02457
</td>
<td style="text-align:left;">
C1QB;C1QC;CFB;APOE;ACTA1
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
cc
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0004252" class="uri">GO:0004252</a>
</td>
<td style="text-align:left;">
serine-type endopeptidase activity
</td>
<td style="text-align:right;">
68
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.51862
</td>
<td style="text-align:right;">
11.56927
</td>
<td style="text-align:right;">
0.00001
</td>
<td style="text-align:right;">
0.01059
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0008236" class="uri">GO:0008236</a>
</td>
<td style="text-align:left;">
serine-type peptidase activity
</td>
<td style="text-align:right;">
81
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.61776
</td>
<td style="text-align:right;">
9.71248
</td>
<td style="text-align:right;">
0.00003
</td>
<td style="text-align:right;">
0.01059
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="GO:0017171" class="uri">GO:0017171</a>
</td>
<td style="text-align:left;">
serine hydrolase activity
</td>
<td style="text-align:right;">
82
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
0.62539
</td>
<td style="text-align:right;">
9.59403
</td>
<td style="text-align:right;">
0.00004
</td>
<td style="text-align:right;">
0.01059
</td>
<td style="text-align:left;">
C1QB;C2;C1QC;C1QA;CFB;CFD
</td>
<td style="text-align:left;">
down
</td>
<td style="text-align:left;">
mf
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="virus-vs-virus--dge.html#cb171-1" tabindex="-1"></a><span class="fu">write.csv</span>(go, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_go.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="kegg-analysis-1" class="section level3 hasAnchor" number="4.2.7">
<h3><span class="header-section-number">4.2.7</span> KEGG Analysis<a href="virus-vs-virus--dge.html#kegg-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="virus-vs-virus--dge.html#cb172-1" tabindex="-1"></a>kegg <span class="ot">&lt;-</span></span>
<span id="cb172-2"><a href="virus-vs-virus--dge.html#cb172-2" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">enrichDatabase =</span> <span class="st">&quot;pathway_KEGG&quot;</span>,</span>
<span id="cb172-3"><a href="virus-vs-virus--dge.html#cb172-3" tabindex="-1"></a>              <span class="at">interestGene =</span> results<span class="sc">$</span>gene, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb172-4"><a href="virus-vs-virus--dge.html#cb172-4" tabindex="-1"></a>              <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb172-5"><a href="virus-vs-virus--dge.html#cb172-5" tabindex="-1"></a>              <span class="at">minNum =</span> <span class="dv">5</span>, <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">isOutput =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<pre><code>## Loading the functional categories...
## Loading the ID list...
## Loading the reference list...
## Performing the enrichment analysis...</code></pre>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="virus-vs-virus--dge.html#cb174-1" tabindex="-1"></a><span class="fu">write.csv</span>(kegg, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_kegg.csv&quot;</span>)</span></code></pre></div>

</div>
</div>
<div id="transcriptome-null-set---edger---females" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Transcriptome Null Set - edgeR - Females<a href="virus-vs-virus--dge.html#transcriptome-null-set---edger---females" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-5" class="section level3 hasAnchor" number="4.3.1">
<h3><span class="header-section-number">4.3.1</span> Libraries<a href="virus-vs-virus--dge.html#libraries-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="virus-vs-virus--dge.html#cb175-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb175-2"><a href="virus-vs-virus--dge.html#cb175-2" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb175-3"><a href="virus-vs-virus--dge.html#cb175-3" tabindex="-1"></a><span class="fu">library</span>(WebGestaltR)</span></code></pre></div>
</div>
<div id="data-4" class="section level3 hasAnchor" number="4.3.2">
<h3><span class="header-section-number">4.3.2</span> Data<a href="virus-vs-virus--dge.html#data-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Unknown = female</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="virus-vs-virus--dge.html#cb176-1" tabindex="-1"></a>SampleInfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/metadata.csv&quot;</span>, </span>
<span id="cb176-2"><a href="virus-vs-virus--dge.html#cb176-2" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-3"><a href="virus-vs-virus--dge.html#cb176-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(disease_stage<span class="sc">==</span><span class="st">&quot;acute&quot;</span> <span class="sc">|</span> disease_stage<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;virus+&quot;</span>, <span class="st">&quot;virus-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb176-4"><a href="virus-vs-virus--dge.html#cb176-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sex<span class="sc">==</span><span class="st">&quot;male&quot;</span>)</span>
<span id="cb176-5"><a href="virus-vs-virus--dge.html#cb176-5" tabindex="-1"></a></span>
<span id="cb176-6"><a href="virus-vs-virus--dge.html#cb176-6" tabindex="-1"></a>Raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;Output Files/txome_genecounts_locid_nohemo.csv&quot;</span>, </span>
<span id="cb176-7"><a href="virus-vs-virus--dge.html#cb176-7" tabindex="-1"></a>                <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>,</span>
<span id="cb176-8"><a href="virus-vs-virus--dge.html#cb176-8" tabindex="-1"></a>                <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb176-9"><a href="virus-vs-virus--dge.html#cb176-9" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb176-10"><a href="virus-vs-virus--dge.html#cb176-10" tabindex="-1"></a>  <span class="fu">select</span>(SampleInfo<span class="sc">$</span>sample)</span></code></pre></div>
</div>
<div id="loop-differential-gene-expression-analysis" class="section level3 hasAnchor" number="4.3.3">
<h3><span class="header-section-number">4.3.3</span> Loop Differential Gene Expression Analysis<a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="configure-inputs-set-up-results-lists" class="section level4 hasAnchor" number="4.3.3.1">
<h4><span class="header-section-number">4.3.3.1</span> Configure inputs &amp; set up results lists<a href="virus-vs-virus--dge.html#configure-inputs-set-up-results-lists" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="virus-vs-virus--dge.html#cb177-1" tabindex="-1"></a>rna <span class="ot">&lt;-</span> SampleInfo<span class="sc">$</span>RNA</span>
<span id="cb177-2"><a href="virus-vs-virus--dge.html#cb177-2" tabindex="-1"></a></span>
<span id="cb177-3"><a href="virus-vs-virus--dge.html#cb177-3" tabindex="-1"></a>degenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb177-4"><a href="virus-vs-virus--dge.html#cb177-4" tabindex="-1"></a></span>
<span id="cb177-5"><a href="virus-vs-virus--dge.html#cb177-5" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-2" class="section level4 hasAnchor" number="4.3.3.2">
<h4><span class="header-section-number">4.3.3.2</span> Loop<a href="virus-vs-virus--dge.html#loop-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Random number generator used to generate 9 for set.seed() function.</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="virus-vs-virus--dge.html#cb178-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">9</span>)</span>
<span id="cb178-2"><a href="virus-vs-virus--dge.html#cb178-2" tabindex="-1"></a></span>
<span id="cb178-3"><a href="virus-vs-virus--dge.html#cb178-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb178-4"><a href="virus-vs-virus--dge.html#cb178-4" tabindex="-1"></a></span>
<span id="cb178-5"><a href="virus-vs-virus--dge.html#cb178-5" tabindex="-1"></a><span class="co">#Shuffle group assignments without replacement</span></span>
<span id="cb178-6"><a href="virus-vs-virus--dge.html#cb178-6" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">sample</span>(rna, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb178-7"><a href="virus-vs-virus--dge.html#cb178-7" tabindex="-1"></a></span>
<span id="cb178-8"><a href="virus-vs-virus--dge.html#cb178-8" tabindex="-1"></a><span class="co"># Create DGE Object</span></span>
<span id="cb178-9"><a href="virus-vs-virus--dge.html#cb178-9" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span>Raw, <span class="at">group=</span>groups)</span>
<span id="cb178-10"><a href="virus-vs-virus--dge.html#cb178-10" tabindex="-1"></a></span>
<span id="cb178-11"><a href="virus-vs-virus--dge.html#cb178-11" tabindex="-1"></a><span class="co"># Filter out lowly expressed genes</span></span>
<span id="cb178-12"><a href="virus-vs-virus--dge.html#cb178-12" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(DGE)</span>
<span id="cb178-13"><a href="virus-vs-virus--dge.html#cb178-13" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> DGE[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb178-14"><a href="virus-vs-virus--dge.html#cb178-14" tabindex="-1"></a></span>
<span id="cb178-15"><a href="virus-vs-virus--dge.html#cb178-15" tabindex="-1"></a>postfilter[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb178-16"><a href="virus-vs-virus--dge.html#cb178-16" tabindex="-1"></a>  DGE <span class="sc">%&gt;%</span> </span>
<span id="cb178-17"><a href="virus-vs-virus--dge.html#cb178-17" tabindex="-1"></a>  <span class="st">&#39;[[&#39;</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb178-18"><a href="virus-vs-virus--dge.html#cb178-18" tabindex="-1"></a>  <span class="fu">rownames</span>()</span>
<span id="cb178-19"><a href="virus-vs-virus--dge.html#cb178-19" tabindex="-1"></a></span>
<span id="cb178-20"><a href="virus-vs-virus--dge.html#cb178-20" tabindex="-1"></a><span class="co"># Normalize based on library sizes &amp; calculate dispersion</span></span>
<span id="cb178-21"><a href="virus-vs-virus--dge.html#cb178-21" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(DGE)</span>
<span id="cb178-22"><a href="virus-vs-virus--dge.html#cb178-22" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateCommonDisp</span>(DGE)</span>
<span id="cb178-23"><a href="virus-vs-virus--dge.html#cb178-23" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTrendedDisp</span>(DGE)</span>
<span id="cb178-24"><a href="virus-vs-virus--dge.html#cb178-24" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTagwiseDisp</span>(DGE)</span>
<span id="cb178-25"><a href="virus-vs-virus--dge.html#cb178-25" tabindex="-1"></a></span>
<span id="cb178-26"><a href="virus-vs-virus--dge.html#cb178-26" tabindex="-1"></a><span class="co"># Test for DEGs</span></span>
<span id="cb178-27"><a href="virus-vs-virus--dge.html#cb178-27" tabindex="-1"></a>et <span class="ot">&lt;-</span> <span class="fu">exactTest</span>(DGE, <span class="at">pair=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))</span>
<span id="cb178-28"><a href="virus-vs-virus--dge.html#cb178-28" tabindex="-1"></a></span>
<span id="cb178-29"><a href="virus-vs-virus--dge.html#cb178-29" tabindex="-1"></a><span class="co"># Select DEGs by fold change &amp; pvalue</span></span>
<span id="cb178-30"><a href="virus-vs-virus--dge.html#cb178-30" tabindex="-1"></a>results <span class="ot">&lt;-</span> </span>
<span id="cb178-31"><a href="virus-vs-virus--dge.html#cb178-31" tabindex="-1"></a>  <span class="fu">topTags</span>(et, <span class="at">n =</span> <span class="fu">dim</span>(DGE)[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb178-32"><a href="virus-vs-virus--dge.html#cb178-32" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb178-33"><a href="virus-vs-virus--dge.html#cb178-33" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb178-34"><a href="virus-vs-virus--dge.html#cb178-34" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb178-35"><a href="virus-vs-virus--dge.html#cb178-35" tabindex="-1"></a></span>
<span id="cb178-36"><a href="virus-vs-virus--dge.html#cb178-36" tabindex="-1"></a><span class="co"># Save output to list</span></span>
<span id="cb178-37"><a href="virus-vs-virus--dge.html#cb178-37" tabindex="-1"></a>degenes[[i]] <span class="ot">&lt;-</span> results<span class="sc">$</span>gene</span>
<span id="cb178-38"><a href="virus-vs-virus--dge.html#cb178-38" tabindex="-1"></a></span>
<span id="cb178-39"><a href="virus-vs-virus--dge.html#cb178-39" tabindex="-1"></a><span class="co"># End loop</span></span>
<span id="cb178-40"><a href="virus-vs-virus--dge.html#cb178-40" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-lists-of-degs" class="section level4 hasAnchor" number="4.3.3.3">
<h4><span class="header-section-number">4.3.3.3</span> Save lists of DEGs<a href="virus-vs-virus--dge.html#save-lists-of-degs" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="virus-vs-virus--dge.html#cb179-1" tabindex="-1"></a><span class="co"># Postfilter List</span></span>
<span id="cb179-2"><a href="virus-vs-virus--dge.html#cb179-2" tabindex="-1"></a>postfilter_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(postfilter[[<span class="dv">1</span>]])</span>
<span id="cb179-3"><a href="virus-vs-virus--dge.html#cb179-3" tabindex="-1"></a><span class="fu">write.csv</span>(postfilter_df, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG_postfilter.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb179-4"><a href="virus-vs-virus--dge.html#cb179-4" tabindex="-1"></a></span>
<span id="cb179-5"><a href="virus-vs-virus--dge.html#cb179-5" tabindex="-1"></a><span class="co"># DEGs</span></span>
<span id="cb179-6"><a href="virus-vs-virus--dge.html#cb179-6" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(degenes, length)))</span>
<span id="cb179-7"><a href="virus-vs-virus--dge.html#cb179-7" tabindex="-1"></a>filled <span class="ot">&lt;-</span></span>
<span id="cb179-8"><a href="virus-vs-virus--dge.html#cb179-8" tabindex="-1"></a>  <span class="fu">lapply</span>(degenes, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb179-9"><a href="virus-vs-virus--dge.html#cb179-9" tabindex="-1"></a>    ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb179-10"><a href="virus-vs-virus--dge.html#cb179-10" tabindex="-1"></a>    <span class="fu">return</span>(ans)})</span>
<span id="cb179-11"><a href="virus-vs-virus--dge.html#cb179-11" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, filled)</span>
<span id="cb179-12"><a href="virus-vs-virus--dge.html#cb179-12" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="go-enrichment-analysis-2" class="section level3 hasAnchor" number="4.3.4">
<h3><span class="header-section-number">4.3.4</span> GO Enrichment Analysis<a href="virus-vs-virus--dge.html#go-enrichment-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="import-data" class="section level4 hasAnchor" number="4.3.4.1">
<h4><span class="header-section-number">4.3.4.1</span> Import data<a href="virus-vs-virus--dge.html#import-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="virus-vs-virus--dge.html#cb180-1" tabindex="-1"></a>degenes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb180-2"><a href="virus-vs-virus--dge.html#cb180-2" tabindex="-1"></a>  <span class="fu">as.list</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb180-3"><a href="virus-vs-virus--dge.html#cb180-3" tabindex="-1"></a>  <span class="fu">lapply</span>(., <span class="cf">function</span>(x) x[<span class="sc">!</span><span class="fu">is.na</span>(x)])</span>
<span id="cb180-4"><a href="virus-vs-virus--dge.html#cb180-4" tabindex="-1"></a></span>
<span id="cb180-5"><a href="virus-vs-virus--dge.html#cb180-5" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_DEG_postfilter.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="set-up-results-lists" class="section level4 hasAnchor" number="4.3.4.2">
<h4><span class="header-section-number">4.3.4.2</span> Set up results lists<a href="virus-vs-virus--dge.html#set-up-results-lists" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="virus-vs-virus--dge.html#cb181-1" tabindex="-1"></a>go_bp <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb181-2"><a href="virus-vs-virus--dge.html#cb181-2" tabindex="-1"></a>go_mf <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb181-3"><a href="virus-vs-virus--dge.html#cb181-3" tabindex="-1"></a>go_cc <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-3" class="section level4 hasAnchor" number="4.3.4.3">
<h4><span class="header-section-number">4.3.4.3</span> Loop<a href="virus-vs-virus--dge.html#loop-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="virus-vs-virus--dge.html#cb182-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb182-2"><a href="virus-vs-virus--dge.html#cb182-2" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> degenes[[i]]</span>
<span id="cb182-3"><a href="virus-vs-virus--dge.html#cb182-3" tabindex="-1"></a>  background <span class="ot">&lt;-</span> postfilter[,<span class="dv">1</span>]</span>
<span id="cb182-4"><a href="virus-vs-virus--dge.html#cb182-4" tabindex="-1"></a>  bp_test <span class="ot">&lt;-</span> </span>
<span id="cb182-5"><a href="virus-vs-virus--dge.html#cb182-5" tabindex="-1"></a>        <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb182-6"><a href="virus-vs-virus--dge.html#cb182-6" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb182-7"><a href="virus-vs-virus--dge.html#cb182-7" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-8"><a href="virus-vs-virus--dge.html#cb182-8" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-9"><a href="virus-vs-virus--dge.html#cb182-9" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb182-10"><a href="virus-vs-virus--dge.html#cb182-10" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb182-11"><a href="virus-vs-virus--dge.html#cb182-11" tabindex="-1"></a>go_bp[[i]] <span class="ot">&lt;-</span> bp_test<span class="sc">$</span>description</span>
<span id="cb182-12"><a href="virus-vs-virus--dge.html#cb182-12" tabindex="-1"></a>  </span>
<span id="cb182-13"><a href="virus-vs-virus--dge.html#cb182-13" tabindex="-1"></a>mf_test <span class="ot">&lt;-</span> </span>
<span id="cb182-14"><a href="virus-vs-virus--dge.html#cb182-14" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb182-15"><a href="virus-vs-virus--dge.html#cb182-15" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb182-16"><a href="virus-vs-virus--dge.html#cb182-16" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-17"><a href="virus-vs-virus--dge.html#cb182-17" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-18"><a href="virus-vs-virus--dge.html#cb182-18" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb182-19"><a href="virus-vs-virus--dge.html#cb182-19" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb182-20"><a href="virus-vs-virus--dge.html#cb182-20" tabindex="-1"></a>go_mf[[i]] <span class="ot">&lt;-</span> mf_test<span class="sc">$</span>description</span>
<span id="cb182-21"><a href="virus-vs-virus--dge.html#cb182-21" tabindex="-1"></a></span>
<span id="cb182-22"><a href="virus-vs-virus--dge.html#cb182-22" tabindex="-1"></a>cc_test <span class="ot">&lt;-</span> </span>
<span id="cb182-23"><a href="virus-vs-virus--dge.html#cb182-23" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb182-24"><a href="virus-vs-virus--dge.html#cb182-24" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb182-25"><a href="virus-vs-virus--dge.html#cb182-25" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-26"><a href="virus-vs-virus--dge.html#cb182-26" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb182-27"><a href="virus-vs-virus--dge.html#cb182-27" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb182-28"><a href="virus-vs-virus--dge.html#cb182-28" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)    </span>
<span id="cb182-29"><a href="virus-vs-virus--dge.html#cb182-29" tabindex="-1"></a>go_cc[[i]] <span class="ot">&lt;-</span> cc_test<span class="sc">$</span>description</span>
<span id="cb182-30"><a href="virus-vs-virus--dge.html#cb182-30" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-files" class="section level4 hasAnchor" number="4.3.4.4">
<h4><span class="header-section-number">4.3.4.4</span> Save files<a href="virus-vs-virus--dge.html#save-files" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="virus-vs-virus--dge.html#cb183-1" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_bp, length)))</span>
<span id="cb183-2"><a href="virus-vs-virus--dge.html#cb183-2" tabindex="-1"></a>go_bp_filled <span class="ot">&lt;-</span></span>
<span id="cb183-3"><a href="virus-vs-virus--dge.html#cb183-3" tabindex="-1"></a>  <span class="fu">lapply</span>(go_bp, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb183-4"><a href="virus-vs-virus--dge.html#cb183-4" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb183-5"><a href="virus-vs-virus--dge.html#cb183-5" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb183-6"><a href="virus-vs-virus--dge.html#cb183-6" tabindex="-1"></a>go_bp_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_bp_filled)</span>
<span id="cb183-7"><a href="virus-vs-virus--dge.html#cb183-7" tabindex="-1"></a><span class="fu">write.csv</span>(go_bp_final, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb183-8"><a href="virus-vs-virus--dge.html#cb183-8" tabindex="-1"></a></span>
<span id="cb183-9"><a href="virus-vs-virus--dge.html#cb183-9" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_mf, length)))</span>
<span id="cb183-10"><a href="virus-vs-virus--dge.html#cb183-10" tabindex="-1"></a>go_mf_filled <span class="ot">&lt;-</span></span>
<span id="cb183-11"><a href="virus-vs-virus--dge.html#cb183-11" tabindex="-1"></a>  <span class="fu">lapply</span>(go_mf, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb183-12"><a href="virus-vs-virus--dge.html#cb183-12" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb183-13"><a href="virus-vs-virus--dge.html#cb183-13" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb183-14"><a href="virus-vs-virus--dge.html#cb183-14" tabindex="-1"></a>go_mf_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_mf_filled)</span>
<span id="cb183-15"><a href="virus-vs-virus--dge.html#cb183-15" tabindex="-1"></a><span class="fu">write.csv</span>(go_mf_final, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb183-16"><a href="virus-vs-virus--dge.html#cb183-16" tabindex="-1"></a></span>
<span id="cb183-17"><a href="virus-vs-virus--dge.html#cb183-17" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_cc, length)))</span>
<span id="cb183-18"><a href="virus-vs-virus--dge.html#cb183-18" tabindex="-1"></a>go_cc_filled <span class="ot">&lt;-</span></span>
<span id="cb183-19"><a href="virus-vs-virus--dge.html#cb183-19" tabindex="-1"></a>  <span class="fu">lapply</span>(go_cc, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb183-20"><a href="virus-vs-virus--dge.html#cb183-20" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb183-21"><a href="virus-vs-virus--dge.html#cb183-21" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb183-22"><a href="virus-vs-virus--dge.html#cb183-22" tabindex="-1"></a>go_cc_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_cc_filled)</span>
<span id="cb183-23"><a href="virus-vs-virus--dge.html#cb183-23" tabindex="-1"></a><span class="fu">write.csv</span>(go_cc_final, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="kegg-analysis-2" class="section level3 hasAnchor" number="4.3.5">
<h3><span class="header-section-number">4.3.5</span> KEGG Analysis<a href="virus-vs-virus--dge.html#kegg-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="set-up-results-lists-1" class="section level4 hasAnchor" number="4.3.5.1">
<h4><span class="header-section-number">4.3.5.1</span> Set up results lists<a href="virus-vs-virus--dge.html#set-up-results-lists-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="virus-vs-virus--dge.html#cb184-1" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-4" class="section level4 hasAnchor" number="4.3.5.2">
<h4><span class="header-section-number">4.3.5.2</span> Loop<a href="virus-vs-virus--dge.html#loop-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="virus-vs-virus--dge.html#cb185-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">164</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb185-2"><a href="virus-vs-virus--dge.html#cb185-2" tabindex="-1"></a>genes <span class="ot">&lt;-</span> degenes[[i]]</span>
<span id="cb185-3"><a href="virus-vs-virus--dge.html#cb185-3" tabindex="-1"></a>background <span class="ot">&lt;-</span> postfilter[,<span class="dv">1</span>]</span>
<span id="cb185-4"><a href="virus-vs-virus--dge.html#cb185-4" tabindex="-1"></a>test <span class="ot">&lt;-</span></span>
<span id="cb185-5"><a href="virus-vs-virus--dge.html#cb185-5" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">enrichDatabase =</span> <span class="st">&quot;pathway_KEGG&quot;</span>,</span>
<span id="cb185-6"><a href="virus-vs-virus--dge.html#cb185-6" tabindex="-1"></a>              <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb185-7"><a href="virus-vs-virus--dge.html#cb185-7" tabindex="-1"></a>              <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb185-8"><a href="virus-vs-virus--dge.html#cb185-8" tabindex="-1"></a>              <span class="at">minNum =</span> <span class="dv">5</span>, <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb185-9"><a href="virus-vs-virus--dge.html#cb185-9" tabindex="-1"></a></span>
<span id="cb185-10"><a href="virus-vs-virus--dge.html#cb185-10" tabindex="-1"></a><span class="fu">ifelse</span>(<span class="fu">is.data.frame</span>(test)<span class="sc">==</span><span class="st">&quot;FALSE&quot;</span>, </span>
<span id="cb185-11"><a href="virus-vs-virus--dge.html#cb185-11" tabindex="-1"></a>       kegg[[i]] <span class="ot">&lt;-</span> <span class="cn">NA</span>, </span>
<span id="cb185-12"><a href="virus-vs-virus--dge.html#cb185-12" tabindex="-1"></a>       kegg[[i]] <span class="ot">&lt;-</span> test<span class="sc">$</span>description)</span>
<span id="cb185-13"><a href="virus-vs-virus--dge.html#cb185-13" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-results-2" class="section level4 hasAnchor" number="4.3.5.3">
<h4><span class="header-section-number">4.3.5.3</span> Save results<a href="virus-vs-virus--dge.html#save-results-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="virus-vs-virus--dge.html#cb186-1" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(kegg, length)))</span>
<span id="cb186-2"><a href="virus-vs-virus--dge.html#cb186-2" tabindex="-1"></a>kegg <span class="ot">&lt;-</span></span>
<span id="cb186-3"><a href="virus-vs-virus--dge.html#cb186-3" tabindex="-1"></a>  <span class="fu">lapply</span>(kegg, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb186-4"><a href="virus-vs-virus--dge.html#cb186-4" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb186-5"><a href="virus-vs-virus--dge.html#cb186-5" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb186-6"><a href="virus-vs-virus--dge.html#cb186-6" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, kegg)</span>
<span id="cb186-7"><a href="virus-vs-virus--dge.html#cb186-7" tabindex="-1"></a><span class="fu">write.csv</span>(kegg, <span class="at">file =</span> <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>

</div>
</div>
</div>
<div id="genome-null-set---edger---females" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Genome Null Set - edgeR - Females<a href="virus-vs-virus--dge.html#genome-null-set---edger---females" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="libraries-6" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Libraries<a href="virus-vs-virus--dge.html#libraries-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="virus-vs-virus--dge.html#cb187-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb187-2"><a href="virus-vs-virus--dge.html#cb187-2" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb187-3"><a href="virus-vs-virus--dge.html#cb187-3" tabindex="-1"></a><span class="fu">library</span>(WebGestaltR)</span></code></pre></div>
</div>
<div id="data-5" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Data<a href="virus-vs-virus--dge.html#data-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Unknown = female</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="virus-vs-virus--dge.html#cb188-1" tabindex="-1"></a>SampleInfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/metadata.csv&quot;</span>, </span>
<span id="cb188-2"><a href="virus-vs-virus--dge.html#cb188-2" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb188-3"><a href="virus-vs-virus--dge.html#cb188-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">RNA=</span><span class="fu">ifelse</span>(disease_stage<span class="sc">==</span><span class="st">&quot;acute&quot;</span> <span class="sc">|</span> disease_stage<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;virus+&quot;</span>, <span class="st">&quot;virus-&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb188-4"><a href="virus-vs-virus--dge.html#cb188-4" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>sex<span class="sc">==</span><span class="st">&quot;male&quot;</span>)</span>
<span id="cb188-5"><a href="virus-vs-virus--dge.html#cb188-5" tabindex="-1"></a></span>
<span id="cb188-6"><a href="virus-vs-virus--dge.html#cb188-6" tabindex="-1"></a>Raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;Output Files/gnome_genecounts_locid_nohemo.csv&quot;</span>, </span>
<span id="cb188-7"><a href="virus-vs-virus--dge.html#cb188-7" tabindex="-1"></a>                <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>,</span>
<span id="cb188-8"><a href="virus-vs-virus--dge.html#cb188-8" tabindex="-1"></a>                <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb188-9"><a href="virus-vs-virus--dge.html#cb188-9" tabindex="-1"></a>                <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb188-10"><a href="virus-vs-virus--dge.html#cb188-10" tabindex="-1"></a>  <span class="fu">select</span>(SampleInfo<span class="sc">$</span>sample)</span></code></pre></div>
</div>
<div id="loop-differential-gene-expression-analysis-1" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> Loop Differential Gene Expression Analysis<a href="virus-vs-virus--dge.html#loop-differential-gene-expression-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="configure-inputs-set-up-results-lists-1" class="section level4 hasAnchor" number="4.4.3.1">
<h4><span class="header-section-number">4.4.3.1</span> Configure inputs &amp; set up results lists<a href="virus-vs-virus--dge.html#configure-inputs-set-up-results-lists-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="virus-vs-virus--dge.html#cb189-1" tabindex="-1"></a>rna <span class="ot">&lt;-</span> SampleInfo<span class="sc">$</span>RNA</span>
<span id="cb189-2"><a href="virus-vs-virus--dge.html#cb189-2" tabindex="-1"></a></span>
<span id="cb189-3"><a href="virus-vs-virus--dge.html#cb189-3" tabindex="-1"></a>degenes <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb189-4"><a href="virus-vs-virus--dge.html#cb189-4" tabindex="-1"></a></span>
<span id="cb189-5"><a href="virus-vs-virus--dge.html#cb189-5" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-5" class="section level4 hasAnchor" number="4.4.3.2">
<h4><span class="header-section-number">4.4.3.2</span> Loop<a href="virus-vs-virus--dge.html#loop-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Random number generator used to generate 9 for set.seed() function.</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="virus-vs-virus--dge.html#cb190-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">9</span>)</span>
<span id="cb190-2"><a href="virus-vs-virus--dge.html#cb190-2" tabindex="-1"></a></span>
<span id="cb190-3"><a href="virus-vs-virus--dge.html#cb190-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb190-4"><a href="virus-vs-virus--dge.html#cb190-4" tabindex="-1"></a></span>
<span id="cb190-5"><a href="virus-vs-virus--dge.html#cb190-5" tabindex="-1"></a><span class="co">#Shuffle group assignments without replacement</span></span>
<span id="cb190-6"><a href="virus-vs-virus--dge.html#cb190-6" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">sample</span>(rna, <span class="at">replace =</span> <span class="cn">FALSE</span>, <span class="at">prob =</span> <span class="cn">NULL</span>)</span>
<span id="cb190-7"><a href="virus-vs-virus--dge.html#cb190-7" tabindex="-1"></a></span>
<span id="cb190-8"><a href="virus-vs-virus--dge.html#cb190-8" tabindex="-1"></a><span class="co"># Create DGE Object</span></span>
<span id="cb190-9"><a href="virus-vs-virus--dge.html#cb190-9" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span>Raw, <span class="at">group=</span>groups)</span>
<span id="cb190-10"><a href="virus-vs-virus--dge.html#cb190-10" tabindex="-1"></a></span>
<span id="cb190-11"><a href="virus-vs-virus--dge.html#cb190-11" tabindex="-1"></a><span class="co"># Filter out lowly expressed genes</span></span>
<span id="cb190-12"><a href="virus-vs-virus--dge.html#cb190-12" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(DGE)</span>
<span id="cb190-13"><a href="virus-vs-virus--dge.html#cb190-13" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> DGE[keep, , keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb190-14"><a href="virus-vs-virus--dge.html#cb190-14" tabindex="-1"></a></span>
<span id="cb190-15"><a href="virus-vs-virus--dge.html#cb190-15" tabindex="-1"></a>postfilter[[i]] <span class="ot">&lt;-</span> </span>
<span id="cb190-16"><a href="virus-vs-virus--dge.html#cb190-16" tabindex="-1"></a>  DGE <span class="sc">%&gt;%</span> </span>
<span id="cb190-17"><a href="virus-vs-virus--dge.html#cb190-17" tabindex="-1"></a>  <span class="st">&#39;[[&#39;</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb190-18"><a href="virus-vs-virus--dge.html#cb190-18" tabindex="-1"></a>  <span class="fu">rownames</span>()</span>
<span id="cb190-19"><a href="virus-vs-virus--dge.html#cb190-19" tabindex="-1"></a></span>
<span id="cb190-20"><a href="virus-vs-virus--dge.html#cb190-20" tabindex="-1"></a><span class="co"># Normalize based on library sizes &amp; calculate dispersion</span></span>
<span id="cb190-21"><a href="virus-vs-virus--dge.html#cb190-21" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(DGE)</span>
<span id="cb190-22"><a href="virus-vs-virus--dge.html#cb190-22" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateCommonDisp</span>(DGE)</span>
<span id="cb190-23"><a href="virus-vs-virus--dge.html#cb190-23" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTrendedDisp</span>(DGE)</span>
<span id="cb190-24"><a href="virus-vs-virus--dge.html#cb190-24" tabindex="-1"></a>DGE <span class="ot">&lt;-</span> <span class="fu">estimateTagwiseDisp</span>(DGE)</span>
<span id="cb190-25"><a href="virus-vs-virus--dge.html#cb190-25" tabindex="-1"></a></span>
<span id="cb190-26"><a href="virus-vs-virus--dge.html#cb190-26" tabindex="-1"></a><span class="co"># Test for DEGs</span></span>
<span id="cb190-27"><a href="virus-vs-virus--dge.html#cb190-27" tabindex="-1"></a>et <span class="ot">&lt;-</span> <span class="fu">exactTest</span>(DGE, <span class="at">pair=</span><span class="fu">c</span>(<span class="st">&quot;virus-&quot;</span>, <span class="st">&quot;virus+&quot;</span>))</span>
<span id="cb190-28"><a href="virus-vs-virus--dge.html#cb190-28" tabindex="-1"></a></span>
<span id="cb190-29"><a href="virus-vs-virus--dge.html#cb190-29" tabindex="-1"></a><span class="co"># Select DEGs by fold change &amp; pvalue</span></span>
<span id="cb190-30"><a href="virus-vs-virus--dge.html#cb190-30" tabindex="-1"></a>results <span class="ot">&lt;-</span> </span>
<span id="cb190-31"><a href="virus-vs-virus--dge.html#cb190-31" tabindex="-1"></a>  <span class="fu">topTags</span>(et, <span class="at">n =</span> <span class="fu">dim</span>(DGE)[<span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb190-32"><a href="virus-vs-virus--dge.html#cb190-32" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb190-33"><a href="virus-vs-virus--dge.html#cb190-33" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span> <span class="sc">&amp;</span> PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb190-34"><a href="virus-vs-virus--dge.html#cb190-34" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb190-35"><a href="virus-vs-virus--dge.html#cb190-35" tabindex="-1"></a></span>
<span id="cb190-36"><a href="virus-vs-virus--dge.html#cb190-36" tabindex="-1"></a><span class="co"># Save output to list</span></span>
<span id="cb190-37"><a href="virus-vs-virus--dge.html#cb190-37" tabindex="-1"></a>degenes[[i]] <span class="ot">&lt;-</span> results<span class="sc">$</span>gene</span>
<span id="cb190-38"><a href="virus-vs-virus--dge.html#cb190-38" tabindex="-1"></a></span>
<span id="cb190-39"><a href="virus-vs-virus--dge.html#cb190-39" tabindex="-1"></a><span class="co"># End loop</span></span>
<span id="cb190-40"><a href="virus-vs-virus--dge.html#cb190-40" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-lists-of-degs-1" class="section level4 hasAnchor" number="4.4.3.3">
<h4><span class="header-section-number">4.4.3.3</span> Save lists of DEGs<a href="virus-vs-virus--dge.html#save-lists-of-degs-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="virus-vs-virus--dge.html#cb191-1" tabindex="-1"></a><span class="co"># Postfilter Lists</span></span>
<span id="cb191-2"><a href="virus-vs-virus--dge.html#cb191-2" tabindex="-1"></a>postfilter_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(postfilter[[<span class="dv">1</span>]])</span>
<span id="cb191-3"><a href="virus-vs-virus--dge.html#cb191-3" tabindex="-1"></a><span class="fu">write.csv</span>(postfilter_df, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG_postfilter.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb191-4"><a href="virus-vs-virus--dge.html#cb191-4" tabindex="-1"></a></span>
<span id="cb191-5"><a href="virus-vs-virus--dge.html#cb191-5" tabindex="-1"></a><span class="co"># DEGs</span></span>
<span id="cb191-6"><a href="virus-vs-virus--dge.html#cb191-6" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(degenes, length)))</span>
<span id="cb191-7"><a href="virus-vs-virus--dge.html#cb191-7" tabindex="-1"></a>filled <span class="ot">&lt;-</span></span>
<span id="cb191-8"><a href="virus-vs-virus--dge.html#cb191-8" tabindex="-1"></a>  <span class="fu">lapply</span>(degenes, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb191-9"><a href="virus-vs-virus--dge.html#cb191-9" tabindex="-1"></a>    ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb191-10"><a href="virus-vs-virus--dge.html#cb191-10" tabindex="-1"></a>    <span class="fu">return</span>(ans)})</span>
<span id="cb191-11"><a href="virus-vs-virus--dge.html#cb191-11" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, filled)</span>
<span id="cb191-12"><a href="virus-vs-virus--dge.html#cb191-12" tabindex="-1"></a><span class="fu">write.csv</span>(df, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="go-enrichment-analysis-3" class="section level3 hasAnchor" number="4.4.4">
<h3><span class="header-section-number">4.4.4</span> GO Enrichment Analysis<a href="virus-vs-virus--dge.html#go-enrichment-analysis-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="import-data-1" class="section level4 hasAnchor" number="4.4.4.1">
<h4><span class="header-section-number">4.4.4.1</span> Import data<a href="virus-vs-virus--dge.html#import-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="virus-vs-virus--dge.html#cb192-1" tabindex="-1"></a>degenes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG.csv&quot;</span>)</span>
<span id="cb192-2"><a href="virus-vs-virus--dge.html#cb192-2" tabindex="-1"></a></span>
<span id="cb192-3"><a href="virus-vs-virus--dge.html#cb192-3" tabindex="-1"></a>postfilter <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_DEG_postfilter.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="set-up-results-lists-2" class="section level4 hasAnchor" number="4.4.4.2">
<h4><span class="header-section-number">4.4.4.2</span> Set up results lists<a href="virus-vs-virus--dge.html#set-up-results-lists-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="virus-vs-virus--dge.html#cb193-1" tabindex="-1"></a>go_bp <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb193-2"><a href="virus-vs-virus--dge.html#cb193-2" tabindex="-1"></a>go_mf <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb193-3"><a href="virus-vs-virus--dge.html#cb193-3" tabindex="-1"></a>go_cc <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-6" class="section level4 hasAnchor" number="4.4.4.3">
<h4><span class="header-section-number">4.4.4.3</span> Loop<a href="virus-vs-virus--dge.html#loop-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="virus-vs-virus--dge.html#cb194-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb194-2"><a href="virus-vs-virus--dge.html#cb194-2" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> degenes[[i]]</span>
<span id="cb194-3"><a href="virus-vs-virus--dge.html#cb194-3" tabindex="-1"></a>  background <span class="ot">&lt;-</span> postfilter[,<span class="dv">1</span>]</span>
<span id="cb194-4"><a href="virus-vs-virus--dge.html#cb194-4" tabindex="-1"></a>  bp_test <span class="ot">&lt;-</span> </span>
<span id="cb194-5"><a href="virus-vs-virus--dge.html#cb194-5" tabindex="-1"></a>        <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb194-6"><a href="virus-vs-virus--dge.html#cb194-6" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Biological_Process&quot;</span>,</span>
<span id="cb194-7"><a href="virus-vs-virus--dge.html#cb194-7" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-8"><a href="virus-vs-virus--dge.html#cb194-8" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-9"><a href="virus-vs-virus--dge.html#cb194-9" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb194-10"><a href="virus-vs-virus--dge.html#cb194-10" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb194-11"><a href="virus-vs-virus--dge.html#cb194-11" tabindex="-1"></a>go_bp[[i]] <span class="ot">&lt;-</span> bp_test<span class="sc">$</span>description</span>
<span id="cb194-12"><a href="virus-vs-virus--dge.html#cb194-12" tabindex="-1"></a></span>
<span id="cb194-13"><a href="virus-vs-virus--dge.html#cb194-13" tabindex="-1"></a>mf_test <span class="ot">&lt;-</span> </span>
<span id="cb194-14"><a href="virus-vs-virus--dge.html#cb194-14" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb194-15"><a href="virus-vs-virus--dge.html#cb194-15" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Molecular_Function&quot;</span>,</span>
<span id="cb194-16"><a href="virus-vs-virus--dge.html#cb194-16" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-17"><a href="virus-vs-virus--dge.html#cb194-17" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-18"><a href="virus-vs-virus--dge.html#cb194-18" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb194-19"><a href="virus-vs-virus--dge.html#cb194-19" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb194-20"><a href="virus-vs-virus--dge.html#cb194-20" tabindex="-1"></a>go_mf[[i]] <span class="ot">&lt;-</span> mf_test<span class="sc">$</span>description</span>
<span id="cb194-21"><a href="virus-vs-virus--dge.html#cb194-21" tabindex="-1"></a></span>
<span id="cb194-22"><a href="virus-vs-virus--dge.html#cb194-22" tabindex="-1"></a>cc_test <span class="ot">&lt;-</span> </span>
<span id="cb194-23"><a href="virus-vs-virus--dge.html#cb194-23" tabindex="-1"></a>    <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod=</span><span class="st">&quot;ORA&quot;</span>, <span class="at">organism=</span><span class="st">&quot;hsapiens&quot;</span>, </span>
<span id="cb194-24"><a href="virus-vs-virus--dge.html#cb194-24" tabindex="-1"></a>                <span class="at">enrichDatabase=</span><span class="st">&quot;geneontology_Cellular_Component&quot;</span>,</span>
<span id="cb194-25"><a href="virus-vs-virus--dge.html#cb194-25" tabindex="-1"></a>                <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-26"><a href="virus-vs-virus--dge.html#cb194-26" tabindex="-1"></a>                <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb194-27"><a href="virus-vs-virus--dge.html#cb194-27" tabindex="-1"></a>                <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdrThr =</span> <span class="fl">0.05</span>, </span>
<span id="cb194-28"><a href="virus-vs-virus--dge.html#cb194-28" tabindex="-1"></a>                <span class="at">isOutput =</span> <span class="cn">FALSE</span>)    </span>
<span id="cb194-29"><a href="virus-vs-virus--dge.html#cb194-29" tabindex="-1"></a>go_cc[[i]] <span class="ot">&lt;-</span> cc_test<span class="sc">$</span>description</span>
<span id="cb194-30"><a href="virus-vs-virus--dge.html#cb194-30" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-files-1" class="section level4 hasAnchor" number="4.4.4.4">
<h4><span class="header-section-number">4.4.4.4</span> Save files<a href="virus-vs-virus--dge.html#save-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="virus-vs-virus--dge.html#cb195-1" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_bp, length)))</span>
<span id="cb195-2"><a href="virus-vs-virus--dge.html#cb195-2" tabindex="-1"></a>go_bp_filled <span class="ot">&lt;-</span></span>
<span id="cb195-3"><a href="virus-vs-virus--dge.html#cb195-3" tabindex="-1"></a>  <span class="fu">lapply</span>(go_bp, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb195-4"><a href="virus-vs-virus--dge.html#cb195-4" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb195-5"><a href="virus-vs-virus--dge.html#cb195-5" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb195-6"><a href="virus-vs-virus--dge.html#cb195-6" tabindex="-1"></a>go_bp_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_bp_filled)</span>
<span id="cb195-7"><a href="virus-vs-virus--dge.html#cb195-7" tabindex="-1"></a><span class="fu">write.csv</span>(go_bp_final, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb195-8"><a href="virus-vs-virus--dge.html#cb195-8" tabindex="-1"></a></span>
<span id="cb195-9"><a href="virus-vs-virus--dge.html#cb195-9" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_mf, length)))</span>
<span id="cb195-10"><a href="virus-vs-virus--dge.html#cb195-10" tabindex="-1"></a>go_mf_filled <span class="ot">&lt;-</span></span>
<span id="cb195-11"><a href="virus-vs-virus--dge.html#cb195-11" tabindex="-1"></a>  <span class="fu">lapply</span>(go_mf, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb195-12"><a href="virus-vs-virus--dge.html#cb195-12" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb195-13"><a href="virus-vs-virus--dge.html#cb195-13" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb195-14"><a href="virus-vs-virus--dge.html#cb195-14" tabindex="-1"></a>go_mf_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_mf_filled)</span>
<span id="cb195-15"><a href="virus-vs-virus--dge.html#cb195-15" tabindex="-1"></a><span class="fu">write.csv</span>(go_mf_final, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb195-16"><a href="virus-vs-virus--dge.html#cb195-16" tabindex="-1"></a></span>
<span id="cb195-17"><a href="virus-vs-virus--dge.html#cb195-17" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(go_cc, length)))</span>
<span id="cb195-18"><a href="virus-vs-virus--dge.html#cb195-18" tabindex="-1"></a>go_cc_filled <span class="ot">&lt;-</span></span>
<span id="cb195-19"><a href="virus-vs-virus--dge.html#cb195-19" tabindex="-1"></a>  <span class="fu">lapply</span>(go_cc, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb195-20"><a href="virus-vs-virus--dge.html#cb195-20" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb195-21"><a href="virus-vs-virus--dge.html#cb195-21" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb195-22"><a href="virus-vs-virus--dge.html#cb195-22" tabindex="-1"></a>go_cc_final <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, go_cc_filled)</span>
<span id="cb195-23"><a href="virus-vs-virus--dge.html#cb195-23" tabindex="-1"></a><span class="fu">write.csv</span>(go_cc_final, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
</div>
<div id="kegg-analysis-3" class="section level3 hasAnchor" number="4.4.5">
<h3><span class="header-section-number">4.4.5</span> KEGG Analysis<a href="virus-vs-virus--dge.html#kegg-analysis-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="set-up-results-lists-3" class="section level4 hasAnchor" number="4.4.5.1">
<h4><span class="header-section-number">4.4.5.1</span> Set up results lists<a href="virus-vs-virus--dge.html#set-up-results-lists-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="virus-vs-virus--dge.html#cb196-1" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="fu">vector</span>(<span class="st">&#39;list&#39;</span>, <span class="dv">1000</span>), <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span></code></pre></div>
</div>
<div id="loop-7" class="section level4 hasAnchor" number="4.4.5.2">
<h4><span class="header-section-number">4.4.5.2</span> Loop<a href="virus-vs-virus--dge.html#loop-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="virus-vs-virus--dge.html#cb197-1" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) {</span>
<span id="cb197-2"><a href="virus-vs-virus--dge.html#cb197-2" tabindex="-1"></a>genes <span class="ot">&lt;-</span> degenes[[i]]</span>
<span id="cb197-3"><a href="virus-vs-virus--dge.html#cb197-3" tabindex="-1"></a>background <span class="ot">&lt;-</span> postfilter[,<span class="dv">1</span>]</span>
<span id="cb197-4"><a href="virus-vs-virus--dge.html#cb197-4" tabindex="-1"></a>test <span class="ot">&lt;-</span></span>
<span id="cb197-5"><a href="virus-vs-virus--dge.html#cb197-5" tabindex="-1"></a>  <span class="fu">WebGestaltR</span>(<span class="at">enrichMethod =</span> <span class="st">&quot;ORA&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">enrichDatabase =</span> <span class="st">&quot;pathway_KEGG&quot;</span>,</span>
<span id="cb197-6"><a href="virus-vs-virus--dge.html#cb197-6" tabindex="-1"></a>              <span class="at">interestGene =</span> genes, <span class="at">interestGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb197-7"><a href="virus-vs-virus--dge.html#cb197-7" tabindex="-1"></a>              <span class="at">referenceGene =</span> background, <span class="at">referenceGeneType =</span> <span class="st">&quot;genesymbol&quot;</span>, </span>
<span id="cb197-8"><a href="virus-vs-virus--dge.html#cb197-8" tabindex="-1"></a>              <span class="at">minNum =</span> <span class="dv">5</span>, <span class="at">sigMethod =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">isOutput =</span> <span class="cn">FALSE</span>)</span>
<span id="cb197-9"><a href="virus-vs-virus--dge.html#cb197-9" tabindex="-1"></a></span>
<span id="cb197-10"><a href="virus-vs-virus--dge.html#cb197-10" tabindex="-1"></a><span class="fu">ifelse</span>(<span class="fu">is.data.frame</span>(test)<span class="sc">==</span><span class="st">&quot;FALSE&quot;</span>, </span>
<span id="cb197-11"><a href="virus-vs-virus--dge.html#cb197-11" tabindex="-1"></a>       kegg[[i]] <span class="ot">&lt;-</span> <span class="cn">NA</span>, </span>
<span id="cb197-12"><a href="virus-vs-virus--dge.html#cb197-12" tabindex="-1"></a>       kegg[[i]] <span class="ot">&lt;-</span> test<span class="sc">$</span>description)</span>
<span id="cb197-13"><a href="virus-vs-virus--dge.html#cb197-13" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="save-results-3" class="section level4 hasAnchor" number="4.4.5.3">
<h4><span class="header-section-number">4.4.5.3</span> Save results<a href="virus-vs-virus--dge.html#save-results-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="virus-vs-virus--dge.html#cb198-1" tabindex="-1"></a>max_length <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(kegg, length)))</span>
<span id="cb198-2"><a href="virus-vs-virus--dge.html#cb198-2" tabindex="-1"></a>kegg <span class="ot">&lt;-</span></span>
<span id="cb198-3"><a href="virus-vs-virus--dge.html#cb198-3" tabindex="-1"></a>  <span class="fu">lapply</span>(kegg, <span class="cf">function</span>(x) {ans <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>,<span class="at">length=</span>max_length);</span>
<span id="cb198-4"><a href="virus-vs-virus--dge.html#cb198-4" tabindex="-1"></a>  ans[<span class="dv">0</span><span class="sc">:</span><span class="fu">length</span>(x)]<span class="ot">&lt;-</span> x;</span>
<span id="cb198-5"><a href="virus-vs-virus--dge.html#cb198-5" tabindex="-1"></a>  <span class="fu">return</span>(ans)})</span>
<span id="cb198-6"><a href="virus-vs-virus--dge.html#cb198-6" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, kegg)</span>
<span id="cb198-7"><a href="virus-vs-virus--dge.html#cb198-7" tabindex="-1"></a><span class="fu">write.csv</span>(kegg, <span class="at">file =</span> <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>

</div>
</div>
</div>
<div id="refine-gokegg" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Refine GO/KEGG<a href="virus-vs-virus--dge.html#refine-gokegg" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>*Based on Null Set results</p>
<div id="libraries-7" class="section level3 hasAnchor" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> Libraries<a href="virus-vs-virus--dge.html#libraries-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="virus-vs-virus--dge.html#cb199-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb199-2"><a href="virus-vs-virus--dge.html#cb199-2" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb199-3"><a href="virus-vs-virus--dge.html#cb199-3" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb199-4"><a href="virus-vs-virus--dge.html#cb199-4" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span>
<span id="cb199-5"><a href="virus-vs-virus--dge.html#cb199-5" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span></code></pre></div>
</div>
<div id="data-6" class="section level3 hasAnchor" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> Data<a href="virus-vs-virus--dge.html#data-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="gokegg-data" class="section level4 hasAnchor" number="4.5.2.1">
<h4><span class="header-section-number">4.5.2.1</span> GO/KEGG Data<a href="virus-vs-virus--dge.html#gokegg-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="virus-vs-virus--dge.html#cb200-1" tabindex="-1"></a>txome_go <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_go.csv&quot;</span>)</span>
<span id="cb200-2"><a href="virus-vs-virus--dge.html#cb200-2" tabindex="-1"></a>t_null_bp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>)</span>
<span id="cb200-3"><a href="virus-vs-virus--dge.html#cb200-3" tabindex="-1"></a>t_null_mf <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>)</span>
<span id="cb200-4"><a href="virus-vs-virus--dge.html#cb200-4" tabindex="-1"></a>t_null_cc <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>)</span>
<span id="cb200-5"><a href="virus-vs-virus--dge.html#cb200-5" tabindex="-1"></a></span>
<span id="cb200-6"><a href="virus-vs-virus--dge.html#cb200-6" tabindex="-1"></a></span>
<span id="cb200-7"><a href="virus-vs-virus--dge.html#cb200-7" tabindex="-1"></a>txome_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_kegg.csv&quot;</span>)</span>
<span id="cb200-8"><a href="virus-vs-virus--dge.html#cb200-8" tabindex="-1"></a>t_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>)</span>
<span id="cb200-9"><a href="virus-vs-virus--dge.html#cb200-9" tabindex="-1"></a></span>
<span id="cb200-10"><a href="virus-vs-virus--dge.html#cb200-10" tabindex="-1"></a>gnome_go <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_go.csv&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb200-11"><a href="virus-vs-virus--dge.html#cb200-11" tabindex="-1"></a>g_null_bp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_bp.csv&quot;</span>)</span>
<span id="cb200-12"><a href="virus-vs-virus--dge.html#cb200-12" tabindex="-1"></a>g_null_mf <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_mf.csv&quot;</span>)</span>
<span id="cb200-13"><a href="virus-vs-virus--dge.html#cb200-13" tabindex="-1"></a>g_null_cc <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_GO_cc.csv&quot;</span>)</span>
<span id="cb200-14"><a href="virus-vs-virus--dge.html#cb200-14" tabindex="-1"></a></span>
<span id="cb200-15"><a href="virus-vs-virus--dge.html#cb200-15" tabindex="-1"></a>gnome_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_kegg.csv&quot;</span>)</span>
<span id="cb200-16"><a href="virus-vs-virus--dge.html#cb200-16" tabindex="-1"></a>g_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_null_kegg.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="raw-gene-expression" class="section level4 hasAnchor" number="4.5.2.2">
<h4><span class="header-section-number">4.5.2.2</span> Raw Gene Expression<a href="virus-vs-virus--dge.html#raw-gene-expression" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="virus-vs-virus--dge.html#cb201-1" tabindex="-1"></a>t_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_normcounts.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb201-2"><a href="virus-vs-virus--dge.html#cb201-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;gene&quot;</span> <span class="ot">=</span> <span class="st">&quot;X&quot;</span>)</span>
<span id="cb201-3"><a href="virus-vs-virus--dge.html#cb201-3" tabindex="-1"></a></span>
<span id="cb201-4"><a href="virus-vs-virus--dge.html#cb201-4" tabindex="-1"></a>g_raw <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_normcounts.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb201-5"><a href="virus-vs-virus--dge.html#cb201-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;gene&quot;</span> <span class="ot">=</span> <span class="st">&quot;X&quot;</span>)</span>
<span id="cb201-6"><a href="virus-vs-virus--dge.html#cb201-6" tabindex="-1"></a></span>
<span id="cb201-7"><a href="virus-vs-virus--dge.html#cb201-7" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Input Files/metadata.csv&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="transcriptome-1" class="section level3 hasAnchor" number="4.5.3">
<h3><span class="header-section-number">4.5.3</span> Transcriptome<a href="virus-vs-virus--dge.html#transcriptome-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="go-term" class="section level4 hasAnchor" number="4.5.3.1">
<h4><span class="header-section-number">4.5.3.1</span> GO Term<a href="virus-vs-virus--dge.html#go-term" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="what-terms-are-overrepresented-in-10-of-nullsets" class="section level5 hasAnchor" number="4.5.3.1.1">
<h5><span class="header-section-number">4.5.3.1.1</span> What terms are overrepresented in &gt;10% of nullsets?<a href="virus-vs-virus--dge.html#what-terms-are-overrepresented-in-10-of-nullsets" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="virus-vs-virus--dge.html#cb202-1" tabindex="-1"></a><span class="co"># BP</span></span>
<span id="cb202-2"><a href="virus-vs-virus--dge.html#cb202-2" tabindex="-1"></a>t_null_bp_10 <span class="ot">&lt;-</span></span>
<span id="cb202-3"><a href="virus-vs-virus--dge.html#cb202-3" tabindex="-1"></a>  t_null_bp <span class="sc">%&gt;%</span> </span>
<span id="cb202-4"><a href="virus-vs-virus--dge.html#cb202-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb202-5"><a href="virus-vs-virus--dge.html#cb202-5" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-6"><a href="virus-vs-virus--dge.html#cb202-6" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-7"><a href="virus-vs-virus--dge.html#cb202-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb202-8"><a href="virus-vs-virus--dge.html#cb202-8" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span>
<span id="cb202-9"><a href="virus-vs-virus--dge.html#cb202-9" tabindex="-1"></a></span>
<span id="cb202-10"><a href="virus-vs-virus--dge.html#cb202-10" tabindex="-1"></a><span class="co"># MF</span></span>
<span id="cb202-11"><a href="virus-vs-virus--dge.html#cb202-11" tabindex="-1"></a>t_null_mf_10 <span class="ot">&lt;-</span></span>
<span id="cb202-12"><a href="virus-vs-virus--dge.html#cb202-12" tabindex="-1"></a>  t_null_mf <span class="sc">%&gt;%</span> </span>
<span id="cb202-13"><a href="virus-vs-virus--dge.html#cb202-13" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb202-14"><a href="virus-vs-virus--dge.html#cb202-14" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-15"><a href="virus-vs-virus--dge.html#cb202-15" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-16"><a href="virus-vs-virus--dge.html#cb202-16" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb202-17"><a href="virus-vs-virus--dge.html#cb202-17" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span>
<span id="cb202-18"><a href="virus-vs-virus--dge.html#cb202-18" tabindex="-1"></a></span>
<span id="cb202-19"><a href="virus-vs-virus--dge.html#cb202-19" tabindex="-1"></a><span class="co"># CC</span></span>
<span id="cb202-20"><a href="virus-vs-virus--dge.html#cb202-20" tabindex="-1"></a>t_null_cc_10 <span class="ot">&lt;-</span></span>
<span id="cb202-21"><a href="virus-vs-virus--dge.html#cb202-21" tabindex="-1"></a>  t_null_cc <span class="sc">%&gt;%</span> </span>
<span id="cb202-22"><a href="virus-vs-virus--dge.html#cb202-22" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb202-23"><a href="virus-vs-virus--dge.html#cb202-23" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-24"><a href="virus-vs-virus--dge.html#cb202-24" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb202-25"><a href="virus-vs-virus--dge.html#cb202-25" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb202-26"><a href="virus-vs-virus--dge.html#cb202-26" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span></code></pre></div>
</div>
<div id="filter-results" class="section level5 hasAnchor" number="4.5.3.1.2">
<h5><span class="header-section-number">4.5.3.1.2</span> Filter results<a href="virus-vs-virus--dge.html#filter-results" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="virus-vs-virus--dge.html#cb203-1" tabindex="-1"></a>final_t_bp <span class="ot">&lt;-</span> </span>
<span id="cb203-2"><a href="virus-vs-virus--dge.html#cb203-2" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb203-3"><a href="virus-vs-virus--dge.html#cb203-3" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb203-4"><a href="virus-vs-virus--dge.html#cb203-4" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> t_null_bp_10<span class="sc">$</span>value)</span>
<span id="cb203-5"><a href="virus-vs-virus--dge.html#cb203-5" tabindex="-1"></a>         </span>
<span id="cb203-6"><a href="virus-vs-virus--dge.html#cb203-6" tabindex="-1"></a>final_t_mf <span class="ot">&lt;-</span></span>
<span id="cb203-7"><a href="virus-vs-virus--dge.html#cb203-7" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb203-8"><a href="virus-vs-virus--dge.html#cb203-8" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;mf&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb203-9"><a href="virus-vs-virus--dge.html#cb203-9" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> t_null_mf_10<span class="sc">$</span>value)</span>
<span id="cb203-10"><a href="virus-vs-virus--dge.html#cb203-10" tabindex="-1"></a></span>
<span id="cb203-11"><a href="virus-vs-virus--dge.html#cb203-11" tabindex="-1"></a>final_t_cc <span class="ot">&lt;-</span></span>
<span id="cb203-12"><a href="virus-vs-virus--dge.html#cb203-12" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb203-13"><a href="virus-vs-virus--dge.html#cb203-13" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb203-14"><a href="virus-vs-virus--dge.html#cb203-14" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> t_null_cc_10<span class="sc">$</span>value)</span></code></pre></div>
</div>
</div>
<div id="kegg-pathway" class="section level4 hasAnchor" number="4.5.3.2">
<h4><span class="header-section-number">4.5.3.2</span> KEGG Pathway<a href="virus-vs-virus--dge.html#kegg-pathway" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="what-pathways-are-overrepreesented-in-10-of-nullsets" class="section level5 hasAnchor" number="4.5.3.2.1">
<h5><span class="header-section-number">4.5.3.2.1</span> What pathways are overrepreesented in &gt;10% of nullsets?<a href="virus-vs-virus--dge.html#what-pathways-are-overrepreesented-in-10-of-nullsets" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb204"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb204-1"><a href="virus-vs-virus--dge.html#cb204-1" tabindex="-1"></a>t_kegg_10 <span class="ot">&lt;-</span></span>
<span id="cb204-2"><a href="virus-vs-virus--dge.html#cb204-2" tabindex="-1"></a>  t_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb204-3"><a href="virus-vs-virus--dge.html#cb204-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb204-4"><a href="virus-vs-virus--dge.html#cb204-4" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb204-5"><a href="virus-vs-virus--dge.html#cb204-5" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb204-6"><a href="virus-vs-virus--dge.html#cb204-6" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb204-7"><a href="virus-vs-virus--dge.html#cb204-7" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span></code></pre></div>
</div>
<div id="filter-results-1" class="section level5 hasAnchor" number="4.5.3.2.2">
<h5><span class="header-section-number">4.5.3.2.2</span> Filter Results<a href="virus-vs-virus--dge.html#filter-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="virus-vs-virus--dge.html#cb205-1" tabindex="-1"></a>txome_kegg_final <span class="ot">&lt;-</span> </span>
<span id="cb205-2"><a href="virus-vs-virus--dge.html#cb205-2" tabindex="-1"></a>  txome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb205-3"><a href="virus-vs-virus--dge.html#cb205-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>description <span class="sc">%in%</span> t_kegg_10<span class="sc">$</span>value)</span>
<span id="cb205-4"><a href="virus-vs-virus--dge.html#cb205-4" tabindex="-1"></a>txome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb205-5"><a href="virus-vs-virus--dge.html#cb205-5" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
</div>
</div>
</div>
<div id="genome-1" class="section level3 hasAnchor" number="4.5.4">
<h3><span class="header-section-number">4.5.4</span> Genome<a href="virus-vs-virus--dge.html#genome-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="go-term-1" class="section level4 hasAnchor" number="4.5.4.1">
<h4><span class="header-section-number">4.5.4.1</span> GO Term<a href="virus-vs-virus--dge.html#go-term-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="what-terms-are-overrepresented-in-10-of-nullsets-1" class="section level5 hasAnchor" number="4.5.4.1.1">
<h5><span class="header-section-number">4.5.4.1.1</span> What terms are overrepresented in &gt;10% of nullsets?<a href="virus-vs-virus--dge.html#what-terms-are-overrepresented-in-10-of-nullsets-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="virus-vs-virus--dge.html#cb206-1" tabindex="-1"></a><span class="co"># BP</span></span>
<span id="cb206-2"><a href="virus-vs-virus--dge.html#cb206-2" tabindex="-1"></a>g_null_bp_10 <span class="ot">&lt;-</span></span>
<span id="cb206-3"><a href="virus-vs-virus--dge.html#cb206-3" tabindex="-1"></a>  g_null_bp <span class="sc">%&gt;%</span> </span>
<span id="cb206-4"><a href="virus-vs-virus--dge.html#cb206-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb206-5"><a href="virus-vs-virus--dge.html#cb206-5" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-6"><a href="virus-vs-virus--dge.html#cb206-6" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-7"><a href="virus-vs-virus--dge.html#cb206-7" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb206-8"><a href="virus-vs-virus--dge.html#cb206-8" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span>
<span id="cb206-9"><a href="virus-vs-virus--dge.html#cb206-9" tabindex="-1"></a></span>
<span id="cb206-10"><a href="virus-vs-virus--dge.html#cb206-10" tabindex="-1"></a><span class="co"># MF</span></span>
<span id="cb206-11"><a href="virus-vs-virus--dge.html#cb206-11" tabindex="-1"></a>g_null_mf_10 <span class="ot">&lt;-</span></span>
<span id="cb206-12"><a href="virus-vs-virus--dge.html#cb206-12" tabindex="-1"></a>  g_null_mf <span class="sc">%&gt;%</span> </span>
<span id="cb206-13"><a href="virus-vs-virus--dge.html#cb206-13" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb206-14"><a href="virus-vs-virus--dge.html#cb206-14" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-15"><a href="virus-vs-virus--dge.html#cb206-15" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-16"><a href="virus-vs-virus--dge.html#cb206-16" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb206-17"><a href="virus-vs-virus--dge.html#cb206-17" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span>
<span id="cb206-18"><a href="virus-vs-virus--dge.html#cb206-18" tabindex="-1"></a></span>
<span id="cb206-19"><a href="virus-vs-virus--dge.html#cb206-19" tabindex="-1"></a><span class="co"># CC</span></span>
<span id="cb206-20"><a href="virus-vs-virus--dge.html#cb206-20" tabindex="-1"></a>g_null_cc_10 <span class="ot">&lt;-</span></span>
<span id="cb206-21"><a href="virus-vs-virus--dge.html#cb206-21" tabindex="-1"></a>  g_null_cc <span class="sc">%&gt;%</span> </span>
<span id="cb206-22"><a href="virus-vs-virus--dge.html#cb206-22" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb206-23"><a href="virus-vs-virus--dge.html#cb206-23" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-24"><a href="virus-vs-virus--dge.html#cb206-24" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb206-25"><a href="virus-vs-virus--dge.html#cb206-25" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb206-26"><a href="virus-vs-virus--dge.html#cb206-26" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span></code></pre></div>
</div>
<div id="filter-results-2" class="section level5 hasAnchor" number="4.5.4.1.2">
<h5><span class="header-section-number">4.5.4.1.2</span> Filter results<a href="virus-vs-virus--dge.html#filter-results-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="virus-vs-virus--dge.html#cb207-1" tabindex="-1"></a>final_g_bp <span class="ot">&lt;-</span> </span>
<span id="cb207-2"><a href="virus-vs-virus--dge.html#cb207-2" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb207-3"><a href="virus-vs-virus--dge.html#cb207-3" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb207-4"><a href="virus-vs-virus--dge.html#cb207-4" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> g_null_bp_10<span class="sc">$</span>value)</span>
<span id="cb207-5"><a href="virus-vs-virus--dge.html#cb207-5" tabindex="-1"></a>         </span>
<span id="cb207-6"><a href="virus-vs-virus--dge.html#cb207-6" tabindex="-1"></a>final_g_mf <span class="ot">&lt;-</span></span>
<span id="cb207-7"><a href="virus-vs-virus--dge.html#cb207-7" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb207-8"><a href="virus-vs-virus--dge.html#cb207-8" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;mf&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb207-9"><a href="virus-vs-virus--dge.html#cb207-9" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> g_null_mf_10<span class="sc">$</span>value)</span>
<span id="cb207-10"><a href="virus-vs-virus--dge.html#cb207-10" tabindex="-1"></a></span>
<span id="cb207-11"><a href="virus-vs-virus--dge.html#cb207-11" tabindex="-1"></a>final_g_cc <span class="ot">&lt;-</span></span>
<span id="cb207-12"><a href="virus-vs-virus--dge.html#cb207-12" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb207-13"><a href="virus-vs-virus--dge.html#cb207-13" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span> <span class="sc">&amp;</span></span>
<span id="cb207-14"><a href="virus-vs-virus--dge.html#cb207-14" tabindex="-1"></a>         <span class="sc">!</span>description <span class="sc">%in%</span> g_null_cc_10<span class="sc">$</span>value)</span></code></pre></div>
</div>
</div>
<div id="kegg-pathway-1" class="section level4 hasAnchor" number="4.5.4.2">
<h4><span class="header-section-number">4.5.4.2</span> KEGG Pathway<a href="virus-vs-virus--dge.html#kegg-pathway-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="what-pathways-are-overrepreesented-in-10-of-nullsets-1" class="section level5 hasAnchor" number="4.5.4.2.1">
<h5><span class="header-section-number">4.5.4.2.1</span> What pathways are overrepreesented in &gt;10% of nullsets?<a href="virus-vs-virus--dge.html#what-pathways-are-overrepreesented-in-10-of-nullsets-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="virus-vs-virus--dge.html#cb208-1" tabindex="-1"></a>g_kegg_10 <span class="ot">&lt;-</span></span>
<span id="cb208-2"><a href="virus-vs-virus--dge.html#cb208-2" tabindex="-1"></a>  g_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb208-3"><a href="virus-vs-virus--dge.html#cb208-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb208-4"><a href="virus-vs-virus--dge.html#cb208-4" tabindex="-1"></a>  <span class="fu">na.omit</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb208-5"><a href="virus-vs-virus--dge.html#cb208-5" tabindex="-1"></a>  <span class="fu">group_by</span>(value) <span class="sc">%&gt;%</span> </span>
<span id="cb208-6"><a href="virus-vs-virus--dge.html#cb208-6" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">count=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb208-7"><a href="virus-vs-virus--dge.html#cb208-7" tabindex="-1"></a>  <span class="fu">filter</span>(count<span class="sc">&gt;</span><span class="dv">100</span>) </span></code></pre></div>
</div>
<div id="filter-results-3" class="section level5 hasAnchor" number="4.5.4.2.2">
<h5><span class="header-section-number">4.5.4.2.2</span> Filter Results<a href="virus-vs-virus--dge.html#filter-results-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="virus-vs-virus--dge.html#cb209-1" tabindex="-1"></a>gnome_kegg_final <span class="ot">&lt;-</span> </span>
<span id="cb209-2"><a href="virus-vs-virus--dge.html#cb209-2" tabindex="-1"></a>  gnome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb209-3"><a href="virus-vs-virus--dge.html#cb209-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>description <span class="sc">%in%</span> g_kegg_10<span class="sc">$</span>value) </span>
<span id="cb209-4"><a href="virus-vs-virus--dge.html#cb209-4" tabindex="-1"></a>gnome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb209-5"><a href="virus-vs-virus--dge.html#cb209-5" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
</div>
</div>
</div>
<div id="go-term-figure" class="section level3 hasAnchor" number="4.5.5">
<h3><span class="header-section-number">4.5.5</span> GO Term Figure<a href="virus-vs-virus--dge.html#go-term-figure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="virus-vs-virus--dge.html#cb210-1" tabindex="-1"></a>t_final <span class="ot">&lt;-</span> </span>
<span id="cb210-2"><a href="virus-vs-virus--dge.html#cb210-2" tabindex="-1"></a>  <span class="fu">rbind</span>(final_t_bp, final_t_mf, final_t_cc) <span class="sc">%&gt;%</span> </span>
<span id="cb210-3"><a href="virus-vs-virus--dge.html#cb210-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span><span class="st">&quot;Transcriptome&quot;</span>)</span>
<span id="cb210-4"><a href="virus-vs-virus--dge.html#cb210-4" tabindex="-1"></a><span class="fu">write.csv</span>(t_final, <span class="st">&quot;Output Files/txome_female_edgeR_viralRNA_go_refined.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb210-5"><a href="virus-vs-virus--dge.html#cb210-5" tabindex="-1"></a></span>
<span id="cb210-6"><a href="virus-vs-virus--dge.html#cb210-6" tabindex="-1"></a>g_final <span class="ot">&lt;-</span></span>
<span id="cb210-7"><a href="virus-vs-virus--dge.html#cb210-7" tabindex="-1"></a>  <span class="fu">rbind</span>(final_g_bp, final_g_mf, final_g_cc) <span class="sc">%&gt;%</span> </span>
<span id="cb210-8"><a href="virus-vs-virus--dge.html#cb210-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">approach=</span> <span class="st">&quot;Genome&quot;</span>)</span>
<span id="cb210-9"><a href="virus-vs-virus--dge.html#cb210-9" tabindex="-1"></a><span class="fu">write.csv</span>(g_final, <span class="st">&quot;Output Files/gnome_female_edgeR_viralRNA_go_refined.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span>
<span id="cb210-10"><a href="virus-vs-virus--dge.html#cb210-10" tabindex="-1"></a></span>
<span id="cb210-11"><a href="virus-vs-virus--dge.html#cb210-11" tabindex="-1"></a>go_all <span class="ot">&lt;-</span></span>
<span id="cb210-12"><a href="virus-vs-virus--dge.html#cb210-12" tabindex="-1"></a>  <span class="fu">rbind</span>(t_final, g_final)</span>
<span id="cb210-13"><a href="virus-vs-virus--dge.html#cb210-13" tabindex="-1"></a></span>
<span id="cb210-14"><a href="virus-vs-virus--dge.html#cb210-14" tabindex="-1"></a>go_fig <span class="ot">&lt;-</span></span>
<span id="cb210-15"><a href="virus-vs-virus--dge.html#cb210-15" tabindex="-1"></a>  go_all <span class="sc">%&gt;%</span> </span>
<span id="cb210-16"><a href="virus-vs-virus--dge.html#cb210-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">go=</span><span class="fu">factor</span>(</span>
<span id="cb210-17"><a href="virus-vs-virus--dge.html#cb210-17" tabindex="-1"></a>           <span class="fu">ifelse</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span>, <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb210-18"><a href="virus-vs-virus--dge.html#cb210-18" tabindex="-1"></a>                  <span class="fu">ifelse</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span>, <span class="st">&quot;CC&quot;</span>, <span class="st">&quot;MF&quot;</span>)),</span>
<span id="cb210-19"><a href="virus-vs-virus--dge.html#cb210-19" tabindex="-1"></a>           <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;BP&quot;</span>, <span class="st">&quot;CC&quot;</span>, <span class="st">&quot;MF&quot;</span>)), </span>
<span id="cb210-20"><a href="virus-vs-virus--dge.html#cb210-20" tabindex="-1"></a>         <span class="at">enrichmentRatio=</span><span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, enrichmentRatio<span class="sc">*-</span><span class="dv">1</span>, enrichmentRatio)) <span class="sc">%&gt;%</span> </span>
<span id="cb210-21"><a href="virus-vs-virus--dge.html#cb210-21" tabindex="-1"></a>  </span>
<span id="cb210-22"><a href="virus-vs-virus--dge.html#cb210-22" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data=</span>., <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">fct_reorder</span>(description, enrichmentRatio, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb210-23"><a href="virus-vs-virus--dge.html#cb210-23" tabindex="-1"></a>                     <span class="at">y=</span>enrichmentRatio, <span class="at">fill=</span><span class="fu">factor</span>(approach))) <span class="sc">+</span></span>
<span id="cb210-24"><a href="virus-vs-virus--dge.html#cb210-24" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="st">&quot;dodge&quot;</span>) <span class="sc">+</span></span>
<span id="cb210-25"><a href="virus-vs-virus--dge.html#cb210-25" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;slategray3&quot;</span>, <span class="st">&quot;slategrey&quot;</span>), <span class="at">name=</span><span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb210-26"><a href="virus-vs-virus--dge.html#cb210-26" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;GO Term&quot;</span>, <span class="at">y=</span><span class="st">&quot;Enrichment Ratio&quot;</span>) <span class="sc">+</span></span>
<span id="cb210-27"><a href="virus-vs-virus--dge.html#cb210-27" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(description) <span class="fu">str_wrap</span>(description, <span class="at">width =</span> <span class="dv">65</span>)) <span class="sc">+</span></span>
<span id="cb210-28"><a href="virus-vs-virus--dge.html#cb210-28" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">n.breaks=</span><span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb210-29"><a href="virus-vs-virus--dge.html#cb210-29" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(go), <span class="at">scales=</span><span class="st">&quot;free_y&quot;</span>, <span class="at">space=</span><span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb210-30"><a href="virus-vs-virus--dge.html#cb210-30" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb210-31"><a href="virus-vs-virus--dge.html#cb210-31" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(), <span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">11</span>), </span>
<span id="cb210-32"><a href="virus-vs-virus--dge.html#cb210-32" tabindex="-1"></a>        <span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb210-33"><a href="virus-vs-virus--dge.html#cb210-33" tabindex="-1"></a>  <span class="fu">coord_flip</span>() </span>
<span id="cb210-34"><a href="virus-vs-virus--dge.html#cb210-34" tabindex="-1"></a>  </span>
<span id="cb210-35"><a href="virus-vs-virus--dge.html#cb210-35" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;Figures/viralRNA_goterm.jpg&quot;</span>, go_fig, <span class="at">width=</span><span class="dv">9</span>, <span class="at">height=</span><span class="dv">8</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
<div id="presentation-figure" class="section level4 hasAnchor" number="4.5.5.1">
<h4><span class="header-section-number">4.5.5.1</span> Presentation figure<a href="virus-vs-virus--dge.html#presentation-figure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="virus-vs-virus--dge.html#cb211-1" tabindex="-1"></a>go_fig_pres <span class="ot">&lt;-</span></span>
<span id="cb211-2"><a href="virus-vs-virus--dge.html#cb211-2" tabindex="-1"></a>  t_final <span class="sc">%&gt;%</span> </span>
<span id="cb211-3"><a href="virus-vs-virus--dge.html#cb211-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">go=</span><span class="fu">factor</span>(</span>
<span id="cb211-4"><a href="virus-vs-virus--dge.html#cb211-4" tabindex="-1"></a>           <span class="fu">ifelse</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span>, <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb211-5"><a href="virus-vs-virus--dge.html#cb211-5" tabindex="-1"></a>                  <span class="fu">ifelse</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span>, <span class="st">&quot;CC&quot;</span>, <span class="st">&quot;MF&quot;</span>)),</span>
<span id="cb211-6"><a href="virus-vs-virus--dge.html#cb211-6" tabindex="-1"></a>           <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;BP&quot;</span>, <span class="st">&quot;CC&quot;</span>, <span class="st">&quot;MF&quot;</span>)), </span>
<span id="cb211-7"><a href="virus-vs-virus--dge.html#cb211-7" tabindex="-1"></a>         <span class="at">enrichmentRatio=</span><span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, enrichmentRatio<span class="sc">*-</span><span class="dv">1</span>, enrichmentRatio)) <span class="sc">%&gt;%</span> </span>
<span id="cb211-8"><a href="virus-vs-virus--dge.html#cb211-8" tabindex="-1"></a>  </span>
<span id="cb211-9"><a href="virus-vs-virus--dge.html#cb211-9" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data=</span>., <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">fct_reorder</span>(description, enrichmentRatio, <span class="at">.desc=</span><span class="cn">TRUE</span>), </span>
<span id="cb211-10"><a href="virus-vs-virus--dge.html#cb211-10" tabindex="-1"></a>                     <span class="at">y=</span>enrichmentRatio)) <span class="sc">+</span></span>
<span id="cb211-11"><a href="virus-vs-virus--dge.html#cb211-11" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">position=</span><span class="st">&quot;dodge&quot;</span>, <span class="at">fill=</span><span class="st">&quot;slategray&quot;</span>) <span class="sc">+</span></span>
<span id="cb211-12"><a href="virus-vs-virus--dge.html#cb211-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;GO Term&quot;</span>, <span class="at">y=</span><span class="st">&quot;Enrichment Ratio&quot;</span>) <span class="sc">+</span></span>
<span id="cb211-13"><a href="virus-vs-virus--dge.html#cb211-13" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="cf">function</span>(description) <span class="fu">str_wrap</span>(description, <span class="at">width =</span> <span class="dv">65</span>)) <span class="sc">+</span></span>
<span id="cb211-14"><a href="virus-vs-virus--dge.html#cb211-14" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">n.breaks=</span><span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb211-15"><a href="virus-vs-virus--dge.html#cb211-15" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(go), <span class="at">scales=</span><span class="st">&quot;free_y&quot;</span>, <span class="at">space=</span><span class="st">&quot;free_y&quot;</span>) <span class="sc">+</span></span>
<span id="cb211-16"><a href="virus-vs-virus--dge.html#cb211-16" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb211-17"><a href="virus-vs-virus--dge.html#cb211-17" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>(), <span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), </span>
<span id="cb211-18"><a href="virus-vs-virus--dge.html#cb211-18" tabindex="-1"></a>        <span class="at">legend.position=</span><span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb211-19"><a href="virus-vs-virus--dge.html#cb211-19" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span>
<span id="cb211-20"><a href="virus-vs-virus--dge.html#cb211-20" tabindex="-1"></a>go_fig_pres</span>
<span id="cb211-21"><a href="virus-vs-virus--dge.html#cb211-21" tabindex="-1"></a></span>
<span id="cb211-22"><a href="virus-vs-virus--dge.html#cb211-22" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename=</span><span class="st">&quot;Figures/viralRNA_goterm_t_pres.jpg&quot;</span>, go_fig_pres, <span class="at">width=</span><span class="dv">10</span>, <span class="at">height=</span><span class="dv">8</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="immunne-go-term-genes" class="section level3 hasAnchor" number="4.5.6">
<h3><span class="header-section-number">4.5.6</span> Immunne GO term Genes<a href="virus-vs-virus--dge.html#immunne-go-term-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="extracting-immune-go-genes" class="section level4 hasAnchor" number="4.5.6.1">
<h4><span class="header-section-number">4.5.6.1</span> Extracting Immune GO Genes<a href="virus-vs-virus--dge.html#extracting-immune-go-genes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Manually look up function of each gene &amp; categorize. Select subset of 16 genes for presentation in manuscript.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="virus-vs-virus--dge.html#cb212-1" tabindex="-1"></a>immune_go_genes <span class="ot">&lt;-</span> </span>
<span id="cb212-2"><a href="virus-vs-virus--dge.html#cb212-2" tabindex="-1"></a>  go_all <span class="sc">%&gt;%</span> </span>
<span id="cb212-3"><a href="virus-vs-virus--dge.html#cb212-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;immune|viral|inflammatory|interferon|leukocyte&quot;</span>, description)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-4"><a href="virus-vs-virus--dge.html#cb212-4" tabindex="-1"></a>  <span class="fu">select</span>(userId) <span class="sc">%&gt;%</span> </span>
<span id="cb212-5"><a href="virus-vs-virus--dge.html#cb212-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genes =</span> <span class="fu">str_split_fixed</span>(userId, <span class="st">&quot;;&quot;</span>, <span class="at">n=</span><span class="dv">18</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-6"><a href="virus-vs-virus--dge.html#cb212-6" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">!</span>userId) <span class="sc">%&gt;%</span> </span>
<span id="cb212-7"><a href="virus-vs-virus--dge.html#cb212-7" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb212-8"><a href="virus-vs-virus--dge.html#cb212-8" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb212-9"><a href="virus-vs-virus--dge.html#cb212-9" tabindex="-1"></a>  <span class="fu">setNames</span>(<span class="st">&quot;genes&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb212-10"><a href="virus-vs-virus--dge.html#cb212-10" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(genes)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-11"><a href="virus-vs-virus--dge.html#cb212-11" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>genes<span class="sc">==</span><span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb212-12"><a href="virus-vs-virus--dge.html#cb212-12" tabindex="-1"></a>  <span class="fu">arrange</span>(genes) <span class="sc">%&gt;%</span> </span>
<span id="cb212-13"><a href="virus-vs-virus--dge.html#cb212-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">description =</span> <span class="fu">c</span>(</span>
<span id="cb212-14"><a href="virus-vs-virus--dge.html#cb212-14" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-15"><a href="virus-vs-virus--dge.html#cb212-15" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-16"><a href="virus-vs-virus--dge.html#cb212-16" tabindex="-1"></a>    <span class="st">&quot;Antiviral&quot;</span>, </span>
<span id="cb212-17"><a href="virus-vs-virus--dge.html#cb212-17" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity &amp; adaptive&quot;</span>, </span>
<span id="cb212-18"><a href="virus-vs-virus--dge.html#cb212-18" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-19"><a href="virus-vs-virus--dge.html#cb212-19" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-20"><a href="virus-vs-virus--dge.html#cb212-20" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>, </span>
<span id="cb212-21"><a href="virus-vs-virus--dge.html#cb212-21" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>, </span>
<span id="cb212-22"><a href="virus-vs-virus--dge.html#cb212-22" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>, </span>
<span id="cb212-23"><a href="virus-vs-virus--dge.html#cb212-23" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>, </span>
<span id="cb212-24"><a href="virus-vs-virus--dge.html#cb212-24" tabindex="-1"></a>    <span class="st">&quot;adaptive&quot;</span>, </span>
<span id="cb212-25"><a href="virus-vs-virus--dge.html#cb212-25" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>, </span>
<span id="cb212-26"><a href="virus-vs-virus--dge.html#cb212-26" tabindex="-1"></a>    <span class="st">&quot;Complement System&quot;</span>,</span>
<span id="cb212-27"><a href="virus-vs-virus--dge.html#cb212-27" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span>,</span>
<span id="cb212-28"><a href="virus-vs-virus--dge.html#cb212-28" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-29"><a href="virus-vs-virus--dge.html#cb212-29" tabindex="-1"></a>    <span class="st">&quot;inflammation&quot;</span>, </span>
<span id="cb212-30"><a href="virus-vs-virus--dge.html#cb212-30" tabindex="-1"></a>    <span class="st">&quot;mhc&quot;</span>, </span>
<span id="cb212-31"><a href="virus-vs-virus--dge.html#cb212-31" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>,</span>
<span id="cb212-32"><a href="virus-vs-virus--dge.html#cb212-32" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span>,</span>
<span id="cb212-33"><a href="virus-vs-virus--dge.html#cb212-33" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span>, </span>
<span id="cb212-34"><a href="virus-vs-virus--dge.html#cb212-34" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-35"><a href="virus-vs-virus--dge.html#cb212-35" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span>, </span>
<span id="cb212-36"><a href="virus-vs-virus--dge.html#cb212-36" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-37"><a href="virus-vs-virus--dge.html#cb212-37" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>,</span>
<span id="cb212-38"><a href="virus-vs-virus--dge.html#cb212-38" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-39"><a href="virus-vs-virus--dge.html#cb212-39" tabindex="-1"></a>    <span class="st">&quot;Antiviral&quot;</span>, </span>
<span id="cb212-40"><a href="virus-vs-virus--dge.html#cb212-40" tabindex="-1"></a>    <span class="st">&quot;Antiviral&quot;</span>, </span>
<span id="cb212-41"><a href="virus-vs-virus--dge.html#cb212-41" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-42"><a href="virus-vs-virus--dge.html#cb212-42" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-43"><a href="virus-vs-virus--dge.html#cb212-43" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-44"><a href="virus-vs-virus--dge.html#cb212-44" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-45"><a href="virus-vs-virus--dge.html#cb212-45" tabindex="-1"></a>    <span class="st">&quot;Antiviral&quot;</span>, </span>
<span id="cb212-46"><a href="virus-vs-virus--dge.html#cb212-46" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span>, </span>
<span id="cb212-47"><a href="virus-vs-virus--dge.html#cb212-47" tabindex="-1"></a>    <span class="st">&quot;inflammation&quot;</span>, </span>
<span id="cb212-48"><a href="virus-vs-virus--dge.html#cb212-48" tabindex="-1"></a>    <span class="st">&quot;extra&quot;</span>, </span>
<span id="cb212-49"><a href="virus-vs-virus--dge.html#cb212-49" tabindex="-1"></a>    <span class="st">&quot;Innate Immunity&quot;</span>, </span>
<span id="cb212-50"><a href="virus-vs-virus--dge.html#cb212-50" tabindex="-1"></a>    <span class="st">&quot;Immunoregulation&quot;</span></span>
<span id="cb212-51"><a href="virus-vs-virus--dge.html#cb212-51" tabindex="-1"></a>  ))</span>
<span id="cb212-52"><a href="virus-vs-virus--dge.html#cb212-52" tabindex="-1"></a></span>
<span id="cb212-53"><a href="virus-vs-virus--dge.html#cb212-53" tabindex="-1"></a>immune_go_genes_subset <span class="ot">&lt;-</span> </span>
<span id="cb212-54"><a href="virus-vs-virus--dge.html#cb212-54" tabindex="-1"></a>  immune_go_genes <span class="sc">%&gt;%</span> </span>
<span id="cb212-55"><a href="virus-vs-virus--dge.html#cb212-55" tabindex="-1"></a>  <span class="fu">filter</span>(description <span class="sc">==</span> <span class="st">&quot;Innate Immunity&quot;</span> <span class="sc">|</span></span>
<span id="cb212-56"><a href="virus-vs-virus--dge.html#cb212-56" tabindex="-1"></a>         description <span class="sc">==</span> <span class="st">&quot;Complement System&quot;</span> <span class="sc">|</span></span>
<span id="cb212-57"><a href="virus-vs-virus--dge.html#cb212-57" tabindex="-1"></a>         description <span class="sc">==</span> <span class="st">&quot;Immunoregulation&quot;</span> <span class="sc">|</span></span>
<span id="cb212-58"><a href="virus-vs-virus--dge.html#cb212-58" tabindex="-1"></a>         description <span class="sc">==</span> <span class="st">&quot;Antiviral&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb212-59"><a href="virus-vs-virus--dge.html#cb212-59" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>genes <span class="sc">==</span> <span class="st">&quot;C2&quot;</span> <span class="sc">&amp;</span> </span>
<span id="cb212-60"><a href="virus-vs-virus--dge.html#cb212-60" tabindex="-1"></a>         <span class="sc">!</span>genes <span class="sc">==</span> <span class="st">&quot;CFD&quot;</span>,</span>
<span id="cb212-61"><a href="virus-vs-virus--dge.html#cb212-61" tabindex="-1"></a>         <span class="sc">!</span>genes <span class="sc">==</span> <span class="st">&quot;HSPA8&quot;</span>,</span>
<span id="cb212-62"><a href="virus-vs-virus--dge.html#cb212-62" tabindex="-1"></a>         <span class="sc">!</span>genes <span class="sc">==</span> <span class="st">&quot;ACTB&quot;</span>, </span>
<span id="cb212-63"><a href="virus-vs-virus--dge.html#cb212-63" tabindex="-1"></a>         <span class="sc">!</span>genes <span class="sc">==</span> <span class="st">&quot;DHX58&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb212-64"><a href="virus-vs-virus--dge.html#cb212-64" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">description =</span> <span class="fu">factor</span>(description, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Innate Immunity&quot;</span>, <span class="st">&quot;Immunoregulation&quot;</span>, <span class="st">&quot;Antiviral&quot;</span>, <span class="st">&quot;Complement System&quot;</span>)))</span></code></pre></div>
</div>
<div id="extract-expression-data" class="section level4 hasAnchor" number="4.5.6.2">
<h4><span class="header-section-number">4.5.6.2</span> Extract Expression Data<a href="virus-vs-virus--dge.html#extract-expression-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="virus-vs-virus--dge.html#cb213-1" tabindex="-1"></a>t_immunegenes_subset <span class="ot">&lt;-</span> </span>
<span id="cb213-2"><a href="virus-vs-virus--dge.html#cb213-2" tabindex="-1"></a>  t_raw <span class="sc">%&gt;%</span> </span>
<span id="cb213-3"><a href="virus-vs-virus--dge.html#cb213-3" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> immune_go_genes_subset<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb213-4"><a href="virus-vs-virus--dge.html#cb213-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to=</span><span class="st">&quot;sample&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb213-5"><a href="virus-vs-virus--dge.html#cb213-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">disease=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> meta<span class="sc">$</span>sample,</span>
<span id="cb213-6"><a href="virus-vs-virus--dge.html#cb213-6" tabindex="-1"></a>                     meta[<span class="fu">match</span>(sample, meta<span class="sc">$</span>sample),<span class="dv">4</span>],</span>
<span id="cb213-7"><a href="virus-vs-virus--dge.html#cb213-7" tabindex="-1"></a>                     sample),</span>
<span id="cb213-8"><a href="virus-vs-virus--dge.html#cb213-8" tabindex="-1"></a>         <span class="at">disease=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;acute&quot;</span>, <span class="st">&quot;A&quot;</span>, </span>
<span id="cb213-9"><a href="virus-vs-virus--dge.html#cb213-9" tabindex="-1"></a>                               <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;P&quot;</span>, </span>
<span id="cb213-10"><a href="virus-vs-virus--dge.html#cb213-10" tabindex="-1"></a>                                      <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;late&quot;</span>, <span class="st">&quot;L&quot;</span>, <span class="st">&quot;C&quot;</span>))),</span>
<span id="cb213-11"><a href="virus-vs-virus--dge.html#cb213-11" tabindex="-1"></a>                        <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;C&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;L&quot;</span>)),</span>
<span id="cb213-12"><a href="virus-vs-virus--dge.html#cb213-12" tabindex="-1"></a>         <span class="at">description =</span> <span class="fu">ifelse</span>(gene <span class="sc">%in%</span> immune_go_genes<span class="sc">$</span>genes, </span>
<span id="cb213-13"><a href="virus-vs-virus--dge.html#cb213-13" tabindex="-1"></a>                              immune_go_genes[<span class="fu">match</span>(gene, immune_go_genes<span class="sc">$</span>genes),<span class="dv">2</span>],</span>
<span id="cb213-14"><a href="virus-vs-virus--dge.html#cb213-14" tabindex="-1"></a>                              gene)) </span>
<span id="cb213-15"><a href="virus-vs-virus--dge.html#cb213-15" tabindex="-1"></a></span>
<span id="cb213-16"><a href="virus-vs-virus--dge.html#cb213-16" tabindex="-1"></a>t_immune_genes_all <span class="ot">&lt;-</span></span>
<span id="cb213-17"><a href="virus-vs-virus--dge.html#cb213-17" tabindex="-1"></a>  t_raw <span class="sc">%&gt;%</span> </span>
<span id="cb213-18"><a href="virus-vs-virus--dge.html#cb213-18" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> immune_go_genes<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb213-19"><a href="virus-vs-virus--dge.html#cb213-19" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to=</span><span class="st">&quot;sample&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb213-20"><a href="virus-vs-virus--dge.html#cb213-20" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">disease=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> meta<span class="sc">$</span>sample,</span>
<span id="cb213-21"><a href="virus-vs-virus--dge.html#cb213-21" tabindex="-1"></a>                     meta[<span class="fu">match</span>(sample, meta<span class="sc">$</span>sample),<span class="dv">4</span>],</span>
<span id="cb213-22"><a href="virus-vs-virus--dge.html#cb213-22" tabindex="-1"></a>                     sample),</span>
<span id="cb213-23"><a href="virus-vs-virus--dge.html#cb213-23" tabindex="-1"></a>         <span class="at">disease=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;acute&quot;</span>, <span class="st">&quot;A&quot;</span>, </span>
<span id="cb213-24"><a href="virus-vs-virus--dge.html#cb213-24" tabindex="-1"></a>                               <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;P&quot;</span>, </span>
<span id="cb213-25"><a href="virus-vs-virus--dge.html#cb213-25" tabindex="-1"></a>                                      <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;late&quot;</span>, <span class="st">&quot;L&quot;</span>, <span class="st">&quot;C&quot;</span>))),</span>
<span id="cb213-26"><a href="virus-vs-virus--dge.html#cb213-26" tabindex="-1"></a>                        <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;C&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;L&quot;</span>)))</span>
<span id="cb213-27"><a href="virus-vs-virus--dge.html#cb213-27" tabindex="-1"></a></span>
<span id="cb213-28"><a href="virus-vs-virus--dge.html#cb213-28" tabindex="-1"></a>g_immune_genes_all <span class="ot">&lt;-</span> </span>
<span id="cb213-29"><a href="virus-vs-virus--dge.html#cb213-29" tabindex="-1"></a>  g_raw <span class="sc">%&gt;%</span> </span>
<span id="cb213-30"><a href="virus-vs-virus--dge.html#cb213-30" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">%in%</span> immune_go_genes<span class="sc">$</span>genes) <span class="sc">%&gt;%</span> </span>
<span id="cb213-31"><a href="virus-vs-virus--dge.html#cb213-31" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>gene, <span class="at">names_to=</span><span class="st">&quot;sample&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb213-32"><a href="virus-vs-virus--dge.html#cb213-32" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">disease=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> meta<span class="sc">$</span>sample,</span>
<span id="cb213-33"><a href="virus-vs-virus--dge.html#cb213-33" tabindex="-1"></a>                     meta[<span class="fu">match</span>(sample, meta<span class="sc">$</span>sample),<span class="dv">4</span>],</span>
<span id="cb213-34"><a href="virus-vs-virus--dge.html#cb213-34" tabindex="-1"></a>                     sample),</span>
<span id="cb213-35"><a href="virus-vs-virus--dge.html#cb213-35" tabindex="-1"></a>         <span class="at">disease=</span><span class="fu">factor</span>(<span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;acute&quot;</span>, <span class="st">&quot;A&quot;</span>, </span>
<span id="cb213-36"><a href="virus-vs-virus--dge.html#cb213-36" tabindex="-1"></a>                               <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;peak&quot;</span>, <span class="st">&quot;P&quot;</span>, </span>
<span id="cb213-37"><a href="virus-vs-virus--dge.html#cb213-37" tabindex="-1"></a>                                      <span class="fu">ifelse</span>(disease<span class="sc">==</span><span class="st">&quot;late&quot;</span>, <span class="st">&quot;L&quot;</span>, <span class="st">&quot;C&quot;</span>))),</span>
<span id="cb213-38"><a href="virus-vs-virus--dge.html#cb213-38" tabindex="-1"></a>                        <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;C&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;P&quot;</span>, <span class="st">&quot;L&quot;</span>))) </span></code></pre></div>
</div>
<div id="plot-for-manuscript" class="section level4 hasAnchor" number="4.5.6.3">
<h4><span class="header-section-number">4.5.6.3</span> Plot for manuscript<a href="virus-vs-virus--dge.html#plot-for-manuscript" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="virus-vs-virus--dge.html#cb214-1" tabindex="-1"></a>t_plot_manuscript <span class="ot">&lt;-</span></span>
<span id="cb214-2"><a href="virus-vs-virus--dge.html#cb214-2" tabindex="-1"></a>  t_immunegenes_subset <span class="sc">%&gt;%</span> </span>
<span id="cb214-3"><a href="virus-vs-virus--dge.html#cb214-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>disease, <span class="at">y=</span>expression)) <span class="sc">+</span></span>
<span id="cb214-4"><a href="virus-vs-virus--dge.html#cb214-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb214-5"><a href="virus-vs-virus--dge.html#cb214-5" tabindex="-1"></a>  <span class="fu">facet_nested_wrap</span>(<span class="fu">vars</span>(description, gene),</span>
<span id="cb214-6"><a href="virus-vs-virus--dge.html#cb214-6" tabindex="-1"></a>                    <span class="at">nest_line=</span><span class="fu">element_line</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>)) <span class="sc">+</span></span>
<span id="cb214-7"><a href="virus-vs-virus--dge.html#cb214-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">ggh4x.facet.nestline =</span> <span class="fu">element_line</span>(<span class="at">linetype =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb214-8"><a href="virus-vs-virus--dge.html#cb214-8" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Influenza Stage&quot;</span>, <span class="at">y =</span> <span class="st">&quot;log2(CPM)&quot;</span>) <span class="sc">+</span></span>
<span id="cb214-9"><a href="virus-vs-virus--dge.html#cb214-9" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb214-10"><a href="virus-vs-virus--dge.html#cb214-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb214-11"><a href="virus-vs-virus--dge.html#cb214-11" tabindex="-1"></a>t_plot_manuscript</span>
<span id="cb214-12"><a href="virus-vs-virus--dge.html#cb214-12" tabindex="-1"></a></span>
<span id="cb214-13"><a href="virus-vs-virus--dge.html#cb214-13" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/T_female_edgeR_immunegenes.jpg&quot;</span>, t_plot_manuscript, </span>
<span id="cb214-14"><a href="virus-vs-virus--dge.html#cb214-14" tabindex="-1"></a>       <span class="at">width=</span><span class="dv">8</span>, <span class="at">height=</span><span class="dv">8</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-for-supplemental" class="section level4 hasAnchor" number="4.5.6.4">
<h4><span class="header-section-number">4.5.6.4</span> Plot for supplemental<a href="virus-vs-virus--dge.html#plot-for-supplemental" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="virus-vs-virus--dge.html#cb215-1" tabindex="-1"></a>t_plot_supplemental <span class="ot">&lt;-</span></span>
<span id="cb215-2"><a href="virus-vs-virus--dge.html#cb215-2" tabindex="-1"></a>  t_immune_genes_all <span class="sc">%&gt;%</span> </span>
<span id="cb215-3"><a href="virus-vs-virus--dge.html#cb215-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>disease, <span class="at">y=</span>expression)) <span class="sc">+</span></span>
<span id="cb215-4"><a href="virus-vs-virus--dge.html#cb215-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb215-5"><a href="virus-vs-virus--dge.html#cb215-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene, <span class="at">scales=</span><span class="st">&quot;free&quot;</span>, <span class="at">ncol=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb215-6"><a href="virus-vs-virus--dge.html#cb215-6" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="cn">NULL</span>,</span>
<span id="cb215-7"><a href="virus-vs-virus--dge.html#cb215-7" tabindex="-1"></a>       <span class="at">x=</span><span class="st">&quot;Influenza Stage&quot;</span>, <span class="at">y=</span><span class="st">&quot;log2(CPM)&quot;</span>) <span class="sc">+</span></span>
<span id="cb215-8"><a href="virus-vs-virus--dge.html#cb215-8" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb215-9"><a href="virus-vs-virus--dge.html#cb215-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb215-10"><a href="virus-vs-virus--dge.html#cb215-10" tabindex="-1"></a></span>
<span id="cb215-11"><a href="virus-vs-virus--dge.html#cb215-11" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/supplemental_T_female_edgeR_immunegenes.jpg&quot;</span>, t_plot_supplemental,</span>
<span id="cb215-12"><a href="virus-vs-virus--dge.html#cb215-12" tabindex="-1"></a>          <span class="at">width=</span><span class="dv">8</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span>
<span id="cb215-13"><a href="virus-vs-virus--dge.html#cb215-13" tabindex="-1"></a></span>
<span id="cb215-14"><a href="virus-vs-virus--dge.html#cb215-14" tabindex="-1"></a>g_plot_supplemental <span class="ot">&lt;-</span></span>
<span id="cb215-15"><a href="virus-vs-virus--dge.html#cb215-15" tabindex="-1"></a>  g_immune_genes_all <span class="sc">%&gt;%</span> </span>
<span id="cb215-16"><a href="virus-vs-virus--dge.html#cb215-16" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>disease, <span class="at">y=</span>expression)) <span class="sc">+</span></span>
<span id="cb215-17"><a href="virus-vs-virus--dge.html#cb215-17" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb215-18"><a href="virus-vs-virus--dge.html#cb215-18" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>gene, <span class="at">scales=</span><span class="st">&quot;free&quot;</span>, <span class="at">ncol=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb215-19"><a href="virus-vs-virus--dge.html#cb215-19" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="cn">NULL</span>,</span>
<span id="cb215-20"><a href="virus-vs-virus--dge.html#cb215-20" tabindex="-1"></a>       <span class="at">x=</span><span class="st">&quot;Influenza Stage&quot;</span>, <span class="at">y=</span><span class="st">&quot;log2(CPM)&quot;</span>) <span class="sc">+</span></span>
<span id="cb215-21"><a href="virus-vs-virus--dge.html#cb215-21" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb215-22"><a href="virus-vs-virus--dge.html#cb215-22" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid=</span><span class="fu">element_blank</span>())</span>
<span id="cb215-23"><a href="virus-vs-virus--dge.html#cb215-23" tabindex="-1"></a></span>
<span id="cb215-24"><a href="virus-vs-virus--dge.html#cb215-24" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;Figures/supplemental_G_female_edgeR_immunegenes.jpg&quot;</span>, g_plot_supplemental,</span>
<span id="cb215-25"><a href="virus-vs-virus--dge.html#cb215-25" tabindex="-1"></a>          <span class="at">width=</span><span class="dv">8</span>, <span class="at">height=</span><span class="dv">10</span>, <span class="at">units=</span><span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="kegg-table---manuscript" class="section level3 hasAnchor" number="4.5.7">
<h3><span class="header-section-number">4.5.7</span> KEGG Table - Manuscript<a href="virus-vs-virus--dge.html#kegg-table---manuscript" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="virus-vs-virus--dge.html#cb216-1" tabindex="-1"></a>t_kegg_table <span class="ot">&lt;-</span> </span>
<span id="cb216-2"><a href="virus-vs-virus--dge.html#cb216-2" tabindex="-1"></a>  txome_kegg_final <span class="sc">%&gt;%</span> </span>
<span id="cb216-3"><a href="virus-vs-virus--dge.html#cb216-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Pairwise Comparison (Approach)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Virus+ vs Virus- (Transcriptome)&quot;</span>,</span>
<span id="cb216-4"><a href="virus-vs-virus--dge.html#cb216-4" tabindex="-1"></a>         <span class="st">`</span><span class="at">KEGG Pathway</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb216-5"><a href="virus-vs-virus--dge.html#cb216-5" tabindex="-1"></a>         <span class="st">`</span><span class="at">Enrichment Ratio (FDR)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">3</span>), <span class="st">&quot; (&quot;</span>, <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">4</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb216-6"><a href="virus-vs-virus--dge.html#cb216-6" tabindex="-1"></a>         <span class="at">Genes=</span>userId) <span class="sc">%&gt;%</span> </span>
<span id="cb216-7"><a href="virus-vs-virus--dge.html#cb216-7" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">Pairwise Comparison (Approach)</span><span class="st">`</span>, <span class="st">`</span><span class="at">KEGG Pathway</span><span class="st">`</span>, <span class="st">`</span><span class="at">Enrichment Ratio (FDR)</span><span class="st">`</span>, Genes)</span>
<span id="cb216-8"><a href="virus-vs-virus--dge.html#cb216-8" tabindex="-1"></a></span>
<span id="cb216-9"><a href="virus-vs-virus--dge.html#cb216-9" tabindex="-1"></a>g_kegg_table <span class="ot">&lt;-</span></span>
<span id="cb216-10"><a href="virus-vs-virus--dge.html#cb216-10" tabindex="-1"></a>  gnome_kegg_final <span class="sc">%&gt;%</span> </span>
<span id="cb216-11"><a href="virus-vs-virus--dge.html#cb216-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Pairwise Comparison (Approach)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;Virus+ vs Virus- (Genome)&quot;</span>,</span>
<span id="cb216-12"><a href="virus-vs-virus--dge.html#cb216-12" tabindex="-1"></a>         <span class="st">`</span><span class="at">KEGG Pathway</span><span class="st">`</span> <span class="ot">=</span> description,</span>
<span id="cb216-13"><a href="virus-vs-virus--dge.html#cb216-13" tabindex="-1"></a>         <span class="st">`</span><span class="at">Enrichment Ratio (FDR)</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">2</span>), <span class="st">&quot; (&quot;</span>, <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">4</span>), <span class="st">&quot;)&quot;</span>),</span>
<span id="cb216-14"><a href="virus-vs-virus--dge.html#cb216-14" tabindex="-1"></a>         <span class="at">Genes=</span>userId) <span class="sc">%&gt;%</span> </span>
<span id="cb216-15"><a href="virus-vs-virus--dge.html#cb216-15" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">Pairwise Comparison (Approach)</span><span class="st">`</span>, <span class="st">`</span><span class="at">KEGG Pathway</span><span class="st">`</span>, <span class="st">`</span><span class="at">Enrichment Ratio (FDR)</span><span class="st">`</span>, Genes)</span>
<span id="cb216-16"><a href="virus-vs-virus--dge.html#cb216-16" tabindex="-1"></a></span>
<span id="cb216-17"><a href="virus-vs-virus--dge.html#cb216-17" tabindex="-1"></a></span>
<span id="cb216-18"><a href="virus-vs-virus--dge.html#cb216-18" tabindex="-1"></a>kegg_table <span class="ot">&lt;-</span> </span>
<span id="cb216-19"><a href="virus-vs-virus--dge.html#cb216-19" tabindex="-1"></a>  <span class="fu">rbind</span>(t_kegg_table, g_kegg_table)</span>
<span id="cb216-20"><a href="virus-vs-virus--dge.html#cb216-20" tabindex="-1"></a></span>
<span id="cb216-21"><a href="virus-vs-virus--dge.html#cb216-21" tabindex="-1"></a><span class="fu">write.csv</span>(kegg_table, <span class="st">&quot;Output Files/manuscript_keggtable.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="supplemental-table" class="section level3 hasAnchor" number="4.5.8">
<h3><span class="header-section-number">4.5.8</span> Supplemental Table<a href="virus-vs-virus--dge.html#supplemental-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="biological-process-4" class="section level4 hasAnchor" number="4.5.8.1">
<h4><span class="header-section-number">4.5.8.1</span> Biological Process<a href="virus-vs-virus--dge.html#biological-process-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="virus-vs-virus--dge.html#cb217-1" tabindex="-1"></a>t_sup_bp <span class="ot">&lt;-</span> </span>
<span id="cb217-2"><a href="virus-vs-virus--dge.html#cb217-2" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb217-3"><a href="virus-vs-virus--dge.html#cb217-3" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb217-4"><a href="virus-vs-virus--dge.html#cb217-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb217-5"><a href="virus-vs-virus--dge.html#cb217-5" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb217-6"><a href="virus-vs-virus--dge.html#cb217-6" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb217-7"><a href="virus-vs-virus--dge.html#cb217-7" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb217-8"><a href="virus-vs-virus--dge.html#cb217-8" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb217-9"><a href="virus-vs-virus--dge.html#cb217-9" tabindex="-1"></a>         <span class="at">Transcriptome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb217-10"><a href="virus-vs-virus--dge.html#cb217-10" tabindex="-1"></a>         <span class="at">Transcriptome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> t_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb217-11"><a href="virus-vs-virus--dge.html#cb217-11" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Transcriptome, Transcriptome_null)</span>
<span id="cb217-12"><a href="virus-vs-virus--dge.html#cb217-12" tabindex="-1"></a></span>
<span id="cb217-13"><a href="virus-vs-virus--dge.html#cb217-13" tabindex="-1"></a></span>
<span id="cb217-14"><a href="virus-vs-virus--dge.html#cb217-14" tabindex="-1"></a>g_sup_bp <span class="ot">&lt;-</span> </span>
<span id="cb217-15"><a href="virus-vs-virus--dge.html#cb217-15" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb217-16"><a href="virus-vs-virus--dge.html#cb217-16" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;bp&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb217-17"><a href="virus-vs-virus--dge.html#cb217-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb217-18"><a href="virus-vs-virus--dge.html#cb217-18" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb217-19"><a href="virus-vs-virus--dge.html#cb217-19" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb217-20"><a href="virus-vs-virus--dge.html#cb217-20" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb217-21"><a href="virus-vs-virus--dge.html#cb217-21" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb217-22"><a href="virus-vs-virus--dge.html#cb217-22" tabindex="-1"></a>         <span class="at">Genome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb217-23"><a href="virus-vs-virus--dge.html#cb217-23" tabindex="-1"></a>         <span class="at">Genome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> g_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb217-24"><a href="virus-vs-virus--dge.html#cb217-24" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Genome, Genome_null)</span>
<span id="cb217-25"><a href="virus-vs-virus--dge.html#cb217-25" tabindex="-1"></a></span>
<span id="cb217-26"><a href="virus-vs-virus--dge.html#cb217-26" tabindex="-1"></a>sup_bp <span class="ot">&lt;-</span> </span>
<span id="cb217-27"><a href="virus-vs-virus--dge.html#cb217-27" tabindex="-1"></a>  <span class="fu">merge</span>(t_sup_bp, g_sup_bp, <span class="at">all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb217-28"><a href="virus-vs-virus--dge.html#cb217-28" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">go=</span><span class="st">&quot;BP&quot;</span>)</span></code></pre></div>
</div>
<div id="cellular-component-4" class="section level4 hasAnchor" number="4.5.8.2">
<h4><span class="header-section-number">4.5.8.2</span> Cellular Component<a href="virus-vs-virus--dge.html#cellular-component-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="virus-vs-virus--dge.html#cb218-1" tabindex="-1"></a>t_sup_cc <span class="ot">&lt;-</span> </span>
<span id="cb218-2"><a href="virus-vs-virus--dge.html#cb218-2" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb218-3"><a href="virus-vs-virus--dge.html#cb218-3" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb218-4"><a href="virus-vs-virus--dge.html#cb218-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb218-5"><a href="virus-vs-virus--dge.html#cb218-5" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb218-6"><a href="virus-vs-virus--dge.html#cb218-6" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb218-7"><a href="virus-vs-virus--dge.html#cb218-7" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb218-8"><a href="virus-vs-virus--dge.html#cb218-8" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb218-9"><a href="virus-vs-virus--dge.html#cb218-9" tabindex="-1"></a>         <span class="at">Transcriptome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb218-10"><a href="virus-vs-virus--dge.html#cb218-10" tabindex="-1"></a>         <span class="at">Transcriptome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> t_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb218-11"><a href="virus-vs-virus--dge.html#cb218-11" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Transcriptome, Transcriptome_null)</span>
<span id="cb218-12"><a href="virus-vs-virus--dge.html#cb218-12" tabindex="-1"></a></span>
<span id="cb218-13"><a href="virus-vs-virus--dge.html#cb218-13" tabindex="-1"></a></span>
<span id="cb218-14"><a href="virus-vs-virus--dge.html#cb218-14" tabindex="-1"></a>g_sup_cc <span class="ot">&lt;-</span> </span>
<span id="cb218-15"><a href="virus-vs-virus--dge.html#cb218-15" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb218-16"><a href="virus-vs-virus--dge.html#cb218-16" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;cc&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb218-17"><a href="virus-vs-virus--dge.html#cb218-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb218-18"><a href="virus-vs-virus--dge.html#cb218-18" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb218-19"><a href="virus-vs-virus--dge.html#cb218-19" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb218-20"><a href="virus-vs-virus--dge.html#cb218-20" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb218-21"><a href="virus-vs-virus--dge.html#cb218-21" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb218-22"><a href="virus-vs-virus--dge.html#cb218-22" tabindex="-1"></a>         <span class="at">Genome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb218-23"><a href="virus-vs-virus--dge.html#cb218-23" tabindex="-1"></a>         <span class="at">Genome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> g_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb218-24"><a href="virus-vs-virus--dge.html#cb218-24" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Genome, Genome_null)</span>
<span id="cb218-25"><a href="virus-vs-virus--dge.html#cb218-25" tabindex="-1"></a></span>
<span id="cb218-26"><a href="virus-vs-virus--dge.html#cb218-26" tabindex="-1"></a>sup_cc <span class="ot">&lt;-</span> </span>
<span id="cb218-27"><a href="virus-vs-virus--dge.html#cb218-27" tabindex="-1"></a>  <span class="fu">merge</span>(t_sup_cc, g_sup_cc, <span class="at">all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb218-28"><a href="virus-vs-virus--dge.html#cb218-28" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">go=</span><span class="st">&quot;CC&quot;</span>)</span></code></pre></div>
</div>
<div id="molecular-function-4" class="section level4 hasAnchor" number="4.5.8.3">
<h4><span class="header-section-number">4.5.8.3</span> Molecular Function<a href="virus-vs-virus--dge.html#molecular-function-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="virus-vs-virus--dge.html#cb219-1" tabindex="-1"></a>t_sup_mf <span class="ot">&lt;-</span> </span>
<span id="cb219-2"><a href="virus-vs-virus--dge.html#cb219-2" tabindex="-1"></a>  txome_go <span class="sc">%&gt;%</span> </span>
<span id="cb219-3"><a href="virus-vs-virus--dge.html#cb219-3" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;mf&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb219-4"><a href="virus-vs-virus--dge.html#cb219-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb219-5"><a href="virus-vs-virus--dge.html#cb219-5" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb219-6"><a href="virus-vs-virus--dge.html#cb219-6" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb219-7"><a href="virus-vs-virus--dge.html#cb219-7" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb219-8"><a href="virus-vs-virus--dge.html#cb219-8" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb219-9"><a href="virus-vs-virus--dge.html#cb219-9" tabindex="-1"></a>         <span class="at">Transcriptome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb219-10"><a href="virus-vs-virus--dge.html#cb219-10" tabindex="-1"></a>         <span class="at">Transcriptome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> t_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb219-11"><a href="virus-vs-virus--dge.html#cb219-11" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Transcriptome, Transcriptome_null)</span>
<span id="cb219-12"><a href="virus-vs-virus--dge.html#cb219-12" tabindex="-1"></a></span>
<span id="cb219-13"><a href="virus-vs-virus--dge.html#cb219-13" tabindex="-1"></a></span>
<span id="cb219-14"><a href="virus-vs-virus--dge.html#cb219-14" tabindex="-1"></a>g_sup_mf <span class="ot">&lt;-</span> </span>
<span id="cb219-15"><a href="virus-vs-virus--dge.html#cb219-15" tabindex="-1"></a>  gnome_go <span class="sc">%&gt;%</span> </span>
<span id="cb219-16"><a href="virus-vs-virus--dge.html#cb219-16" tabindex="-1"></a>  <span class="fu">filter</span>(go<span class="sc">==</span><span class="st">&quot;mf&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb219-17"><a href="virus-vs-virus--dge.html#cb219-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GOTerm =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb219-18"><a href="virus-vs-virus--dge.html#cb219-18" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb219-19"><a href="virus-vs-virus--dge.html#cb219-19" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb219-20"><a href="virus-vs-virus--dge.html#cb219-20" tabindex="-1"></a>         <span class="at">enrichmentRatio =</span> <span class="fu">ifelse</span>(direction<span class="sc">==</span><span class="st">&quot;down&quot;</span>, <span class="fu">paste</span>(<span class="st">&quot;-&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>),</span>
<span id="cb219-21"><a href="virus-vs-virus--dge.html#cb219-21" tabindex="-1"></a>                                  <span class="fu">paste</span>(<span class="st">&quot;+&quot;</span>, enrichmentRatio, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)),</span>
<span id="cb219-22"><a href="virus-vs-virus--dge.html#cb219-22" tabindex="-1"></a>         <span class="at">Genome =</span> <span class="fu">paste</span>(enrichmentRatio, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb219-23"><a href="virus-vs-virus--dge.html#cb219-23" tabindex="-1"></a>         <span class="at">Genome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> g_null_bp_10<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb219-24"><a href="virus-vs-virus--dge.html#cb219-24" tabindex="-1"></a>  <span class="fu">select</span>(GOTerm, Genome, Genome_null)</span>
<span id="cb219-25"><a href="virus-vs-virus--dge.html#cb219-25" tabindex="-1"></a></span>
<span id="cb219-26"><a href="virus-vs-virus--dge.html#cb219-26" tabindex="-1"></a>sup_mf <span class="ot">&lt;-</span> </span>
<span id="cb219-27"><a href="virus-vs-virus--dge.html#cb219-27" tabindex="-1"></a>  <span class="fu">merge</span>(t_sup_mf, g_sup_mf, <span class="at">all=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb219-28"><a href="virus-vs-virus--dge.html#cb219-28" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">go=</span><span class="st">&quot;MF&quot;</span>)</span></code></pre></div>
</div>
<div id="save-go-tables" class="section level4 hasAnchor" number="4.5.8.4">
<h4><span class="header-section-number">4.5.8.4</span> Save GO Tables<a href="virus-vs-virus--dge.html#save-go-tables" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="virus-vs-virus--dge.html#cb220-1" tabindex="-1"></a>sup <span class="ot">&lt;-</span> </span>
<span id="cb220-2"><a href="virus-vs-virus--dge.html#cb220-2" tabindex="-1"></a>  <span class="fu">rbind</span>(sup_bp, sup_cc, sup_mf)</span>
<span id="cb220-3"><a href="virus-vs-virus--dge.html#cb220-3" tabindex="-1"></a></span>
<span id="cb220-4"><a href="virus-vs-virus--dge.html#cb220-4" tabindex="-1"></a>sup <span class="sc">%&gt;%</span> </span>
<span id="cb220-5"><a href="virus-vs-virus--dge.html#cb220-5" tabindex="-1"></a>  <span class="fu">kable</span>()</span>
<span id="cb220-6"><a href="virus-vs-virus--dge.html#cb220-6" tabindex="-1"></a></span>
<span id="cb220-7"><a href="virus-vs-virus--dge.html#cb220-7" tabindex="-1"></a><span class="fu">write.csv</span>(sup, <span class="st">&quot;Output Files/supplemental_female_edgeR_viralRNA_go.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="kegg" class="section level4 hasAnchor" number="4.5.8.5">
<h4><span class="header-section-number">4.5.8.5</span> KEGG<a href="virus-vs-virus--dge.html#kegg" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="virus-vs-virus--dge.html#cb221-1" tabindex="-1"></a>t_sup_kegg <span class="ot">&lt;-</span> </span>
<span id="cb221-2"><a href="virus-vs-virus--dge.html#cb221-2" tabindex="-1"></a>  txome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb221-3"><a href="virus-vs-virus--dge.html#cb221-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb221-4"><a href="virus-vs-virus--dge.html#cb221-4" tabindex="-1"></a>         <span class="at">Enrichment =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb221-5"><a href="virus-vs-virus--dge.html#cb221-5" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb221-6"><a href="virus-vs-virus--dge.html#cb221-6" tabindex="-1"></a>         <span class="at">Transcriptome =</span> <span class="fu">paste</span>(Enrichment, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb221-7"><a href="virus-vs-virus--dge.html#cb221-7" tabindex="-1"></a>         <span class="at">Transcriptome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> t_kegg<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb221-8"><a href="virus-vs-virus--dge.html#cb221-8" tabindex="-1"></a>         <span class="at">Transcriptome_Genes =</span> userId) <span class="sc">%&gt;%</span> </span>
<span id="cb221-9"><a href="virus-vs-virus--dge.html#cb221-9" tabindex="-1"></a>  <span class="fu">select</span>(KEGG, Transcriptome, Transcriptome_null, Transcriptome_Genes)</span>
<span id="cb221-10"><a href="virus-vs-virus--dge.html#cb221-10" tabindex="-1"></a></span>
<span id="cb221-11"><a href="virus-vs-virus--dge.html#cb221-11" tabindex="-1"></a>g_sup_kegg <span class="ot">&lt;-</span> </span>
<span id="cb221-12"><a href="virus-vs-virus--dge.html#cb221-12" tabindex="-1"></a>  gnome_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb221-13"><a href="virus-vs-virus--dge.html#cb221-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">paste</span>(description, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, geneSet, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb221-14"><a href="virus-vs-virus--dge.html#cb221-14" tabindex="-1"></a>         <span class="at">Enrichment =</span> <span class="fu">round</span>(enrichmentRatio, <span class="at">digits=</span><span class="dv">1</span>),</span>
<span id="cb221-15"><a href="virus-vs-virus--dge.html#cb221-15" tabindex="-1"></a>         <span class="at">FDR =</span> <span class="fu">round</span>(FDR, <span class="at">digits=</span><span class="dv">3</span>),</span>
<span id="cb221-16"><a href="virus-vs-virus--dge.html#cb221-16" tabindex="-1"></a>         <span class="at">Genome =</span> <span class="fu">paste</span>(Enrichment, <span class="st">&quot; &quot;</span>, <span class="st">&quot;(&quot;</span>, FDR, <span class="st">&quot;)&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>), </span>
<span id="cb221-17"><a href="virus-vs-virus--dge.html#cb221-17" tabindex="-1"></a>         <span class="at">Genome_null =</span> <span class="fu">ifelse</span>(description <span class="sc">%in%</span> g_kegg<span class="sc">$</span>value, <span class="st">&quot;x&quot;</span>, <span class="st">&quot;&quot;</span>),</span>
<span id="cb221-18"><a href="virus-vs-virus--dge.html#cb221-18" tabindex="-1"></a>         <span class="at">Genome_Genes =</span> userId) <span class="sc">%&gt;%</span> </span>
<span id="cb221-19"><a href="virus-vs-virus--dge.html#cb221-19" tabindex="-1"></a>  <span class="fu">select</span>(KEGG, Genome, Genome_null, Genome_Genes)</span>
<span id="cb221-20"><a href="virus-vs-virus--dge.html#cb221-20" tabindex="-1"></a></span>
<span id="cb221-21"><a href="virus-vs-virus--dge.html#cb221-21" tabindex="-1"></a></span>
<span id="cb221-22"><a href="virus-vs-virus--dge.html#cb221-22" tabindex="-1"></a>sup_kegg <span class="ot">&lt;-</span> </span>
<span id="cb221-23"><a href="virus-vs-virus--dge.html#cb221-23" tabindex="-1"></a>  <span class="fu">merge</span>(t_sup_kegg, g_sup_kegg, <span class="at">all=</span><span class="cn">TRUE</span>)</span>
<span id="cb221-24"><a href="virus-vs-virus--dge.html#cb221-24" tabindex="-1"></a></span>
<span id="cb221-25"><a href="virus-vs-virus--dge.html#cb221-25" tabindex="-1"></a>sup_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb221-26"><a href="virus-vs-virus--dge.html#cb221-26" tabindex="-1"></a>  <span class="fu">kable</span>()</span>
<span id="cb221-27"><a href="virus-vs-virus--dge.html#cb221-27" tabindex="-1"></a></span>
<span id="cb221-28"><a href="virus-vs-virus--dge.html#cb221-28" tabindex="-1"></a><span class="fu">write.csv</span>(sup_kegg, <span class="st">&quot;Output Files/supplemental_female_edgeR_viralRNA_kegg.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">FALSE</span>)</span></code></pre></div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="male-v-female-dge.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="influenza-stage-dge.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": false
});
});
</script>

</body>

</html>
